# Re-use of bulk RNA-seq methods for spatial data exercise

Instructor: Leo

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">New in <a href="https://twitter.com/ScienceMagazine?ref_src=twsrc%5Etfw">@sciencemagazine</a>: our work from <a href="https://twitter.com/LieberInstitute?ref_src=twsrc%5Etfw">@LieberInstitute</a> <a href="https://twitter.com/hashtag/spatialDLPFC?src=hash&amp;ref_src=twsrc%5Etfw">#spatialDLPFC</a> applies <a href="https://twitter.com/hashtag/snRNAseq?src=hash&amp;ref_src=twsrc%5Etfw">#snRNAseq</a> and <a href="https://twitter.com/hashtag/Visium?src=hash&amp;ref_src=twsrc%5Etfw">#Visium</a> spatial transcriptomic in the DLPFC to better understand anatomical structure and cellular populations in the human brain <a href="https://twitter.com/hashtag/PsychENCODE?src=hash&amp;ref_src=twsrc%5Etfw">#PsychENCODE</a> <a href="https://t.co/DKZqmG4YDi">https://t.co/DKZqmG4YDi</a> <a href="https://t.co/Tjp2OjTo63">https://t.co/Tjp2OjTo63</a> <a href="https://t.co/vQbjts2JtQ">pic.twitter.com/vQbjts2JtQ</a></p>&mdash; Louise Huuki-Myers (@lahuuki) <a href="https://twitter.com/lahuuki/status/1793714294334648638?ref_src=twsrc%5Etfw">May 23, 2024</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

## Spatial registration

In 2023, Louise A. Huuki-Myers contributed a new vignette to `spatialLIBD` as noted on the package news / changelog: http://research.libd.org/spatialLIBD/news/index.html#spatiallibd-1132.

You should be able to run without any issues the code Louise explained at http://research.libd.org/spatialLIBD/articles/guide_to_spatial_registration.html. This same information is displayed at https://bioconductor.org/packages/release/data/experiment/vignettes/spatialLIBD/inst/doc/guide_to_spatial_registration.html.

```{r "download_modeling_results"}
## get reference layer enrichment statistics
layer_modeling_results <- spatialLIBD::fetch_data(type = "modeling_results")
```

If the above doesn't work, related to the `curl` issue [we previously discussed](https://lcolladotor.github.io/cshl_rstats_genome_scale_2024/interactive-summarizedexperiment-visualizations.html#setting-up-the-data), then use this workaround:

```{r "layer_modeling_results_workaround"}
tmp_modeling_results <- tempfile("modeling_results.RData")
download.file(
    "https://www.dropbox.com/s/se6rrgb9yhm5gfh/Human_DLPFC_Visium_modeling_results.Rdata?dl=1",
    tmp_modeling_results,
    mode = "wb"
)
load(tmp_modeling_results, verbose = TRUE)
layer_modeling_results <- modeling_results
```



<div style="background-color:#E0EEE0; padding:20px; font-family: sans-serif">

This journal club style video of the main results of the `spatialDLPFC` paper does explain the basics of *spatial registration*:

<iframe width="560" height="315" src="https://www.youtube.com/embed/EhP5-mhw29w?si=QFY3joJ7qEWFBXf-" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>

For more on `spatialDLPFC`, check this second video about the supplementary results:

<iframe width="560" height="315" src="https://www.youtube.com/embed/vSjXCni8Ndc?si=FqM85j1zPB8fOsT5" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>

</div>

## Exercise

<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}
</style>

<p class="exercise">
**Exercise**:
Follow the vignette on spatial registration. Do the results change when you use `cutoff_merge_ratio = 0.1`? What is this argument controlling?
</p>
