<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 biocthis introduction | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="10 biocthis introduction | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="10 biocthis introduction | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 biocthis introduction | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="research-talks.html"/>
<link rel="next" href="scrna-seq-data-analysis-overview.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.2</b> Exercises</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.3</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
<li class="chapter" data-level="2.1.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment"><i class="fa fa-check"></i><b>2.1.3</b> SpatialExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
<li class="chapter" data-level="2.4.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#explore-the-data"><i class="fa fa-check"></i><b>2.4.2</b> Explore the Data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features"><i class="fa fa-check"></i><b>2.5</b> Introduction to Advanced iSEE Features</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression"><i class="fa fa-check"></i><b>2.5.1</b> Selecting Cells Based on a Single Gene Expression</a></li>
<li class="chapter" data-level="2.5.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression"><i class="fa fa-check"></i><b>2.5.2</b> Using a Single Plot for Two Gene Co-Expression</a></li>
<li class="chapter" data-level="2.5.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes"><i class="fa fa-check"></i><b>2.5.3</b> Selecting Cells Based on the Co-Expression of Two or more Genes</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.6</b> References</a></li>
<li class="chapter" data-level="2.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#community"><i class="fa fa-check"></i><b>2.7</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-1"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-2"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>9</b> Research talks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>9.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="9.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>9.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="9.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>9.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>10</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>10.1</b> Related past materials</a></li>
<li class="chapter" data-level="10.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>10.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="10.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>10.3</b> Live demo</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#example-function"><i class="fa fa-check"></i><b>10.3.1</b> Example function</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-3"><i class="fa fa-check"></i><b>10.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>11</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="11.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing"><i class="fa fa-check"></i><b>11.1</b> Single cell RNA sequencing</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#pre-processing-of-scrna-seq-data-before-r"><i class="fa fa-check"></i><b>11.1.1</b> Pre-processing of scRNA-seq Data (Before R)</a></li>
<li class="chapter" data-level="11.1.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#different-technologies"><i class="fa fa-check"></i><b>11.1.2</b> Different Technologies</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basic-workflow"><i class="fa fa-check"></i><b>11.2</b> Basic Workflow</a></li>
<li class="chapter" data-level="11.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>11.3</b> The SingleCellExperiment class</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#data-loading"><i class="fa fa-check"></i><b>11.3.1</b> Data Loading</a></li>
<li class="chapter" data-level="11.3.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basics-of-your-sce"><i class="fa fa-check"></i><b>11.3.2</b> Basics of your SCE</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>11.4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#common-choices-of-qc-metrics"><i class="fa fa-check"></i><b>11.4.1</b> Common choices of QC metrics</a></li>
<li class="chapter" data-level="11.4.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#identifying-low-quality-cells"><i class="fa fa-check"></i><b>11.4.2</b> Identifying low-quality cells</a></li>
<li class="chapter" data-level="11.4.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#checking-diagnostic-plots"><i class="fa fa-check"></i><b>11.4.3</b> Checking diagnostic plots</a></li>
<li class="chapter" data-level="11.4.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#removing-low-quality-cells"><i class="fa fa-check"></i><b>11.4.4</b> Removing low-quality cells</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>11.5</b> Normalization</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#library-size-normalization"><i class="fa fa-check"></i><b>11.5.1</b> Library size normalization</a></li>
<li class="chapter" data-level="11.5.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-deconvolution"><i class="fa fa-check"></i><b>11.5.2</b> Normalization by deconvolution</a></li>
<li class="chapter" data-level="11.5.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-spike-ins"><i class="fa fa-check"></i><b>11.5.3</b> Normalization by spike-ins</a></li>
<li class="chapter" data-level="11.5.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#scaling-and-log-transforming"><i class="fa fa-check"></i><b>11.5.4</b> Scaling and log-transforming</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>11.6</b> Feature selection</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-per-gene-variation"><i class="fa fa-check"></i><b>11.6.1</b> Quantifying per-gene variation</a></li>
<li class="chapter" data-level="11.6.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-spike-ins"><i class="fa fa-check"></i><b>11.6.2</b> Quantifying technical noise (spike-ins)</a></li>
<li class="chapter" data-level="11.6.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-mean-variance-trend"><i class="fa fa-check"></i><b>11.6.3</b> Quantifying technical noise (mean-variance trend)</a></li>
<li class="chapter" data-level="11.6.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#handling-batch-effects"><i class="fa fa-check"></i><b>11.6.4</b> Handling batch effects</a></li>
<li class="chapter" data-level="11.6.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#selecting-highly-variable-genes"><i class="fa fa-check"></i><b>11.6.5</b> Selecting highly variable genes</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>11.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#principal-components-analysis"><i class="fa fa-check"></i><b>11.7.1</b> Principal components analysis</a></li>
<li class="chapter" data-level="11.7.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#choosing-the-number-of-pcs"><i class="fa fa-check"></i><b>11.7.2</b> Choosing the number of PCs</a></li>
<li class="chapter" data-level="11.7.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#visualizing-the-pcs"><i class="fa fa-check"></i><b>11.7.3</b> Visualizing the PCs</a></li>
<li class="chapter" data-level="11.7.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#non-linear-methods-for-visualization"><i class="fa fa-check"></i><b>11.7.4</b> Non-linear methods for visualization</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>11.8</b> Clustering</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#graph-based-clustering"><i class="fa fa-check"></i><b>11.8.1</b> Graph-based clustering</a></li>
<li class="chapter" data-level="11.8.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#adjusting-the-parameters"><i class="fa fa-check"></i><b>11.8.2</b> Adjusting the parameters</a></li>
<li class="chapter" data-level="11.8.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#hierarchical-clustering"><i class="fa fa-check"></i><b>11.8.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="11.8.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#subclustering"><i class="fa fa-check"></i><b>11.8.4</b> Subclustering</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>11.9</b> Marker gene detection</a>
<ul>
<li class="chapter" data-level="11.9.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#scoring-markers-by-pairwise-comparisons"><i class="fa fa-check"></i><b>11.9.1</b> Scoring markers by pairwise comparisons</a></li>
<li class="chapter" data-level="11.9.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#effect-sizes-for-pairwise-comparisons"><i class="fa fa-check"></i><b>11.9.2</b> Effect sizes for pairwise comparisons</a></li>
<li class="chapter" data-level="11.9.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#summaries-of-pairwise-effects"><i class="fa fa-check"></i><b>11.9.3</b> Summaries of pairwise effects</a></li>
<li class="chapter" data-level="11.9.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#choose-top-markers"><i class="fa fa-check"></i><b>11.9.4</b> Choose top markers</a></li>
<li class="chapter" data-level="11.9.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#using-a-log-fold-change-threshold"><i class="fa fa-check"></i><b>11.9.5</b> Using a log-fold change threshold</a></li>
<li class="chapter" data-level="11.9.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#handling-blocking-factors"><i class="fa fa-check"></i><b>11.9.6</b> Handling blocking factors</a></li>
<li class="chapter" data-level="11.9.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#deconvobuddies"><i class="fa fa-check"></i><b>11.9.7</b> Deconvobuddies</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>11.10</b> Cell type annotation</a>
<ul>
<li class="chapter" data-level="11.10.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#assigning-cell-labels-from-reference-data"><i class="fa fa-check"></i><b>11.10.1</b> Assigning cell labels from reference data</a></li>
<li class="chapter" data-level="11.10.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#getting-ready-again"><i class="fa fa-check"></i><b>11.10.2</b> Getting ready again</a></li>
<li class="chapter" data-level="11.10.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#using-existing-references"><i class="fa fa-check"></i><b>11.10.3</b> Using existing references</a></li>
<li class="chapter" data-level="11.10.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#assigning-cell-labels-from-gene-sets"><i class="fa fa-check"></i><b>11.10.4</b> Assigning cell labels from gene sets</a></li>
</ul></li>
<li class="chapter" data-level="11.11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#references-4"><i class="fa fa-check"></i><b>11.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>12</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>12.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>12.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>12.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>12.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>13</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="13.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>13.1</b> Spatial registration</a></li>
<li class="chapter" data-level="13.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>13.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>14</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="14.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>14.1</b> here</a></li>
<li class="chapter" data-level="14.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>14.2</b> Usethis</a></li>
<li class="chapter" data-level="14.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>14.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>14.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="14.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>14.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="14.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>14.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="14.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>14.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>14.4</b> R websites</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>14.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="14.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>14.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="14.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>14.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="14.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>14.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>14.5</b> postcards</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>14.5.1</b> Installation</a></li>
<li class="chapter" data-level="14.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>14.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>14.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>14.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="14.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>14.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="14.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>14.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="14.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>14.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="14.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>14.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-5"><i class="fa fa-check"></i><b>14.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="biocthis-introduction" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">10</span> biocthis introduction<a href="biocthis-introduction.html#biocthis-introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Leo</p>
<div id="related-past-materials" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Related past materials<a href="biocthis-introduction.html#related-past-materials" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I’ve taught a lot about <code>biocthis</code> over the years. Here’s a 2020 video:</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/aMTxkYsM-8o" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen>
</iframe>
<script defer class="speakerdeck-embed" data-id="f2e58b59b83244179f062c2852af25ff" data-ratio="1.77777777777778" src="//speakerdeck.com/assets/embed.js"></script>
<p>and more recently, these are the <a href="https://docs.google.com/document/d/117p0sKFKpMSZy90ZWc56_J-MhwfSMEnMVkqAVqZStaA/edit?usp=sharing"><em>LIBD rstats club 2023-03-10 notes</em></a>.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/Rbadjup630w?si=MEvPQQVKeVs7H_5E" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen>
</iframe>
</div>
<div id="biocthis-main-commands" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> biocthis main commands<a href="biocthis-introduction.html#biocthis-main-commands" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><a href="https://bioconductor.org/packages/biocthis" class="uri">https://bioconductor.org/packages/biocthis</a></li>
<li><code>pkgdown</code> documentation website: <a href="https://lcolladotor.github.io/biocthis/" class="uri">https://lcolladotor.github.io/biocthis/</a>
<ul>
<li><code>biocthis::use_bioc_pkg_templates()</code> documentation: <a href="https://lcolladotor.github.io/biocthis/reference/use_bioc_pkg_templates.html" class="uri">https://lcolladotor.github.io/biocthis/reference/use_bioc_pkg_templates.html</a></li>
</ul></li>
</ul>
<p>These are the main steps you will need to know to make a Bioconductor package
with <code>biocthis</code>:</p>
<ol style="list-style-type: decimal">
<li>You first will need to create a package using a command from <code>usethis</code>.</li>
</ol>
<ul>
<li>For example: <code>usethis::create_package("~/Desktop/cshl2024pkg")</code></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Now that you have a package, we can use <code>biocthis</code> to create 4 template R
scripts that will guide you and help you make the full structure for a
Bioconductor R package.</li>
</ol>
<ul>
<li>On your new R package (<code>cshl2024pkg</code>), we can now use <code>biocthis::use_bioc_pkg_templates()</code>.</li>
</ul>
<p>In part these commands were born out of my own self interest to make it easier
to make new packages instead of copy-pasting the contents of an older one, then
manually adjusting all the pieces for a new package. See
<a href="https://lcolladotor.github.io/pkgs/" class="uri">https://lcolladotor.github.io/pkgs/</a> for the list of all the R packages I’ve been
involved in.</p>
</div>
<div id="live-demo" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Live demo<a href="biocthis-introduction.html#live-demo" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here is the live demo result <a href="https://github.com/lcolladotor/cshl2024pkg/" class="uri">https://github.com/lcolladotor/cshl2024pkg/</a> with
its companion documentation website at
<a href="https://lcolladotor.github.io/cshl2024pkg/" class="uri">https://lcolladotor.github.io/cshl2024pkg/</a>. You might also want to check the
2023 version at <a href="https://github.com/lcolladotor/cshl2024pkg/" class="uri">https://github.com/lcolladotor/cshl2024pkg/</a>.</p>
<p>Check the git commit history at
<a href="https://github.com/lcolladotor/cshl2024pkg/commits/devel" class="uri">https://github.com/lcolladotor/cshl2024pkg/commits/devel</a> and the GitHub Actions
history at <a href="https://github.com/lcolladotor/cshl2024pkg/actions" class="uri">https://github.com/lcolladotor/cshl2024pkg/actions</a>. We can see at
<a href="https://app.codecov.io/gh/lcolladotor/cshl2024pkg" class="uri">https://app.codecov.io/gh/lcolladotor/cshl2024pkg</a> the <em>code coverage</em> results
for this demonstration package.</p>
<div id="example-function" class="section level3 hasAnchor" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> Example function<a href="biocthis-introduction.html#example-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s have a function to work with: <code>weekday_praise()</code>.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="biocthis-introduction.html#cb118-1" tabindex="-1"></a>weekday_praise <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">date =</span> <span class="fu">Sys.Date</span>()) {</span>
<span id="cb118-2"><a href="biocthis-introduction.html#cb118-2" tabindex="-1"></a>    date <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(date)</span>
<span id="cb118-3"><a href="biocthis-introduction.html#cb118-3" tabindex="-1"></a>    date_weekday <span class="ot">&lt;-</span> <span class="fu">weekdays</span>(date)</span>
<span id="cb118-4"><a href="biocthis-introduction.html#cb118-4" tabindex="-1"></a>    <span class="fu">paste0</span>(date_weekday, <span class="st">&quot;: &quot;</span>, praise<span class="sc">::</span><span class="fu">praise</span>())</span>
<span id="cb118-5"><a href="biocthis-introduction.html#cb118-5" tabindex="-1"></a>}</span>
<span id="cb118-6"><a href="biocthis-introduction.html#cb118-6" tabindex="-1"></a><span class="fu">weekday_praise</span>()</span>
<span id="cb118-7"><a href="biocthis-introduction.html#cb118-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tuesday: You are perfect!&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="biocthis-introduction.html#cb119-1" tabindex="-1"></a><span class="fu">weekday_praise</span>(<span class="st">&quot;2024-06-09&quot;</span>)</span>
<span id="cb119-2"><a href="biocthis-introduction.html#cb119-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Sunday: You are excellent!&quot;</span></span></code></pre></div>
<p>Here’s the full code for the function and its documentation.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="biocthis-introduction.html#cb120-1" tabindex="-1"></a><span class="co">#&#39; Praise a weekday</span></span>
<span id="cb120-2"><a href="biocthis-introduction.html#cb120-2" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-3"><a href="biocthis-introduction.html#cb120-3" tabindex="-1"></a><span class="co">#&#39; Given a date, figure out which weekday it was, then write a positive</span></span>
<span id="cb120-4"><a href="biocthis-introduction.html#cb120-4" tabindex="-1"></a><span class="co">#&#39; message.</span></span>
<span id="cb120-5"><a href="biocthis-introduction.html#cb120-5" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-6"><a href="biocthis-introduction.html#cb120-6" tabindex="-1"></a><span class="co">#&#39; @param date A `base::Date` object or a `character()` in a format that can be</span></span>
<span id="cb120-7"><a href="biocthis-introduction.html#cb120-7" tabindex="-1"></a><span class="co">#&#39; converted to a `base::Date` object with `base::as.Date()`.</span></span>
<span id="cb120-8"><a href="biocthis-introduction.html#cb120-8" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-9"><a href="biocthis-introduction.html#cb120-9" tabindex="-1"></a><span class="co">#&#39; @importFrom praise praise</span></span>
<span id="cb120-10"><a href="biocthis-introduction.html#cb120-10" tabindex="-1"></a><span class="co">#&#39; @export</span></span>
<span id="cb120-11"><a href="biocthis-introduction.html#cb120-11" tabindex="-1"></a><span class="co">#&#39; @examples</span></span>
<span id="cb120-12"><a href="biocthis-introduction.html#cb120-12" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-13"><a href="biocthis-introduction.html#cb120-13" tabindex="-1"></a><span class="co">#&#39; ## Praise the current weekday</span></span>
<span id="cb120-14"><a href="biocthis-introduction.html#cb120-14" tabindex="-1"></a><span class="co">#&#39; weekday_praise()</span></span>
<span id="cb120-15"><a href="biocthis-introduction.html#cb120-15" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-16"><a href="biocthis-introduction.html#cb120-16" tabindex="-1"></a><span class="co">#&#39; ## Praise the date we started teaching</span></span>
<span id="cb120-17"><a href="biocthis-introduction.html#cb120-17" tabindex="-1"></a><span class="co">#&#39; weekday_praise(&quot;2024-06-09&quot;)</span></span>
<span id="cb120-18"><a href="biocthis-introduction.html#cb120-18" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-19"><a href="biocthis-introduction.html#cb120-19" tabindex="-1"></a><span class="co">#&#39; ## Praise the current weekday in a reproducible way</span></span>
<span id="cb120-20"><a href="biocthis-introduction.html#cb120-20" tabindex="-1"></a><span class="co">#&#39; set.seed(20240610)</span></span>
<span id="cb120-21"><a href="biocthis-introduction.html#cb120-21" tabindex="-1"></a><span class="co">#&#39; weekday_praise()</span></span>
<span id="cb120-22"><a href="biocthis-introduction.html#cb120-22" tabindex="-1"></a><span class="co">#&#39;</span></span>
<span id="cb120-23"><a href="biocthis-introduction.html#cb120-23" tabindex="-1"></a><span class="co">#&#39; ## Verify that it&#39;s reproducible</span></span>
<span id="cb120-24"><a href="biocthis-introduction.html#cb120-24" tabindex="-1"></a><span class="co">#&#39; set.seed(20240610)</span></span>
<span id="cb120-25"><a href="biocthis-introduction.html#cb120-25" tabindex="-1"></a><span class="co">#&#39; weekday_praise()</span></span>
<span id="cb120-26"><a href="biocthis-introduction.html#cb120-26" tabindex="-1"></a>weekday_praise <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">date =</span> <span class="fu">Sys.Date</span>()) {</span>
<span id="cb120-27"><a href="biocthis-introduction.html#cb120-27" tabindex="-1"></a>    date <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(date)</span>
<span id="cb120-28"><a href="biocthis-introduction.html#cb120-28" tabindex="-1"></a>    date_weekday <span class="ot">&lt;-</span> <span class="fu">weekdays</span>(date)</span>
<span id="cb120-29"><a href="biocthis-introduction.html#cb120-29" tabindex="-1"></a>    <span class="fu">paste0</span>(date_weekday, <span class="st">&quot;: &quot;</span>, praise<span class="sc">::</span><span class="fu">praise</span>())</span>
<span id="cb120-30"><a href="biocthis-introduction.html#cb120-30" tabindex="-1"></a>}</span></code></pre></div>
<p>Here’s a test for our function too.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="biocthis-introduction.html#cb121-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;testthat&quot;</span>)</span>
<span id="cb121-2"><a href="biocthis-introduction.html#cb121-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb121-3"><a href="biocthis-introduction.html#cb121-3" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;testthat&#39;</span></span>
<span id="cb121-4"><a href="biocthis-introduction.html#cb121-4" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:rlang&#39;:</span></span>
<span id="cb121-5"><a href="biocthis-introduction.html#cb121-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb121-6"><a href="biocthis-introduction.html#cb121-6" tabindex="-1"></a><span class="co">#&gt;     is_false, is_null, is_true</span></span>
<span id="cb121-7"><a href="biocthis-introduction.html#cb121-7" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:Hmisc&#39;:</span></span>
<span id="cb121-8"><a href="biocthis-introduction.html#cb121-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb121-9"><a href="biocthis-introduction.html#cb121-9" tabindex="-1"></a><span class="co">#&gt;     describe</span></span></code></pre></div>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="biocthis-introduction.html#cb122-1" tabindex="-1"></a></span>
<span id="cb122-2"><a href="biocthis-introduction.html#cb122-2" tabindex="-1"></a><span class="do">## Verify that we get the result we wanted</span></span>
<span id="cb122-3"><a href="biocthis-introduction.html#cb122-3" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20240610</span>)</span>
<span id="cb122-4"><a href="biocthis-introduction.html#cb122-4" tabindex="-1"></a><span class="fu">expect_equal</span>(<span class="fu">weekday_praise</span>(<span class="st">&quot;2024-06-09&quot;</span>), <span class="st">&quot;Sunday: You are wondrous!&quot;</span>)</span>
<span id="cb122-5"><a href="biocthis-introduction.html#cb122-5" tabindex="-1"></a></span>
<span id="cb122-6"><a href="biocthis-introduction.html#cb122-6" tabindex="-1"></a><span class="do">## Verify that we get an error if the input is not correct</span></span>
<span id="cb122-7"><a href="biocthis-introduction.html#cb122-7" tabindex="-1"></a><span class="fu">expect_error</span>(<span class="fu">weekday_praise</span>(<span class="st">&quot;240609&quot;</span>))</span>
<span id="cb122-8"><a href="biocthis-introduction.html#cb122-8" tabindex="-1"></a></span>
<span id="cb122-9"><a href="biocthis-introduction.html#cb122-9" tabindex="-1"></a><span class="do">## Should work for a vector input</span></span>
<span id="cb122-10"><a href="biocthis-introduction.html#cb122-10" tabindex="-1"></a><span class="fu">expect_equal</span>(<span class="fu">length</span>(<span class="fu">weekday_praise</span>(<span class="fu">c</span>(<span class="st">&quot;2024-06-09&quot;</span>, <span class="st">&quot;2024-06-10&quot;</span>))), <span class="dv">2</span><span class="dt">L</span>)</span></code></pre></div>
</div>
</div>
<div id="community-3" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Community<a href="biocthis-introduction.html#community-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For more materials on R/Bioconductor package development check <a href="http://contributions.bioconductor.org/" class="uri">http://contributions.bioconductor.org/</a>.</p>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
I’m on a Friday night mood now enjoying <a href="https://twitter.com/lmwebr?ref_src=twsrc%5Etfw"><span class="citation">@lmwebr</span></a>’s <a href="https://twitter.com/hashtag/OSTA?src=hash&amp;ref_src=twsrc%5Etfw">#OSTA</a> workshop 🔥, feeling grateful 🙏🏽 to everyone who nominated me for the <a href="https://twitter.com/hashtag/BioC2021?src=hash&amp;ref_src=twsrc%5Etfw">#BioC2021</a> community award 🥇&amp; celebrating 🍺 <a href="https://t.co/2oFLdGO3Uh">https://t.co/2oFLdGO3Uh</a><br><br>See you in <a href="https://twitter.com/hashtag/BioC2022?src=hash&amp;ref_src=twsrc%5Etfw">#BioC2022</a>🤞🏽 <a href="https://twitter.com/Bioconductor?ref_src=twsrc%5Etfw"><span class="citation">@Bioconductor</span></a> <a href="https://twitter.com/hashtag/rstats?src=hash&amp;ref_src=twsrc%5Etfw">#rstats</a> <a href="https://twitter.com/CDSBMexico?ref_src=twsrc%5Etfw"><span class="citation">@CDSBMexico</span></a> <a href="https://t.co/0SGHDfiRCs">https://t.co/0SGHDfiRCs</a> <a href="https://t.co/UmM9nMP2W2">pic.twitter.com/UmM9nMP2W2</a>
</p>
— 🇲🇽 Leonardo Collado-Torres (<span class="citation">@lcolladotor</span>) <a href="https://twitter.com/lcolladotor/status/1423805153724059649?ref_src=twsrc%5Etfw">August 7, 2021</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
<p><code>biocthis</code> is one of the reasons for my 2021 Bioconductor community award :-)</p>
<p>Do you want to play an active role? Join the <a href="https://community-bioc.slack.com/archives/C04NPB74Y78"><code>cloud-working-group</code></a> Slack channel.</p>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="research-talks.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="scrna-seq-data-analysis-overview.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
