<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 DGE model building with variancePartition | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="7 DGE model building with variancePartition | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="7 DGE model building with variancePartition | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 DGE model building with variancePartition | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="interpreting-model-coefficients-with-exploremodelmatrix.html"/>
<link rel="next" href="differential-gene-expression-exercise.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.2</b> Exercises</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.3</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
<li class="chapter" data-level="2.1.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment"><i class="fa fa-check"></i><b>2.1.3</b> SpatialExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
<li class="chapter" data-level="2.4.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#explore-the-data"><i class="fa fa-check"></i><b>2.4.2</b> Explore the Data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features"><i class="fa fa-check"></i><b>2.5</b> Introduction to Advanced iSEE Features</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression"><i class="fa fa-check"></i><b>2.5.1</b> Selecting Cells Based on a Single Gene Expression</a></li>
<li class="chapter" data-level="2.5.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression"><i class="fa fa-check"></i><b>2.5.2</b> Using a Single Plot for Two Gene Co-Expression</a></li>
<li class="chapter" data-level="2.5.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes"><i class="fa fa-check"></i><b>2.5.3</b> Selecting Cells Based on the Co-Expression of Two or more Genes</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.6</b> References</a></li>
<li class="chapter" data-level="2.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#community"><i class="fa fa-check"></i><b>2.7</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-1"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-2"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>9</b> Research talks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>9.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="9.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>9.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="9.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>9.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>10</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>10.1</b> Related past materials</a></li>
<li class="chapter" data-level="10.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>10.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="10.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>10.3</b> Live demo</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#example-function"><i class="fa fa-check"></i><b>10.3.1</b> Example function</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-3"><i class="fa fa-check"></i><b>10.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>11</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="11.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing"><i class="fa fa-check"></i><b>11.1</b> Single cell RNA sequencing</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#pre-processing-of-scrna-seq-data-before-r"><i class="fa fa-check"></i><b>11.1.1</b> Pre-processing of scRNA-seq Data (Before R)</a></li>
<li class="chapter" data-level="11.1.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#different-technologies"><i class="fa fa-check"></i><b>11.1.2</b> Different Technologies</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basic-workflow"><i class="fa fa-check"></i><b>11.2</b> Basic Workflow</a></li>
<li class="chapter" data-level="11.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>11.3</b> The SingleCellExperiment class</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#data-loading"><i class="fa fa-check"></i><b>11.3.1</b> Data Loading</a></li>
<li class="chapter" data-level="11.3.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basics-of-your-sce"><i class="fa fa-check"></i><b>11.3.2</b> Basics of your SCE</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>11.4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#common-choices-of-qc-metrics"><i class="fa fa-check"></i><b>11.4.1</b> Common choices of QC metrics</a></li>
<li class="chapter" data-level="11.4.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#identifying-low-quality-cells"><i class="fa fa-check"></i><b>11.4.2</b> Identifying low-quality cells</a></li>
<li class="chapter" data-level="11.4.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#checking-diagnostic-plots"><i class="fa fa-check"></i><b>11.4.3</b> Checking diagnostic plots</a></li>
<li class="chapter" data-level="11.4.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#removing-low-quality-cells"><i class="fa fa-check"></i><b>11.4.4</b> Removing low-quality cells</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>11.5</b> Normalization</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#library-size-normalization"><i class="fa fa-check"></i><b>11.5.1</b> Library size normalization</a></li>
<li class="chapter" data-level="11.5.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-deconvolution"><i class="fa fa-check"></i><b>11.5.2</b> Normalization by deconvolution</a></li>
<li class="chapter" data-level="11.5.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-spike-ins"><i class="fa fa-check"></i><b>11.5.3</b> Normalization by spike-ins</a></li>
<li class="chapter" data-level="11.5.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#scaling-and-log-transforming"><i class="fa fa-check"></i><b>11.5.4</b> Scaling and log-transforming</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>11.6</b> Feature selection</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-per-gene-variation"><i class="fa fa-check"></i><b>11.6.1</b> Quantifying per-gene variation</a></li>
<li class="chapter" data-level="11.6.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-spike-ins"><i class="fa fa-check"></i><b>11.6.2</b> Quantifying technical noise (spike-ins)</a></li>
<li class="chapter" data-level="11.6.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-mean-variance-trend"><i class="fa fa-check"></i><b>11.6.3</b> Quantifying technical noise (mean-variance trend)</a></li>
<li class="chapter" data-level="11.6.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#handling-batch-effects"><i class="fa fa-check"></i><b>11.6.4</b> Handling batch effects</a></li>
<li class="chapter" data-level="11.6.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#selecting-highly-variable-genes"><i class="fa fa-check"></i><b>11.6.5</b> Selecting highly variable genes</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>11.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#principal-components-analysis"><i class="fa fa-check"></i><b>11.7.1</b> Principal components analysis</a></li>
<li class="chapter" data-level="11.7.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#choosing-the-number-of-pcs"><i class="fa fa-check"></i><b>11.7.2</b> Choosing the number of PCs</a></li>
<li class="chapter" data-level="11.7.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#visualizing-the-pcs"><i class="fa fa-check"></i><b>11.7.3</b> Visualizing the PCs</a></li>
<li class="chapter" data-level="11.7.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#non-linear-methods-for-visualization"><i class="fa fa-check"></i><b>11.7.4</b> Non-linear methods for visualization</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>11.8</b> Clustering</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#graph-based-clustering"><i class="fa fa-check"></i><b>11.8.1</b> Graph-based clustering</a></li>
<li class="chapter" data-level="11.8.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#adjusting-the-parameters"><i class="fa fa-check"></i><b>11.8.2</b> Adjusting the parameters</a></li>
<li class="chapter" data-level="11.8.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#hierarchical-clustering"><i class="fa fa-check"></i><b>11.8.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="11.8.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#subclustering"><i class="fa fa-check"></i><b>11.8.4</b> Subclustering</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>11.9</b> Marker gene detection</a>
<ul>
<li class="chapter" data-level="11.9.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#scoring-markers-by-pairwise-comparisons"><i class="fa fa-check"></i><b>11.9.1</b> Scoring markers by pairwise comparisons</a></li>
<li class="chapter" data-level="11.9.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#effect-sizes-for-pairwise-comparisons"><i class="fa fa-check"></i><b>11.9.2</b> Effect sizes for pairwise comparisons</a></li>
<li class="chapter" data-level="11.9.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#summaries-of-pairwise-effects"><i class="fa fa-check"></i><b>11.9.3</b> Summaries of pairwise effects</a></li>
<li class="chapter" data-level="11.9.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#choose-top-markers"><i class="fa fa-check"></i><b>11.9.4</b> Choose top markers</a></li>
<li class="chapter" data-level="11.9.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#handling-blocking-factors"><i class="fa fa-check"></i><b>11.9.5</b> Handling blocking factors</a></li>
<li class="chapter" data-level="11.9.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#deconvobuddies"><i class="fa fa-check"></i><b>11.9.6</b> Deconvobuddies</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>11.10</b> Cell type annotation</a>
<ul>
<li class="chapter" data-level="11.10.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#assigning-cell-labels-from-reference-data"><i class="fa fa-check"></i><b>11.10.1</b> Assigning cell labels from reference data</a></li>
<li class="chapter" data-level="11.10.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#getting-ready-again"><i class="fa fa-check"></i><b>11.10.2</b> Getting ready again</a></li>
<li class="chapter" data-level="11.10.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#using-existing-references"><i class="fa fa-check"></i><b>11.10.3</b> Using existing references</a></li>
<li class="chapter" data-level="11.10.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#assigning-cell-labels-from-gene-sets"><i class="fa fa-check"></i><b>11.10.4</b> Assigning cell labels from gene sets</a></li>
</ul></li>
<li class="chapter" data-level="11.11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#references-4"><i class="fa fa-check"></i><b>11.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>12</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>12.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>12.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>12.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>12.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>13</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="13.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>13.1</b> Spatial registration</a></li>
<li class="chapter" data-level="13.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>13.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>14</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="14.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>14.1</b> here</a></li>
<li class="chapter" data-level="14.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>14.2</b> Usethis</a></li>
<li class="chapter" data-level="14.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>14.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>14.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="14.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>14.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="14.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>14.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="14.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>14.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>14.4</b> R websites</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>14.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="14.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>14.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="14.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>14.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="14.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>14.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>14.5</b> postcards</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>14.5.1</b> Installation</a></li>
<li class="chapter" data-level="14.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>14.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>14.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>14.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="14.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>14.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="14.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>14.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="14.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>14.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="14.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>14.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-5"><i class="fa fa-check"></i><b>14.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dge-model-building-with-variancepartition" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> DGE model building with <em>variancePartition</em><a href="dge-model-building-with-variancepartition.html#dge-model-building-with-variancepartition" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Daianna González-Padilla</p>
<p>After having processed RNA-seq data and assessed the quality and the variability of the samples the next step for DGE is to explore the variance in the expression of the genes themselves according to sample groups, or in other words, to quantify the contribution of the multiple sample variables in the gene expression variation.</p>
<p>To determine which variables are the major drivers of expression variability, and importantly to define if the technical variability of RNA-seq data is low enough to study the condition of interest, we can implement an analysis of variance partition. <a href="https://www.bioconductor.org/packages/variancePartition"><em>variancePartition</em></a> is a package that decomposes for each gene the expression variation into fractions of variance explained (FVE) by the sample variables in the experimental design of high-throughput genomics studies [1].</p>
<p>In order to exemplify how to implement this analysis and the type of conclusions that can be drawn from it, we’ll use bulk RNA-seq data from the <a href="https://bioconductor.org/packages/smokingMouse"><em>smokingMouse</em></a> package.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="dge-model-building-with-variancepartition.html#cb100-1" tabindex="-1"></a><span class="do">## Load the container package for this type of data</span></span>
<span id="cb100-2"><a href="dge-model-building-with-variancepartition.html#cb100-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</span>
<span id="cb100-3"><a href="dge-model-building-with-variancepartition.html#cb100-3" tabindex="-1"></a></span>
<span id="cb100-4"><a href="dge-model-building-with-variancepartition.html#cb100-4" tabindex="-1"></a><span class="do">## Connect to ExperimentHub</span></span>
<span id="cb100-5"><a href="dge-model-building-with-variancepartition.html#cb100-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ExperimentHub&quot;</span>)</span>
<span id="cb100-6"><a href="dge-model-building-with-variancepartition.html#cb100-6" tabindex="-1"></a>eh <span class="ot">&lt;-</span> ExperimentHub<span class="sc">::</span><span class="fu">ExperimentHub</span>()</span>
<span id="cb100-7"><a href="dge-model-building-with-variancepartition.html#cb100-7" tabindex="-1"></a></span>
<span id="cb100-8"><a href="dge-model-building-with-variancepartition.html#cb100-8" tabindex="-1"></a><span class="do">## Load the datasets of the package</span></span>
<span id="cb100-9"><a href="dge-model-building-with-variancepartition.html#cb100-9" tabindex="-1"></a>myfiles <span class="ot">&lt;-</span> <span class="fu">query</span>(eh, <span class="st">&quot;smokingMouse&quot;</span>)</span>
<span id="cb100-10"><a href="dge-model-building-with-variancepartition.html#cb100-10" tabindex="-1"></a></span>
<span id="cb100-11"><a href="dge-model-building-with-variancepartition.html#cb100-11" tabindex="-1"></a><span class="do">## Download the mouse gene data</span></span>
<span id="cb100-12"><a href="dge-model-building-with-variancepartition.html#cb100-12" tabindex="-1"></a>rse_gene <span class="ot">&lt;-</span> myfiles[[<span class="st">&quot;EH8313&quot;</span>]]</span>
<span id="cb100-13"><a href="dge-model-building-with-variancepartition.html#cb100-13" tabindex="-1"></a></span>
<span id="cb100-14"><a href="dge-model-building-with-variancepartition.html#cb100-14" tabindex="-1"></a><span class="do">## Keep samples from nicotine experiment and pups only</span></span>
<span id="cb100-15"><a href="dge-model-building-with-variancepartition.html#cb100-15" tabindex="-1"></a>rse_gene_nic <span class="ot">&lt;-</span> rse_gene[</span>
<span id="cb100-16"><a href="dge-model-building-with-variancepartition.html#cb100-16" tabindex="-1"></a>    ,</span>
<span id="cb100-17"><a href="dge-model-building-with-variancepartition.html#cb100-17" tabindex="-1"></a>    <span class="fu">which</span>(rse_gene<span class="sc">$</span>Expt <span class="sc">==</span> <span class="st">&quot;Nicotine&quot;</span> <span class="sc">&amp;</span> rse_gene<span class="sc">$</span>Age <span class="sc">==</span> <span class="st">&quot;Pup&quot;</span>)</span>
<span id="cb100-18"><a href="dge-model-building-with-variancepartition.html#cb100-18" tabindex="-1"></a>]</span>
<span id="cb100-19"><a href="dge-model-building-with-variancepartition.html#cb100-19" tabindex="-1"></a></span>
<span id="cb100-20"><a href="dge-model-building-with-variancepartition.html#cb100-20" tabindex="-1"></a><span class="do">## Use expressed genes only (i.e. that passed the filtering step)</span></span>
<span id="cb100-21"><a href="dge-model-building-with-variancepartition.html#cb100-21" tabindex="-1"></a>rse_gene_filt <span class="ot">&lt;-</span> rse_gene_nic[</span>
<span id="cb100-22"><a href="dge-model-building-with-variancepartition.html#cb100-22" tabindex="-1"></a>    <span class="fu">rowData</span>(rse_gene_nic)<span class="sc">$</span>retained_after_feature_filtering,</span>
<span id="cb100-23"><a href="dge-model-building-with-variancepartition.html#cb100-23" tabindex="-1"></a>]</span>
<span id="cb100-24"><a href="dge-model-building-with-variancepartition.html#cb100-24" tabindex="-1"></a></span>
<span id="cb100-25"><a href="dge-model-building-with-variancepartition.html#cb100-25" tabindex="-1"></a><span class="do">## Keep samples that passed QC and manual sample filtering steps (all passed)</span></span>
<span id="cb100-26"><a href="dge-model-building-with-variancepartition.html#cb100-26" tabindex="-1"></a>rse_gene_filt <span class="ot">&lt;-</span> rse_gene_filt[</span>
<span id="cb100-27"><a href="dge-model-building-with-variancepartition.html#cb100-27" tabindex="-1"></a>    ,</span>
<span id="cb100-28"><a href="dge-model-building-with-variancepartition.html#cb100-28" tabindex="-1"></a>    rse_gene_filt<span class="sc">$</span>retained_after_QC_sample_filtering <span class="sc">&amp;</span></span>
<span id="cb100-29"><a href="dge-model-building-with-variancepartition.html#cb100-29" tabindex="-1"></a>        rse_gene_filt<span class="sc">$</span>retained_after_manual_sample_filtering</span>
<span id="cb100-30"><a href="dge-model-building-with-variancepartition.html#cb100-30" tabindex="-1"></a>]</span></code></pre></div>
<div id="canonical-correlation-analysis" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Canonical Correlation Analysis<a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Prior to the variance partition analysis, evaluating the correlation between sample variables is crucial because highly correlated variables can produce unstable estimates of the variance fractions and impede the identification of the variables that really contribute to the expression variation. There are at least two problems with correlated variables:</p>
<ul>
<li>If two variables are highly correlated we could incorrectly determine that one of them contributes to gene expression changes when it was actually not explanatory but just correlated with a real contributory variable.</li>
<li>The part of variance explained by a biologically relevant variable can be reduced by the apparent contributions of correlated variables, if for example, they contain very similar information (i.e. are redundant variables).</li>
</ul>
<p>Additionally, the analysis is better performed with simpler models, specially when we have a limited number of samples in the study.</p>
<p>Hence, to remove such variables we must first identify them. We will perform a Canonical Correlation Analysis (CCA) with <code>canCorPairs()</code> that assesses the degree to which the variables co-vary and contain the same information. With CCA, linear combinations that maximize the correlation between variable sets are estimated. CCA is just like a normal correlation analysis between 2 vectors but it can accommodate matrices as well (variable sets). Note that CCA returns correlations values between 0 and 1 [2].</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="dge-model-building-with-variancepartition.html#cb101-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;variancePartition&quot;</span>)</span>
<span id="cb101-2"><a href="dge-model-building-with-variancepartition.html#cb101-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb101-3"><a href="dge-model-building-with-variancepartition.html#cb101-3" tabindex="-1"></a></span>
<span id="cb101-4"><a href="dge-model-building-with-variancepartition.html#cb101-4" tabindex="-1"></a><span class="do">## Plot heatmap of correlations</span></span>
<span id="cb101-5"><a href="dge-model-building-with-variancepartition.html#cb101-5" tabindex="-1"></a><span class="do">## Define all variables to examine; remove those with single values</span></span>
<span id="cb101-6"><a href="dge-model-building-with-variancepartition.html#cb101-6" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="er">~</span> Group <span class="sc">+</span> Sex <span class="sc">+</span> plate <span class="sc">+</span> flowcell <span class="sc">+</span> mitoRate <span class="sc">+</span> overallMapRate <span class="sc">+</span> totalAssignedGene <span class="sc">+</span> rRNA_rate <span class="sc">+</span> sum <span class="sc">+</span> detected <span class="sc">+</span> ERCCsumLogErr</span>
<span id="cb101-7"><a href="dge-model-building-with-variancepartition.html#cb101-7" tabindex="-1"></a></span>
<span id="cb101-8"><a href="dge-model-building-with-variancepartition.html#cb101-8" tabindex="-1"></a><span class="do">## Measure correlations</span></span>
<span id="cb101-9"><a href="dge-model-building-with-variancepartition.html#cb101-9" tabindex="-1"></a>CCA <span class="ot">&lt;-</span> <span class="fu">canCorPairs</span>(formula, <span class="fu">colData</span>(rse_gene_filt))</span>
<span id="cb101-10"><a href="dge-model-building-with-variancepartition.html#cb101-10" tabindex="-1"></a><span class="do">## Heatmap</span></span>
<span id="cb101-11"><a href="dge-model-building-with-variancepartition.html#cb101-11" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb101-12"><a href="dge-model-building-with-variancepartition.html#cb101-12" tabindex="-1"></a>    CCA, <span class="do">## data</span></span>
<span id="cb101-13"><a href="dge-model-building-with-variancepartition.html#cb101-13" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">hcl.colors</span>(<span class="dv">50</span>, <span class="st">&quot;YlOrRd&quot;</span>, <span class="at">rev =</span> <span class="cn">TRUE</span>), <span class="do">## color scale</span></span>
<span id="cb101-14"><a href="dge-model-building-with-variancepartition.html#cb101-14" tabindex="-1"></a>    <span class="at">fontsize =</span> <span class="dv">8</span>, <span class="do">## text size</span></span>
<span id="cb101-15"><a href="dge-model-building-with-variancepartition.html#cb101-15" tabindex="-1"></a>    <span class="at">border_color =</span> <span class="st">&quot;black&quot;</span>, <span class="do">## border color for heatmap cells</span></span>
<span id="cb101-16"><a href="dge-model-building-with-variancepartition.html#cb101-16" tabindex="-1"></a>    <span class="at">cellwidth =</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">&quot;cm&quot;</span>), <span class="do">## height of cells</span></span>
<span id="cb101-17"><a href="dge-model-building-with-variancepartition.html#cb101-17" tabindex="-1"></a>    <span class="at">cellheight =</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">&quot;cm&quot;</span>) <span class="do">## width of cells</span></span>
<span id="cb101-18"><a href="dge-model-building-with-variancepartition.html#cb101-18" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/CCA-1.png" width="672" /></p>
<style>
p.alert {
background-color: #FFE4E1;
padding: 14px;
border: 0px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<p class="alert">
⚠️ <strong>Very important</strong>: always inspect if there are any correlated variables with the one of interest in your study! This is extremely important as correlated variables could represent confounding factors and/or hinder the detection of significant DE events, thus yielding to misleading results.
</p>
<p>Importantly, <code>Group</code> is not highly correlated with any other variable in this study, but <code>overallMapRate</code> is correlated with <code>rRNA_rate</code>, library preparation <code>plate</code>, and the sequencing <code>flowcell</code>; <code>sum</code> (library size) and <code>detected</code> (number of expressed genes) are also correlated. For a detailed definition of these variables check <a href="https://lcolladotor.github.io/cshl_rstats_genome_scale_2023/smokingmouse-rse.html">here</a>.</p>
<p class="exercise">
📝 <strong>Exercise 1</strong>: Run a CCA analysis and determine which pairs of variables in your dataset are correlated. Is there any correlated variable with your variable of interest?
</p>
<p>Depending on your results there’s sometimes convenient to dig a little deeper into the relationship between correlated variables and to analyze these metrics among our control and experimental samples. Let’s work on that!</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="dge-model-building-with-variancepartition.html#cb102-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb102-2"><a href="dge-model-building-with-variancepartition.html#cb102-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;cowplot&quot;</span>)</span>
<span id="cb102-3"><a href="dge-model-building-with-variancepartition.html#cb102-3" tabindex="-1"></a></span>
<span id="cb102-4"><a href="dge-model-building-with-variancepartition.html#cb102-4" tabindex="-1"></a><span class="do">## Boxplots/Scatterplots/Barplots for each pair of correlated variables</span></span>
<span id="cb102-5"><a href="dge-model-building-with-variancepartition.html#cb102-5" tabindex="-1"></a></span>
<span id="cb102-6"><a href="dge-model-building-with-variancepartition.html#cb102-6" tabindex="-1"></a>corr_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(sample_var1, sample_var2, sample_color) {</span>
<span id="cb102-7"><a href="dge-model-building-with-variancepartition.html#cb102-7" tabindex="-1"></a>    <span class="do">## Define sample colors by variable</span></span>
<span id="cb102-8"><a href="dge-model-building-with-variancepartition.html#cb102-8" tabindex="-1"></a>    colors <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb102-9"><a href="dge-model-building-with-variancepartition.html#cb102-9" tabindex="-1"></a>        <span class="st">&quot;Group&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span> <span class="ot">=</span> <span class="st">&quot;brown2&quot;</span>, <span class="st">&quot;Experimental&quot;</span> <span class="ot">=</span> <span class="st">&quot;deepskyblue3&quot;</span>),</span>
<span id="cb102-10"><a href="dge-model-building-with-variancepartition.html#cb102-10" tabindex="-1"></a>        <span class="st">&quot;Sex&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink1&quot;</span>, <span class="st">&quot;M&quot;</span> <span class="ot">=</span> <span class="st">&quot;dodgerblue&quot;</span>),</span>
<span id="cb102-11"><a href="dge-model-building-with-variancepartition.html#cb102-11" tabindex="-1"></a>        <span class="st">&quot;plate&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Plate1&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkorange&quot;</span>, <span class="st">&quot;Plate2&quot;</span> <span class="ot">=</span> <span class="st">&quot;lightskyblue&quot;</span>, <span class="st">&quot;Plate3&quot;</span> <span class="ot">=</span> <span class="st">&quot;deeppink1&quot;</span>),</span>
<span id="cb102-12"><a href="dge-model-building-with-variancepartition.html#cb102-12" tabindex="-1"></a>        <span class="st">&quot;flowcell&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb102-13"><a href="dge-model-building-with-variancepartition.html#cb102-13" tabindex="-1"></a>            <span class="st">&quot;HKCG7DSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;chartreuse2&quot;</span>, <span class="st">&quot;HKCMHDSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;magenta&quot;</span>,</span>
<span id="cb102-14"><a href="dge-model-building-with-variancepartition.html#cb102-14" tabindex="-1"></a>            <span class="st">&quot;HKCNKDSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;turquoise3&quot;</span>, <span class="st">&quot;HKCTMDSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;tomato&quot;</span></span>
<span id="cb102-15"><a href="dge-model-building-with-variancepartition.html#cb102-15" tabindex="-1"></a>        )</span>
<span id="cb102-16"><a href="dge-model-building-with-variancepartition.html#cb102-16" tabindex="-1"></a>    )</span>
<span id="cb102-17"><a href="dge-model-building-with-variancepartition.html#cb102-17" tabindex="-1"></a></span>
<span id="cb102-18"><a href="dge-model-building-with-variancepartition.html#cb102-18" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">colData</span>(rse_gene_filt)</span>
<span id="cb102-19"><a href="dge-model-building-with-variancepartition.html#cb102-19" tabindex="-1"></a></span>
<span id="cb102-20"><a href="dge-model-building-with-variancepartition.html#cb102-20" tabindex="-1"></a>    <span class="do">## a) Barplots for categorical variable vs categorical variable</span></span>
<span id="cb102-21"><a href="dge-model-building-with-variancepartition.html#cb102-21" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(data[, sample_var1]) <span class="sc">==</span> <span class="st">&quot;character&quot;</span> <span class="sc">&amp;</span> <span class="fu">class</span>(data[, sample_var2]) <span class="sc">==</span> <span class="st">&quot;character&quot;</span>) {</span>
<span id="cb102-22"><a href="dge-model-building-with-variancepartition.html#cb102-22" tabindex="-1"></a>        <span class="do">## y-axis label</span></span>
<span id="cb102-23"><a href="dge-model-building-with-variancepartition.html#cb102-23" tabindex="-1"></a>        y_label <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;Number of samples from each &quot;</span>, sample_var2, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb102-24"><a href="dge-model-building-with-variancepartition.html#cb102-24" tabindex="-1"></a></span>
<span id="cb102-25"><a href="dge-model-building-with-variancepartition.html#cb102-25" tabindex="-1"></a>        <span class="do">## Stacked barplot with counts for 2nd variable</span></span>
<span id="cb102-26"><a href="dge-model-building-with-variancepartition.html#cb102-26" tabindex="-1"></a>        plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">as.data.frame</span>(data), <span class="fu">aes</span>(</span>
<span id="cb102-27"><a href="dge-model-building-with-variancepartition.html#cb102-27" tabindex="-1"></a>            <span class="at">x =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var1),</span>
<span id="cb102-28"><a href="dge-model-building-with-variancepartition.html#cb102-28" tabindex="-1"></a>            <span class="at">fill =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var2)</span>
<span id="cb102-29"><a href="dge-model-building-with-variancepartition.html#cb102-29" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb102-30"><a href="dge-model-building-with-variancepartition.html#cb102-30" tabindex="-1"></a>            <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">&quot;stack&quot;</span>) <span class="sc">+</span></span>
<span id="cb102-31"><a href="dge-model-building-with-variancepartition.html#cb102-31" tabindex="-1"></a>            <span class="do">## Colors by 2nd variable</span></span>
<span id="cb102-32"><a href="dge-model-building-with-variancepartition.html#cb102-32" tabindex="-1"></a>            <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> colors[[sample_var2]]) <span class="sc">+</span></span>
<span id="cb102-33"><a href="dge-model-building-with-variancepartition.html#cb102-33" tabindex="-1"></a>            <span class="do">## Show sample counts on stacked bars</span></span>
<span id="cb102-34"><a href="dge-model-building-with-variancepartition.html#cb102-34" tabindex="-1"></a>            <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)),</span>
<span id="cb102-35"><a href="dge-model-building-with-variancepartition.html#cb102-35" tabindex="-1"></a>                <span class="at">stat =</span> <span class="st">&quot;count&quot;</span>,</span>
<span id="cb102-36"><a href="dge-model-building-with-variancepartition.html#cb102-36" tabindex="-1"></a>                <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>), <span class="at">colour =</span> <span class="st">&quot;gray20&quot;</span>, <span class="at">size =</span> <span class="dv">3</span></span>
<span id="cb102-37"><a href="dge-model-building-with-variancepartition.html#cb102-37" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb102-38"><a href="dge-model-building-with-variancepartition.html#cb102-38" tabindex="-1"></a>            <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb102-39"><a href="dge-model-building-with-variancepartition.html#cb102-39" tabindex="-1"></a>            <span class="fu">labs</span>(</span>
<span id="cb102-40"><a href="dge-model-building-with-variancepartition.html#cb102-40" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;Corr: &quot;</span>, <span class="fu">signif</span>(CCA[sample_var1, sample_var2], <span class="at">digits =</span> <span class="dv">3</span>)),</span>
<span id="cb102-41"><a href="dge-model-building-with-variancepartition.html#cb102-41" tabindex="-1"></a>                <span class="at">y =</span> y_label</span>
<span id="cb102-42"><a href="dge-model-building-with-variancepartition.html#cb102-42" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb102-43"><a href="dge-model-building-with-variancepartition.html#cb102-43" tabindex="-1"></a>            <span class="fu">theme</span>(</span>
<span id="cb102-44"><a href="dge-model-building-with-variancepartition.html#cb102-44" tabindex="-1"></a>                <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb102-45"><a href="dge-model-building-with-variancepartition.html#cb102-45" tabindex="-1"></a>                <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">6</span>)),</span>
<span id="cb102-46"><a href="dge-model-building-with-variancepartition.html#cb102-46" tabindex="-1"></a>                <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>),</span>
<span id="cb102-47"><a href="dge-model-building-with-variancepartition.html#cb102-47" tabindex="-1"></a>                <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb102-48"><a href="dge-model-building-with-variancepartition.html#cb102-48" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb102-49"><a href="dge-model-building-with-variancepartition.html#cb102-49" tabindex="-1"></a>            )</span>
<span id="cb102-50"><a href="dge-model-building-with-variancepartition.html#cb102-50" tabindex="-1"></a>    }</span>
<span id="cb102-51"><a href="dge-model-building-with-variancepartition.html#cb102-51" tabindex="-1"></a></span>
<span id="cb102-52"><a href="dge-model-building-with-variancepartition.html#cb102-52" tabindex="-1"></a>    <span class="do">## b) Boxplots for categorical variable vs continuous variable</span></span>
<span id="cb102-53"><a href="dge-model-building-with-variancepartition.html#cb102-53" tabindex="-1"></a>    <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">class</span>(data[, sample_var1]) <span class="sc">==</span> <span class="st">&quot;character&quot;</span> <span class="sc">&amp;</span> <span class="fu">class</span>(data[, sample_var2]) <span class="sc">==</span> <span class="st">&quot;numeric&quot;</span>) {</span>
<span id="cb102-54"><a href="dge-model-building-with-variancepartition.html#cb102-54" tabindex="-1"></a>        plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">as.data.frame</span>(data), <span class="at">mapping =</span> <span class="fu">aes</span>(</span>
<span id="cb102-55"><a href="dge-model-building-with-variancepartition.html#cb102-55" tabindex="-1"></a>            <span class="at">x =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var1),</span>
<span id="cb102-56"><a href="dge-model-building-with-variancepartition.html#cb102-56" tabindex="-1"></a>            <span class="at">y =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var2),</span>
<span id="cb102-57"><a href="dge-model-building-with-variancepartition.html#cb102-57" tabindex="-1"></a>            <span class="at">color =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var1)</span>
<span id="cb102-58"><a href="dge-model-building-with-variancepartition.html#cb102-58" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb102-59"><a href="dge-model-building-with-variancepartition.html#cb102-59" tabindex="-1"></a>            <span class="fu">geom_boxplot</span>(<span class="at">size =</span> <span class="fl">0.25</span>, <span class="at">width =</span> <span class="fl">0.32</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">outlier.color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb102-60"><a href="dge-model-building-with-variancepartition.html#cb102-60" tabindex="-1"></a>            <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.15</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb102-61"><a href="dge-model-building-with-variancepartition.html#cb102-61" tabindex="-1"></a>            <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">geom =</span> <span class="st">&quot;line&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.4</span>, <span class="at">span =</span> <span class="fl">0.3</span>, <span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>), <span class="at">color =</span> <span class="st">&quot;orangered3&quot;</span>) <span class="sc">+</span></span>
<span id="cb102-62"><a href="dge-model-building-with-variancepartition.html#cb102-62" tabindex="-1"></a>            <span class="fu">scale_color_manual</span>(<span class="at">values =</span> colors[[sample_var1]]) <span class="sc">+</span></span>
<span id="cb102-63"><a href="dge-model-building-with-variancepartition.html#cb102-63" tabindex="-1"></a>            <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb102-64"><a href="dge-model-building-with-variancepartition.html#cb102-64" tabindex="-1"></a>            <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb102-65"><a href="dge-model-building-with-variancepartition.html#cb102-65" tabindex="-1"></a>            <span class="fu">labs</span>(</span>
<span id="cb102-66"><a href="dge-model-building-with-variancepartition.html#cb102-66" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;Corr: &quot;</span>, <span class="fu">signif</span>(CCA[sample_var1, sample_var2], <span class="at">digits =</span> <span class="dv">3</span>)), <span class="at">y =</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, sample_var2),</span>
<span id="cb102-67"><a href="dge-model-building-with-variancepartition.html#cb102-67" tabindex="-1"></a>                <span class="at">x =</span> sample_var1</span>
<span id="cb102-68"><a href="dge-model-building-with-variancepartition.html#cb102-68" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb102-69"><a href="dge-model-building-with-variancepartition.html#cb102-69" tabindex="-1"></a>            <span class="fu">theme</span>(</span>
<span id="cb102-70"><a href="dge-model-building-with-variancepartition.html#cb102-70" tabindex="-1"></a>                <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb102-71"><a href="dge-model-building-with-variancepartition.html#cb102-71" tabindex="-1"></a>                <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">6</span>)),</span>
<span id="cb102-72"><a href="dge-model-building-with-variancepartition.html#cb102-72" tabindex="-1"></a>                <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>),</span>
<span id="cb102-73"><a href="dge-model-building-with-variancepartition.html#cb102-73" tabindex="-1"></a>                <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb102-74"><a href="dge-model-building-with-variancepartition.html#cb102-74" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb102-75"><a href="dge-model-building-with-variancepartition.html#cb102-75" tabindex="-1"></a>            )</span>
<span id="cb102-76"><a href="dge-model-building-with-variancepartition.html#cb102-76" tabindex="-1"></a>    }</span>
<span id="cb102-77"><a href="dge-model-building-with-variancepartition.html#cb102-77" tabindex="-1"></a></span>
<span id="cb102-78"><a href="dge-model-building-with-variancepartition.html#cb102-78" tabindex="-1"></a>    <span class="do">## c) Scatterplots for continuous variable vs continuous variable</span></span>
<span id="cb102-79"><a href="dge-model-building-with-variancepartition.html#cb102-79" tabindex="-1"></a>    <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">class</span>(data[, sample_var1]) <span class="sc">==</span> <span class="st">&quot;numeric&quot;</span> <span class="sc">&amp;</span> <span class="fu">class</span>(data[, sample_var2]) <span class="sc">==</span> <span class="st">&quot;numeric&quot;</span>) {</span>
<span id="cb102-80"><a href="dge-model-building-with-variancepartition.html#cb102-80" tabindex="-1"></a>        plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(data), <span class="fu">aes</span>(</span>
<span id="cb102-81"><a href="dge-model-building-with-variancepartition.html#cb102-81" tabindex="-1"></a>            <span class="at">x =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var1),</span>
<span id="cb102-82"><a href="dge-model-building-with-variancepartition.html#cb102-82" tabindex="-1"></a>            <span class="at">y =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var2),</span>
<span id="cb102-83"><a href="dge-model-building-with-variancepartition.html#cb102-83" tabindex="-1"></a>            <span class="at">color =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_color)</span>
<span id="cb102-84"><a href="dge-model-building-with-variancepartition.html#cb102-84" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb102-85"><a href="dge-model-building-with-variancepartition.html#cb102-85" tabindex="-1"></a>            <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb102-86"><a href="dge-model-building-with-variancepartition.html#cb102-86" tabindex="-1"></a>            <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">geom =</span> <span class="st">&quot;line&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.6</span>, <span class="at">span =</span> <span class="fl">0.25</span>, <span class="at">color =</span> <span class="st">&quot;orangered3&quot;</span>) <span class="sc">+</span></span>
<span id="cb102-87"><a href="dge-model-building-with-variancepartition.html#cb102-87" tabindex="-1"></a>            <span class="do">## Color by sample_color variable</span></span>
<span id="cb102-88"><a href="dge-model-building-with-variancepartition.html#cb102-88" tabindex="-1"></a>            <span class="fu">scale_color_manual</span>(<span class="at">name =</span> sample_color, <span class="at">values =</span> colors[[sample_color]]) <span class="sc">+</span></span>
<span id="cb102-89"><a href="dge-model-building-with-variancepartition.html#cb102-89" tabindex="-1"></a>            <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb102-90"><a href="dge-model-building-with-variancepartition.html#cb102-90" tabindex="-1"></a>            <span class="fu">labs</span>(</span>
<span id="cb102-91"><a href="dge-model-building-with-variancepartition.html#cb102-91" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;Corr: &quot;</span>, <span class="fu">signif</span>(CCA[sample_var1, sample_var2], <span class="at">digits =</span> <span class="dv">3</span>)),</span>
<span id="cb102-92"><a href="dge-model-building-with-variancepartition.html#cb102-92" tabindex="-1"></a>                <span class="at">y =</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, sample_var2),</span>
<span id="cb102-93"><a href="dge-model-building-with-variancepartition.html#cb102-93" tabindex="-1"></a>                <span class="at">x =</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, sample_var1)</span>
<span id="cb102-94"><a href="dge-model-building-with-variancepartition.html#cb102-94" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb102-95"><a href="dge-model-building-with-variancepartition.html#cb102-95" tabindex="-1"></a>            <span class="fu">theme</span>(</span>
<span id="cb102-96"><a href="dge-model-building-with-variancepartition.html#cb102-96" tabindex="-1"></a>                <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb102-97"><a href="dge-model-building-with-variancepartition.html#cb102-97" tabindex="-1"></a>                <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">6</span>)),</span>
<span id="cb102-98"><a href="dge-model-building-with-variancepartition.html#cb102-98" tabindex="-1"></a>                <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>),</span>
<span id="cb102-99"><a href="dge-model-building-with-variancepartition.html#cb102-99" tabindex="-1"></a>                <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb102-100"><a href="dge-model-building-with-variancepartition.html#cb102-100" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb102-101"><a href="dge-model-building-with-variancepartition.html#cb102-101" tabindex="-1"></a>            )</span>
<span id="cb102-102"><a href="dge-model-building-with-variancepartition.html#cb102-102" tabindex="-1"></a>    }</span>
<span id="cb102-103"><a href="dge-model-building-with-variancepartition.html#cb102-103" tabindex="-1"></a></span>
<span id="cb102-104"><a href="dge-model-building-with-variancepartition.html#cb102-104" tabindex="-1"></a>    <span class="fu">return</span>(plot)</span>
<span id="cb102-105"><a href="dge-model-building-with-variancepartition.html#cb102-105" tabindex="-1"></a>}</span></code></pre></div>
<p>As shown below, <code>Group</code> and <code>plate</code> are moderately correlated given that 14 of the 23 (60.8%) control samples and 11 of the 19 (57.9%) exposed samples were in the first and second plate for library preparation, respectively.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="dge-model-building-with-variancepartition.html#cb103-1" tabindex="-1"></a><span class="do">## Correlation plot for Group and plate</span></span>
<span id="cb103-2"><a href="dge-model-building-with-variancepartition.html#cb103-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">corr_plots</span>(<span class="st">&quot;Group&quot;</span>, <span class="st">&quot;plate&quot;</span>, <span class="cn">NULL</span>)</span>
<span id="cb103-3"><a href="dge-model-building-with-variancepartition.html#cb103-3" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">5.5</span>, <span class="dv">1</span>, <span class="fl">5.5</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>We can also observe that even though QC metrics such as <code>overallMapRate</code> and <code>rRNA_rate</code> are correlated, there’s no distinction between control and exposed samples for these variables.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="dge-model-building-with-variancepartition.html#cb104-1" tabindex="-1"></a><span class="do">## Correlation plot for overallMapRate and rRNA_rate</span></span>
<span id="cb104-2"><a href="dge-model-building-with-variancepartition.html#cb104-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">corr_plots</span>(<span class="st">&quot;overallMapRate&quot;</span>, <span class="st">&quot;rRNA_rate&quot;</span>, <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb104-3"><a href="dge-model-building-with-variancepartition.html#cb104-3" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="fl">3.5</span>, <span class="dv">2</span>, <span class="fl">3.5</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Moreover, the correlation between <code>overallMapRate</code> and the library preparation <code>plate</code> is mainly given by the plate 1 samples that have lower rates, similar to what occurs with the samples from the first flowcell.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="dge-model-building-with-variancepartition.html#cb105-1" tabindex="-1"></a><span class="do">## Correlation plot for overallMapRate and plate</span></span>
<span id="cb105-2"><a href="dge-model-building-with-variancepartition.html#cb105-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">corr_plots</span>(<span class="st">&quot;plate&quot;</span>, <span class="st">&quot;overallMapRate&quot;</span>, <span class="cn">NULL</span>)</span>
<span id="cb105-3"><a href="dge-model-building-with-variancepartition.html#cb105-3" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">2</span>, <span class="dv">5</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="dge-model-building-with-variancepartition.html#cb106-1" tabindex="-1"></a><span class="do">## Correlation plot for overallMapRate and flowcell</span></span>
<span id="cb106-2"><a href="dge-model-building-with-variancepartition.html#cb106-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">corr_plots</span>(<span class="st">&quot;flowcell&quot;</span>, <span class="st">&quot;overallMapRate&quot;</span>, <span class="cn">NULL</span>)</span>
<span id="cb106-3"><a href="dge-model-building-with-variancepartition.html#cb106-3" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">2</span>, <span class="dv">5</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>Interestingly, control samples seem to present more expressed genes than exposed samples for a given library size,
however none of these variables is correlated with <code>Group</code>.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="dge-model-building-with-variancepartition.html#cb107-1" tabindex="-1"></a><span class="do">## Correlation plots for sum and detected</span></span>
<span id="cb107-2"><a href="dge-model-building-with-variancepartition.html#cb107-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">corr_plots</span>(<span class="st">&quot;sum&quot;</span>, <span class="st">&quot;detected&quot;</span>, <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb107-3"><a href="dge-model-building-with-variancepartition.html#cb107-3" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="fl">3.5</span>, <span class="dv">2</span>, <span class="fl">3.5</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div style="background-color:#FFFFFF; padding:20px; font-family: sans-serif; border: 2px solid black">
<p>❓ Now look at the following plot. <strong>Why is it important that experimental and control samples are distributed throughout all sequencing flowcells?</strong></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="dge-model-building-with-variancepartition.html#cb108-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">corr_plots</span>(<span class="st">&quot;Group&quot;</span>, <span class="st">&quot;flowcell&quot;</span>, <span class="cn">NULL</span>)</span>
<span id="cb108-2"><a href="dge-model-building-with-variancepartition.html#cb108-2" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(p)</span>
<span id="cb108-3"><a href="dge-model-building-with-variancepartition.html#cb108-3" tabindex="-1"></a>plots <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.5</span>, <span class="dv">5</span>, <span class="fl">0.5</span>, <span class="dv">5</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p><strong><em>Hint</em></strong>: What would happen if all experimental samples were in one flowcell and all controls in another?</p>
</div>
<p>
</p>
<p>After identifying which variables are correlated and exploring the metrics of control and experimental samples the next is to determine which variable from each pair of correlated variables should be discarded and which one included in the models. How do we discern which ones to keep? As recommended in the <em>variancePartition</em> user’s guide [2], initially we can fit a linear model to the expression data of each gene taking all sample variables and then investigate which ones explain higher percentages of variance for many genes. But first let’s review how <em>variancePartition</em> works.</p>
</div>
<div id="fit-model-and-extract-fraction-of-variance-explained" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Fit model and extract fraction of variance explained<a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Briefly, what <em>variancePartition</em> does is to fit a linear model for each gene separately and to compute the fraction of the total data variance explained by each variable of the study design, as well as by the residuals, using the <code>calcVarPart()</code> function. These computed fractions of variation explained (FVE) summarize the contribution of each variable and naturally sum to 1 [1].</p>
<p><em>variancePartition</em> fits two types of models:</p>
<ol style="list-style-type: decimal">
<li><strong>Linear mixed model (LMM)</strong> where all categorical variables are modeled as <strong>random effects</strong> and all continuous variables are <strong>fixed effects</strong>. The function <code>lmer()</code> from <em>lme4</em> is used to fit this model.</li>
</ol>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="dge-model-building-with-variancepartition.html#cb109-1" tabindex="-1"></a><span class="do">## Fit LMM specifying the existence of random effects with &#39;(1| )&#39; </span></span>
<span id="cb109-2"><a href="dge-model-building-with-variancepartition.html#cb109-2" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmer</span>(expr <span class="sc">~</span> a <span class="sc">+</span> b <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>c), <span class="at">data=</span>data)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Fixed effects model, which is basically the <strong>standard linear model (LM)</strong>, where all variables are modeled as fixed effects. The function <code>lm()</code> is used to fit this model.</li>
</ol>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="dge-model-building-with-variancepartition.html#cb110-1" tabindex="-1"></a><span class="do">## Fit LM modeling all variables as fixed effects</span></span>
<span id="cb110-2"><a href="dge-model-building-with-variancepartition.html#cb110-2" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(expr <span class="sc">~</span> a <span class="sc">+</span> b <span class="sc">+</span> c, <span class="at">data=</span>data)</span></code></pre></div>
<p>In our case, the function will be modeled as a mixed model since we have both effects.</p>
<style>
p.question{
background-color: #E3E3E3;
padding: 20px;
border: 1px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<p class="question">
❓ <strong>What are random and fixed effects?</strong>
Categorical variables are usually modeled as <strong>random effects</strong>, i.e., variables such as flowcell, plate, donor, etc. whose levels are “randomly chosen or selected from a larger population”. These levels are not of interest by themselves but the grouping of the samples by them. Random effects correspond to those variables whose effect on the expression of a gene varies according to its sample groups/levels. On the other hand, continuous variables can be modeled as <strong>fixed effects</strong>. These are sample-level variables that preserve their impact on the expression of a gene irrespective of the sample.
</p>
<p class="question">
❓ <strong>Why is this effect distinction important?</strong>
Because when we have clustered data, like gene expression values grouped by sample sex, batch, etc. we are violating the relevant assumption of independence, making an incorrect inference when using a <strong>general linear model (GLM)</strong>. If we have clustered data where the variables’ values have distinct effects on gene expression, we must work with an extension of GLM, i.e. with the <strong>linear mixed model (LMM)</strong> that contains a mix of both fixed and random effects [3].
</p>
<p><strong>Linear mixed model fit</strong></p>
<aside>
<p>1️⃣ After fitting a linear model to the expression data of each gene we obtain the predicted expression of the genes given by the estimated coefficients of the variables:</p>
<p><span class="math inline">\(\hat y =\sum_{1}^j\beta_{j}X_{j} + \sum_{1}^k\alpha_{k}Z_{k}\)</span> ← the expression of a gene across all samples is given by the samples’ values in the <span class="math inline">\(j\)</span> fixed effects and <span class="math inline">\(k\)</span> random effects. Therefore, the gene expression in the sample <span class="math inline">\(i\)</span> is given by <span class="math inline">\(\hat y_i =\sum_{1}^j\beta_{j}X_{ij} + \sum_{1}^k\alpha_{k}Z_{ik}\)</span>.</p>
<p>Then <span class="math inline">\(y=\hat y+\epsilon\)</span>, which means that the true (observed) expression value is given by the predicted value plus an error term (<span class="math inline">\(\epsilon\)</span>), also called noise or residual:</p>
<p><span class="math display">\[y =\sum_{1}^j\beta_{j}X_{j} + \sum_{1}^k\alpha_{k}Z_{k} + \epsilon\]</span></p>
<ul>
<li><span class="math inline">\(X_j\)</span> is the vector of the values of the samples in the <span class="math inline">\(j\)</span>th fixed effect.</li>
<li><span class="math inline">\(\beta_j\)</span> is the predicted coefficient of the fixed effect <span class="math inline">\(j\)</span>.</li>
</ul>
<p><span class="math display">\[
X_j\beta_j=  \ \ _{n \ \ samples}\stackrel{j^{th}\ \ fixed\ \ effect }{\begin{bmatrix} X_{1j} \\ ... \\ X_{(n-1)j} \\ X_{nj}  \end{bmatrix}}\beta_j = {\begin{bmatrix}  X_{1j}\beta_j \\ ... \\ X_{(n-1)j}\beta_j \\ X_{nj} \beta_j \end{bmatrix}}
\]</span></p>
<ul>
<li><span class="math inline">\(Z_k\)</span> is the vector of values the samples have for the <span class="math inline">\(k\)</span>th random effect.</li>
<li><span class="math inline">\(\alpha_k\)</span> is the predicted coefficient of the random effect <span class="math inline">\(k\)</span>. These are drawn from a normal distribution <span class="math inline">\(∼N(0, \sigma_{\alpha_k}^2 )\)</span>.</li>
</ul>
<p><span class="math display">\[
Z_k\alpha_k= \ \ _{n \ \ samples}\stackrel{{k^{th}\ \ random\ \ effect }}{\begin{bmatrix}  Z_{1k} \\ ... \\ Z_{(n-1)k}\\ Z_{nk} \end{bmatrix}}\alpha_k = {\begin{bmatrix} Z_{1k}\alpha_k \\ ... \\ Z_{(n-1)k}\alpha_k\\ Z_{nk}\alpha_k \end{bmatrix}}
\]</span></p>
<p><span class="math inline">\(\epsilon\)</span> is the noise term which is <span class="math inline">\(y-\hat y\)</span>, the difference between the observed and predicted expression and is also drawn from <span class="math inline">\(∼N(0, \sigma_{\epsilon}^2 )\)</span>.</p>
<p>Expanding,</p>
<p><span class="math display">\[
y= {\begin{bmatrix}  X_{11}\beta_1 \\ ... \\ X_{(n-1)1}\beta_1 \\ X_{n1} \beta_1 \end{bmatrix}} + {\begin{bmatrix}  X_{12}\beta_2 \\ ... \\ X_{(n-1)2}\beta_2 \\ X_{n2} \beta_2 \end{bmatrix}} + ... +{\begin{bmatrix}  X_{1j}\beta_j \\ ... \\ X_{(n-1)j}\beta_j \\ X_{nj} \beta_j \end{bmatrix}} + {\begin{bmatrix} Z_{11}\alpha_1 \\ ... \\ Z_{(n-1)1}\alpha_1\\ Z_{n1}\alpha_1 \end{bmatrix}} +{\begin{bmatrix} Z_{12}\alpha_2 \\ ... \\ Z_{(n-1)2}\alpha_2\\ Z_{n2}\alpha_2 \end{bmatrix}}
\]</span>
<span class="math display">\[
+ ... + {\begin{bmatrix} Z_{1k}\alpha_k \\ ... \\ Z_{(n-1)k}\alpha_k\\ Z_{nk}\alpha_k \end{bmatrix}} + {\begin{bmatrix} \epsilon_1 \\ ... \\ \epsilon_{(n-1)} \\ \epsilon_n \end{bmatrix}}
\]</span></p>
All parameters are estimated with maximum likelihood, the default method in the <em>variancePartition</em> software when random effects are specified because it performs best in simulations.
</aside>
<aside>
<p>2️⃣ Then, <code>calcVarPart()</code> computes for each fixed effect</p>
<p><span class="math inline">\(\sum_{i=1}^n(\beta_{j}X_{ij}-\bar{\beta_{j}X_{j}})^2=var(\beta_{j}X_{j})(n-1)\)</span>, the squared sum of the predicted expression values of a gene in the <span class="math inline">\(n\)</span> samples only taking into account the variable <span class="math inline">\(j\)</span> in the regression model: <span class="math inline">\(\hat y = \beta_{j}X_{j}\)</span>. Each of these squared sums is scaled by additional factors but to simplify let’s just explain this analysis in terms of the variance (that is proportional to the squared sum):</p>
<ul>
<li>The variance explained by the <span class="math inline">\(j\)</span>th fixed effect is: <mark style= "background-color: #FEF5E7"><span class="math inline">\(\sigma_{\beta_j}^2=var(X_j{\beta_j})\)</span></mark></li>
</ul>
<p>For random effects the variances are computed by variance component estimates with <code>VarCorr()</code> from <code>nmle</code>:</p>
<ul>
<li>The variance of the <span class="math inline">\(k\)</span>th random effect is <mark style= "background-color: #D6EAF8"><span class="math inline">\(\sigma_{\alpha_k}^2=var(Z_k{\alpha_k})\)</span></mark></li>
</ul>
<p>The total variance of the expression values is calculated by <span class="math inline">\(\sum_{i=1}^n(y_i - \bar { y})^2=var(y)(n-1)\)</span>, where <span class="math inline">\(y_i = \sum_{1}^j\beta_{j}X_{ij} + \sum_{1}^k\alpha_{k}Z_{ik} + \epsilon_i\)</span> considering all variables in the model and the error:</p>
<ul>
<li>The total variance is:</li>
</ul>
<p><span class="math inline">\(var(y)= \sigma_{Total}^2= var(X_{1}\beta_1)+var(X_{2}\beta_2)+...+var(X_{j}\beta_j)+var(Z_{1}\alpha_1)+var(Z_{2}\alpha_2)+...+var(Z_{k}\alpha_k)+var(\epsilon)=\)</span>
<span class="math inline">\(\sum_1^jvar(X_j\beta_j)+\sum_1^kvar(Z_k\alpha_k)+var(\epsilon)=\)</span></p>
<p><mark style= "background-color: #F9EBEA"><span class="math inline">\(\sigma_{Total}^2=\sum_1^j{ \sigma_{\beta_j}^2} + \sum_1^k{ \sigma_{\alpha_k}^2} + \sigma_{\epsilon}^2\)</span></mark></p>
</aside>
<aside>
<p>3️⃣ Finally, it computes:</p>
<ul>
<li>The fraction of the total data variance explained by the <span class="math inline">\(j\)</span>th fixed effect is <mark style= "background-color: #FEF5E7"><span class="math inline">\(\sigma_{\beta_j}^2\)</span></mark> / <mark style= "background-color: #F9EBEA"><span class="math inline">\(\sigma_{Total}^2\)</span></mark></li>
<li>The fraction of the total data variance explained by the <span class="math inline">\(k\)</span>th random effect is <mark style= "background-color: #D6EAF8"><span class="math inline">\(\sigma_{\alpha_k}^2\)</span></mark> / <mark style= "background-color: #F9EBEA"><span class="math inline">\(\sigma_{Total}^2\)</span></mark></li>
</ul>
<p>Note that <span class="math inline">\(y=\hat y+\epsilon\)</span> because the expression can’t be completely described by a straight line, so not all the variation of <span class="math inline">\(y\)</span> can be explained by the variation of the sample variables, instead <span class="math inline">\(var(y)=var(\hat y)+var(\epsilon)=var(\hat y) + \sigma_{\epsilon}^2\)</span>, where <span class="math inline">\(\sigma_{\epsilon}^2=\sum_{i=1}^n(\hat y_i -y_i)^2/n-1\)</span>.</p>
<ul>
<li>The residual variance is <mark style= "background-color: #E8DAEF"><span class="math inline">\(\sigma_{\epsilon}^2\)</span></mark> / <mark style= "background-color: #F9EBEA"><span class="math inline">\(\sigma_{Total}^2\)</span></mark> ; this is the variance that the model (with the included covariates) couldn’t explain.</li>
</ul>
</aside>
<style>
p.link{
background-color: #FFFFFF;
padding: 10px;
border: 0px solid black;
margin-left: 0px;
border-radius: 1px;
font-size: 13px;
font-family: sans-serif;
}
</style>
<p class="link">
👉🏼 Source code of <code>calcVarPart()</code> <a href="https://rdrr.io/github/GabrielHoffman/variancePartition/src/R/calcVarPart.R#sym-getVarianceComponents">here</a>.
</p>
<p>Once we have reviewed what <em>variancePartition</em> computes and how, we can use it to quantify the FVE for each variable.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="dge-model-building-with-variancepartition.html#cb111-1" tabindex="-1"></a><span class="do">## Fit a linear mixed model (LMM) that takes continuous variables as fixed effects and categorical variables as random effects</span></span>
<span id="cb111-2"><a href="dge-model-building-with-variancepartition.html#cb111-2" tabindex="-1"></a></span>
<span id="cb111-3"><a href="dge-model-building-with-variancepartition.html#cb111-3" tabindex="-1"></a>varPartAnalysis <span class="ot">&lt;-</span> <span class="cf">function</span>(formula) {</span>
<span id="cb111-4"><a href="dge-model-building-with-variancepartition.html#cb111-4" tabindex="-1"></a>    <span class="do">## Ignore genes with variance 0</span></span>
<span id="cb111-5"><a href="dge-model-building-with-variancepartition.html#cb111-5" tabindex="-1"></a>    genes_var_zero <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">apply</span>(<span class="fu">assays</span>(rse_gene_filt)<span class="sc">$</span>logcounts, <span class="dv">1</span>, var) <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb111-6"><a href="dge-model-building-with-variancepartition.html#cb111-6" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(genes_var_zero) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb111-7"><a href="dge-model-building-with-variancepartition.html#cb111-7" tabindex="-1"></a>        rse_gene_filt <span class="ot">&lt;-</span> rse_gene_filt[<span class="sc">-</span>genes_var_zero, ]</span>
<span id="cb111-8"><a href="dge-model-building-with-variancepartition.html#cb111-8" tabindex="-1"></a>    }</span>
<span id="cb111-9"><a href="dge-model-building-with-variancepartition.html#cb111-9" tabindex="-1"></a></span>
<span id="cb111-10"><a href="dge-model-building-with-variancepartition.html#cb111-10" tabindex="-1"></a>    <span class="do">## Loop over each gene to fit the model and extract variance explained by each variable</span></span>
<span id="cb111-11"><a href="dge-model-building-with-variancepartition.html#cb111-11" tabindex="-1"></a>    varPart <span class="ot">&lt;-</span> <span class="fu">fitExtractVarPartModel</span>(<span class="fu">assays</span>(rse_gene_filt)<span class="sc">$</span>logcounts, formula, <span class="fu">colData</span>(rse_gene_filt))</span>
<span id="cb111-12"><a href="dge-model-building-with-variancepartition.html#cb111-12" tabindex="-1"></a></span>
<span id="cb111-13"><a href="dge-model-building-with-variancepartition.html#cb111-13" tabindex="-1"></a>    <span class="co"># Sort variables by median fraction of variance explained (FVE)</span></span>
<span id="cb111-14"><a href="dge-model-building-with-variancepartition.html#cb111-14" tabindex="-1"></a>    vp <span class="ot">&lt;-</span> <span class="fu">sortCols</span>(varPart)</span>
<span id="cb111-15"><a href="dge-model-building-with-variancepartition.html#cb111-15" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">plotVarPart</span>(vp)</span>
<span id="cb111-16"><a href="dge-model-building-with-variancepartition.html#cb111-16" tabindex="-1"></a></span>
<span id="cb111-17"><a href="dge-model-building-with-variancepartition.html#cb111-17" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(p, vp))</span>
<span id="cb111-18"><a href="dge-model-building-with-variancepartition.html#cb111-18" tabindex="-1"></a>}</span></code></pre></div>
<p>In the following violin plots, we have the % of variance explained in the expression of each gene by each covariate, based on the model with all variables. Of our pairs of correlated variables, <code>rRNA_rate</code> has the highest median FVE and thus, should be included in the models for DGE, whereas variables correlated with it (<code>overallMapRate</code>) must be removed. Furthermore, library preparation <code>plate</code> must be excluded as it is correlated with <code>Group</code>.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="dge-model-building-with-variancepartition.html#cb112-1" tabindex="-1"></a><span class="do">#####  Fit model with all variables  #####</span></span>
<span id="cb112-2"><a href="dge-model-building-with-variancepartition.html#cb112-2" tabindex="-1"></a></span>
<span id="cb112-3"><a href="dge-model-building-with-variancepartition.html#cb112-3" tabindex="-1"></a><span class="co"># sum, detected, and ERCCsumLogErr are not included as they are in very different scales!</span></span>
<span id="cb112-4"><a href="dge-model-building-with-variancepartition.html#cb112-4" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="er">~</span> (<span class="dv">1</span> <span class="sc">|</span> Group) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Sex) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> plate) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> flowcell) <span class="sc">+</span> mitoRate <span class="sc">+</span> overallMapRate <span class="sc">+</span></span>
<span id="cb112-5"><a href="dge-model-building-with-variancepartition.html#cb112-5" tabindex="-1"></a>    totalAssignedGene <span class="sc">+</span> rRNA_rate</span>
<span id="cb112-6"><a href="dge-model-building-with-variancepartition.html#cb112-6" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">varPartAnalysis</span>(formula)[[<span class="dv">1</span>]]</span>
<span id="cb112-7"><a href="dge-model-building-with-variancepartition.html#cb112-7" tabindex="-1"></a>plot <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb112-8"><a href="dge-model-building-with-variancepartition.html#cb112-8" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb112-9"><a href="dge-model-building-with-variancepartition.html#cb112-9" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb112-10"><a href="dge-model-building-with-variancepartition.html#cb112-10" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="fl">7.5</span>))</span>
<span id="cb112-11"><a href="dge-model-building-with-variancepartition.html#cb112-11" tabindex="-1"></a>)</span></code></pre></div>
<figure>
<img src="Figures/varPart_all.png" width="800px" align=center />
</figure>
<p class="alert">
⚠️ Note that some variables such as the library size and the number of detected genes that are in different orders of magnitude to the rest of the QC metrics and categorical variables are not included in this analysis as they can impact the model predictions and the interpretability of the regression results [4]. These variables can be analyzed only after rescaling.
</p>
<p>After re-running the analysis without the previous correlated variables, now <code>Group</code> contribution increases but so does the residual source, i.e., the % of gene expression variance that the model couldn’t explain increases, although the increase is rather low. This occurs because when we remove independent variables to a regression equation, we can explain less of the variance of the dependent variable [3]. That’s the price to pay when dropping variables, but it is convenient when we don’t have many samples for the model to determine variable unique contributions.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="dge-model-building-with-variancepartition.html#cb113-1" tabindex="-1"></a><span class="do">#####  Fit model without correlated variables  #####</span></span>
<span id="cb113-2"><a href="dge-model-building-with-variancepartition.html#cb113-2" tabindex="-1"></a></span>
<span id="cb113-3"><a href="dge-model-building-with-variancepartition.html#cb113-3" tabindex="-1"></a><span class="do">## Pup plots without overallMapRate and plate</span></span>
<span id="cb113-4"><a href="dge-model-building-with-variancepartition.html#cb113-4" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="er">~</span> (<span class="dv">1</span> <span class="sc">|</span> Group) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Sex) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> flowcell) <span class="sc">+</span> mitoRate <span class="sc">+</span> overallMapRate <span class="sc">+</span> totalAssignedGene</span>
<span id="cb113-5"><a href="dge-model-building-with-variancepartition.html#cb113-5" tabindex="-1"></a>varPart <span class="ot">&lt;-</span> <span class="fu">varPartAnalysis</span>(formula)</span>
<span id="cb113-6"><a href="dge-model-building-with-variancepartition.html#cb113-6" tabindex="-1"></a>varPart_data <span class="ot">&lt;-</span> varPart[[<span class="dv">2</span>]]</span>
<span id="cb113-7"><a href="dge-model-building-with-variancepartition.html#cb113-7" tabindex="-1"></a>plot <span class="ot">&lt;-</span> varPart[[<span class="dv">1</span>]]</span>
<span id="cb113-8"><a href="dge-model-building-with-variancepartition.html#cb113-8" tabindex="-1"></a>plot <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb113-9"><a href="dge-model-building-with-variancepartition.html#cb113-9" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb113-10"><a href="dge-model-building-with-variancepartition.html#cb113-10" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb113-11"><a href="dge-model-building-with-variancepartition.html#cb113-11" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="fl">7.5</span>))</span>
<span id="cb113-12"><a href="dge-model-building-with-variancepartition.html#cb113-12" tabindex="-1"></a>)</span></code></pre></div>
<figure>
<img src="Figures/varPart_subset.png" width="800px" align=center />
</figure>
<p class="exercise">
📝 <strong>Exercise 2</strong>: Perform a variance partition analysis and determine which of your correlated variables have higher contributions in gene expression variance. Based on that, select a set of variables to model gene expression for DGE.
</p>
<p>But what does it mean that a variable explains a high percentage of the expression variation of a gene? In the following section will visualize the existing relationships between the gene expression values in the samples and the sample-level variables.</p>
</div>
<div id="examine-the-expression-of-most-affected-genes-by-each-sample-variable" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Examine the expression of most affected genes by each sample variable<a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the plots presented below we can appreciate the expression levels across samples of the most affected genes by each variable, i.e., the genes for which the variable explains the highest percentages of variance, plotted against the sample values for the same variable. Observe the strong correlations that exist for the sample variables and the gene expression of such affected genes, which ends up causing these variables to explain high percentages of gene expression variation and which obligate us to adjust for them in the models.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="dge-model-building-with-variancepartition.html#cb114-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;rlang&quot;</span>)</span>
<span id="cb114-2"><a href="dge-model-building-with-variancepartition.html#cb114-2" tabindex="-1"></a></span>
<span id="cb114-3"><a href="dge-model-building-with-variancepartition.html#cb114-3" tabindex="-1"></a><span class="do">## Plot of gene expression lognorm counts vs. sample variable</span></span>
<span id="cb114-4"><a href="dge-model-building-with-variancepartition.html#cb114-4" tabindex="-1"></a>plot_gene_expr <span class="ot">&lt;-</span> <span class="cf">function</span>(sample_var, gene_id) {</span>
<span id="cb114-5"><a href="dge-model-building-with-variancepartition.html#cb114-5" tabindex="-1"></a>    colors <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb114-6"><a href="dge-model-building-with-variancepartition.html#cb114-6" tabindex="-1"></a>        <span class="st">&quot;Group&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span> <span class="ot">=</span> <span class="st">&quot;brown2&quot;</span>, <span class="st">&quot;Experimental&quot;</span> <span class="ot">=</span> <span class="st">&quot;deepskyblue3&quot;</span>),</span>
<span id="cb114-7"><a href="dge-model-building-with-variancepartition.html#cb114-7" tabindex="-1"></a>        <span class="st">&quot;Age&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Adult&quot;</span> <span class="ot">=</span> <span class="st">&quot;slateblue3&quot;</span>, <span class="st">&quot;Pup&quot;</span> <span class="ot">=</span> <span class="st">&quot;yellow3&quot;</span>),</span>
<span id="cb114-8"><a href="dge-model-building-with-variancepartition.html#cb114-8" tabindex="-1"></a>        <span class="st">&quot;Sex&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink1&quot;</span>, <span class="st">&quot;M&quot;</span> <span class="ot">=</span> <span class="st">&quot;dodgerblue&quot;</span>),</span>
<span id="cb114-9"><a href="dge-model-building-with-variancepartition.html#cb114-9" tabindex="-1"></a>        <span class="st">&quot;Pregnancy&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Yes&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkorchid3&quot;</span>, <span class="st">&quot;No&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkolivegreen4&quot;</span>),</span>
<span id="cb114-10"><a href="dge-model-building-with-variancepartition.html#cb114-10" tabindex="-1"></a>        <span class="st">&quot;plate&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Plate1&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkorange&quot;</span>, <span class="st">&quot;Plate2&quot;</span> <span class="ot">=</span> <span class="st">&quot;lightskyblue&quot;</span>, <span class="st">&quot;Plate3&quot;</span> <span class="ot">=</span> <span class="st">&quot;deeppink1&quot;</span>),</span>
<span id="cb114-11"><a href="dge-model-building-with-variancepartition.html#cb114-11" tabindex="-1"></a>        <span class="st">&quot;flowcell&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb114-12"><a href="dge-model-building-with-variancepartition.html#cb114-12" tabindex="-1"></a>            <span class="st">&quot;HKCG7DSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;chartreuse2&quot;</span>, <span class="st">&quot;HKCMHDSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;magenta&quot;</span>, <span class="st">&quot;HKCNKDSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;turquoise3&quot;</span>,</span>
<span id="cb114-13"><a href="dge-model-building-with-variancepartition.html#cb114-13" tabindex="-1"></a>            <span class="st">&quot;HKCTMDSXX&quot;</span> <span class="ot">=</span> <span class="st">&quot;tomato&quot;</span></span>
<span id="cb114-14"><a href="dge-model-building-with-variancepartition.html#cb114-14" tabindex="-1"></a>        )</span>
<span id="cb114-15"><a href="dge-model-building-with-variancepartition.html#cb114-15" tabindex="-1"></a>    )</span>
<span id="cb114-16"><a href="dge-model-building-with-variancepartition.html#cb114-16" tabindex="-1"></a></span>
<span id="cb114-17"><a href="dge-model-building-with-variancepartition.html#cb114-17" tabindex="-1"></a>    <span class="do">## Lognorm counts of the gene across samples</span></span>
<span id="cb114-18"><a href="dge-model-building-with-variancepartition.html#cb114-18" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">colData</span>(rse_gene_filt)</span>
<span id="cb114-19"><a href="dge-model-building-with-variancepartition.html#cb114-19" tabindex="-1"></a>    data<span class="sc">$</span>gene_expr <span class="ot">&lt;-</span> <span class="fu">assays</span>(rse_gene_filt)<span class="sc">$</span>logcounts[gene_id, ]</span>
<span id="cb114-20"><a href="dge-model-building-with-variancepartition.html#cb114-20" tabindex="-1"></a></span>
<span id="cb114-21"><a href="dge-model-building-with-variancepartition.html#cb114-21" tabindex="-1"></a>    <span class="do">## Percentage of variance explained by the variable</span></span>
<span id="cb114-22"><a href="dge-model-building-with-variancepartition.html#cb114-22" tabindex="-1"></a>    percentage <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="sc">*</span> <span class="fu">signif</span>(varPart_data[gene_id, sample_var], <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb114-23"><a href="dge-model-building-with-variancepartition.html#cb114-23" tabindex="-1"></a></span>
<span id="cb114-24"><a href="dge-model-building-with-variancepartition.html#cb114-24" tabindex="-1"></a>    <span class="do">## Boxplots for categorical variables</span></span>
<span id="cb114-25"><a href="dge-model-building-with-variancepartition.html#cb114-25" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(data[, sample_var]) <span class="sc">==</span> <span class="st">&quot;character&quot;</span>) {</span>
<span id="cb114-26"><a href="dge-model-building-with-variancepartition.html#cb114-26" tabindex="-1"></a>        plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">as.data.frame</span>(data), <span class="at">mapping =</span> <span class="fu">aes</span>(</span>
<span id="cb114-27"><a href="dge-model-building-with-variancepartition.html#cb114-27" tabindex="-1"></a>            <span class="at">x =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var),</span>
<span id="cb114-28"><a href="dge-model-building-with-variancepartition.html#cb114-28" tabindex="-1"></a>            <span class="at">y =</span> gene_expr, <span class="at">color =</span> <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(sample_var)</span>
<span id="cb114-29"><a href="dge-model-building-with-variancepartition.html#cb114-29" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb114-30"><a href="dge-model-building-with-variancepartition.html#cb114-30" tabindex="-1"></a>            <span class="fu">geom_boxplot</span>(<span class="at">size =</span> <span class="fl">0.25</span>, <span class="at">width =</span> <span class="fl">0.32</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">outlier.color =</span> <span class="st">&quot;#FFFFFFFF&quot;</span>) <span class="sc">+</span></span>
<span id="cb114-31"><a href="dge-model-building-with-variancepartition.html#cb114-31" tabindex="-1"></a>            <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.15</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb114-32"><a href="dge-model-building-with-variancepartition.html#cb114-32" tabindex="-1"></a>            <span class="fu">stat_smooth</span>(<span class="at">geom =</span> <span class="st">&quot;line&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.4</span>, <span class="at">span =</span> <span class="fl">0.3</span>, <span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>), <span class="at">color =</span> <span class="st">&quot;orangered3&quot;</span>) <span class="sc">+</span></span>
<span id="cb114-33"><a href="dge-model-building-with-variancepartition.html#cb114-33" tabindex="-1"></a>            <span class="fu">scale_color_manual</span>(<span class="at">values =</span> colors[[sample_var]]) <span class="sc">+</span></span>
<span id="cb114-34"><a href="dge-model-building-with-variancepartition.html#cb114-34" tabindex="-1"></a>            <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb114-35"><a href="dge-model-building-with-variancepartition.html#cb114-35" tabindex="-1"></a>            <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb114-36"><a href="dge-model-building-with-variancepartition.html#cb114-36" tabindex="-1"></a>            <span class="fu">labs</span>(</span>
<span id="cb114-37"><a href="dge-model-building-with-variancepartition.html#cb114-37" tabindex="-1"></a>                <span class="at">title =</span> gene_id,</span>
<span id="cb114-38"><a href="dge-model-building-with-variancepartition.html#cb114-38" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;Variance explained: &quot;</span>, percentage, <span class="st">&quot;%&quot;</span>),</span>
<span id="cb114-39"><a href="dge-model-building-with-variancepartition.html#cb114-39" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">&quot;lognorm counts&quot;</span>, <span class="at">x =</span> sample_var</span>
<span id="cb114-40"><a href="dge-model-building-with-variancepartition.html#cb114-40" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb114-41"><a href="dge-model-building-with-variancepartition.html#cb114-41" tabindex="-1"></a>            <span class="fu">theme</span>(</span>
<span id="cb114-42"><a href="dge-model-building-with-variancepartition.html#cb114-42" tabindex="-1"></a>                <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb114-43"><a href="dge-model-building-with-variancepartition.html#cb114-43" tabindex="-1"></a>                <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">6</span>)),</span>
<span id="cb114-44"><a href="dge-model-building-with-variancepartition.html#cb114-44" tabindex="-1"></a>                <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="fl">7.5</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb114-45"><a href="dge-model-building-with-variancepartition.html#cb114-45" tabindex="-1"></a>                <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>),</span>
<span id="cb114-46"><a href="dge-model-building-with-variancepartition.html#cb114-46" tabindex="-1"></a>                <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb114-47"><a href="dge-model-building-with-variancepartition.html#cb114-47" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb114-48"><a href="dge-model-building-with-variancepartition.html#cb114-48" tabindex="-1"></a>            )</span>
<span id="cb114-49"><a href="dge-model-building-with-variancepartition.html#cb114-49" tabindex="-1"></a>    }</span>
<span id="cb114-50"><a href="dge-model-building-with-variancepartition.html#cb114-50" tabindex="-1"></a></span>
<span id="cb114-51"><a href="dge-model-building-with-variancepartition.html#cb114-51" tabindex="-1"></a>    <span class="do">## Scatterplots for continuous variables</span></span>
<span id="cb114-52"><a href="dge-model-building-with-variancepartition.html#cb114-52" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb114-53"><a href="dge-model-building-with-variancepartition.html#cb114-53" tabindex="-1"></a>        colors <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb114-54"><a href="dge-model-building-with-variancepartition.html#cb114-54" tabindex="-1"></a>            <span class="st">&quot;mitoRate&quot;</span> <span class="ot">=</span> <span class="st">&quot;khaki3&quot;</span>, <span class="st">&quot;overallMapRate&quot;</span> <span class="ot">=</span> <span class="st">&quot;turquoise&quot;</span>, <span class="st">&quot;totalAssignedGene&quot;</span> <span class="ot">=</span> <span class="st">&quot;plum2&quot;</span>, <span class="st">&quot;rRNA_rate&quot;</span> <span class="ot">=</span> <span class="st">&quot;orange3&quot;</span>,</span>
<span id="cb114-55"><a href="dge-model-building-with-variancepartition.html#cb114-55" tabindex="-1"></a>            <span class="st">&quot;sum&quot;</span> <span class="ot">=</span> <span class="st">&quot;palegreen3&quot;</span>, <span class="st">&quot;detected&quot;</span> <span class="ot">=</span> <span class="st">&quot;skyblue2&quot;</span>, <span class="st">&quot;ERCCsumLogErr&quot;</span> <span class="ot">=</span> <span class="st">&quot;slateblue1&quot;</span></span>
<span id="cb114-56"><a href="dge-model-building-with-variancepartition.html#cb114-56" tabindex="-1"></a>        )</span>
<span id="cb114-57"><a href="dge-model-building-with-variancepartition.html#cb114-57" tabindex="-1"></a></span>
<span id="cb114-58"><a href="dge-model-building-with-variancepartition.html#cb114-58" tabindex="-1"></a>        plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(data), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">eval</span>(<span class="fu">parse_expr</span>(sample_var)), <span class="at">y =</span> gene_expr)) <span class="sc">+</span></span>
<span id="cb114-59"><a href="dge-model-building-with-variancepartition.html#cb114-59" tabindex="-1"></a>            <span class="fu">geom_point</span>(<span class="at">color =</span> colors[[sample_var]], <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb114-60"><a href="dge-model-building-with-variancepartition.html#cb114-60" tabindex="-1"></a>            <span class="fu">stat_smooth</span>(<span class="at">geom =</span> <span class="st">&quot;line&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">size =</span> <span class="fl">0.4</span>, <span class="at">span =</span> <span class="fl">0.25</span>, <span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">color =</span> <span class="st">&quot;orangered3&quot;</span>) <span class="sc">+</span></span>
<span id="cb114-61"><a href="dge-model-building-with-variancepartition.html#cb114-61" tabindex="-1"></a>            <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb114-62"><a href="dge-model-building-with-variancepartition.html#cb114-62" tabindex="-1"></a>            <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb114-63"><a href="dge-model-building-with-variancepartition.html#cb114-63" tabindex="-1"></a>            <span class="fu">labs</span>(</span>
<span id="cb114-64"><a href="dge-model-building-with-variancepartition.html#cb114-64" tabindex="-1"></a>                <span class="at">title =</span> gene_id,</span>
<span id="cb114-65"><a href="dge-model-building-with-variancepartition.html#cb114-65" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">&quot;Variance explained: &quot;</span>, percentage, <span class="st">&quot;%&quot;</span>),</span>
<span id="cb114-66"><a href="dge-model-building-with-variancepartition.html#cb114-66" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">&quot;lognorm counts&quot;</span>, <span class="at">x =</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, sample_var)</span>
<span id="cb114-67"><a href="dge-model-building-with-variancepartition.html#cb114-67" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb114-68"><a href="dge-model-building-with-variancepartition.html#cb114-68" tabindex="-1"></a>            <span class="fu">theme</span>(</span>
<span id="cb114-69"><a href="dge-model-building-with-variancepartition.html#cb114-69" tabindex="-1"></a>                <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.1</span>, <span class="fl">0.4</span>, <span class="fl">0.1</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb114-70"><a href="dge-model-building-with-variancepartition.html#cb114-70" tabindex="-1"></a>                <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">7</span>)),</span>
<span id="cb114-71"><a href="dge-model-building-with-variancepartition.html#cb114-71" tabindex="-1"></a>                <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">6</span>)),</span>
<span id="cb114-72"><a href="dge-model-building-with-variancepartition.html#cb114-72" tabindex="-1"></a>                <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="fl">7.5</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb114-73"><a href="dge-model-building-with-variancepartition.html#cb114-73" tabindex="-1"></a>                <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">&quot;gray40&quot;</span>),</span>
<span id="cb114-74"><a href="dge-model-building-with-variancepartition.html#cb114-74" tabindex="-1"></a>                <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb114-75"><a href="dge-model-building-with-variancepartition.html#cb114-75" tabindex="-1"></a>                <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb114-76"><a href="dge-model-building-with-variancepartition.html#cb114-76" tabindex="-1"></a>            )</span>
<span id="cb114-77"><a href="dge-model-building-with-variancepartition.html#cb114-77" tabindex="-1"></a>    }</span>
<span id="cb114-78"><a href="dge-model-building-with-variancepartition.html#cb114-78" tabindex="-1"></a></span>
<span id="cb114-79"><a href="dge-model-building-with-variancepartition.html#cb114-79" tabindex="-1"></a>    <span class="fu">return</span>(plot)</span>
<span id="cb114-80"><a href="dge-model-building-with-variancepartition.html#cb114-80" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="dge-model-building-with-variancepartition.html#cb115-1" tabindex="-1"></a><span class="do">## Function to plot gene expression vs sample variable data for top 3 most affected genes</span></span>
<span id="cb115-2"><a href="dge-model-building-with-variancepartition.html#cb115-2" tabindex="-1"></a></span>
<span id="cb115-3"><a href="dge-model-building-with-variancepartition.html#cb115-3" tabindex="-1"></a>plot_gene_expr_sample <span class="ot">&lt;-</span> <span class="cf">function</span>(sample_var) {</span>
<span id="cb115-4"><a href="dge-model-building-with-variancepartition.html#cb115-4" tabindex="-1"></a>    <span class="do">## Top 3 genes most affected by sample variable</span></span>
<span id="cb115-5"><a href="dge-model-building-with-variancepartition.html#cb115-5" tabindex="-1"></a>    affected_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(varPart_data[<span class="fu">order</span>(varPart_data[, sample_var], <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ][<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, ])</span>
<span id="cb115-6"><a href="dge-model-building-with-variancepartition.html#cb115-6" tabindex="-1"></a></span>
<span id="cb115-7"><a href="dge-model-building-with-variancepartition.html#cb115-7" tabindex="-1"></a>    <span class="do">## Plots</span></span>
<span id="cb115-8"><a href="dge-model-building-with-variancepartition.html#cb115-8" tabindex="-1"></a>    plots <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb115-9"><a href="dge-model-building-with-variancepartition.html#cb115-9" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(affected_genes)) {</span>
<span id="cb115-10"><a href="dge-model-building-with-variancepartition.html#cb115-10" tabindex="-1"></a>        plots[[i]] <span class="ot">&lt;-</span> <span class="fu">plot_gene_expr</span>(sample_var, affected_genes[i])</span>
<span id="cb115-11"><a href="dge-model-building-with-variancepartition.html#cb115-11" tabindex="-1"></a>    }</span>
<span id="cb115-12"><a href="dge-model-building-with-variancepartition.html#cb115-12" tabindex="-1"></a>    <span class="fu">plot_grid</span>(plots[[<span class="dv">1</span>]], plots[[<span class="dv">2</span>]], plots[[<span class="dv">3</span>]], <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb115-13"><a href="dge-model-building-with-variancepartition.html#cb115-13" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="dge-model-building-with-variancepartition.html#cb116-1" tabindex="-1"></a><span class="do">## Plots for top affected genes by &#39;overallMapRate&#39;</span></span>
<span id="cb116-2"><a href="dge-model-building-with-variancepartition.html#cb116-2" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">plot_gene_expr_sample</span>(<span class="st">&quot;overallMapRate&quot;</span>)</span>
<span id="cb116-3"><a href="dge-model-building-with-variancepartition.html#cb116-3" tabindex="-1"></a>plots <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb116-4"><a href="dge-model-building-with-variancepartition.html#cb116-4" tabindex="-1"></a></span>
<span id="cb116-5"><a href="dge-model-building-with-variancepartition.html#cb116-5" tabindex="-1"></a><span class="do">## Plots for top affected genes by &#39;totalAssignedGene&#39;</span></span>
<span id="cb116-6"><a href="dge-model-building-with-variancepartition.html#cb116-6" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">plot_gene_expr_sample</span>(<span class="st">&quot;totalAssignedGene&quot;</span>)</span>
<span id="cb116-7"><a href="dge-model-building-with-variancepartition.html#cb116-7" tabindex="-1"></a>plots <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb116-8"><a href="dge-model-building-with-variancepartition.html#cb116-8" tabindex="-1"></a></span>
<span id="cb116-9"><a href="dge-model-building-with-variancepartition.html#cb116-9" tabindex="-1"></a><span class="do">## Plots for top affected genes by &#39;Group&#39;</span></span>
<span id="cb116-10"><a href="dge-model-building-with-variancepartition.html#cb116-10" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">plot_gene_expr_sample</span>(<span class="st">&quot;Group&quot;</span>)</span>
<span id="cb116-11"><a href="dge-model-building-with-variancepartition.html#cb116-11" tabindex="-1"></a>plots <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb116-12"><a href="dge-model-building-with-variancepartition.html#cb116-12" tabindex="-1"></a></span>
<span id="cb116-13"><a href="dge-model-building-with-variancepartition.html#cb116-13" tabindex="-1"></a><span class="do">## Plots for top affected genes by &#39;Sex&#39; (genes in sexual chrs)</span></span>
<span id="cb116-14"><a href="dge-model-building-with-variancepartition.html#cb116-14" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">plot_gene_expr_sample</span>(<span class="st">&quot;Sex&quot;</span>)</span>
<span id="cb116-15"><a href="dge-model-building-with-variancepartition.html#cb116-15" tabindex="-1"></a>plots <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<figure>
<img src="Figures/overallMapRate_affected_genes.png" width="800px" align=center />
</figure>
<figure>
<img src="Figures/totalAssignedGene_affected_genes.png" width="800px" align=center />
</figure>
<figure>
<img src="Figures/Group_affected_genes.png" width="800px" align=center />
</figure>
<figure>
<img src="Figures/Sex_affected_genes.png" width="800px" align=center />
</figure>
<style>
p.exercise {
background-color: #FFFAFA;
padding: 15px;
border: 2px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<p class="exercise">
📝 <strong>Exercise 3</strong>: What % of variance does <code>Group</code> explain for the gene ENSMUSG00000042348.10? Create the boxplots for its counts in control and experimental samples. Is it more likely that the gene is up-regulated or down-regulated?
</p>
<p class="exercise">
📝 <strong>Exercise 4</strong>: Do the same for the gene ENSMUSG00000064372.1. What do you observe in terms of variance percentage and sample differences?
</p>
</div>
<div id="references-3" class="section level2 unnumbered hasAnchor">
<h2>References<a href="dge-model-building-with-variancepartition.html#references-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>Hoffman, G. E., &amp; Schadt, E. E. (2016). variancePartition: interpreting drivers of variation in complex gene expression studies.<em>BMC bioinformatics</em>, 17(1), 1-13.</p></li>
<li><p>Hoffman, G. (2022). variancePartition: Quantifying and interpreting drivers of variation in multilevel gene expression experiments.</p></li>
<li><p>van den Berg, S. M. (2022). Analysing data using linear models. Web site: <a href="https://bookdown.org/pingapang9/linear_models_bookdown/" class="uri">https://bookdown.org/pingapang9/linear_models_bookdown/</a></p></li>
<li><p>Simoiu, C. &amp; Savage, J. (2016). A bag of tips and tricks for dealing with scale issues. Web site: <a href="https://rpubs.com/jimsavage/scale_issues" class="uri">https://rpubs.com/jimsavage/scale_issues</a></p></li>
</ol>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="interpreting-model-coefficients-with-exploremodelmatrix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-gene-expression-exercise.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
