<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 Differential gene expression analysis with limma | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="10 Differential gene expression analysis with limma | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="10 Differential gene expression analysis with limma | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 Differential gene expression analysis with limma | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="interpreting-model-coefficients-with-exploremodelmatrix.html"/>
<link rel="next" href="differential-gene-expression-exercise.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html"><i class="fa fa-check"></i><b>1</b> SPEAQeasy introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#overview-slides"><i class="fa fa-check"></i><b>1.1</b> 2022-04-20 overview slides</a></li>
<li class="chapter" data-level="1.2" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#speaqeasy-main-links"><i class="fa fa-check"></i><b>1.2</b> SPEAQeasy main links</a></li>
<li class="chapter" data-level="1.3" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#pipeline-outputs"><i class="fa fa-check"></i><b>1.3</b> Pipeline outputs</a></li>
<li class="chapter" data-level="1.4" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#exercises"><i class="fa fa-check"></i><b>1.4</b> Exercises</a></li>
<li class="chapter" data-level="1.5" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#solutions"><i class="fa fa-check"></i><b>1.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html"><i class="fa fa-check"></i><b>2</b> smokingMouse study</a>
<ul>
<li class="chapter" data-level="2.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#overview-1"><i class="fa fa-check"></i><b>2.2</b> Overview</a></li>
<li class="chapter" data-level="2.3" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#goals"><i class="fa fa-check"></i><b>2.3</b> Goals</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#main-objective"><i class="fa fa-check"></i><b>2.3.1</b> Main objective</a></li>
<li class="chapter" data-level="2.3.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#secondary-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Secondary objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#study-design"><i class="fa fa-check"></i><b>2.4</b> Study design</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html"><i class="fa fa-check"></i><b>3</b> SummarizedExperiment review</a>
<ul>
<li class="chapter" data-level="3.1" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#quiz"><i class="fa fa-check"></i><b>3.2</b> Quiz</a></li>
<li class="chapter" data-level="3.3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#exercises-1"><i class="fa fa-check"></i><b>3.3</b> Exercises</a></li>
<li class="chapter" data-level="3.4" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#solutions-1"><i class="fa fa-check"></i><b>3.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html"><i class="fa fa-check"></i><b>4</b> smokingMouse RSE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#download-data"><i class="fa fa-check"></i><b>4.1</b> Download data</a></li>
<li class="chapter" data-level="4.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#data-overview"><i class="fa fa-check"></i><b>4.2</b> Data overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#assays"><i class="fa fa-check"></i><b>4.2.1</b> Assays</a></li>
<li class="chapter" data-level="4.2.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#sample-data"><i class="fa fa-check"></i><b>4.2.2</b> Sample data</a></li>
<li class="chapter" data-level="4.2.3" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#gene-information"><i class="fa fa-check"></i><b>4.2.3</b> Gene Information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="isee-introduction.html"><a href="isee-introduction.html"><i class="fa fa-check"></i><b>5</b> iSEE introduction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="isee-introduction.html"><a href="isee-introduction.html#toy-rse-data"><i class="fa fa-check"></i><b>5.1</b> Toy RSE data</a></li>
<li class="chapter" data-level="5.2" data-path="isee-introduction.html"><a href="isee-introduction.html#isee"><i class="fa fa-check"></i><b>5.2</b> iSEE</a></li>
<li class="chapter" data-level="5.3" data-path="isee-introduction.html"><a href="isee-introduction.html#exercise-with-data-from-spatiallibd"><i class="fa fa-check"></i><b>5.3</b> Exercise with data from spatialLIBD</a></li>
<li class="chapter" data-level="5.4" data-path="isee-introduction.html"><a href="isee-introduction.html#community"><i class="fa fa-check"></i><b>5.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html"><i class="fa fa-check"></i><b>6</b> RSE plotting basics with smokingMouse</a>
<ul>
<li class="chapter" data-level="6.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#download-the-smokingmouse-data"><i class="fa fa-check"></i><b>6.1</b> Download the smokingMouse data</a></li>
<li class="chapter" data-level="6.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-preparation"><i class="fa fa-check"></i><b>6.2</b> Data preparation</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-normalization"><i class="fa fa-check"></i><b>6.2.1</b> Data normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#gene-filtering"><i class="fa fa-check"></i><b>6.2.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>6.3</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="6.4" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#quality-control-analysis"><i class="fa fa-check"></i><b>6.4</b> Quality Control Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#evaluate-qc-metrics-for-groups-of-samples"><i class="fa fa-check"></i><b>6.4.1</b> Evaluate QC metrics for groups of samples</a></li>
<li class="chapter" data-level="6.4.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#examine-relationships-between-qc-metrics-of-samples"><i class="fa fa-check"></i><b>6.4.2</b> Examine relationships between QC metrics of samples</a></li>
<li class="chapter" data-level="6.4.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#qc-sample-filtering"><i class="fa fa-check"></i><b>6.4.3</b> QC sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#miscellaneous-about-cpm-values"><i class="fa fa-check"></i><b>6.5</b> Miscellaneous about CPM values</a></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>7</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="7.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>7.1</b> Related past materials</a></li>
<li class="chapter" data-level="7.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>7.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="7.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>7.3</b> Live demo</a></li>
<li class="chapter" data-level="7.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-1"><i class="fa fa-check"></i><b>7.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>8</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>8.1</b> recount projects</a></li>
<li class="chapter" data-level="8.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>8.2</b> Using recount3</a></li>
<li class="chapter" data-level="8.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>8.3</b> Exercise</a></li>
<li class="chapter" data-level="8.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-2"><i class="fa fa-check"></i><b>8.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>9</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="9.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>9.1</b> Model objects in R</a></li>
<li class="chapter" data-level="9.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>9.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>9.3</b> Example 1</a></li>
<li class="chapter" data-level="9.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>9.4</b> Example 2</a></li>
<li class="chapter" data-level="9.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>9.5</b> Example 3</a></li>
<li class="chapter" data-level="9.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>9.6</b> Exercise</a></li>
<li class="chapter" data-level="9.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>9.7</b> To learn more</a></li>
<li class="chapter" data-level="9.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-3"><i class="fa fa-check"></i><b>9.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html"><i class="fa fa-check"></i><b>10</b> Differential gene expression analysis with limma</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#srp045638-data"><i class="fa fa-check"></i><b>10.1</b> SRP045638 data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#data-normalization-1"><i class="fa fa-check"></i><b>10.2</b> Data normalization</a></li>
<li class="chapter" data-level="10.3" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#differential-expression"><i class="fa fa-check"></i><b>10.3</b> Differential expression</a></li>
<li class="chapter" data-level="10.4" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#visualizing-degs"><i class="fa fa-check"></i><b>10.4</b> Visualizing DEGs</a></li>
<li class="chapter" data-level="10.5" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#community-4"><i class="fa fa-check"></i><b>10.5</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>11</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>11.1</b> Recap</a></li>
<li class="chapter" data-level="11.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>11.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>12</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="12.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>12.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="12.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>12.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="12.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>12.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>13</b> Expression heatmaps with ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="13.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#introduction-1"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#complexheatmap"><i class="fa fa-check"></i><b>13.2</b> ComplexHeatmap</a></li>
<li class="chapter" data-level="13.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#heatmap-with-expression-data"><i class="fa fa-check"></i><b>13.3</b> Heatmap with expression data</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#download-data-1"><i class="fa fa-check"></i><b>13.3.1</b> Download data</a></li>
<li class="chapter" data-level="13.3.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#prepare-the-data"><i class="fa fa-check"></i><b>13.3.2</b> Prepare the data</a></li>
<li class="chapter" data-level="13.3.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#create-annotations"><i class="fa fa-check"></i><b>13.3.3</b> Create annotations</a></li>
<li class="chapter" data-level="13.3.4" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#plot-our-heatmap"><i class="fa fa-check"></i><b>13.3.4</b> Plot our heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="complexheatmap-exercise-solution.html"><a href="complexheatmap-exercise-solution.html"><i class="fa fa-check"></i><b>14</b> ComplexHeatmap exercise solution</a></li>
<li class="chapter" data-level="15" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html"><i class="fa fa-check"></i><b>15</b> Exercise with variancePartition and ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="15.1" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html#steps"><i class="fa fa-check"></i><b>15.1</b> Steps:</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>16</b> Miscellaneous</a>
<ul>
<li class="chapter" data-level="16.1" data-path="miscellaneous.html"><a href="miscellaneous.html#postcards"><i class="fa fa-check"></i><b>16.1</b> postcards</a></li>
<li class="chapter" data-level="16.2" data-path="miscellaneous.html"><a href="miscellaneous.html#organizing-your-work"><i class="fa fa-check"></i><b>16.2</b> Organizing your work</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><i class="fa fa-check"></i><b>17</b> Introduction to the world of spatially-resolved transcriptomics with Visium data</a>
<ul>
<li class="chapter" data-level="17.1" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html#community-5"><i class="fa fa-check"></i><b>17.1</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><i class="fa fa-check"></i><b>18</b> Downloading public Visium spatial data and visualizing it with spatialLIBD</a>
<ul>
<li class="chapter" data-level="18.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#download-data-from-10x-genomics"><i class="fa fa-check"></i><b>18.1</b> Download data from 10x Genomics</a></li>
<li class="chapter" data-level="18.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#add-gene-annotation-information"><i class="fa fa-check"></i><b>18.2</b> Add gene annotation information</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-10x"><i class="fa fa-check"></i><b>18.2.1</b> From 10x</a></li>
<li class="chapter" data-level="18.2.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-gencode"><i class="fa fa-check"></i><b>18.2.2</b> From Gencode</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#wrapper-functions-for-reading-the-data"><i class="fa fa-check"></i><b>18.3</b> Wrapper functions for reading the data</a></li>
<li class="chapter" data-level="18.4" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#visualize-spatial-plots"><i class="fa fa-check"></i><b>18.4</b> Visualize spatial plots</a></li>
<li class="chapter" data-level="18.5" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#exercise-3"><i class="fa fa-check"></i><b>18.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>19</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="19.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>19.1</b> Spatial registration</a></li>
<li class="chapter" data-level="19.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-4"><i class="fa fa-check"></i><b>19.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
<li class="chapter" data-level="20" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>20</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="20.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>20.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>20.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="20.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis-1"><i class="fa fa-check"></i><b>20.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>20.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="20.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>20.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>21</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="21.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>21.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="21.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>21.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>21.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="21.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>21.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="21.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>21.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="21.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>21.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="21.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>21.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>21.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="21.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>21.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="21.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps"><i class="fa fa-check"></i><b>21.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>22</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="22.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>22.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="22.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>22.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>22.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="22.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>22.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>23</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="23.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>23.1</b> here</a></li>
<li class="chapter" data-level="23.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>23.2</b> Usethis</a></li>
<li class="chapter" data-level="23.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>23.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="23.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>23.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="23.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>23.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="23.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>23.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="23.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>23.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>23.4</b> R websites</a>
<ul>
<li class="chapter" data-level="23.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>23.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="23.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>23.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="23.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>23.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="23.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>23.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards-1"><i class="fa fa-check"></i><b>23.5</b> postcards</a>
<ul>
<li class="chapter" data-level="23.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>23.5.1</b> Installation</a></li>
<li class="chapter" data-level="23.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>23.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="23.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>23.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="23.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>23.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="23.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>23.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="23.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>23.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="23.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>23.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="23.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>23.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-4"><i class="fa fa-check"></i><b>23.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>24</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="24.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>24.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="24.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>24.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="24.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>24.1.2</b> SingleCellExperiment</a></li>
</ul></li>
<li class="chapter" data-level="24.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>24.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="24.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>24.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="24.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>24.3.1</b> Header</a></li>
<li class="chapter" data-level="24.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>24.3.2</b> Panel types</a></li>
<li class="chapter" data-level="24.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>24.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="24.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>24.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="24.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>24.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="24.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>24.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="24.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>24.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="24.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>24.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="24.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>24.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="24.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>24.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="24.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps-1"><i class="fa fa-check"></i><b>24.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="24.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>24.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="24.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>24.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="24.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>24.4.1</b> Setting up the data</a></li>
</ul></li>
<li class="chapter" data-level="24.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references-5"><i class="fa fa-check"></i><b>24.5</b> References</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>25</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="25.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing-overview"><i class="fa fa-check"></i><b>25.1</b> Single cell RNA sequencing overview</a></li>
<li class="chapter" data-level="25.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>25.2</b> The SingleCellExperiment class</a></li>
<li class="chapter" data-level="25.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>25.3</b> Quality Control</a></li>
<li class="chapter" data-level="25.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>25.4</b> Normalization</a></li>
<li class="chapter" data-level="25.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>25.5</b> Feature selection</a></li>
<li class="chapter" data-level="25.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>25.6</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="25.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>25.7</b> Clustering</a></li>
<li class="chapter" data-level="25.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>25.8</b> Marker gene detection</a></li>
<li class="chapter" data-level="25.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>25.9</b> Cell type annotation</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-gene-expression-analysis-with-limma" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">10</span> Differential gene expression analysis with limma<a href="differential-gene-expression-analysis-with-limma.html#differential-gene-expression-analysis-with-limma" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Leo</p>
<div id="srp045638-data" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> SRP045638 data<a href="differential-gene-expression-analysis-with-limma.html#srp045638-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll use data from <a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRP045638" class="uri">https://www.ncbi.nlm.nih.gov/sra/?term=SRP045638</a> processed and made available through the <code>recount3</code> project. First, we need to download the data with the same commands we saw earlier.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="differential-gene-expression-analysis-with-limma.html#cb120-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;recount3&quot;</span>)</span>
<span id="cb120-2"><a href="differential-gene-expression-analysis-with-limma.html#cb120-2" tabindex="-1"></a></span>
<span id="cb120-3"><a href="differential-gene-expression-analysis-with-limma.html#cb120-3" tabindex="-1"></a>human_projects <span class="ot">&lt;-</span> <span class="fu">available_projects</span>()</span>
<span id="cb120-4"><a href="differential-gene-expression-analysis-with-limma.html#cb120-4" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:48.744515 caching file sra.recount_project.MD.gz.</span></span>
<span id="cb120-5"><a href="differential-gene-expression-analysis-with-limma.html#cb120-5" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:49.029739 caching file gtex.recount_project.MD.gz.</span></span>
<span id="cb120-6"><a href="differential-gene-expression-analysis-with-limma.html#cb120-6" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:49.328728 caching file tcga.recount_project.MD.gz.</span></span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="differential-gene-expression-analysis-with-limma.html#cb121-1" tabindex="-1"></a></span>
<span id="cb121-2"><a href="differential-gene-expression-analysis-with-limma.html#cb121-2" tabindex="-1"></a>rse_gene_SRP045638 <span class="ot">&lt;-</span> <span class="fu">create_rse</span>(</span>
<span id="cb121-3"><a href="differential-gene-expression-analysis-with-limma.html#cb121-3" tabindex="-1"></a>    <span class="fu">subset</span>(</span>
<span id="cb121-4"><a href="differential-gene-expression-analysis-with-limma.html#cb121-4" tabindex="-1"></a>        human_projects,</span>
<span id="cb121-5"><a href="differential-gene-expression-analysis-with-limma.html#cb121-5" tabindex="-1"></a>        project <span class="sc">==</span> <span class="st">&quot;SRP045638&quot;</span> <span class="sc">&amp;</span> project_type <span class="sc">==</span> <span class="st">&quot;data_sources&quot;</span></span>
<span id="cb121-6"><a href="differential-gene-expression-analysis-with-limma.html#cb121-6" tabindex="-1"></a>    )</span>
<span id="cb121-7"><a href="differential-gene-expression-analysis-with-limma.html#cb121-7" tabindex="-1"></a>)</span>
<span id="cb121-8"><a href="differential-gene-expression-analysis-with-limma.html#cb121-8" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:51.209506 downloading and reading the metadata.</span></span>
<span id="cb121-9"><a href="differential-gene-expression-analysis-with-limma.html#cb121-9" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:51.504816 caching file sra.sra.SRP045638.MD.gz.</span></span>
<span id="cb121-10"><a href="differential-gene-expression-analysis-with-limma.html#cb121-10" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/38/SRP045638/sra.sra.SRP045638.MD.gz&#39;</span></span>
<span id="cb121-11"><a href="differential-gene-expression-analysis-with-limma.html#cb121-11" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:52.428957 caching file sra.recount_project.SRP045638.MD.gz.</span></span>
<span id="cb121-12"><a href="differential-gene-expression-analysis-with-limma.html#cb121-12" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/38/SRP045638/sra.recount_project.SRP045638.MD.gz&#39;</span></span>
<span id="cb121-13"><a href="differential-gene-expression-analysis-with-limma.html#cb121-13" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:53.390673 caching file sra.recount_qc.SRP045638.MD.gz.</span></span>
<span id="cb121-14"><a href="differential-gene-expression-analysis-with-limma.html#cb121-14" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/38/SRP045638/sra.recount_qc.SRP045638.MD.gz&#39;</span></span>
<span id="cb121-15"><a href="differential-gene-expression-analysis-with-limma.html#cb121-15" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:54.260688 caching file sra.recount_seq_qc.SRP045638.MD.gz.</span></span>
<span id="cb121-16"><a href="differential-gene-expression-analysis-with-limma.html#cb121-16" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/38/SRP045638/sra.recount_seq_qc.SRP045638.MD.gz&#39;</span></span>
<span id="cb121-17"><a href="differential-gene-expression-analysis-with-limma.html#cb121-17" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:55.053198 caching file sra.recount_pred.SRP045638.MD.gz.</span></span>
<span id="cb121-18"><a href="differential-gene-expression-analysis-with-limma.html#cb121-18" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/metadata/38/SRP045638/sra.recount_pred.SRP045638.MD.gz&#39;</span></span>
<span id="cb121-19"><a href="differential-gene-expression-analysis-with-limma.html#cb121-19" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:55.696828 downloading and reading the feature information.</span></span>
<span id="cb121-20"><a href="differential-gene-expression-analysis-with-limma.html#cb121-20" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:55.94076 caching file human.gene_sums.G026.gtf.gz.</span></span>
<span id="cb121-21"><a href="differential-gene-expression-analysis-with-limma.html#cb121-21" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:56.372368 downloading and reading the counts: 66 samples across 63856 features.</span></span>
<span id="cb121-22"><a href="differential-gene-expression-analysis-with-limma.html#cb121-22" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:56.606645 caching file sra.gene_sums.SRP045638.G026.gz.</span></span>
<span id="cb121-23"><a href="differential-gene-expression-analysis-with-limma.html#cb121-23" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;http://duffel.rail.bio/recount3/human/data_sources/sra/gene_sums/38/SRP045638/sra.gene_sums.SRP045638.G026.gz&#39;</span></span>
<span id="cb121-24"><a href="differential-gene-expression-analysis-with-limma.html#cb121-24" tabindex="-1"></a><span class="co">#&gt; 2024-06-06 18:06:59.02981 constructing the RangedSummarizedExperiment (rse) object.</span></span></code></pre></div>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="differential-gene-expression-analysis-with-limma.html#cb122-1" tabindex="-1"></a><span class="fu">assay</span>(rse_gene_SRP045638, <span class="st">&quot;counts&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">compute_read_counts</span>(rse_gene_SRP045638)</span></code></pre></div>
<p>Now that we have the data and have computed the read counts (which will be needed for downstream analyses), we can use <code>expand_sra_attributes()</code> to make it easier to use the sample information in downstream analyses. However, we have to resolve some issues with this data first.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="differential-gene-expression-analysis-with-limma.html#cb123-1" tabindex="-1"></a><span class="do">## Can you notice the problem with the sample information?</span></span>
<span id="cb123-2"><a href="differential-gene-expression-analysis-with-limma.html#cb123-2" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra.sample_attributes[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb123-3"><a href="differential-gene-expression-analysis-with-limma.html#cb123-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;age;;67.78|biomaterial_provider;;LIBD|BioSampleModel;;Human|dev_stage;;Fetal|disease;;Control|Fraction;;total|isolate;;DLPFC|race;;AA|RIN;;8.3|sex;;female|tissue;;DLPFC&quot;</span></span>
<span id="cb123-4"><a href="differential-gene-expression-analysis-with-limma.html#cb123-4" tabindex="-1"></a><span class="co">#&gt; [2] &quot;age;;40.42|biomaterial_provider;;LIBD|BioSampleModel;;Human|disease;;Control|Fraction;;total|isolate;;DLPFC|race;;AA|RIN;;8.4|sex;;male|tissue;;DLPFC&quot;                   </span></span>
<span id="cb123-5"><a href="differential-gene-expression-analysis-with-limma.html#cb123-5" tabindex="-1"></a><span class="co">#&gt; [3] &quot;age;;41.58|biomaterial_provider;;LIBD|BioSampleModel;;Human|disease;;control|Fraction;;total|isolate;;R2869|race;;AA|RIN;;8.7|sex;;male|tissue;;DLPFC&quot;</span></span></code></pre></div>
<p>Lets resolve the issue we detected with this sample information. We’ll resolve this by eliminating some information that is only present in a subset of samples and that we don’t need.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="differential-gene-expression-analysis-with-limma.html#cb124-1" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra.sample_attributes <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;dev_stage;;Fetal</span><span class="sc">\\</span><span class="st">|&quot;</span>, <span class="st">&quot;&quot;</span>, rse_gene_SRP045638<span class="sc">$</span>sra.sample_attributes)</span>
<span id="cb124-2"><a href="differential-gene-expression-analysis-with-limma.html#cb124-2" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra.sample_attributes[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb124-3"><a href="differential-gene-expression-analysis-with-limma.html#cb124-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;age;;67.78|biomaterial_provider;;LIBD|BioSampleModel;;Human|disease;;Control|Fraction;;total|isolate;;DLPFC|race;;AA|RIN;;8.3|sex;;female|tissue;;DLPFC&quot;</span></span>
<span id="cb124-4"><a href="differential-gene-expression-analysis-with-limma.html#cb124-4" tabindex="-1"></a><span class="co">#&gt; [2] &quot;age;;40.42|biomaterial_provider;;LIBD|BioSampleModel;;Human|disease;;Control|Fraction;;total|isolate;;DLPFC|race;;AA|RIN;;8.4|sex;;male|tissue;;DLPFC&quot;  </span></span>
<span id="cb124-5"><a href="differential-gene-expression-analysis-with-limma.html#cb124-5" tabindex="-1"></a><span class="co">#&gt; [3] &quot;age;;41.58|biomaterial_provider;;LIBD|BioSampleModel;;Human|disease;;control|Fraction;;total|isolate;;R2869|race;;AA|RIN;;8.7|sex;;male|tissue;;DLPFC&quot;</span></span></code></pre></div>
<p>Now we can continue our work using similar code from the one we used earlier.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="differential-gene-expression-analysis-with-limma.html#cb125-1" tabindex="-1"></a>rse_gene_SRP045638 <span class="ot">&lt;-</span> <span class="fu">expand_sra_attributes</span>(rse_gene_SRP045638)</span>
<span id="cb125-2"><a href="differential-gene-expression-analysis-with-limma.html#cb125-2" tabindex="-1"></a></span>
<span id="cb125-3"><a href="differential-gene-expression-analysis-with-limma.html#cb125-3" tabindex="-1"></a><span class="fu">colData</span>(rse_gene_SRP045638)[</span>
<span id="cb125-4"><a href="differential-gene-expression-analysis-with-limma.html#cb125-4" tabindex="-1"></a>    ,</span>
<span id="cb125-5"><a href="differential-gene-expression-analysis-with-limma.html#cb125-5" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">&quot;^sra_attribute&quot;</span>, <span class="fu">colnames</span>(<span class="fu">colData</span>(rse_gene_SRP045638)))</span>
<span id="cb125-6"><a href="differential-gene-expression-analysis-with-limma.html#cb125-6" tabindex="-1"></a>]</span>
<span id="cb125-7"><a href="differential-gene-expression-analysis-with-limma.html#cb125-7" tabindex="-1"></a><span class="co">#&gt; DataFrame with 66 rows and 10 columns</span></span>
<span id="cb125-8"><a href="differential-gene-expression-analysis-with-limma.html#cb125-8" tabindex="-1"></a><span class="co">#&gt;            sra_attribute.age sra_attribute.biomaterial_provider sra_attribute.BioSampleModel sra_attribute.disease</span></span>
<span id="cb125-9"><a href="differential-gene-expression-analysis-with-limma.html#cb125-9" tabindex="-1"></a><span class="co">#&gt;                  &lt;character&gt;                        &lt;character&gt;                  &lt;character&gt;           &lt;character&gt;</span></span>
<span id="cb125-10"><a href="differential-gene-expression-analysis-with-limma.html#cb125-10" tabindex="-1"></a><span class="co">#&gt; SRR2071341             67.78                               LIBD                        Human               Control</span></span>
<span id="cb125-11"><a href="differential-gene-expression-analysis-with-limma.html#cb125-11" tabindex="-1"></a><span class="co">#&gt; SRR2071345             40.42                               LIBD                        Human               Control</span></span>
<span id="cb125-12"><a href="differential-gene-expression-analysis-with-limma.html#cb125-12" tabindex="-1"></a><span class="co">#&gt; SRR2071346             41.58                               LIBD                        Human               control</span></span>
<span id="cb125-13"><a href="differential-gene-expression-analysis-with-limma.html#cb125-13" tabindex="-1"></a><span class="co">#&gt; SRR2071347             44.17                               LIBD                        Human               control</span></span>
<span id="cb125-14"><a href="differential-gene-expression-analysis-with-limma.html#cb125-14" tabindex="-1"></a><span class="co">#&gt;            sra_attribute.Fraction sra_attribute.isolate sra_attribute.race sra_attribute.RIN sra_attribute.sex</span></span>
<span id="cb125-15"><a href="differential-gene-expression-analysis-with-limma.html#cb125-15" tabindex="-1"></a><span class="co">#&gt;                       &lt;character&gt;           &lt;character&gt;        &lt;character&gt;       &lt;character&gt;       &lt;character&gt;</span></span>
<span id="cb125-16"><a href="differential-gene-expression-analysis-with-limma.html#cb125-16" tabindex="-1"></a><span class="co">#&gt; SRR2071341                  total                 DLPFC                 AA               8.3            female</span></span>
<span id="cb125-17"><a href="differential-gene-expression-analysis-with-limma.html#cb125-17" tabindex="-1"></a><span class="co">#&gt; SRR2071345                  total                 DLPFC                 AA               8.4              male</span></span>
<span id="cb125-18"><a href="differential-gene-expression-analysis-with-limma.html#cb125-18" tabindex="-1"></a><span class="co">#&gt; SRR2071346                  total                 R2869                 AA               8.7              male</span></span>
<span id="cb125-19"><a href="differential-gene-expression-analysis-with-limma.html#cb125-19" tabindex="-1"></a><span class="co">#&gt; SRR2071347                  total                 R3098                 AA               5.3            female</span></span>
<span id="cb125-20"><a href="differential-gene-expression-analysis-with-limma.html#cb125-20" tabindex="-1"></a><span class="co">#&gt;            sra_attribute.tissue</span></span>
<span id="cb125-21"><a href="differential-gene-expression-analysis-with-limma.html#cb125-21" tabindex="-1"></a><span class="co">#&gt;                     &lt;character&gt;</span></span>
<span id="cb125-22"><a href="differential-gene-expression-analysis-with-limma.html#cb125-22" tabindex="-1"></a><span class="co">#&gt; SRR2071341                DLPFC</span></span>
<span id="cb125-23"><a href="differential-gene-expression-analysis-with-limma.html#cb125-23" tabindex="-1"></a><span class="co">#&gt; SRR2071345                DLPFC</span></span>
<span id="cb125-24"><a href="differential-gene-expression-analysis-with-limma.html#cb125-24" tabindex="-1"></a><span class="co">#&gt; SRR2071346                DLPFC</span></span>
<span id="cb125-25"><a href="differential-gene-expression-analysis-with-limma.html#cb125-25" tabindex="-1"></a><span class="co">#&gt; SRR2071347                DLPFC</span></span>
<span id="cb125-26"><a href="differential-gene-expression-analysis-with-limma.html#cb125-26" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 7 rows ]</span></span></code></pre></div>
<p>As well use the sampel information for building our statistical model, it will important that we make sure that it is on the correct format R expects later on.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="differential-gene-expression-analysis-with-limma.html#cb126-1" tabindex="-1"></a><span class="do">## Recast character vectors into numeric or factor ones</span></span>
<span id="cb126-2"><a href="differential-gene-expression-analysis-with-limma.html#cb126-2" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra_attribute.age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(rse_gene_SRP045638<span class="sc">$</span>sra_attribute.age)</span>
<span id="cb126-3"><a href="differential-gene-expression-analysis-with-limma.html#cb126-3" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra_attribute.disease <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">tolower</span>(rse_gene_SRP045638<span class="sc">$</span>sra_attribute.disease))</span>
<span id="cb126-4"><a href="differential-gene-expression-analysis-with-limma.html#cb126-4" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra_attribute.RIN <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(rse_gene_SRP045638<span class="sc">$</span>sra_attribute.RIN)</span>
<span id="cb126-5"><a href="differential-gene-expression-analysis-with-limma.html#cb126-5" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>sra_attribute.sex <span class="ot">&lt;-</span> <span class="fu">factor</span>(rse_gene_SRP045638<span class="sc">$</span>sra_attribute.sex)</span>
<span id="cb126-6"><a href="differential-gene-expression-analysis-with-limma.html#cb126-6" tabindex="-1"></a></span>
<span id="cb126-7"><a href="differential-gene-expression-analysis-with-limma.html#cb126-7" tabindex="-1"></a><span class="do">## Summary of our variables of interest</span></span>
<span id="cb126-8"><a href="differential-gene-expression-analysis-with-limma.html#cb126-8" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(rse_gene_SRP045638)[</span>
<span id="cb126-9"><a href="differential-gene-expression-analysis-with-limma.html#cb126-9" tabindex="-1"></a>    ,</span>
<span id="cb126-10"><a href="differential-gene-expression-analysis-with-limma.html#cb126-10" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">&quot;^sra_attribute.[age|disease|RIN|sex]&quot;</span>, <span class="fu">colnames</span>(<span class="fu">colData</span>(rse_gene_SRP045638)))</span>
<span id="cb126-11"><a href="differential-gene-expression-analysis-with-limma.html#cb126-11" tabindex="-1"></a>]))</span>
<span id="cb126-12"><a href="differential-gene-expression-analysis-with-limma.html#cb126-12" tabindex="-1"></a><span class="co">#&gt;  sra_attribute.age sra_attribute.disease sra_attribute.isolate sra_attribute.RIN sra_attribute.sex</span></span>
<span id="cb126-13"><a href="differential-gene-expression-analysis-with-limma.html#cb126-13" tabindex="-1"></a><span class="co">#&gt;  Min.   :-0.4986   control:66            Length:66             Min.   :5.30      female:22        </span></span>
<span id="cb126-14"><a href="differential-gene-expression-analysis-with-limma.html#cb126-14" tabindex="-1"></a><span class="co">#&gt;  1st Qu.: 0.3424                         Class :character      1st Qu.:8.00      male  :44        </span></span>
<span id="cb126-15"><a href="differential-gene-expression-analysis-with-limma.html#cb126-15" tabindex="-1"></a><span class="co">#&gt;  Median :14.9000                         Mode  :character      Median :8.30                       </span></span>
<span id="cb126-16"><a href="differential-gene-expression-analysis-with-limma.html#cb126-16" tabindex="-1"></a><span class="co">#&gt;  Mean   :22.6286                                               Mean   :8.15                       </span></span>
<span id="cb126-17"><a href="differential-gene-expression-analysis-with-limma.html#cb126-17" tabindex="-1"></a><span class="co">#&gt;  3rd Qu.:41.2900                                               3rd Qu.:8.70                       </span></span>
<span id="cb126-18"><a href="differential-gene-expression-analysis-with-limma.html#cb126-18" tabindex="-1"></a><span class="co">#&gt;  Max.   :73.9100                                               Max.   :9.60</span></span></code></pre></div>
<p><a href="https://r4ds.had.co.nz/explore-intro.html"><img src="images/data-science-explore.png" width="800px" /></a></p>
<p>We’ll now create a few variables from this sample information so we can use them in our analysis.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="differential-gene-expression-analysis-with-limma.html#cb127-1" tabindex="-1"></a><span class="do">## We&#39;ll want to look for differences between prenatal and postnatal samples</span></span>
<span id="cb127-2"><a href="differential-gene-expression-analysis-with-limma.html#cb127-2" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>prenatal <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">ifelse</span>(rse_gene_SRP045638<span class="sc">$</span>sra_attribute.age <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="st">&quot;prenatal&quot;</span>, <span class="st">&quot;postnatal&quot;</span>))</span>
<span id="cb127-3"><a href="differential-gene-expression-analysis-with-limma.html#cb127-3" tabindex="-1"></a><span class="fu">table</span>(rse_gene_SRP045638<span class="sc">$</span>prenatal)</span>
<span id="cb127-4"><a href="differential-gene-expression-analysis-with-limma.html#cb127-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb127-5"><a href="differential-gene-expression-analysis-with-limma.html#cb127-5" tabindex="-1"></a><span class="co">#&gt; postnatal  prenatal </span></span>
<span id="cb127-6"><a href="differential-gene-expression-analysis-with-limma.html#cb127-6" tabindex="-1"></a><span class="co">#&gt;        56        10</span></span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="differential-gene-expression-analysis-with-limma.html#cb128-1" tabindex="-1"></a></span>
<span id="cb128-2"><a href="differential-gene-expression-analysis-with-limma.html#cb128-2" tabindex="-1"></a><span class="do">## http://rna.recount.bio/docs/quality-check-fields.html</span></span>
<span id="cb128-3"><a href="differential-gene-expression-analysis-with-limma.html#cb128-3" tabindex="-1"></a>rse_gene_SRP045638<span class="sc">$</span>assigned_gene_prop <span class="ot">&lt;-</span> rse_gene_SRP045638<span class="sc">$</span>recount_qc.gene_fc_count_all.assigned <span class="sc">/</span> rse_gene_SRP045638<span class="sc">$</span>recount_qc.gene_fc_count_all.total</span>
<span id="cb128-4"><a href="differential-gene-expression-analysis-with-limma.html#cb128-4" tabindex="-1"></a><span class="fu">summary</span>(rse_gene_SRP045638<span class="sc">$</span>assigned_gene_prop)</span>
<span id="cb128-5"><a href="differential-gene-expression-analysis-with-limma.html#cb128-5" tabindex="-1"></a><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span id="cb128-6"><a href="differential-gene-expression-analysis-with-limma.html#cb128-6" tabindex="-1"></a><span class="co">#&gt;  0.1942  0.7004  0.7591  0.7170  0.7991  0.8493</span></span></code></pre></div>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="differential-gene-expression-analysis-with-limma.html#cb129-1" tabindex="-1"></a><span class="fu">with</span>(<span class="fu">colData</span>(rse_gene_SRP045638), <span class="fu">plot</span>(assigned_gene_prop, sra_attribute.RIN))</span></code></pre></div>
<p><img src="_main_files/figure-html/new_variables-1.png" width="672" /></p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="differential-gene-expression-analysis-with-limma.html#cb130-1" tabindex="-1"></a></span>
<span id="cb130-2"><a href="differential-gene-expression-analysis-with-limma.html#cb130-2" tabindex="-1"></a><span class="do">## Hm... lets check if there is a difference between these two groups</span></span>
<span id="cb130-3"><a href="differential-gene-expression-analysis-with-limma.html#cb130-3" tabindex="-1"></a><span class="fu">with</span>(<span class="fu">colData</span>(rse_gene_SRP045638), <span class="fu">tapply</span>(assigned_gene_prop, prenatal, summary))</span>
<span id="cb130-4"><a href="differential-gene-expression-analysis-with-limma.html#cb130-4" tabindex="-1"></a><span class="co">#&gt; $postnatal</span></span>
<span id="cb130-5"><a href="differential-gene-expression-analysis-with-limma.html#cb130-5" tabindex="-1"></a><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span id="cb130-6"><a href="differential-gene-expression-analysis-with-limma.html#cb130-6" tabindex="-1"></a><span class="co">#&gt;  0.1942  0.7072  0.7719  0.7179  0.8017  0.8493 </span></span>
<span id="cb130-7"><a href="differential-gene-expression-analysis-with-limma.html#cb130-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb130-8"><a href="differential-gene-expression-analysis-with-limma.html#cb130-8" tabindex="-1"></a><span class="co">#&gt; $prenatal</span></span>
<span id="cb130-9"><a href="differential-gene-expression-analysis-with-limma.html#cb130-9" tabindex="-1"></a><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span id="cb130-10"><a href="differential-gene-expression-analysis-with-limma.html#cb130-10" tabindex="-1"></a><span class="co">#&gt;  0.6856  0.7004  0.7088  0.7116  0.7259  0.7347</span></span></code></pre></div>
<p>We can next drop some samples that we consider of low quality as well as genes that have low expression levels.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="differential-gene-expression-analysis-with-limma.html#cb131-1" tabindex="-1"></a><span class="do">## Lets save our full object for now in case we change our minds later on</span></span>
<span id="cb131-2"><a href="differential-gene-expression-analysis-with-limma.html#cb131-2" tabindex="-1"></a>rse_gene_SRP045638_unfiltered <span class="ot">&lt;-</span> rse_gene_SRP045638</span>
<span id="cb131-3"><a href="differential-gene-expression-analysis-with-limma.html#cb131-3" tabindex="-1"></a></span>
<span id="cb131-4"><a href="differential-gene-expression-analysis-with-limma.html#cb131-4" tabindex="-1"></a><span class="do">## Lets drop some bad samples. On a real analysis, you would likely use</span></span>
<span id="cb131-5"><a href="differential-gene-expression-analysis-with-limma.html#cb131-5" tabindex="-1"></a><span class="do">## some statistical method for identifying outliers such as scuttle::isOutlier()</span></span>
<span id="cb131-6"><a href="differential-gene-expression-analysis-with-limma.html#cb131-6" tabindex="-1"></a><span class="fu">hist</span>(rse_gene_SRP045638<span class="sc">$</span>assigned_gene_prop)</span></code></pre></div>
<p><img src="_main_files/figure-html/filter_rse-1.png" width="672" /></p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="differential-gene-expression-analysis-with-limma.html#cb132-1" tabindex="-1"></a><span class="fu">table</span>(rse_gene_SRP045638<span class="sc">$</span>assigned_gene_prop <span class="sc">&lt;</span> <span class="fl">0.3</span>)</span>
<span id="cb132-2"><a href="differential-gene-expression-analysis-with-limma.html#cb132-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb132-3"><a href="differential-gene-expression-analysis-with-limma.html#cb132-3" tabindex="-1"></a><span class="co">#&gt; FALSE  TRUE </span></span>
<span id="cb132-4"><a href="differential-gene-expression-analysis-with-limma.html#cb132-4" tabindex="-1"></a><span class="co">#&gt;    65     1</span></span></code></pre></div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="differential-gene-expression-analysis-with-limma.html#cb133-1" tabindex="-1"></a>rse_gene_SRP045638 <span class="ot">&lt;-</span> rse_gene_SRP045638[, rse_gene_SRP045638<span class="sc">$</span>assigned_gene_prop <span class="sc">&gt;</span> <span class="fl">0.3</span>]</span>
<span id="cb133-2"><a href="differential-gene-expression-analysis-with-limma.html#cb133-2" tabindex="-1"></a></span>
<span id="cb133-3"><a href="differential-gene-expression-analysis-with-limma.html#cb133-3" tabindex="-1"></a><span class="do">## Lets compute the mean expression levels.</span></span>
<span id="cb133-4"><a href="differential-gene-expression-analysis-with-limma.html#cb133-4" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb133-5"><a href="differential-gene-expression-analysis-with-limma.html#cb133-5" tabindex="-1"></a><span class="do">## Note: in a real analysis we would likely do this with RPKMs or CPMs instead</span></span>
<span id="cb133-6"><a href="differential-gene-expression-analysis-with-limma.html#cb133-6" tabindex="-1"></a><span class="do">## of counts. That is, we would use one of the following options:</span></span>
<span id="cb133-7"><a href="differential-gene-expression-analysis-with-limma.html#cb133-7" tabindex="-1"></a><span class="co"># edgeR::filterByExpr() https://bioconductor.org/packages/edgeR/ https://rdrr.io/bioc/edgeR/man/filterByExpr.html</span></span>
<span id="cb133-8"><a href="differential-gene-expression-analysis-with-limma.html#cb133-8" tabindex="-1"></a><span class="co"># genefilter::genefilter() https://bioconductor.org/packages/genefilter/ https://rdrr.io/bioc/genefilter/man/genefilter.html</span></span>
<span id="cb133-9"><a href="differential-gene-expression-analysis-with-limma.html#cb133-9" tabindex="-1"></a><span class="co"># jaffelab::expression_cutoff() http://research.libd.org/jaffelab/reference/expression_cutoff.html</span></span>
<span id="cb133-10"><a href="differential-gene-expression-analysis-with-limma.html#cb133-10" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb133-11"><a href="differential-gene-expression-analysis-with-limma.html#cb133-11" tabindex="-1"></a>gene_means <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(rse_gene_SRP045638, <span class="st">&quot;counts&quot;</span>))</span>
<span id="cb133-12"><a href="differential-gene-expression-analysis-with-limma.html#cb133-12" tabindex="-1"></a><span class="fu">summary</span>(gene_means)</span>
<span id="cb133-13"><a href="differential-gene-expression-analysis-with-limma.html#cb133-13" tabindex="-1"></a><span class="co">#&gt;      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. </span></span>
<span id="cb133-14"><a href="differential-gene-expression-analysis-with-limma.html#cb133-14" tabindex="-1"></a><span class="co">#&gt;       0.0       0.1       2.5     815.2     170.2 1362047.9</span></span></code></pre></div>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="differential-gene-expression-analysis-with-limma.html#cb134-1" tabindex="-1"></a></span>
<span id="cb134-2"><a href="differential-gene-expression-analysis-with-limma.html#cb134-2" tabindex="-1"></a><span class="do">## We can now drop genes with low expression levels</span></span>
<span id="cb134-3"><a href="differential-gene-expression-analysis-with-limma.html#cb134-3" tabindex="-1"></a>rse_gene_SRP045638 <span class="ot">&lt;-</span> rse_gene_SRP045638[gene_means <span class="sc">&gt;</span> <span class="fl">0.1</span>, ]</span>
<span id="cb134-4"><a href="differential-gene-expression-analysis-with-limma.html#cb134-4" tabindex="-1"></a></span>
<span id="cb134-5"><a href="differential-gene-expression-analysis-with-limma.html#cb134-5" tabindex="-1"></a><span class="do">## Final dimensions of our RSE object</span></span>
<span id="cb134-6"><a href="differential-gene-expression-analysis-with-limma.html#cb134-6" tabindex="-1"></a><span class="fu">dim</span>(rse_gene_SRP045638)</span>
<span id="cb134-7"><a href="differential-gene-expression-analysis-with-limma.html#cb134-7" tabindex="-1"></a><span class="co">#&gt; [1] 46929    65</span></span></code></pre></div>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="differential-gene-expression-analysis-with-limma.html#cb135-1" tabindex="-1"></a></span>
<span id="cb135-2"><a href="differential-gene-expression-analysis-with-limma.html#cb135-2" tabindex="-1"></a><span class="do">## Percent of genes that we retained:</span></span>
<span id="cb135-3"><a href="differential-gene-expression-analysis-with-limma.html#cb135-3" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">nrow</span>(rse_gene_SRP045638) <span class="sc">/</span> <span class="fu">nrow</span>(rse_gene_SRP045638_unfiltered) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb135-4"><a href="differential-gene-expression-analysis-with-limma.html#cb135-4" tabindex="-1"></a><span class="co">#&gt; [1] 73.49</span></span></code></pre></div>
<p>We are now ready to continue with the differential expression analysis. Well, almost! 😅</p>
</div>
<div id="data-normalization-1" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Data normalization<a href="differential-gene-expression-analysis-with-limma.html#data-normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Read the <em>A hypothetical scenario</em> in one of the <code>edgeR</code> papers <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25#Sec2" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25#Sec2</a> to understand the concept of <em>composition bias</em>.</li>
<li>This concept is still relevant nowadays with single cell RNA-seq (scRNA-seq) data as you can see at <a href="http://bioconductor.org/books/3.16/OSCA.multisample/multi-sample-comparisons.html#performing-the-de-analysis" class="uri">http://bioconductor.org/books/3.16/OSCA.multisample/multi-sample-comparisons.html#performing-the-de-analysis</a>. In that chapter they describe a series of steps for re-using bulk RNA-seq methods with scRNA-seq data.</li>
</ul>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="differential-gene-expression-analysis-with-limma.html#cb136-1" tabindex="-1"></a><span class="do">## Use edgeR::calcNormFactors() to address the composition bias</span></span>
<span id="cb136-2"><a href="differential-gene-expression-analysis-with-limma.html#cb136-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;edgeR&quot;</span>)</span>
<span id="cb136-3"><a href="differential-gene-expression-analysis-with-limma.html#cb136-3" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(</span>
<span id="cb136-4"><a href="differential-gene-expression-analysis-with-limma.html#cb136-4" tabindex="-1"></a>    <span class="at">counts =</span> <span class="fu">assay</span>(rse_gene_SRP045638, <span class="st">&quot;counts&quot;</span>),</span>
<span id="cb136-5"><a href="differential-gene-expression-analysis-with-limma.html#cb136-5" tabindex="-1"></a>    <span class="at">genes =</span> <span class="fu">rowData</span>(rse_gene_SRP045638)</span>
<span id="cb136-6"><a href="differential-gene-expression-analysis-with-limma.html#cb136-6" tabindex="-1"></a>)</span>
<span id="cb136-7"><a href="differential-gene-expression-analysis-with-limma.html#cb136-7" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span></code></pre></div>
</div>
<div id="differential-expression" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Differential expression<a href="differential-gene-expression-analysis-with-limma.html#differential-expression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First of all, lets define our differential expression model. Typically, we would explore the data more to check that there are no other quality control issues with our samples and to explore in more detail the relationship between our sample phenotype variables.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="differential-gene-expression-analysis-with-limma.html#cb137-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb137-2"><a href="differential-gene-expression-analysis-with-limma.html#cb137-2" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(rse_gene_SRP045638)), <span class="fu">aes</span>(<span class="at">y =</span> assigned_gene_prop, <span class="at">x =</span> prenatal)) <span class="sc">+</span></span>
<span id="cb137-3"><a href="differential-gene-expression-analysis-with-limma.html#cb137-3" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb137-4"><a href="differential-gene-expression-analysis-with-limma.html#cb137-4" tabindex="-1"></a>    <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb137-5"><a href="differential-gene-expression-analysis-with-limma.html#cb137-5" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Assigned Gene Prop&quot;</span>) <span class="sc">+</span></span>
<span id="cb137-6"><a href="differential-gene-expression-analysis-with-limma.html#cb137-6" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Age Group&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/explore_gene_prop_by_age-1.png" width="672" /></p>
<p>For example, we would explore the contribution of different variables to the gene expression variability we observe using the <a href="https://bioconductor.org/packages/variancePartition"><code>variancePartition</code></a> and <a href="https://bioconductor.org/packages/scater"><code>scater</code></a> Bioconductor packages, among others. We’ll do more of this tomorrow but you can also check these <a href="https://docs.google.com/document/d/1hil3zwPN6BW6HlwldLbM1FdlLIBWKFNXRUqJJEK_-eY/edit">LIBD rstats club notes</a>.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/OdNU5LUOHng" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p>For now, we’ll use the following stastistical model.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="differential-gene-expression-analysis-with-limma.html#cb138-1" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> prenatal <span class="sc">+</span> sra_attribute.RIN <span class="sc">+</span> sra_attribute.sex <span class="sc">+</span> assigned_gene_prop,</span>
<span id="cb138-2"><a href="differential-gene-expression-analysis-with-limma.html#cb138-2" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">colData</span>(rse_gene_SRP045638)</span>
<span id="cb138-3"><a href="differential-gene-expression-analysis-with-limma.html#cb138-3" tabindex="-1"></a>)</span>
<span id="cb138-4"><a href="differential-gene-expression-analysis-with-limma.html#cb138-4" tabindex="-1"></a><span class="fu">colnames</span>(mod)</span>
<span id="cb138-5"><a href="differential-gene-expression-analysis-with-limma.html#cb138-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;(Intercept)&quot;           &quot;prenatalprenatal&quot;      &quot;sra_attribute.RIN&quot;     &quot;sra_attribute.sexmale&quot;</span></span>
<span id="cb138-6"><a href="differential-gene-expression-analysis-with-limma.html#cb138-6" tabindex="-1"></a><span class="co">#&gt; [5] &quot;assigned_gene_prop&quot;</span></span></code></pre></div>
<p>Now that we have a model, we can use <code>limma</code> to actually compute the differential expression statistics and extract the results.</p>
<script defer class="speakerdeck-embed" data-slide="57" data-id="3c32410b600740abb4724486e83ebd30" data-ratio="1.77725118483412" src="//speakerdeck.com/assets/embed.js"></script>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="differential-gene-expression-analysis-with-limma.html#cb139-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;limma&quot;</span>)</span>
<span id="cb139-2"><a href="differential-gene-expression-analysis-with-limma.html#cb139-2" tabindex="-1"></a>vGene <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, mod, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/run_limma-1.png" width="672" /></p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="differential-gene-expression-analysis-with-limma.html#cb140-1" tabindex="-1"></a></span>
<span id="cb140-2"><a href="differential-gene-expression-analysis-with-limma.html#cb140-2" tabindex="-1"></a>eb_results <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(<span class="fu">lmFit</span>(vGene))</span>
<span id="cb140-3"><a href="differential-gene-expression-analysis-with-limma.html#cb140-3" tabindex="-1"></a></span>
<span id="cb140-4"><a href="differential-gene-expression-analysis-with-limma.html#cb140-4" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">topTable</span>(</span>
<span id="cb140-5"><a href="differential-gene-expression-analysis-with-limma.html#cb140-5" tabindex="-1"></a>    eb_results,</span>
<span id="cb140-6"><a href="differential-gene-expression-analysis-with-limma.html#cb140-6" tabindex="-1"></a>    <span class="at">coef =</span> <span class="dv">2</span>,</span>
<span id="cb140-7"><a href="differential-gene-expression-analysis-with-limma.html#cb140-7" tabindex="-1"></a>    <span class="at">number =</span> <span class="fu">nrow</span>(rse_gene_SRP045638),</span>
<span id="cb140-8"><a href="differential-gene-expression-analysis-with-limma.html#cb140-8" tabindex="-1"></a>    <span class="at">sort.by =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb140-9"><a href="differential-gene-expression-analysis-with-limma.html#cb140-9" tabindex="-1"></a>)</span>
<span id="cb140-10"><a href="differential-gene-expression-analysis-with-limma.html#cb140-10" tabindex="-1"></a><span class="fu">dim</span>(de_results)</span>
<span id="cb140-11"><a href="differential-gene-expression-analysis-with-limma.html#cb140-11" tabindex="-1"></a><span class="co">#&gt; [1] 46929    16</span></span></code></pre></div>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="differential-gene-expression-analysis-with-limma.html#cb141-1" tabindex="-1"></a><span class="fu">head</span>(de_results)</span>
<span id="cb141-2"><a href="differential-gene-expression-analysis-with-limma.html#cb141-2" tabindex="-1"></a><span class="co">#&gt;                    source type bp_length phase           gene_id                          gene_type gene_name level</span></span>
<span id="cb141-3"><a href="differential-gene-expression-analysis-with-limma.html#cb141-3" tabindex="-1"></a><span class="co">#&gt; ENSG00000223972.5  HAVANA gene      1735    NA ENSG00000223972.5 transcribed_unprocessed_pseudogene   DDX11L1     2</span></span>
<span id="cb141-4"><a href="differential-gene-expression-analysis-with-limma.html#cb141-4" tabindex="-1"></a><span class="co">#&gt; ENSG00000278267.1 ENSEMBL gene        68    NA ENSG00000278267.1                              miRNA MIR6859-1     3</span></span>
<span id="cb141-5"><a href="differential-gene-expression-analysis-with-limma.html#cb141-5" tabindex="-1"></a><span class="co">#&gt; ENSG00000227232.5  HAVANA gene      1351    NA ENSG00000227232.5             unprocessed_pseudogene    WASH7P     2</span></span>
<span id="cb141-6"><a href="differential-gene-expression-analysis-with-limma.html#cb141-6" tabindex="-1"></a><span class="co">#&gt;                            havana_gene  tag      logFC   AveExpr         t      P.Value    adj.P.Val         B</span></span>
<span id="cb141-7"><a href="differential-gene-expression-analysis-with-limma.html#cb141-7" tabindex="-1"></a><span class="co">#&gt; ENSG00000223972.5 OTTHUMG00000000961.2 &lt;NA&gt; -0.4522185 -3.444164 -1.637034 1.064080e-01 1.348462e-01 -5.554484</span></span>
<span id="cb141-8"><a href="differential-gene-expression-analysis-with-limma.html#cb141-8" tabindex="-1"></a><span class="co">#&gt; ENSG00000278267.1                 &lt;NA&gt; &lt;NA&gt;  1.1158860 -1.311645  6.123756 5.743771e-08 2.105690e-07  7.763912</span></span>
<span id="cb141-9"><a href="differential-gene-expression-analysis-with-limma.html#cb141-9" tabindex="-1"></a><span class="co">#&gt; ENSG00000227232.5 OTTHUMG00000000958.1 &lt;NA&gt;  0.6860298  3.641131  5.644804 3.816951e-07 1.247133e-06  5.500784</span></span>
<span id="cb141-10"><a href="differential-gene-expression-analysis-with-limma.html#cb141-10" tabindex="-1"></a><span class="co">#&gt;  [ reached &#39;max&#39; / getOption(&quot;max.print&quot;) -- omitted 3 rows ]</span></span></code></pre></div>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="differential-gene-expression-analysis-with-limma.html#cb142-1" tabindex="-1"></a></span>
<span id="cb142-2"><a href="differential-gene-expression-analysis-with-limma.html#cb142-2" tabindex="-1"></a><span class="do">## Differentially expressed genes between pre and post natal with FDR &lt; 5%</span></span>
<span id="cb142-3"><a href="differential-gene-expression-analysis-with-limma.html#cb142-3" tabindex="-1"></a><span class="fu">table</span>(de_results<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb142-4"><a href="differential-gene-expression-analysis-with-limma.html#cb142-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb142-5"><a href="differential-gene-expression-analysis-with-limma.html#cb142-5" tabindex="-1"></a><span class="co">#&gt; FALSE  TRUE </span></span>
<span id="cb142-6"><a href="differential-gene-expression-analysis-with-limma.html#cb142-6" tabindex="-1"></a><span class="co">#&gt; 12851 34078</span></span></code></pre></div>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="differential-gene-expression-analysis-with-limma.html#cb143-1" tabindex="-1"></a></span>
<span id="cb143-2"><a href="differential-gene-expression-analysis-with-limma.html#cb143-2" tabindex="-1"></a><span class="do">## We can now visualize the resulting differential expression results</span></span>
<span id="cb143-3"><a href="differential-gene-expression-analysis-with-limma.html#cb143-3" tabindex="-1"></a><span class="fu">plotMA</span>(eb_results, <span class="at">coef =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/run_limma-2.png" width="672" /></p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="differential-gene-expression-analysis-with-limma.html#cb144-1" tabindex="-1"></a></span>
<span id="cb144-2"><a href="differential-gene-expression-analysis-with-limma.html#cb144-2" tabindex="-1"></a><span class="do">## We can also make a volcano plot</span></span>
<span id="cb144-3"><a href="differential-gene-expression-analysis-with-limma.html#cb144-3" tabindex="-1"></a><span class="fu">volcanoplot</span>(eb_results, <span class="at">coef =</span> <span class="dv">2</span>, <span class="at">highlight =</span> <span class="dv">3</span>, <span class="at">names =</span> de_results<span class="sc">$</span>gene_name)</span></code></pre></div>
<p><img src="_main_files/figure-html/run_limma-3.png" width="672" /></p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="differential-gene-expression-analysis-with-limma.html#cb145-1" tabindex="-1"></a>de_results[de_results<span class="sc">$</span>gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;ZSCAN2&quot;</span>, <span class="st">&quot;VASH2&quot;</span>, <span class="st">&quot;KIAA0922&quot;</span>), ]</span>
<span id="cb145-2"><a href="differential-gene-expression-analysis-with-limma.html#cb145-2" tabindex="-1"></a><span class="co">#&gt;                    source type bp_length phase            gene_id      gene_type gene_name level          havana_gene</span></span>
<span id="cb145-3"><a href="differential-gene-expression-analysis-with-limma.html#cb145-3" tabindex="-1"></a><span class="co">#&gt; ENSG00000143494.15 HAVANA gene      9086    NA ENSG00000143494.15 protein_coding     VASH2     2 OTTHUMG00000036925.5</span></span>
<span id="cb145-4"><a href="differential-gene-expression-analysis-with-limma.html#cb145-4" tabindex="-1"></a><span class="co">#&gt; ENSG00000176371.13 HAVANA gene      4878    NA ENSG00000176371.13 protein_coding    ZSCAN2     1 OTTHUMG00000074027.5</span></span>
<span id="cb145-5"><a href="differential-gene-expression-analysis-with-limma.html#cb145-5" tabindex="-1"></a><span class="co">#&gt; ENSG00000121210.15 HAVANA gene      6393    NA ENSG00000121210.15 protein_coding  KIAA0922     2 OTTHUMG00000153244.5</span></span>
<span id="cb145-6"><a href="differential-gene-expression-analysis-with-limma.html#cb145-6" tabindex="-1"></a><span class="co">#&gt;                     tag    logFC  AveExpr        t      P.Value    adj.P.Val         B</span></span>
<span id="cb145-7"><a href="differential-gene-expression-analysis-with-limma.html#cb145-7" tabindex="-1"></a><span class="co">#&gt; ENSG00000143494.15 &lt;NA&gt; 5.444271 1.876990 37.85216 2.561452e-46 6.010319e-42  95.10126</span></span>
<span id="cb145-8"><a href="differential-gene-expression-analysis-with-limma.html#cb145-8" tabindex="-1"></a><span class="co">#&gt; ENSG00000176371.13 &lt;NA&gt; 2.735639 2.751109 36.59344 2.137022e-45 3.342943e-41  92.99949</span></span>
<span id="cb145-9"><a href="differential-gene-expression-analysis-with-limma.html#cb145-9" tabindex="-1"></a><span class="co">#&gt; ENSG00000121210.15 &lt;NA&gt; 3.283050 2.945269 41.89725 4.235912e-49 1.987871e-44 101.31124</span></span></code></pre></div>
<ul>
<li><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=ZSCAN2" class="uri">https://www.genecards.org/cgi-bin/carddisp.pl?gene=ZSCAN2</a></li>
<li><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=VASH2" class="uri">https://www.genecards.org/cgi-bin/carddisp.pl?gene=VASH2</a></li>
<li><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=KIAA0922" class="uri">https://www.genecards.org/cgi-bin/carddisp.pl?gene=KIAA0922</a></li>
</ul>
</div>
<div id="visualizing-degs" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Visualizing DEGs<a href="differential-gene-expression-analysis-with-limma.html#visualizing-degs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>From <code>vGene$E</code> we can extract the normalized expression values that <code>limma-voom</code> computed. We can check the top 50 differentially expressed genes (DEGs) for example.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="differential-gene-expression-analysis-with-limma.html#cb146-1" tabindex="-1"></a><span class="do">## Extract the normalized expression values from our limma-voom result</span></span>
<span id="cb146-2"><a href="differential-gene-expression-analysis-with-limma.html#cb146-2" tabindex="-1"></a><span class="do">## from earlier</span></span>
<span id="cb146-3"><a href="differential-gene-expression-analysis-with-limma.html#cb146-3" tabindex="-1"></a>exprs_heatmap <span class="ot">&lt;-</span> vGene<span class="sc">$</span>E[<span class="fu">rank</span>(de_results<span class="sc">$</span>adj.P.Val) <span class="sc">&lt;=</span> <span class="dv">50</span>, ]</span>
<span id="cb146-4"><a href="differential-gene-expression-analysis-with-limma.html#cb146-4" tabindex="-1"></a></span>
<span id="cb146-5"><a href="differential-gene-expression-analysis-with-limma.html#cb146-5" tabindex="-1"></a><span class="do">## We can now build a table with information about our samples and</span></span>
<span id="cb146-6"><a href="differential-gene-expression-analysis-with-limma.html#cb146-6" tabindex="-1"></a><span class="do">## then make the names a bit more friendly by making them easier to</span></span>
<span id="cb146-7"><a href="differential-gene-expression-analysis-with-limma.html#cb146-7" tabindex="-1"></a><span class="do">## understand</span></span>
<span id="cb146-8"><a href="differential-gene-expression-analysis-with-limma.html#cb146-8" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(rse_gene_SRP045638)[, <span class="fu">c</span>(<span class="st">&quot;prenatal&quot;</span>, <span class="st">&quot;sra_attribute.RIN&quot;</span>, <span class="st">&quot;sra_attribute.sex&quot;</span>)])</span>
<span id="cb146-9"><a href="differential-gene-expression-analysis-with-limma.html#cb146-9" tabindex="-1"></a><span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AgeGroup&quot;</span>, <span class="st">&quot;RIN&quot;</span>, <span class="st">&quot;Sex&quot;</span>)</span>
<span id="cb146-10"><a href="differential-gene-expression-analysis-with-limma.html#cb146-10" tabindex="-1"></a></span>
<span id="cb146-11"><a href="differential-gene-expression-analysis-with-limma.html#cb146-11" tabindex="-1"></a><span class="do">## Next, we can make a basic heatmap</span></span>
<span id="cb146-12"><a href="differential-gene-expression-analysis-with-limma.html#cb146-12" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb146-13"><a href="differential-gene-expression-analysis-with-limma.html#cb146-13" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb146-14"><a href="differential-gene-expression-analysis-with-limma.html#cb146-14" tabindex="-1"></a>    exprs_heatmap,</span>
<span id="cb146-15"><a href="differential-gene-expression-analysis-with-limma.html#cb146-15" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb146-16"><a href="differential-gene-expression-analysis-with-limma.html#cb146-16" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb146-17"><a href="differential-gene-expression-analysis-with-limma.html#cb146-17" tabindex="-1"></a>    <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb146-18"><a href="differential-gene-expression-analysis-with-limma.html#cb146-18" tabindex="-1"></a>    <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb146-19"><a href="differential-gene-expression-analysis-with-limma.html#cb146-19" tabindex="-1"></a>    <span class="at">annotation_col =</span> df</span>
<span id="cb146-20"><a href="differential-gene-expression-analysis-with-limma.html#cb146-20" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/pheatmap-1.png" width="672" /></p>
<p>We’ll learn more about gene expression heatmaps tomorrow!</p>
<p>Overall, these DEG results are not as surprising since there is a huge difference between pre and post natal gene expression in the human DLPFC. We can see that more clearly with a MDS (multidimensional scaling) plot just like its described <a href="http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#unsupervised-clustering-of-samples">in the limma workflow</a>. Tomas will teach you more about dimension reduction tomorrow.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="differential-gene-expression-analysis-with-limma.html#cb147-1" tabindex="-1"></a><span class="do">## For nicer colors</span></span>
<span id="cb147-2"><a href="differential-gene-expression-analysis-with-limma.html#cb147-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;RColorBrewer&quot;</span>)</span>
<span id="cb147-3"><a href="differential-gene-expression-analysis-with-limma.html#cb147-3" tabindex="-1"></a></span>
<span id="cb147-4"><a href="differential-gene-expression-analysis-with-limma.html#cb147-4" tabindex="-1"></a><span class="do">## Mapping age groups into colors</span></span>
<span id="cb147-5"><a href="differential-gene-expression-analysis-with-limma.html#cb147-5" tabindex="-1"></a>col.group <span class="ot">&lt;-</span> df<span class="sc">$</span>AgeGroup</span>
<span id="cb147-6"><a href="differential-gene-expression-analysis-with-limma.html#cb147-6" tabindex="-1"></a><span class="fu">levels</span>(col.group) <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="fu">nlevels</span>(col.group), <span class="st">&quot;Set1&quot;</span>)</span>
<span id="cb147-7"><a href="differential-gene-expression-analysis-with-limma.html#cb147-7" tabindex="-1"></a><span class="co">#&gt; Warning in brewer.pal(nlevels(col.group), &quot;Set1&quot;): minimal value for n is 3, returning requested palette with 3 different levels</span></span></code></pre></div>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="differential-gene-expression-analysis-with-limma.html#cb148-1" tabindex="-1"></a>col.group <span class="ot">&lt;-</span> <span class="fu">as.character</span>(col.group)</span>
<span id="cb148-2"><a href="differential-gene-expression-analysis-with-limma.html#cb148-2" tabindex="-1"></a></span>
<span id="cb148-3"><a href="differential-gene-expression-analysis-with-limma.html#cb148-3" tabindex="-1"></a><span class="do">## MDS by age groups</span></span>
<span id="cb148-4"><a href="differential-gene-expression-analysis-with-limma.html#cb148-4" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotMDS</span>(vGene<span class="sc">$</span>E, <span class="at">labels =</span> df<span class="sc">$</span>AgeGroup, <span class="at">col =</span> col.group)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_mds-1.png" width="672" /></p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="differential-gene-expression-analysis-with-limma.html#cb149-1" tabindex="-1"></a></span>
<span id="cb149-2"><a href="differential-gene-expression-analysis-with-limma.html#cb149-2" tabindex="-1"></a><span class="do">## Mapping Sex values into colors</span></span>
<span id="cb149-3"><a href="differential-gene-expression-analysis-with-limma.html#cb149-3" tabindex="-1"></a>col.sex <span class="ot">&lt;-</span> df<span class="sc">$</span>Sex</span>
<span id="cb149-4"><a href="differential-gene-expression-analysis-with-limma.html#cb149-4" tabindex="-1"></a><span class="fu">levels</span>(col.sex) <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="fu">nlevels</span>(col.sex), <span class="st">&quot;Dark2&quot;</span>)</span>
<span id="cb149-5"><a href="differential-gene-expression-analysis-with-limma.html#cb149-5" tabindex="-1"></a><span class="co">#&gt; Warning in brewer.pal(nlevels(col.sex), &quot;Dark2&quot;): minimal value for n is 3, returning requested palette with 3 different levels</span></span></code></pre></div>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="differential-gene-expression-analysis-with-limma.html#cb150-1" tabindex="-1"></a>col.sex <span class="ot">&lt;-</span> <span class="fu">as.character</span>(col.sex)</span>
<span id="cb150-2"><a href="differential-gene-expression-analysis-with-limma.html#cb150-2" tabindex="-1"></a></span>
<span id="cb150-3"><a href="differential-gene-expression-analysis-with-limma.html#cb150-3" tabindex="-1"></a><span class="do">## MDS by Sex</span></span>
<span id="cb150-4"><a href="differential-gene-expression-analysis-with-limma.html#cb150-4" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotMDS</span>(vGene<span class="sc">$</span>E, <span class="at">labels =</span> df<span class="sc">$</span>Sex, <span class="at">col =</span> col.sex)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_mds-2.png" width="672" /></p>
<p>A lot of times, running the differential expression analysis is not the hard part. Building your model and having all the covariates you need can take much more work!</p>
<script defer class="speakerdeck-embed" data-slide="56" data-id="3c32410b600740abb4724486e83ebd30" data-ratio="1.77725118483412" src="//speakerdeck.com/assets/embed.js"></script>
</div>
<div id="community-4" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Community<a href="differential-gene-expression-analysis-with-limma.html#community-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Some <code>edgeR</code> and <code>limma</code> authors:</p>
<ul>
<li><a href="https://twitter.com/mritchieau" class="uri">https://twitter.com/mritchieau</a></li>
<li><a href="https://twitter.com/davisjmcc" class="uri">https://twitter.com/davisjmcc</a></li>
<li><a href="https://twitter.com/markrobinsonca" class="uri">https://twitter.com/markrobinsonca</a></li>
<li><a href="https://twitter.com/AliciaOshlack" class="uri">https://twitter.com/AliciaOshlack</a></li>
</ul>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
If you've ever been dazed by design matrices or confused by contrasts when performing gene expression analysis in limma, the new article by Charity Law is for you <a href="https://t.co/ZSMOA20tdm">https://t.co/ZSMOA20tdm</a> <a href="https://twitter.com/hashtag/bioconductor?src=hash&amp;ref_src=twsrc%5Etfw">#bioconductor</a> <a href="https://twitter.com/hashtag/rstats?src=hash&amp;ref_src=twsrc%5Etfw">#rstats</a> (1/2)
</p>
— Matt Ritchie (<span class="citation">@mritchieau</span>) <a href="https://twitter.com/mritchieau/status/1338639551128952832?ref_src=twsrc%5Etfw">December 15, 2020</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="interpreting-model-coefficients-with-exploremodelmatrix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-gene-expression-exercise.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
