<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Differential Gene Expression analysis with limma-voom | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="5 Differential Gene Expression analysis with limma-voom | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Differential Gene Expression analysis with limma-voom | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Differential Gene Expression analysis with limma-voom | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="differential-gene-expression-analysis-overview.html"/>
<link rel="next" href="interpreting-model-coefficients-with-exploremodelmatrix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.2</b> Exercises</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.3</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
<li class="chapter" data-level="2.1.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment"><i class="fa fa-check"></i><b>2.1.3</b> SpatialExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
<li class="chapter" data-level="2.4.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#explore-the-data"><i class="fa fa-check"></i><b>2.4.2</b> Explore the Data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features"><i class="fa fa-check"></i><b>2.5</b> Introduction to Advanced iSEE Features</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression"><i class="fa fa-check"></i><b>2.5.1</b> Selecting Cells Based on a Single Gene Expression</a></li>
<li class="chapter" data-level="2.5.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression"><i class="fa fa-check"></i><b>2.5.2</b> Using a Single Plot for Two Gene Co-Expression</a></li>
<li class="chapter" data-level="2.5.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes"><i class="fa fa-check"></i><b>2.5.3</b> Selecting Cells Based on the Co-Expression of Two or more Genes</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.6</b> References</a></li>
<li class="chapter" data-level="2.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#community"><i class="fa fa-check"></i><b>2.7</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-1"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-2"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>9</b> Research talks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>9.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="9.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>9.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="9.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>9.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>10</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>10.1</b> Related past materials</a></li>
<li class="chapter" data-level="10.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>10.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="10.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>10.3</b> Live demo</a></li>
<li class="chapter" data-level="10.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-3"><i class="fa fa-check"></i><b>10.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>11</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="11.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing"><i class="fa fa-check"></i><b>11.1</b> Single cell RNA sequencing</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#pre-processing-of-scrna-seq-data-before-r"><i class="fa fa-check"></i><b>11.1.1</b> Pre-processing of scRNA-seq Data (Before R)</a></li>
<li class="chapter" data-level="11.1.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#different-technologies"><i class="fa fa-check"></i><b>11.1.2</b> Different Technologies</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basic-workflow"><i class="fa fa-check"></i><b>11.2</b> Basic Workflow</a></li>
<li class="chapter" data-level="11.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>11.3</b> The SingleCellExperiment class</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#data-loading"><i class="fa fa-check"></i><b>11.3.1</b> Data Loading</a></li>
<li class="chapter" data-level="11.3.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basics-of-your-sce"><i class="fa fa-check"></i><b>11.3.2</b> Basics of your SCE</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>11.4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#common-choices-of-qc-metrics"><i class="fa fa-check"></i><b>11.4.1</b> Common choices of QC metrics</a></li>
<li class="chapter" data-level="11.4.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#identifying-low-quality-cells"><i class="fa fa-check"></i><b>11.4.2</b> Identifying low-quality cells</a></li>
<li class="chapter" data-level="11.4.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#checking-diagnostic-plots"><i class="fa fa-check"></i><b>11.4.3</b> Checking diagnostic plots</a></li>
<li class="chapter" data-level="11.4.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#removing-low-quality-cells"><i class="fa fa-check"></i><b>11.4.4</b> Removing low-quality cells</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>11.5</b> Normalization</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#library-size-normalization"><i class="fa fa-check"></i><b>11.5.1</b> Library size normalization</a></li>
<li class="chapter" data-level="11.5.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-deconvolution"><i class="fa fa-check"></i><b>11.5.2</b> Normalization by deconvolution</a></li>
<li class="chapter" data-level="11.5.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-spike-ins"><i class="fa fa-check"></i><b>11.5.3</b> Normalization by spike-ins</a></li>
<li class="chapter" data-level="11.5.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#scaling-and-log-transforming"><i class="fa fa-check"></i><b>11.5.4</b> Scaling and log-transforming</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>11.6</b> Feature selection</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-per-gene-variation"><i class="fa fa-check"></i><b>11.6.1</b> Quantifying per-gene variation</a></li>
<li class="chapter" data-level="11.6.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-spike-ins"><i class="fa fa-check"></i><b>11.6.2</b> Quantifying technical noise (spike-ins)</a></li>
<li class="chapter" data-level="11.6.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-mean-variance-trend"><i class="fa fa-check"></i><b>11.6.3</b> Quantifying technical noise (mean-variance trend)</a></li>
<li class="chapter" data-level="11.6.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#handling-batch-effects"><i class="fa fa-check"></i><b>11.6.4</b> Handling batch effects</a></li>
<li class="chapter" data-level="11.6.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#selecting-highly-variable-genes"><i class="fa fa-check"></i><b>11.6.5</b> Selecting highly variable genes</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>11.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#principal-components-analysis"><i class="fa fa-check"></i><b>11.7.1</b> Principal components analysis</a></li>
<li class="chapter" data-level="11.7.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#choosing-the-number-of-pcs"><i class="fa fa-check"></i><b>11.7.2</b> Choosing the number of PCs</a></li>
<li class="chapter" data-level="11.7.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#visualizing-the-pcs"><i class="fa fa-check"></i><b>11.7.3</b> Visualizing the PCs</a></li>
<li class="chapter" data-level="11.7.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#non-linear-methods-for-visualization"><i class="fa fa-check"></i><b>11.7.4</b> Non-linear methods for visualization</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>11.8</b> Clustering</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#graph-based-clustering"><i class="fa fa-check"></i><b>11.8.1</b> Graph-based clustering</a></li>
<li class="chapter" data-level="11.8.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#adjusting-the-parameters"><i class="fa fa-check"></i><b>11.8.2</b> Adjusting the parameters</a></li>
<li class="chapter" data-level="11.8.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#hierarchical-clustering"><i class="fa fa-check"></i><b>11.8.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="11.8.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#subclustering"><i class="fa fa-check"></i><b>11.8.4</b> Subclustering</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>11.9</b> Marker gene detection</a></li>
<li class="chapter" data-level="11.10" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>11.10</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>12</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>12.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>12.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>12.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>12.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>13</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="13.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>13.1</b> Spatial registration</a></li>
<li class="chapter" data-level="13.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>13.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>14</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="14.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>14.1</b> here</a></li>
<li class="chapter" data-level="14.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>14.2</b> Usethis</a></li>
<li class="chapter" data-level="14.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>14.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>14.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="14.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>14.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="14.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>14.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="14.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>14.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>14.4</b> R websites</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>14.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="14.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>14.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="14.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>14.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="14.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>14.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>14.5</b> postcards</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>14.5.1</b> Installation</a></li>
<li class="chapter" data-level="14.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>14.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>14.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>14.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="14.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>14.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="14.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>14.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="14.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>14.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="14.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>14.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-4"><i class="fa fa-check"></i><b>14.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-gene-expression-analysis-with-limma-voom" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em><a href="differential-gene-expression-analysis-with-limma-voom.html#differential-gene-expression-analysis-with-limma-voom" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Daianna González Padilla</p>
<p>In this chapter you’ll learn how DGE analysis is performed under the empirical Bayes framework of the popular <em>limma</em>-<em>voom</em> pipeline, highlighting key assumptions and concepts, and main differences with other methodologies.</p>
<div id="nb-based-dge-methods" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> NB-based DGE methods?<a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>An initial central point of discussion around DGE method development is how to model the distribution of the reads. Many methods model the read counts (<span class="math inline">\(y_{k,ij}\)</span>, non-negative integers) of a gene <span class="math inline">\(i\)</span> in the <span class="math inline">\(j\)</span> samples of condition <span class="math inline">\(k\)</span> through the Poisson or the Negative Binomial (NB) distribution. Of these, NB is often preferred as it allows the mean (<span class="math inline">\(\mu\)</span>) and the variance (<span class="math inline">\(\sigma\)</span>) of the reads to be different, compared to the Poisson distribution where <span class="math inline">\(\mu\)</span>=<span class="math inline">\(\sigma\)</span>. This is of particular importance as controlling the variance allows to account for variability in the gene expression levels across biological samples [1].</p>
<figure>
<img src="Figures/NB_DGE_methods.png" width="800px" align=center />
<figcaption style="color: gray; line-height: 0.9; text-align: justify">
<p><font size="-1.8">
<b>Figure 1</b>: <b> NB-distributed read counts. </b> Modeling of read counts for gene <span class="math inline">\(i\)</span> in the samples of the first and second conditions based on the NB model. Modified from <a href="https://pubmed.ncbi.nlm.nih.gov/31456901/">Li, W. V., &amp; Li, J. J. (2018)</a>.</p>
</font>
</figcaption>
</figure>
<p>Estimating the NB distribution parameters is necessary to assess DE of each gene <span class="math inline">\(i\)</span> between any two conditions <span class="math inline">\(k=1,2\)</span> (<strong>Figure 1</strong>). Bayesian models are used defining prior distributions and relationships of such parameters. Briefly, after 1) estimating gene-wise NB parameters, 2) the mean-variance relationship across all genes can be used to shrink the gene variance estimations borrowing information from all genes or incorporating prior knowledge, something advantageous when sample sizes are small. 3) A statistical test is used to assess for each gene <span class="math inline">\(i\)</span> if its true expression in the first and second condition (<span class="math inline">\(\theta_{1i}\)</span> and <span class="math inline">\(\theta_{2i}\)</span>) is the same (null hypothesis) or differs (alternative hypothesis):</p>
<ul>
<li><span class="math inline">\(H_0: \theta_{1i}=\theta_{2i}\)</span></li>
<li><span class="math inline">\(H_1: \theta_{1i}≠\theta_{2i}\)</span>, where the <span class="math inline">\(\theta_{i}\)</span>’s are parameters included in the mean of the NB distributions (<span class="math inline">\(\mu\)</span>).</li>
</ul>
<p>4) The test statistic is computed for each gene and 5) its associated <em>p</em>-value is calculated based on the null distribution. 6) Finally, <em>p</em>-values are corrected for multiple-testing and DEGs are determined based on an adjusted <em>p</em>-values cutoff [1].</p>
<p>Examples of popular methods based on the NB distribution are <a href="https://bioconductor.org/packages/edgeR"><em>edgeR</em></a> and <a href="https://bioconductor.org/packages/DESeq2"><em>DESeq2</em></a>.</p>
<p>Nevertheless, one limitation NB-based methods face is that they set dispersion of the data as a known and global parameter, ignoring observation-specific variation and importantly, there’s a reduced number of statistical methods for count distributions compared to the normal distribution [1,2]. Here, we’ll focus on <a href="https://bioconductor.org/packages/limma"><em>limma</em></a> that does not rely on a certain distribution but rather works on <span class="math inline">\(log_2(cpm)\)</span> (CPM: counts per million) and fits linear models for DGE enabling the incorporation of additional predictors to model gene expression, a feature specially valuable for complex experimental settings.</p>
</div>
<div id="limma-voom-pipeline" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> <em>limma</em>-<em>voom</em> pipeline<a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://bioconductor.org/packages/limma"><em>limma</em></a> is a package for the analysis of gene expression data arising from microarray or RNA-seq technologies. It has features that make the analyses stable even for experiments with small number of arrays or samples —this is achieved by borrowing information across genes. It is specially designed for analyzing complex experiments with a variety of experimental conditions and predictors [3].</p>
<p>Usually, <em>limma</em> DGE analysis is carried out in five main steps, the last four of them completed by <em>limma</em> <code>R</code> functions, as described below. We’ll use bulk RNA-seq data from the <a href="https://bioconductor.org/packages/smokingMouse"><em>smokingMouse</em></a> package to exemplify these steps.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-1" tabindex="-1"></a><span class="do">## Load the container package for RSE</span></span>
<span id="cb52-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</span>
<span id="cb52-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-3" tabindex="-1"></a></span>
<span id="cb52-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-4" tabindex="-1"></a><span class="do">## Connect to ExperimentHub</span></span>
<span id="cb52-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ExperimentHub&quot;</span>)</span>
<span id="cb52-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-6" tabindex="-1"></a>eh <span class="ot">&lt;-</span> ExperimentHub<span class="sc">::</span><span class="fu">ExperimentHub</span>()</span>
<span id="cb52-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-7" tabindex="-1"></a></span>
<span id="cb52-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-8" tabindex="-1"></a><span class="do">## Load package datasets</span></span>
<span id="cb52-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-9" tabindex="-1"></a>myfiles <span class="ot">&lt;-</span> <span class="fu">query</span>(eh, <span class="st">&quot;smokingMouse&quot;</span>)</span>
<span id="cb52-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-10" tabindex="-1"></a></span>
<span id="cb52-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-11" tabindex="-1"></a><span class="do">## Download the mouse gene data</span></span>
<span id="cb52-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-12" tabindex="-1"></a>rse_gene <span class="ot">&lt;-</span> myfiles[[<span class="st">&quot;EH8313&quot;</span>]]</span>
<span id="cb52-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-13" tabindex="-1"></a></span>
<span id="cb52-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-14" tabindex="-1"></a><span class="do">## Samples from the nicotine experiment and from pups only</span></span>
<span id="cb52-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-15" tabindex="-1"></a>rse_gene_nic <span class="ot">&lt;-</span> rse_gene[, <span class="fu">which</span>(rse_gene<span class="sc">$</span>Expt <span class="sc">==</span> <span class="st">&quot;Nicotine&quot;</span> <span class="sc">&amp;</span> rse_gene<span class="sc">$</span>Age <span class="sc">==</span> <span class="st">&quot;Pup&quot;</span>)]</span>
<span id="cb52-16"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-16" tabindex="-1"></a></span>
<span id="cb52-17"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-17" tabindex="-1"></a><span class="do">## Retain only expressed genes (passed the filtering step)</span></span>
<span id="cb52-18"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-18" tabindex="-1"></a>rse_gene_filt <span class="ot">&lt;-</span> rse_gene_nic[</span>
<span id="cb52-19"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-19" tabindex="-1"></a>    <span class="fu">rowData</span>(rse_gene_nic)<span class="sc">$</span>retained_after_feature_filtering,</span>
<span id="cb52-20"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb52-20" tabindex="-1"></a>]</span></code></pre></div>
<p>Let’s explore a little the data.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb53-1" tabindex="-1"></a><span class="do">## Data dimensions: number of genes and samples</span></span>
<span id="cb53-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb53-2" tabindex="-1"></a><span class="fu">dim</span>(rse_gene_filt)</span>
<span id="cb53-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb53-3" tabindex="-1"></a><span class="co">#&gt; [1] 19974    42</span></span></code></pre></div>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-1" tabindex="-1"></a></span>
<span id="cb54-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-2" tabindex="-1"></a><span class="do">## Raw counts for first 3 genes in the first 5 samples</span></span>
<span id="cb54-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-3" tabindex="-1"></a><span class="fu">assays</span>(rse_gene_filt)<span class="sc">$</span>counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb54-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-4" tabindex="-1"></a><span class="co">#&gt;                       [,1] [,2] [,3] [,4] [,5]</span></span>
<span id="cb54-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5  2652 2107 1978 2691 1833</span></span>
<span id="cb54-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1    15   15    9   15   13</span></span>
<span id="cb54-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb54-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13   10    7   28   11    8</span></span></code></pre></div>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-1" tabindex="-1"></a></span>
<span id="cb55-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-2" tabindex="-1"></a><span class="do">## Log-normalized counts for first 3 genes in the first 5 samples</span></span>
<span id="cb55-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-3" tabindex="-1"></a><span class="fu">assays</span>(rse_gene_filt)<span class="sc">$</span>logcounts[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb55-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-4" tabindex="-1"></a><span class="co">#&gt;                            [,1]      [,2]       [,3]      [,4]      [,5]</span></span>
<span id="cb55-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5   5.639967  5.953457  5.4923034  5.903313  5.800879</span></span>
<span id="cb55-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1  -1.747878 -1.130265 -2.1809593 -1.517393 -1.282590</span></span>
<span id="cb55-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb55-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13 -2.295096 -2.173926 -0.6153596 -1.941338 -1.948814</span></span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-1" tabindex="-1"></a></span>
<span id="cb56-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-2" tabindex="-1"></a><span class="do">## Data for the first 2 samples</span></span>
<span id="cb56-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-3" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(rse_gene_filt), <span class="dv">2</span>)</span>
<span id="cb56-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-4" tabindex="-1"></a><span class="co">#&gt; DataFrame with 2 rows and 71 columns</span></span>
<span id="cb56-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-5" tabindex="-1"></a><span class="co">#&gt;           SAMPLE_ID FQCbasicStats perBaseQual perTileQual  perSeqQual perBaseContent   GCcontent    Ncontent</span></span>
<span id="cb56-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-6" tabindex="-1"></a><span class="co">#&gt;   SeqLengthDist SeqDuplication OverrepSeqs AdapterContent KmerContent SeqLength_R1 percentGC_R1 phred15-19_R1</span></span>
<span id="cb56-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-7" tabindex="-1"></a><span class="co">#&gt;   phred65-69_R1 phred115-119_R1 phred150-151_R1 phredGT30_R1 phredGT35_R1 Adapter65-69_R1 Adapter100-104_R1</span></span>
<span id="cb56-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-8" tabindex="-1"></a><span class="co">#&gt;   Adapter140_R1 SeqLength_R2 percentGC_R2 phred15-19_R2 phred65-69_R2 phred115-119_R2 phred150-151_R2 phredGT30_R2</span></span>
<span id="cb56-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-9" tabindex="-1"></a><span class="co">#&gt;   phredGT35_R2 Adapter65-69_R2 Adapter100-104_R2 Adapter140_R2 ERCCsumLogErr                bamFile   trimmed  numReads</span></span>
<span id="cb56-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-10" tabindex="-1"></a><span class="co">#&gt;   numMapped numUnmapped overallMapRate concordMapRate totalMapped mitoMapped  mitoRate totalAssignedGene  rRNA_rate</span></span>
<span id="cb56-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-11" tabindex="-1"></a><span class="co">#&gt;        Tissue         Age         Sex        Expt        Group    Pregnant       plate    location concentration</span></span>
<span id="cb56-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-12" tabindex="-1"></a><span class="co">#&gt;        medium        date   Pregnancy    flowcell       sum  detected subsets_Mito_sum subsets_Mito_detected</span></span>
<span id="cb56-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-13" tabindex="-1"></a><span class="co">#&gt;   subsets_Mito_percent subsets_Ribo_sum subsets_Ribo_detected subsets_Ribo_percent retained_after_QC_sample_filtering</span></span>
<span id="cb56-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-14" tabindex="-1"></a><span class="co">#&gt;   retained_after_manual_sample_filtering</span></span>
<span id="cb56-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb56-15" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 3 rows ]</span></span></code></pre></div>
<p class="exercise">
📝 <strong>Exercise 1</strong>: in order for you to perform a DGE analysis, locate your own RNA-seq datasets if you have any, or download expression data from a study of your interest and build a <code>RSE</code> object using <em>recount3</em> (see <strong>Chapter 3: recount3 introduction</strong>). A third option you have is to download gene expression data from the <a href="http://research.libd.org/smokingMouse/index.html"><em>smokingMouse</em></a> package used here. A fourth option is to download data from GEO as Sean Davis will explain next. We’ll have more time tomorrow for doing this exercise with data of your choosing.
<p/>
<div id="model.matrix" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> <code>model.matrix()</code><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>limma</em> fits a linear model to the expression data of each gene (response variable), modeling the systematic part of the data by sample-level covariates (predictors).</p>
<style>
p.exercise {
background-color: #FFFAFA;
padding: 15px;
border: 2px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<style>
p.info {
background-color: #FFFFF0;
padding: 20px;
border: 1px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<style>
p.conclusion {
background-color: #EEE9E9;
padding: 20px;
border: 1px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}

</style>
<style>
p.question{
background-color: #E3E3E3;
padding: 20px;
border: 1px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<style>
p.link{
background-color: #FFFFFF;
padding: 10px;
border: 0px solid black;
margin-left: 0px;
border-radius: 1px;
font-size: 13px;
font-family: sans-serif;
}
</style>
<style>
p.comment {
background-color: #F0F0F0;
padding: 20px;
border: 0px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<style>
p.alert {
background-color: #FFE4E1;
padding: 14px;
border: 0px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<style>
p.success {
background-color: #E0EEE0;
padding: 14px;
border: 0px solid black;
margin-left: 0px;
border-radius: 1px;
font-family: sans-serif;
}
</style>
<p class="info">
💡 A model is a specification of how a set of variables relate to each other. In the case of a linear model, it is a linear equation that describes how the dependent or response variable is explained by the independent variables, also called predictors. A regression analysis with more than one independent variable is called <strong>multiple regression</strong>. Regression with only one independent variable is called <strong>simple regression</strong> [4].
</p>
<p>The <em>limma</em> model is specified with a <strong>design matrix</strong>, also known as <strong>model matrix</strong> or <strong>regressor matrix</strong>, often denoted by <span class="math inline">\(X\)</span>. This is a matrix of values for explanatory variables of the samples: rows correspond to samples and columns to sample variables.</p>
<p>Say that the values the <span class="math inline">\(i\)</span>th sample take in the <span class="math inline">\(h\)</span> covariates are <span class="math inline">\(X_{ih}\)</span>’s and their coefficients are <span class="math inline">\(\beta_{h}\)</span>’s. The predicted expression of a gene in the <span class="math inline">\(i\)</span>th sample is given by <span class="math inline">\(\hat y_i =\beta_0 + \sum_{1}^h\beta_{h}X_{ih}\)</span>.</p>
<p><span class="math display">\[
\hat y = X\beta=\displaystyle {\begin{bmatrix} \hat y_{1}\\ \hat y
_{2}\\ \hat y_{3}\\...\\ \hat y_{n-1}\\ \hat y_{n}\end{bmatrix}}={\begin{bmatrix}1&amp;X_{11}&amp;X_{12}&amp;X_{13}&amp;\cdots&amp;X_{1,h-1}&amp;X_{1h}\\1&amp;X_{21}&amp;X_{22}&amp;X_{23}&amp;\cdots&amp;X_{2,h-1}&amp;X_{2h}\\1&amp;X_{31}&amp;X_{32}&amp;X_{33}&amp;\cdots&amp;X_{3,h-1}&amp;X_{3h} \\ \vdots &amp; \vdots &amp; \vdots &amp; \vdots &amp; \ddots &amp; \vdots &amp; \vdots \\1&amp;X_{n-1,1}&amp;X_{n-1,2}&amp;X_{n-1,3}&amp;\cdots&amp;X_{n-1,h-1}&amp;X_{n-1,h} \\1&amp;X_{n,1}&amp;X_{n,2}&amp;X_{n,3}&amp;\cdots&amp;X_{n,h-1}&amp;X_{n,h} \end{bmatrix}}{\begin{bmatrix}\beta _{0}\\\beta _{1}\\\beta _{2}\\\beta_{3}\\...\\\beta_{h-1}\\\beta_{h}\end{bmatrix}}
\]</span>
where <span class="math inline">\(n\)</span> is the number of samples.</p>
<p>In the first step we create this matrix using <code>model.matrix()</code> that receives a formula with the variables to include in the models and the sample data.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-1" tabindex="-1"></a><span class="do">## Define formula</span></span>
<span id="cb57-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-2" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="er">~</span> Group <span class="sc">+</span> Sex <span class="sc">+</span> flowcell <span class="sc">+</span> mitoRate <span class="sc">+</span> overallMapRate <span class="sc">+</span> totalAssignedGene <span class="sc">+</span> detected <span class="sc">+</span> ERCCsumLogErr</span>
<span id="cb57-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-3" tabindex="-1"></a></span>
<span id="cb57-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-4" tabindex="-1"></a><span class="do">## Model matrix</span></span>
<span id="cb57-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-5" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(formula, <span class="at">data =</span> <span class="fu">colData</span>(rse_gene_filt))</span>
<span id="cb57-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-6" tabindex="-1"></a><span class="fu">head</span>(model)</span>
<span id="cb57-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-7" tabindex="-1"></a><span class="co">#&gt;   (Intercept) GroupExperimental SexM flowcellHKCMHDSXX flowcellHKCNKDSXX flowcellHKCTMDSXX   mitoRate overallMapRate</span></span>
<span id="cb57-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-8" tabindex="-1"></a><span class="co">#&gt; 1           1                 0    0                 0                 1                 0 0.03876995         0.9811</span></span>
<span id="cb57-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-9" tabindex="-1"></a><span class="co">#&gt; 2           1                 1    0                 0                 1                 0 0.03337699         0.9791</span></span>
<span id="cb57-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-10" tabindex="-1"></a><span class="co">#&gt; 3           1                 0    1                 0                 1                 0 0.03606147         0.9825</span></span>
<span id="cb57-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-11" tabindex="-1"></a><span class="co">#&gt; 4           1                 1    1                 1                 0                 0 0.03962591         0.9855</span></span>
<span id="cb57-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-12" tabindex="-1"></a><span class="co">#&gt;   totalAssignedGene detected ERCCsumLogErr</span></span>
<span id="cb57-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-13" tabindex="-1"></a><span class="co">#&gt; 1         0.7715862    26545     -67.33211</span></span>
<span id="cb57-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-14" tabindex="-1"></a><span class="co">#&gt; 2         0.7778447    24545     -66.38868</span></span>
<span id="cb57-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-15" tabindex="-1"></a><span class="co">#&gt; 3         0.7870034    25640     -58.89350</span></span>
<span id="cb57-16"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-16" tabindex="-1"></a><span class="co">#&gt; 4         0.7786461    25905     -84.91929</span></span>
<span id="cb57-17"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb57-17" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 2 rows ]</span></span></code></pre></div>
<p class="question">
❓ <strong>Which variables to include as covariates in the models?</strong> A straightforward strategy is to keep the model as <strong>simple as possible</strong> and after fitting the model evaluate the comparisons of interest [3]. In <strong>Chapter 7</strong> we will discuss how correlation and variance partition analyses can help us to set up the best models.
</p>
<p class="alert">
⚠️ <strong>Very important</strong>: always check which condition group is set as the reference in you model for the coefficient/contrast of interest (column named as <code>[Coefficient_name][Reference_Group]</code>; corresponding reference group set to 1) as this determines if a DEG is up or downregulated in the given condition compared to the other.
</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb58-1" tabindex="-1"></a><span class="do">## Comparison of interest: Group</span></span>
<span id="cb58-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb58-2" tabindex="-1"></a>coef <span class="ot">&lt;-</span> <span class="st">&quot;GroupExperimental&quot;</span></span></code></pre></div>
<p class="exercise">
📝 <strong>Exercise 2</strong>: identify the sample data of your study and create the respective design matrix. Which is the reference group for your main variable of interest?
Tomorrow we will learn how to use <code>ExploreModelMatrix</code> for helping us interpret coefficients.
<p/>
</div>
<div id="voom" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> <code>voom()</code><a href="differential-gene-expression-analysis-with-limma-voom.html#voom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Compared to NB-based methods, <em>limma</em> works with <span class="math inline">\(log2(cpm)\)</span> which are approximately normally distributed (as we have seen) and thus, opens the possibility to leverage a wide range of normal-based statistical tools not available for count distributions, including methods developed for microarray data. However, <em>limma</em> doesn’t assume nor require data to follow a normal distribution, but it does apply normal-based microarray-like statistical methods to RNA-seq read counts [2].</p>
<blockquote>
“… limma does not make any assumption that the data appears normal in a histogram.”
<div align="right">
- Gordon Smyth, author of <em>limma</em>, in the <a href="https://support.bioconductor.org/p/9140296/#9156131">Bioconductor support website</a> 2021.
</div>
</blockquote>
<p>The benefit of using <span class="math inline">\(log2(cpm)\)</span>, however, is not immediate. One limitation for the direct application of normal-based methods to log-counts is that reads counts have unequal variabilities even after a log-transformation depending on the count sizes: probability distributions for counts are naturally heteroscedastic, with log-cpm not having constant variances (larger variances for larger counts) [2]. It has been proposed that to design powerful statistical analysis for RNA-seq, it is more important to model the relationship between the mean and the variance in the data than to specify which probabilistic distribution to use for the counts [2]. And importantly, converting count data taking such relationship into account does open up access to their analysis with normal-based methods. That’s why we use <code>voom()</code>.</p>
<p>What <code>voom()</code> does is:</p>
<ol style="list-style-type: decimal">
<li><p>First, to compute log-cpm. Log-normalized expression for gene <span class="math inline">\(g\)</span> in sample <span class="math inline">\(i\)</span> (<span class="math inline">\(y_{gi}\)</span>) is given by<br />
<span class="math display">\[
y_{gi}=log_2(\frac{r_{gi} + 0.5}{R_i + 1.0} \times 10^6)
\]</span>
where <span class="math inline">\(r_{gi}\)</span> is the raw count for the gene in the sample and <span class="math inline">\(R_i\)</span> the library size of the sample.
We add +0.5 to the counts to avoid log of zero and +1 to the library size to
ensure that <span class="math inline">\(\frac{r_{gi}+0.5}{R_i+1}\)</span> is strictly less than 1 (if <span class="math inline">\(r_{gi} = R_i\)</span>).</p></li>
<li><p>A linear model is fitted to gene log-cpm values by ordinary least squares as:
<span class="math display">\[ E(y_{gi})=\mu_{gi}=X_i\beta_g \]</span>
where <span class="math inline">\(E(y_{gi})\)</span> is the expected expression of gene <span class="math inline">\(g\)</span> in sample <span class="math inline">\(i\)</span>, <span class="math inline">\(X_i\)</span> is the vector with the sample values for the covariates and <span class="math inline">\(\beta_g\)</span> the vector of covariate coefficients for the gene. As a result, we have the estimated <span class="math inline">\(\hat\beta_g\)</span>, the fitted log-cpm’s <span class="math inline">\(\hat\mu_{gi}=X_i\hat\beta_g\)</span> and the residual standard deviations <span class="math inline">\(s_g\)</span>.</p></li>
<li><p>Then it estimates the mean-variance trend of the data by fitting a smooth curve to the <span class="math inline">\(\sqrt s_g\)</span> of the genes presented as a function of the average gene expression (in log-counts, not log-cpm). The <span class="math inline">\(\sqrt s_g\)</span>’s are used because they are symmetrically distributed. Log-counts typically show a decreasing mean-variance trend.</p></li>
<li><p><code>voom()</code> then predicts the standard deviation of each individual normalized observation <span class="math inline">\(y_{gi}\)</span> (<em>limma</em>-trend does that at the gene level) using this trend curve: the fitted log-count of each observation is mapped to the curve and its <span class="math inline">\(\sqrt s_{gi}\)</span>value is obtained. The observation weights are <span class="math inline">\(w_{gi}=\frac{1}{s_{gi}^2}\)</span>.</p></li>
</ol>
<figure>
<img src="Figures/voom.png" width="800px" align=center />
<figcaption style="color: gray; line-height: 0.9; text-align: justify">
<p><font size="-1.8">
<b>Figure 2</b>: <b> <code>voom()</code> procedure to estimate observation-level variance weights for <em>limma</em>. </b> Extracted from the original voom publication ( <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29">Law, C. W. <em>et al</em>. 2018</a>).</p>
</font>
</figcaption>
</figure>
<ol start="5" style="list-style-type: decimal">
<li>Log-cpm (<span class="math inline">\(y_{gi}\)</span>) and associated weights (<span class="math inline">\(w_{gi}\)</span>) can then be entered into the <em>limma</em> framework for linear modeling. These weights are used in the linear modeling to adjust for count heteroscedasticity [2].</li>
</ol>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;limma&quot;</span>)</span>
<span id="cb59-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-2" tabindex="-1"></a></span>
<span id="cb59-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-3" tabindex="-1"></a><span class="do">## voom():</span></span>
<span id="cb59-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-4" tabindex="-1"></a><span class="co">#   1. Transform counts to log2(cpm)</span></span>
<span id="cb59-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-5" tabindex="-1"></a><span class="co">#      ----------------------------------------------------------------------------</span></span>
<span id="cb59-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-6" tabindex="-1"></a><span class="co"># .     |   Note we passed voom() raw counts as input, not the lognorm counts!!!   |</span></span>
<span id="cb59-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-7" tabindex="-1"></a><span class="co">#      ----------------------------------------------------------------------------</span></span>
<span id="cb59-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-8" tabindex="-1"></a><span class="co">#   2. Estimate mean-variance relationship for each gene</span></span>
<span id="cb59-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-9" tabindex="-1"></a><span class="co">#   3. Compute observation weights for limma (next step)</span></span>
<span id="cb59-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-10" tabindex="-1"></a></span>
<span id="cb59-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb59-11" tabindex="-1"></a>vGene <span class="ot">&lt;-</span> <span class="fu">voom</span>(<span class="fu">assay</span>(rse_gene_filt), <span class="at">design =</span> model, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/voom-1.png" width="672" /></p>
<p>Let’s explore the outpus of this function.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb60-1" tabindex="-1"></a><span class="do">## Returned data</span></span>
<span id="cb60-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb60-2" tabindex="-1"></a><span class="fu">names</span>(vGene)</span>
<span id="cb60-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb60-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;E&quot;       &quot;weights&quot; &quot;design&quot;  &quot;targets&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb61-1" tabindex="-1"></a></span>
<span id="cb61-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb61-2" tabindex="-1"></a><span class="do">## E: contains the computed log(cpm)</span></span>
<span id="cb61-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb61-3" tabindex="-1"></a><span class="fu">dim</span>(vGene<span class="sc">$</span>E)</span>
<span id="cb61-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb61-4" tabindex="-1"></a><span class="co">#&gt; [1] 19974    42</span></span></code></pre></div>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-1" tabindex="-1"></a>vGene<span class="sc">$</span>E[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb62-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-2" tabindex="-1"></a><span class="co">#&gt;                            [,1]       [,2]       [,3]      [,4]      [,5]</span></span>
<span id="cb62-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-3" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5   5.906572  6.1425731  5.7434780  6.133741  6.061250</span></span>
<span id="cb62-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1  -1.512368 -0.9445475 -1.9587859 -1.306258 -1.024247</span></span>
<span id="cb62-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13 -2.074247 -1.9918532 -0.3738234 -1.736892 -1.691672</span></span>
<span id="cb62-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025902.13  1.446325  1.2611275  1.3707154  1.419026  1.688471</span></span>
<span id="cb62-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb62-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000098104.1   1.572354  1.2408075  1.4727667  1.404882  1.533748</span></span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb63-1" tabindex="-1"></a></span>
<span id="cb63-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb63-2" tabindex="-1"></a><span class="do">## weights: contains the computed variance weight for each observation</span></span>
<span id="cb63-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb63-3" tabindex="-1"></a><span class="fu">dim</span>(vGene<span class="sc">$</span>weights)</span>
<span id="cb63-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb63-4" tabindex="-1"></a><span class="co">#&gt; [1] 19974    42</span></span></code></pre></div>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-1" tabindex="-1"></a>vGene<span class="sc">$</span>weights[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb64-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-2" tabindex="-1"></a><span class="co">#&gt;            [,1]       [,2]       [,3]       [,4]       [,5]</span></span>
<span id="cb64-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-3" tabindex="-1"></a><span class="co">#&gt; [1,] 143.326885 117.323375 139.214140 141.247546 128.818305</span></span>
<span id="cb64-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-4" tabindex="-1"></a><span class="co">#&gt; [2,]   4.255525   4.277395   2.698902   5.113520   3.377285</span></span>
<span id="cb64-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-5" tabindex="-1"></a><span class="co">#&gt; [3,]   4.009671   3.341317   5.555186   4.020098   2.546810</span></span>
<span id="cb64-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-6" tabindex="-1"></a><span class="co">#&gt; [4,]  20.584769  15.108579  15.521441  19.219652  16.893714</span></span>
<span id="cb64-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb64-7" tabindex="-1"></a><span class="co">#&gt; [5,]  22.473314  16.369739  18.359068  17.691839  14.325510</span></span></code></pre></div>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-1" tabindex="-1"></a></span>
<span id="cb65-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-2" tabindex="-1"></a><span class="do">## design: is the provided design matrix</span></span>
<span id="cb65-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-3" tabindex="-1"></a><span class="fu">head</span>(vGene<span class="sc">$</span>design)</span>
<span id="cb65-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-4" tabindex="-1"></a><span class="co">#&gt;   (Intercept) GroupExperimental SexM flowcellHKCMHDSXX flowcellHKCNKDSXX flowcellHKCTMDSXX   mitoRate overallMapRate</span></span>
<span id="cb65-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-5" tabindex="-1"></a><span class="co">#&gt; 1           1                 0    0                 0                 1                 0 0.03876995         0.9811</span></span>
<span id="cb65-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-6" tabindex="-1"></a><span class="co">#&gt; 2           1                 1    0                 0                 1                 0 0.03337699         0.9791</span></span>
<span id="cb65-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-7" tabindex="-1"></a><span class="co">#&gt; 3           1                 0    1                 0                 1                 0 0.03606147         0.9825</span></span>
<span id="cb65-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-8" tabindex="-1"></a><span class="co">#&gt; 4           1                 1    1                 1                 0                 0 0.03962591         0.9855</span></span>
<span id="cb65-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-9" tabindex="-1"></a><span class="co">#&gt;   totalAssignedGene detected ERCCsumLogErr</span></span>
<span id="cb65-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-10" tabindex="-1"></a><span class="co">#&gt; 1         0.7715862    26545     -67.33211</span></span>
<span id="cb65-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-11" tabindex="-1"></a><span class="co">#&gt; 2         0.7778447    24545     -66.38868</span></span>
<span id="cb65-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-12" tabindex="-1"></a><span class="co">#&gt; 3         0.7870034    25640     -58.89350</span></span>
<span id="cb65-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-13" tabindex="-1"></a><span class="co">#&gt; 4         0.7786461    25905     -84.91929</span></span>
<span id="cb65-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb65-14" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 2 rows ]</span></span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb66-1" tabindex="-1"></a></span>
<span id="cb66-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb66-2" tabindex="-1"></a><span class="do">## targets: the sample library sizes used to compute log(cpm) in the first step</span></span>
<span id="cb66-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb66-3" tabindex="-1"></a><span class="fu">dim</span>(vGene<span class="sc">$</span>targets)</span>
<span id="cb66-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb66-4" tabindex="-1"></a><span class="co">#&gt; [1] 42  1</span></span></code></pre></div>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-1" tabindex="-1"></a><span class="fu">head</span>(vGene<span class="sc">$</span>targets)</span>
<span id="cb67-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-2" tabindex="-1"></a><span class="co">#&gt;   lib.size</span></span>
<span id="cb67-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-3" tabindex="-1"></a><span class="co">#&gt; 1 44218086</span></span>
<span id="cb67-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-4" tabindex="-1"></a><span class="co">#&gt; 2 29831069</span></span>
<span id="cb67-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-5" tabindex="-1"></a><span class="co">#&gt; 3 36929795</span></span>
<span id="cb67-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-6" tabindex="-1"></a><span class="co">#&gt; 4 38331383</span></span>
<span id="cb67-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-7" tabindex="-1"></a><span class="co">#&gt; 5 27457620</span></span>
<span id="cb67-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb67-8" tabindex="-1"></a><span class="co">#&gt; 6 27113922</span></span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb68-1" tabindex="-1"></a></span>
<span id="cb68-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb68-2" tabindex="-1"></a><span class="fu">identical</span>(vGene<span class="sc">$</span>targets<span class="sc">$</span>lib.size, <span class="fu">colSums</span>(<span class="fu">assay</span>(rse_gene_filt)))</span>
<span id="cb68-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb68-3" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p class="conclusion">
➡️ In summary, <code>voom()</code> estimates non-parametrically the global mean-variance trend of the count data based on the expression of the genes and uses that to predict the variance of each individual expression observation (each log-cpm value) based on their predicted count sizes. The predicted variances are then associated as inverse weights to each observation that when used in linear modeling eliminate the log-cpm mean-variance trend [2].
</p>
<div style="background-color:#E0EEE0; padding:20px; font-family: sans-serif">
<p>
👉🏼 <strong>Advantages</strong>:
<ul>
<li>
✅ <code>voom()</code> estimates the mean-variance relationship in a non-parametric way.
</li>
<blockquote class="ludwig">
“The parametric advantages of the Poisson or NB distributions are mitigated by the fact that the observed mean-variance relationship of RNA-seq data does not perfectly match the theoretical mean-variance relationships inherent in these distributions. While the quadratic mean-variance relationship of the NB distribution captures most of the mean-variance trend, the NB dispersion still shows a non-ignorable trend with gene abundance.” [2]
</blockquote>
<li>
✅ Since <code>voom()</code> is a method to adapt count data to normal models, these give access to tractable empirical Bayes distribution theory.
</li>
<li>
✅ The use of normal distribution approaches and variance modeling is supported by generalized linear model theory.
</li>
</ul>
</p>
</div>
<p>
</p>
<p class="exercise">
📝 <strong>Exercise 3</strong>: compute the <span class="math inline">\(log2(cpm)\)</span> and the residual variance weights for each observation in your data using <code>voom()</code>.
<p/>
</div>
<div id="lmfit" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> <code>lmFit()</code><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This <em>limma</em> function fits a <strong>multiple linear model</strong> to the expression of each gene by weighted or generalized least squares to estimate the coefficients of the sample covariates which correspond to the logFC’s comparing gene expression between sample groups.</p>
<p><strong>Ordinary least squares (OLS)</strong></p>
<p>This is used to estimate the coefficients of a linear regression by minimizing the residual sum of squares [5].</p>
<div style="text-align: center">
<figure>
<img src="Figures/lmFit.png" width="400px" align=center/>
<figcaption style="color: gray; line-height: 0.9; text-align: justify; caption-side: bottom">
<p><font size="-1.8">
<b>Figure 3</b>: <b> Graphical representation of the OLS method for simple regression analysis</b>. Source: Gulve, A. (2020). Ordinary Least Square (OLS) Method for Linear Regression.</p>
</font>
</figcaption>
</figure>
</div>
<p>For simplicity, let’s work with one gene and say we have <span class="math inline">\(n\)</span> samples. The fitted expression of the gene in the <span class="math inline">\(j\)</span>th sample is <span class="math inline">\(\hat y_j =\beta_{0} + \sum_{1}^h\beta_{h}X_{jh}\)</span> , where <span class="math inline">\(\beta_h\)</span> is the coefficient for the <span class="math inline">\(h\)</span>th covariate and <span class="math inline">\(X_{jh}\)</span> the value the <span class="math inline">\(j\)</span>th sample takes for the <span class="math inline">\(h\)</span>th covariate. It can also be written as <span class="math inline">\(\hat y_j =\sum_{0}^h\beta_{h}X_{jh}\)</span> if <span class="math inline">\(X_{j0}=1\)</span>.</p>
<p>So we have an overdetermined system of <span class="math inline">\(n\)</span> linear equations and <span class="math inline">\(h\)</span> unknown parameters with <span class="math inline">\(n&gt;h\)</span>:</p>
<p><span class="math inline">\(\hat y_j =\sum_{0}^h\beta_{h}X_{jh}\)</span> with <span class="math inline">\(j=(1,2, ..., n)\)</span>.</p>
<p>Such system usually has no exact solution, so we need to estimate the coefficients that better fit the data in a linear regression. The problem is reduced to solving a quadratic minimization problem:</p>
<p><span class="math inline">\(\hat \beta=arg \ _\beta\ min \ \ S(\beta)\)</span> where <span class="math inline">\(S(\beta)=\sum_j(y_j -\hat y_j)^2=RSS\)</span> (residual sum of squares).</p>
<div style="background-color:#FFFFF0; padding:20px; font-family: sans-serif; border: 1px solid black">
<p>💡 We can think of these <span class="math inline">\(\beta\)</span>’s as differences in the fitted (expected) expression of a gene. Say we have two binary categorical variables in the model (<span class="math inline">\(X_1\)</span> and <span class="math inline">\(X_2\)</span>), then the expected gene expression in a sample is <span class="math inline">\(E(y|X_1, X_2) =\hat y =\beta_{0} + \beta_1X_1+\beta_2X_2\)</span>, where <span class="math inline">\(X_1\)</span> and <span class="math inline">\(X_2\)</span> equal to 1 or 0. Then we have the following 4 combinations:</p>
<ul>
<li><mark style= "background-color: #FAECF8"> <span class="math inline">\(E(y|X_1=1, X_2=1) = \mu_{12}=\beta_{0} + \beta_1+\beta_2\)</span> </mark></li>
<li><mark style= "background-color: #FCF3CF"> <span class="math inline">\(E(y|X_1=1, X_2=0) =\mu_{1}=\beta_{0} + \beta_1\)</span> </mark></li>
<li><mark style= "background-color: #FAECF8"> <span class="math inline">\(E(y|X_1=0, X_2=1) =\mu_{2}=\beta_{0} + \beta_2\)</span> </mark></li>
<li><mark style= "background-color: #FCF3CF"> <span class="math inline">\(E(y|X_1=0, X_2=0) =\mu_{0}=\beta_{0}\)</span> </mark></li>
</ul>
<p>So <span class="math inline">\(\beta_1=\)</span> <mark style= "background-color: #FCF3CF"><span class="math inline">\(\mu_1-\mu_0\)</span></mark> <span class="math inline">\(=\)</span> <mark style= "background-color: #FAECF8"> <span class="math inline">\(\mu_{12}-\mu_2\)</span> </mark> and <span class="math inline">\(\beta_2=\)</span> <mark style= "background-color: #FAECF8"><span class="math inline">\(\mu_2\)</span></mark><span class="math inline">\(-\)</span> <mark style= "background-color: #FCF3CF"><span class="math inline">\(\mu_0\)</span></mark>. Say our variable of interest is <span class="math inline">\(\beta_1\)</span>. Then what we are testing is if the expected expression of a gene is different when <span class="math inline">\(X_1=1\)</span> (in the first sample group) and <span class="math inline">\(X_1=0\)</span> (in the second sample group), fixing <span class="math inline">\(X_2\)</span> in either 1 or 0.</p>
</div>
<p>
</p>
<p><strong>Generalized least squares (GLS)</strong></p>
<p>Is a generalization of OLS that allows for heteroskedasticity and correlation between the residuals [6].</p>
<p><strong>Weighted least squares (WLS)</strong></p>
<p>In this case the function to be minimized becomes the weighted sum of the squared residuals: squared residuals are weighted by the reciprocal of their variance so that more noisy observations have less weight. That’s what we used <code>voom()</code> for.</p>
<p><code>lmFit()</code> returns a fitted model object with the estimated <strong>coefficients</strong>, <strong>standard errors</strong> (<span class="math inline">\(SE=sd/\sqrt n\)</span>) and <strong>residual standard errors/deviations</strong> (<span class="math inline">\(RSE=s_g=\sqrt {RSS/ n-2}\)</span>) for each gene. Depending on the arguments and correlations in the data, this function calls one of the following functions to fit a linear model for each gene [7]:</p>
<ul>
<li><code>mrlm</code>: for a robust regression if <code>method="robust”</code>.</li>
<li><code>gls.series</code>: GLS estimator if <code>method="ls”</code> and a correlation structure has been specified.</li>
<li><code>lm.series</code>: OLS method if <code>method="ls”</code> and there is no correlation structure.</li>
</ul>
<p>For the <code>weights</code> argument of <code>lmFit()</code>, the precision weights for the observations previously computed are extracted from the <code>voom()</code> output.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-1" tabindex="-1"></a><span class="do">## lmFit():</span></span>
<span id="cb69-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-2" tabindex="-1"></a><span class="co">#   1. Fit linear model for each gene to estimate logFCs</span></span>
<span id="cb69-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-3" tabindex="-1"></a>fitGene <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(vGene)</span>
<span id="cb69-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-4" tabindex="-1"></a></span>
<span id="cb69-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-5" tabindex="-1"></a><span class="do">## Corroborate &quot;ls&quot; method was applied</span></span>
<span id="cb69-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-6" tabindex="-1"></a>fitGene<span class="sc">$</span>method</span>
<span id="cb69-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb69-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ls&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-1" tabindex="-1"></a></span>
<span id="cb70-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-2" tabindex="-1"></a><span class="do">## Explore outputs: estimated coefficients (logFCs)</span></span>
<span id="cb70-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-3" tabindex="-1"></a><span class="fu">head</span>(fitGene<span class="sc">$</span>coefficients)</span>
<span id="cb70-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-4" tabindex="-1"></a><span class="co">#&gt;                       (Intercept) GroupExperimental        SexM flowcellHKCMHDSXX flowcellHKCNKDSXX flowcellHKCTMDSXX</span></span>
<span id="cb70-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5   -35.637900       -0.05125195  0.05690091       -0.47469588       -0.38545404       -0.66545820</span></span>
<span id="cb70-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1    37.943310        0.72450620  0.19887963       -0.20803712       -0.40926270       -0.10900553</span></span>
<span id="cb70-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13  -43.586603        0.17256694  0.28895786       -0.04476551        0.15257245       -0.06949759</span></span>
<span id="cb70-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-8" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025902.13    5.657837       -0.05025788 -0.04808144       -0.18732331       -0.26041436        0.07364071</span></span>
<span id="cb70-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-9" tabindex="-1"></a><span class="co">#&gt;                         mitoRate overallMapRate totalAssignedGene      detected ERCCsumLogErr</span></span>
<span id="cb70-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-10" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5  -11.487040      37.443647          4.753830  6.442499e-05 -0.0043851842</span></span>
<span id="cb70-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-11" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1  -21.871815      27.323603        -79.223111 -1.026687e-04  0.0166190950</span></span>
<span id="cb70-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-12" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13  16.727251       7.179484         42.367908  8.535604e-05  0.0133260060</span></span>
<span id="cb70-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-13" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025902.13 -24.538883      11.174017        -18.548282 -1.231546e-05 -0.0092194951</span></span>
<span id="cb70-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb70-14" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 2 rows ]</span></span></code></pre></div>
<div style="background-color:#FFFFF0; padding:20px; font-family: sans-serif; border: 1px solid black">
<p>💡 <strong>Interaction terms in linear models</strong></p>
<p>There may be cases where we want to assess gene expression differences between 2 conditions within more than one specific group; for example if we were interested in knowing what are the effects of a treatment (<span class="math inline">\(X_1=1\)</span> for treatment and 0 for controls) in <mark style= "background-color: #FAECF8">females</mark> and <mark style= "background-color: #E0FFFF">males</mark> separately (<span class="math inline">\(X_2=1\)</span> for females and 0 for males). In such cases we can fit an <strong>interaction model</strong> in which we include the product of <span class="math inline">\(X_1\)</span> and <span class="math inline">\(X_2\)</span> so that <span class="math inline">\(X_1X_2=1\)</span> if a sample comes from a female that was treated and 0 otherwise:</p>
<p><span class="math display">\[E(y|X_1, X_2) =\beta_{0} + \beta_1X_1+\beta_2X_2 + \beta_3X_1X_2\]</span></p>
<ul>
<li><mark style= "background-color: #FAECF8"> <span class="math inline">\(E(y|X_1=1, X_2=1) =\mu_{12} =\beta_{0} + \beta_1+\beta_2+\beta_3\)</span> </mark></li>
<li><mark style= "background-color: #E0FFFF"> <span class="math inline">\(E(y|X_1=1, X_2=0) =\mu_{1} =\beta_{0} + \beta_1\)</span> </mark></li>
<li><mark style= "background-color: #FAECF8"> <span class="math inline">\(E(y|X_1=0, X_2=1) =\mu_{2} =\beta_{0} + \beta_2\)</span> </mark></li>
<li><mark style= "background-color: #E0FFFF"> <span class="math inline">\(E(y|X_1=0, X_2=0) =\mu_{0} =\beta_{0}\)</span> </mark></li>
</ul>
<p><span class="math inline">\(\beta_1 + \beta_3=\)</span> <mark style= "background-color: #FAECF8"><span class="math inline">\(\mu_{12}-\mu_2\)</span></mark> which is the expression difference between treated and control female samples (<span class="math inline">\(X_2=1\)</span>) and <span class="math inline">\(\beta_1 =\)</span> <mark style= "background-color: #E0FFFF"><span class="math inline">\(\mu_{1}-\mu_0\)</span></mark> for male samples (<span class="math inline">\(X_2=0\)</span>). Finally <span class="math inline">\(\beta_3\)</span>, called the <strong>interaction term</strong>, is (<mark style= "background-color: #FAECF8"><span class="math inline">\(\mu_{12}-\mu_2\)</span></mark>)<span class="math inline">\(-\)</span>(<mark style= "background-color: #E0FFFF"><span class="math inline">\(\mu_1-\mu_0\)</span></mark>), described as the difference in gene expression changes driven by the treatment in females compared to males [8].</p>
</div>
<p>
</p>
<p class="exercise">
📝 <strong>Exercise 4</strong>: fit a linear regression model to the expression data of your genes and extract the coefficients for the included covariates.
<p/>
</div>
<div id="ebayes" class="section level3 hasAnchor" number="5.2.4">
<h3><span class="header-section-number">5.2.4</span> <code>eBayes()</code><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we want to assess if the differences in gene expression between the sample groups are statistically significant. Initially, we can think of comparing the mean expression of a gene in the sample groups (e.g. cases and controls) which can be handled applying a <strong>two-sample <em>t</em>-test</strong> assuming that the values in both groups have an approximately normal distribution.</p>
<p>Here we use the <em>t</em>-score (<strong><em>t</em>-stats</strong>) to define if the difference in the means is statistically significant based on a <em>t</em>-distribution.</p>
<p>The <em>t</em>-stats is given by:</p>
<p><span class="math display">\[
t=\frac{\bar x_1 - \bar x_2}{\sqrt{\frac{s_1^2}{n_1}+\frac{s_2^2}{n_2}}}
\]</span></p>
<p>where <span class="math inline">\(\bar x_1\)</span> and <span class="math inline">\(\bar x_2\)</span> are the means of the expression values of a gene in the first and second sample groups, <span class="math inline">\(s_1\)</span> and <span class="math inline">\(s_2\)</span> are the sample standard deviations of gene expression in the same groups, and <span class="math inline">\(n_1\)</span>, <span class="math inline">\(n_2\)</span> the corresponding sample group sizes:</p>
<p><span class="math inline">\(s_1 = \sqrt{\frac{\sum_{i=1}^ {n_1} (x_i-\bar x_1)^2}{n_1-1}}\)</span> and <span class="math inline">\(s_2 = \sqrt{\frac{\sum_{j=1}^ {n_2} (x_j-\bar x_2)^2}{n_2-1}}\)</span>, with <span class="math inline">\(x_i\)</span> and <span class="math inline">\(x_j\)</span> the gene expression values in the samples of group 1 and 2, respectively.</p>
<p class="question">
➡️ Note that we say <em>sample</em> means and <em>sample</em> standard deviations because they are estimators of the population parameters, computed based on the data that we have.
</p>
<p>We can think of this <em>t</em>-stats as a ratio of signal and noise. The numerator contains the difference between the two means, taken as the <strong>signal</strong> for DE. The denominator corresponds to the <strong>standard error</strong> and represents the <strong>noise</strong> in terms of gene expression variance within the sample groups. This represents how spread out the signal is [9]. In that way, the <em>t</em>-stats is a measure of how strong is the DE signal. Once computed, the <em>t</em>-stats have an associated <em>p</em>-value based on a Student <em>t</em>-distribution under the null hypothesis (<span class="math inline">\(H_o\)</span>: <span class="math inline">\(\bar x_1 - \bar x_2=0\)</span>).</p>
<p>This is exactly what we can get using <code>lm()</code>:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-1" tabindex="-1"></a><span class="do">## Lognorm expression of first gene</span></span>
<span id="cb71-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-2" tabindex="-1"></a>rse_gene_one_gene <span class="ot">&lt;-</span> rse_gene_filt[<span class="dv">1</span>, ]</span>
<span id="cb71-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-3" tabindex="-1"></a><span class="fu">colData</span>(rse_gene_one_gene) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">colData</span>(rse_gene_one_gene),</span>
<span id="cb71-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-4" tabindex="-1"></a>    <span class="st">&quot;lognorm_expr&quot;</span> <span class="ot">=</span> <span class="fu">assays</span>(rse_gene_one_gene)<span class="sc">$</span>logcounts[<span class="dv">1</span>, ]</span>
<span id="cb71-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-5" tabindex="-1"></a>)</span>
<span id="cb71-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-6" tabindex="-1"></a></span>
<span id="cb71-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-7" tabindex="-1"></a></span>
<span id="cb71-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-8" tabindex="-1"></a><span class="do">## Fit simple linear model</span></span>
<span id="cb71-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-9" tabindex="-1"></a>formula <span class="ot">&lt;-</span> lognorm_expr <span class="sc">~</span> Group</span>
<span id="cb71-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-10" tabindex="-1"></a>lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(formula, <span class="at">data =</span> <span class="fu">colData</span>(rse_gene_one_gene))</span>
<span id="cb71-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-11" tabindex="-1"></a><span class="fu">summary</span>(lm)</span>
<span id="cb71-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb71-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-13" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb71-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-14" tabindex="-1"></a><span class="co">#&gt; lm(formula = formula, data = colData(rse_gene_one_gene))</span></span>
<span id="cb71-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb71-16"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-16" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb71-17"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-17" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb71-18"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-18" tabindex="-1"></a><span class="co">#&gt; -1.05368 -0.06304  0.03012  0.10254  0.24844 </span></span>
<span id="cb71-19"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb71-20"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-20" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb71-21"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-21" tabindex="-1"></a><span class="co">#&gt;                   Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb71-22"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-22" tabindex="-1"></a><span class="co">#&gt; (Intercept)        5.75377    0.04502 127.800   &lt;2e-16 ***</span></span>
<span id="cb71-23"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-23" tabindex="-1"></a><span class="co">#&gt; GroupExperimental -0.04292    0.06694  -0.641    0.525    </span></span>
<span id="cb71-24"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-24" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb71-25"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-25" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb71-26"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-26" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb71-27"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-27" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 0.2159 on 40 degrees of freedom</span></span>
<span id="cb71-28"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-28" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.01017,    Adjusted R-squared:  -0.01457 </span></span>
<span id="cb71-29"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb71-29" tabindex="-1"></a><span class="co">#&gt; F-statistic: 0.4111 on 1 and 40 DF,  p-value: 0.525</span></span></code></pre></div>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-1" tabindex="-1"></a></span>
<span id="cb72-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-2" tabindex="-1"></a><span class="do">## Two sample t-test</span></span>
<span id="cb72-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-3" tabindex="-1"></a><span class="fu">t.test</span>(formula, <span class="at">data =</span> <span class="fu">colData</span>(rse_gene_one_gene), <span class="at">var.equal =</span> <span class="cn">TRUE</span>)</span>
<span id="cb72-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb72-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-5" tabindex="-1"></a><span class="co">#&gt;  Two Sample t-test</span></span>
<span id="cb72-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb72-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-7" tabindex="-1"></a><span class="co">#&gt; data:  lognorm_expr by Group</span></span>
<span id="cb72-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-8" tabindex="-1"></a><span class="co">#&gt; t = 0.64121, df = 40, p-value = 0.525</span></span>
<span id="cb72-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-9" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true difference in means between group Control and group Experimental is not equal to 0</span></span>
<span id="cb72-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-10" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb72-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-11" tabindex="-1"></a><span class="co">#&gt;  -0.09236465  0.17820636</span></span>
<span id="cb72-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-12" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb72-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-13" tabindex="-1"></a><span class="co">#&gt;      mean in group Control mean in group Experimental </span></span>
<span id="cb72-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb72-14" tabindex="-1"></a><span class="co">#&gt;                   5.753765                   5.710845</span></span></code></pre></div>
<p class="info">
💡 Sample sizes are critical! Larger sample sizes increase the power of the tests and reduce the false discovery rate (FDR) as they decrease the denominator of the <em>t</em>-stats (increasing their values) and slight differences can then be detected.
</p>
<p class="alert">
⚠️ Now consider that for genes with small variances in their expression the <em>t</em>-stats will be greater and we could be detecting non-DEGs as DE (false positives).
</p>
<p>But two things must be considered at least when working with gene expression data:</p>
<ol style="list-style-type: decimal">
<li>The first is that expression values are usually not normally distributed.</li>
<li>Second, the distributions and variances of expression values vary across genes and conditions.</li>
</ol>
<p class="question">
➡️ With that in mind, inference at the individual gene level can be addressed borrowing information from all the genes in the experiment through a Bayes or empirical Bayes method that produces more powerful tests.
</p>
<p>The idea of <strong>Bayesian statistics</strong> is to give unknown quantities a prior distribution, considering each feature as a member of a population of features such as genes. More specifically, <strong>empirical Bayes methods</strong> are procedures for statistical inference in which the (empirical) prior distribution is estimated from the population of all features (from the data) [8]; in standard Bayesian methods this prior distribution is fixed before observing any data [10].</p>
<p>Inspired by the work of Lönnstedt and Speed (2002) in which a simple expression for the posterior odds of differential expression for each gene was computed using a parametric empirical Bayes approach, <a href="https://doi.org/10.2202/1544-6115.1027">Smyth, G. K. (2004)</a> generalized this model for its application to experiments with any numbers of samples and conditions and reformulated the posterior odds statistic in terms of a <strong>moderated <em>t</em>-statistic</strong> in which the posterior residual standard deviations are used instead of the ordinary ones, eliminating the requirement of knowing the non-null prior guess for the proportion of differentially expressed genes required in the log-odds [11]. Let’s see how it proceeds.</p>
<p>First, for each gene <span class="math inline">\(g\)</span> we have a vector with the expression values in the <span class="math inline">\(n\)</span> samples:</p>
<p><span class="math display">\[
y_{g}= (y_{g1}, y_{g2}, ..., y_{gn})
\]</span></p>
<p>We already know that the expected (predicted) gene expression in the samples is <span class="math inline">\(E(y_{g})=X\alpha_g\)</span> with <span class="math inline">\(X\)</span> the design matrix and <span class="math inline">\(\alpha_g\)</span> the vector of the coefficients for the <span class="math inline">\(h\)</span> covariates in the model; of these the ones of biological interest are the <span class="math inline">\(\beta_g\)</span>’s (contrasts of interest).</p>
<p>Then, as previously described, a linear model is fitted to the expression data for each gene to obtain the <strong>coefficient estimators</strong> (<span class="math inline">\(\hat \alpha_g\)</span>) (as well as <span class="math inline">\(\hat \beta_g\)</span>), the <strong>residual sample variances</strong> (<span class="math inline">\(s_g^2\)</span>) as estimators of the (true but unknown) residual variances (<span class="math inline">\(\sigma_g^2\)</span>), and the estimated covariance matrices.</p>
<p>Two relevant considerations here:</p>
<ol style="list-style-type: decimal">
<li>The expression values are not necessarily assumed to be normally distributed.</li>
<li>The linear model is not assumed to be necessarily by least squares.</li>
</ol>
<p>However, there are two assumptions:</p>
<ol style="list-style-type: decimal">
<li>The contrast estimators <span class="math inline">\(\hat \beta_g\)</span> are assumed to be approximately normally distributed with mean <span class="math inline">\(\beta_g\)</span>.</li>
<li>The residual sample variances (<span class="math inline">\(s_g^2\)</span>) are assumed to follow approximately a scaled chisquare distribution.</li>
</ol>
<p>Under such assumptions the ordinary <em>t</em>-stats for the covariate <span class="math inline">\(j\)</span> in the gene <span class="math inline">\(g\)</span> is defined by:</p>
<p><span class="math display">\[
t_{gj}=\frac{\hat \beta_{gj}}{s_g u_{gj}}=\frac{\hat \beta_{gj}}{SE(\hat \beta_{gj})}
\]</span></p>
<p>with <span class="math inline">\(s_g\)</span> the residual sample standard deviation of the gene and <span class="math inline">\(u_{gj}\)</span> the unscaled standard deviation. <span class="math inline">\(SE\)</span> stands for standard error.</p>
<p>The key step in the empirical Bayes approach of <em>limma</em> is to leverage the information across all genes by defining prior distributions for the unknown coefficients <span class="math inline">\(\beta_{gj}\)</span> and residual variance <span class="math inline">\(\sigma_g^2\)</span> of the genes.</p>
<ul>
<li><p>For <span class="math inline">\(\beta_{gj}\)</span> it models the prior distribution of the coefficients that are not zero, i.e. is the expected distribution of the <strong>logFC</strong> of the genes that are DE, is given by:
<span class="math display">\[\beta_{gj}|\sigma_{g}^2, \beta_{gj}≠0  \sim N(0, v_{0j}\sigma_g^2)\]</span></p>
<blockquote>
“Saying that the betas have prior information centered around zero implies that we are ignorant of the sign (<strong>+</strong>/<strong>-</strong>) of the beta.”
<div align="right">
– Vincent Carey (personal communication)
</div>
</blockquote></li>
</ul>
<p>
</p>
<ul>
<li>For the residual variances what <em>limma</em> does is to take the residual sample variances of all genes (<span class="math inline">\(s_g^2\)</span>’s) and estimate the empirical parameters of the gamma distribution it is assumed that they follow. Specifically, <span class="math inline">\(\frac{1}{\sigma_g^2}\)</span> is modeled by a scaled chisquare (gamma) prior distribution with mean <span class="math inline">\(\frac{1}{s_0^2}\)</span> and <span class="math inline">\(d_0\)</span> degrees of freedom, describing how the residual variances are expected to vary across genes:
<span class="math display">\[\frac{1}{\sigma_g^2} \sim \frac{1}{d_0s_0^2}\chi_{d_0}^2\]</span></li>
</ul>
<p>What we want to do next is not only to take the mean of the residual variances in the distribution (prior mean<span class="math inline">\(\frac{1}{s_0^2}\)</span>) but to estimate each gene residual variance as a Bayes predictor: as a weighted average of the <strong>prior mean</strong> (<span class="math inline">\(\frac{1}{s_0^2}\)</span>) and the observed <strong>sample variance</strong> (<span class="math inline">\(s_g^2\)</span>) of each gene. This is called the <strong>moderated variance</strong> and what is graphically happening is that we are pulling the observed gene variances towards the prior mean variance: large variances are reduced and the |<em>t</em>-stats| increases (more powerful <em>t</em>-test for those genes) and small variances are increased, decreasing the |<em>t</em>-stats| and the power of the <em>t</em>-test. Under this model the <strong>posterior residual sample variance</strong> or posterior residual variance estimator (<span class="math inline">\(\tilde s_g^2\)</span>) is:</p>
<p><span class="math display">\[
\tilde s_g^2=E(\sigma_g^2|s_g^2)=\frac{d_0s_0^2 + d_gs_g^2}{d_0+d_g}
\]</span></p>
<p>Moderation is somehow like having larger sample sizes for the estimation of variance given that the moderated variances are (on average) closer to the population variance than the original sample variances.</p>
<p>The <strong>moderated <em>t</em>-statistic</strong> can be now defined in terms of this <strong>posterior residual sample standard deviations</strong> instead of the usual ones:</p>
<p><span class="math display">\[
\tilde t_{gj}=\frac{\hat \beta_{gj}}{\tilde s_g u_{gj}}
\]</span></p>
<p>These moderated <em>t</em>-stats follow a <em>t</em>-distribution under the null hypothesis (<span class="math inline">\(H_o:B_{gj}=0\)</span>) with degrees of freedom <span class="math inline">\(d_g+d_0\)</span> and the associated <em>p</em>-values can be computed based on such distribution.</p>
<p>As previously stated, with this redefined formula, large <em>t</em>-stats merely from very small <span class="math inline">\(s_g\)</span>’s are avoided. This results in increased power and reduced false non-discovery rate (FNR) (non detected DEGs) and the number of DEGs obtained increases [8].</p>
<p>In the end we say we have moderated the residual sample standard deviations of each gene in the <em>t</em>-stats denominator by using the distribution of all of them across the population of genes.</p>
<div style="background-color:#E0EEE0; padding:14px; font-family: sans-serif">
✅ The approach of using the posterior values results in shrinking the gene-wise residual sample variances (<span class="math inline">\(s_g^2\)</span>) towards the prior mean, making a more stable inference when a small number of samples is available.
</div>
<p>
</p>
<p><code>eBayes()</code> will implement this empirical Bayes model to compute for each gene and for each contrast these moderated <em>t</em>-statistic and their unadjusted <em>p</em>-values. Additionally, it returns moderated F-statistic and log-odds of differential expression. The moderated F-statistic tests whether any of the contrasts for a gene is non-zero (<span class="math inline">\(H_0:B_{g}=0\)</span>), i.e., whether that gene is differentially expressed for any contrast; it is similar to the ordinary F-statistic from analysis of variance (ANOVAR). The <em>t</em>-test does that for each individual contrast <span class="math inline">\(j\)</span> (<span class="math inline">\(H_0:B_{gj}=0\)</span>).</p>
<p class="link">
👉🏼 Check more about F-stats and other statistics computed by <code>eBayes()</code> here: <a href="https://support.bioconductor.org/p/6124/" class="uri">https://support.bioconductor.org/p/6124/</a>.
</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-1" tabindex="-1"></a><span class="do">## eBayes()</span></span>
<span id="cb73-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-2" tabindex="-1"></a><span class="do">## 1. Compute the empirical Bayes statistics for DE</span></span>
<span id="cb73-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-3" tabindex="-1"></a>eBGene <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fitGene)</span>
<span id="cb73-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-4" tabindex="-1"></a></span>
<span id="cb73-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-5" tabindex="-1"></a><span class="do">## Outputs of interest:</span></span>
<span id="cb73-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-6" tabindex="-1"></a><span class="do">## s2.prior -&gt; prior residual variance (prior mean 1/s0^2)</span></span>
<span id="cb73-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-7" tabindex="-1"></a><span class="do">##             in prior distribution of residual variances</span></span>
<span id="cb73-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-8" tabindex="-1"></a>eBGene<span class="sc">$</span>s2.prior</span>
<span id="cb73-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb73-9" tabindex="-1"></a><span class="co">#&gt; [1] 0.78987</span></span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb74-1" tabindex="-1"></a></span>
<span id="cb74-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb74-2" tabindex="-1"></a><span class="do">## df.prior -&gt; degrees of freedom d0 in prior distribution</span></span>
<span id="cb74-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb74-3" tabindex="-1"></a><span class="do">##             of residual variances</span></span>
<span id="cb74-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb74-4" tabindex="-1"></a>eBGene<span class="sc">$</span>df.prior</span>
<span id="cb74-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb74-5" tabindex="-1"></a><span class="co">#&gt; [1] 4.913248</span></span></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb75-1" tabindex="-1"></a></span>
<span id="cb75-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb75-2" tabindex="-1"></a><span class="do">## s2.post -&gt; posterior residual sample variances of the genes (~sg^2)</span></span>
<span id="cb75-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb75-3" tabindex="-1"></a><span class="fu">length</span>(eBGene<span class="sc">$</span>s2.post)</span>
<span id="cb75-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb75-4" tabindex="-1"></a><span class="co">#&gt; [1] 19974</span></span></code></pre></div>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb76-1" tabindex="-1"></a><span class="fu">head</span>(eBGene<span class="sc">$</span>s2.post)</span>
<span id="cb76-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb76-2" tabindex="-1"></a><span class="co">#&gt; [1] 2.3397702 0.7092520 1.1613995 0.9579389 0.7390718 0.4996251</span></span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb77-1" tabindex="-1"></a></span>
<span id="cb77-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb77-2" tabindex="-1"></a><span class="do">##  t -&gt; moderated t-stats of the genes for each contrast</span></span>
<span id="cb77-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb77-3" tabindex="-1"></a><span class="fu">dim</span>(eBGene<span class="sc">$</span>t)</span>
<span id="cb77-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb77-4" tabindex="-1"></a><span class="co">#&gt; [1] 19974    11</span></span></code></pre></div>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-1" tabindex="-1"></a>eBGene<span class="sc">$</span>t[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb78-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-2" tabindex="-1"></a><span class="co">#&gt;                       (Intercept) GroupExperimental       SexM flowcellHKCMHDSXX flowcellHKCNKDSXX</span></span>
<span id="cb78-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-3" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5   -4.4458336        -1.0615386  1.2467597        -4.8896590        -4.2555598</span></span>
<span id="cb78-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1    1.5930925         4.5298885  1.3144392        -0.6934431        -1.3930231</span></span>
<span id="cb78-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13  -1.2896585         0.8392518  1.4887297        -0.1047634         0.3829694</span></span>
<span id="cb78-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025902.13   0.4035855        -0.5885950 -0.5891347        -1.0922124        -1.6131630</span></span>
<span id="cb78-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb78-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000098104.1    0.7120536        -0.3614893 -1.0031002         0.6473100         1.0165548</span></span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb79-1" tabindex="-1"></a></span>
<span id="cb79-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb79-2" tabindex="-1"></a><span class="do">## p.value: corresponding unadjusted p-values of moderated t-stats</span></span>
<span id="cb79-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb79-3" tabindex="-1"></a><span class="fu">dim</span>(eBGene<span class="sc">$</span>p.value)</span>
<span id="cb79-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb79-4" tabindex="-1"></a><span class="co">#&gt; [1] 19974    11</span></span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-1" tabindex="-1"></a>eBGene<span class="sc">$</span>p.value[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb80-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-2" tabindex="-1"></a><span class="co">#&gt;                        (Intercept) GroupExperimental      SexM flowcellHKCMHDSXX flowcellHKCNKDSXX</span></span>
<span id="cb80-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-3" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5  8.086618e-05      2.955320e-01 0.2205562      2.114989e-05      0.0001425797</span></span>
<span id="cb80-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-4" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102331.1  1.199041e-01      6.283899e-05 0.1970317      4.924924e-01      0.1721763827</span></span>
<span id="cb80-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-5" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025900.13 2.054130e-01      4.068800e-01 0.1452882      9.171465e-01      0.7039999597</span></span>
<span id="cb80-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-6" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000025902.13 6.889106e-01      5.598170e-01 0.5594589      2.820128e-01      0.1154631942</span></span>
<span id="cb80-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb80-7" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000098104.1  4.810321e-01      7.198519e-01 0.3225237      5.215478e-01      0.3161651149</span></span></code></pre></div>
<p class="exercise">
📝 <strong>Exercise 5</strong>: obtain the moderated <em>t</em>-stats and associated <em>p</em>-values of all genes in you data for all covariates included in your model.
<p/>
<p>
</p>
</div>
<div id="toptable" class="section level3 hasAnchor" number="5.2.5">
<h3><span class="header-section-number">5.2.5</span> <code>topTable()</code><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This function is also provided by <em>limma</em> and summarizes the results of the linear model, performs hypothesis tests and adjusts the <em>p</em>-values for multiple testing [12]. Among the summary statistics presented, it returns the log2FCs, moderated <em>t</em>-statistics, <em>p</em>-values, and FDR-adjusted <em>p</em>-values of the genes for a given contrast of interest. The default form of <em>p</em>-value adjustment is the <strong>Benjamini and Hochberg’s</strong> method to control the false discovery rate (FDR) which assumes independence between genes.</p>
<p><strong>Relevant concepts</strong>:</p>
<ul>
<li><strong><em>q</em>-value</strong> → is the <strong>FDR-adjusted <em>p</em>-value</strong> used to control the <strong>False Discovery Rate (FDR)</strong> that is the expected proportion of false discoveries among the discoveries (DEGs). Selecting discoveries as those being below <span class="math inline">\(\alpha\)</span> in <em>q</em>-value, we control FDR ≤ <span class="math inline">\(\alpha\)</span>.</li>
</ul>
<p>Now we have the final statistics to determine wich genes are DE.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-1" tabindex="-1"></a><span class="do">## topTable()</span></span>
<span id="cb81-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-2" tabindex="-1"></a><span class="do">## 1. Obtain gene-wise DE stats for Group (Nicotine vs Ctrl)</span></span>
<span id="cb81-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-3" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">topTable</span>(eBGene, <span class="at">coef =</span> coef, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">number =</span> <span class="fu">nrow</span>(rse_gene_filt), <span class="at">sort.by =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb81-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-4" tabindex="-1"></a></span>
<span id="cb81-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-5" tabindex="-1"></a><span class="do">## Outputs for each gene and for the coeff selected (Group):</span></span>
<span id="cb81-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-6" tabindex="-1"></a><span class="do">##  logFC: log2-fold-changes</span></span>
<span id="cb81-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-7" tabindex="-1"></a><span class="fu">head</span>(top_genes<span class="sc">$</span>logFC)</span>
<span id="cb81-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb81-8" tabindex="-1"></a><span class="co">#&gt; [1] -0.05125195  0.72450620  0.17256694 -0.05025788 -0.02726320 -0.02684710</span></span></code></pre></div>
<p>In limma the <span class="math inline">\(\beta_{gj}\)</span>’s are the logFC’s:</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb82-1" tabindex="-1"></a><span class="fu">setdiff</span>(top_genes<span class="sc">$</span>logFC, eBGene<span class="sc">$</span>coefficients[, <span class="st">&quot;GroupExperimental&quot;</span>])</span>
<span id="cb82-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb82-2" tabindex="-1"></a><span class="co">#&gt; numeric(0)</span></span></code></pre></div>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb83-1" tabindex="-1"></a><span class="do">##  t: moderated t-stats</span></span>
<span id="cb83-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb83-2" tabindex="-1"></a><span class="fu">head</span>(top_genes<span class="sc">$</span>t)</span>
<span id="cb83-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb83-3" tabindex="-1"></a><span class="co">#&gt; [1] -1.0615386  4.5298885  0.8392518 -0.5885950 -0.3614893 -1.0959528</span></span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb84-1" tabindex="-1"></a></span>
<span id="cb84-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb84-2" tabindex="-1"></a><span class="do">## . P.value: unadjusted p-values of t-stats</span></span>
<span id="cb84-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb84-3" tabindex="-1"></a><span class="fu">head</span>(top_genes<span class="sc">$</span>P.Value)</span>
<span id="cb84-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb84-4" tabindex="-1"></a><span class="co">#&gt; [1] 2.955320e-01 6.283899e-05 4.068800e-01 5.598170e-01 7.198519e-01 2.803946e-01</span></span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb85-1" tabindex="-1"></a></span>
<span id="cb85-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb85-2" tabindex="-1"></a><span class="do">##  adj.P.Val: p-values adjusted to control the FDR</span></span>
<span id="cb85-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb85-3" tabindex="-1"></a><span class="fu">head</span>(top_genes<span class="sc">$</span>adj.P.Val)</span>
<span id="cb85-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb85-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.53854173 0.00412576 0.63704026 0.75340755 0.85943342 0.52324928</span></span></code></pre></div>
<p>After running all these 5 steps, one main initial plot we have to look at is the histogram of the <em>p</em>-values of the moderated <em>t</em>-stats of the genes. If there were DEGs, we’d expect to see a flat distribution of <em>p</em>-values corresponding to non-DEGs and a peak near <em>p</em>=0 for DEGs (for which we reject the null hypothesis). If this peak is absent but a uniform distribution still appears, DEGs might be detected after correcting for multiple testing.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb86-1" tabindex="-1"></a><span class="do">## Histogram of unadjusted p-values</span></span>
<span id="cb86-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb86-2" tabindex="-1"></a><span class="fu">hist</span>(top_genes<span class="sc">$</span>P.Value, <span class="at">xlab =</span> <span class="st">&quot;p-values&quot;</span>, <span class="at">main =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/hist_p-1.png" width="672" /></p>
<p>If very different <em>p</em>-value distributions are obtained from the uniform one, the best we can do is trying to explore if there are specific groups of genes (e.g. lowly-expressed genes) presenting such variable <em>p</em>-values and revisiting the assumptions and considerations of the statistical tests implemented [13].</p>
<p class="exercise">
📝 <strong>Exercise 6</strong>: obtain the DE logFCs, <em>t</em>-stats, <em>p</em>-values, and adjusted <em>p</em>-values of the genes for a given constrast/covariate under study.
</p>
</div>
</div>
<div id="de-visualization" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> DE visualization<a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>DEGs are identified defining a significance threshold (on the adjusted <em>p</em>-values). Let’s quantify the number of DEGs for nicotine exposure in pup brain and visualize their expression and DE statistics.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb87-1" tabindex="-1"></a><span class="do">## DEGs for FDR&lt;0.05</span></span>
<span id="cb87-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb87-2" tabindex="-1"></a>de_genes <span class="ot">&lt;-</span> top_genes[<span class="fu">which</span>(top_genes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>), ]</span>
<span id="cb87-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb87-3" tabindex="-1"></a></span>
<span id="cb87-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb87-4" tabindex="-1"></a><span class="do">## Number of DEGs</span></span>
<span id="cb87-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb87-5" tabindex="-1"></a><span class="fu">dim</span>(de_genes)</span>
<span id="cb87-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb87-6" tabindex="-1"></a><span class="co">#&gt; [1] 1895    6</span></span></code></pre></div>
<div id="volcano-plots" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Volcano plots<a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A very practical and useful plot to graphically represent DEGs and visualize their expression differences between conditions is a volcano plot. This is a scatter plot of the logFC’s of the genes in the x-axis vs their adjusted <em>p</em>-values in a -log scale in the y-axis.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb88-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-2" tabindex="-1"></a></span>
<span id="cb88-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-3" tabindex="-1"></a><span class="do">## Define up- and down-regulated DEGs, and non-DEGs</span></span>
<span id="cb88-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-4" tabindex="-1"></a>FDR <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb88-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-5" tabindex="-1"></a>DE <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb88-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-6" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(top_genes)[<span class="dv">1</span>]) {</span>
<span id="cb88-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-7" tabindex="-1"></a>    <span class="cf">if</span> (top_genes<span class="sc">$</span>adj.P.Val[i] <span class="sc">&gt;</span> FDR) {</span>
<span id="cb88-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-8" tabindex="-1"></a>        DE <span class="ot">&lt;-</span> <span class="fu">append</span>(DE, <span class="st">&quot;n.s.&quot;</span>)</span>
<span id="cb88-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-9" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb88-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-10" tabindex="-1"></a>        <span class="cf">if</span> (top_genes<span class="sc">$</span>logFC[i] <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb88-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-11" tabindex="-1"></a>            DE <span class="ot">&lt;-</span> <span class="fu">append</span>(DE, <span class="st">&quot;Up&quot;</span>)</span>
<span id="cb88-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-12" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb88-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-13" tabindex="-1"></a>            DE <span class="ot">&lt;-</span> <span class="fu">append</span>(DE, <span class="st">&quot;Down&quot;</span>)</span>
<span id="cb88-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-14" tabindex="-1"></a>        }</span>
<span id="cb88-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-15" tabindex="-1"></a>    }</span>
<span id="cb88-16"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-16" tabindex="-1"></a>}</span>
<span id="cb88-17"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-17" tabindex="-1"></a>top_genes<span class="sc">$</span>DE <span class="ot">&lt;-</span> DE</span>
<span id="cb88-18"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-18" tabindex="-1"></a></span>
<span id="cb88-19"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-19" tabindex="-1"></a><span class="do">## Colors, sizes and transparencies for up &amp; down DEGs and non-DEGs</span></span>
<span id="cb88-20"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-20" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Up&quot;</span> <span class="ot">=</span> <span class="st">&quot;indianred2&quot;</span>, <span class="st">&quot;Down&quot;</span> <span class="ot">=</span> <span class="st">&quot;steelblue2&quot;</span>, <span class="st">&quot;n.s.&quot;</span> <span class="ot">=</span> <span class="st">&quot;grey&quot;</span>)</span>
<span id="cb88-21"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-21" tabindex="-1"></a>sizes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Up&quot;</span> <span class="ot">=</span> <span class="fl">1.3</span>, <span class="st">&quot;Down&quot;</span> <span class="ot">=</span> <span class="fl">1.3</span>, <span class="st">&quot;n.s.&quot;</span> <span class="ot">=</span> <span class="fl">0.8</span>)</span>
<span id="cb88-22"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-22" tabindex="-1"></a>alphas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Up&quot;</span> <span class="ot">=</span> <span class="fl">0.4</span>, <span class="st">&quot;Down&quot;</span> <span class="ot">=</span> <span class="fl">0.6</span>, <span class="st">&quot;n.s.&quot;</span> <span class="ot">=</span> <span class="fl">0.5</span>)</span>
<span id="cb88-23"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-23" tabindex="-1"></a></span>
<span id="cb88-24"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-24" tabindex="-1"></a><span class="do">## Plot volcano plot</span></span>
<span id="cb88-25"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-25" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb88-26"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-26" tabindex="-1"></a>    <span class="at">data =</span> top_genes,</span>
<span id="cb88-27"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-27" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb88-28"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-28" tabindex="-1"></a>        <span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(adj.P.Val),</span>
<span id="cb88-29"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-29" tabindex="-1"></a>        <span class="at">color =</span> DE,</span>
<span id="cb88-30"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-30" tabindex="-1"></a>        <span class="at">fill =</span> DE,</span>
<span id="cb88-31"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-31" tabindex="-1"></a>        <span class="at">size =</span> DE,</span>
<span id="cb88-32"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-32" tabindex="-1"></a>        <span class="at">alpha =</span> DE</span>
<span id="cb88-33"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-33" tabindex="-1"></a>    )</span>
<span id="cb88-34"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-34" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb88-35"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-35" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb88-36"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-36" tabindex="-1"></a>    <span class="fu">geom_hline</span>(</span>
<span id="cb88-37"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-37" tabindex="-1"></a>        <span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR),</span>
<span id="cb88-38"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-38" tabindex="-1"></a>        <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;gray35&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.5</span></span>
<span id="cb88-39"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-39" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb88-40"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-40" tabindex="-1"></a>    <span class="fu">geom_vline</span>(</span>
<span id="cb88-41"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-41" tabindex="-1"></a>        <span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb88-42"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-42" tabindex="-1"></a>        <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;gray35&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.5</span></span>
<span id="cb88-43"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-43" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb88-44"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-44" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;-log10(FDR)&quot;</span>, <span class="at">x =</span> <span class="st">&quot;logFC(Nicotine vs Control)&quot;</span>) <span class="sc">+</span></span>
<span id="cb88-45"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-45" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb88-46"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-46" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cols, <span class="at">name =</span> <span class="st">&quot;Differential expression&quot;</span>) <span class="sc">+</span></span>
<span id="cb88-47"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-47" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> cols, <span class="at">name =</span> <span class="st">&quot;Differential expression&quot;</span>) <span class="sc">+</span></span>
<span id="cb88-48"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-48" tabindex="-1"></a>    <span class="fu">scale_size_manual</span>(<span class="at">values =</span> sizes, <span class="at">name =</span> <span class="st">&quot;Differential expression&quot;</span>) <span class="sc">+</span></span>
<span id="cb88-49"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-49" tabindex="-1"></a>    <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> alphas, <span class="at">name =</span> <span class="st">&quot;Differential expression&quot;</span>) <span class="sc">+</span></span>
<span id="cb88-50"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-50" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb88-51"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-51" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb88-52"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-52" tabindex="-1"></a>        <span class="at">legend.key.height =</span> <span class="fu">unit</span>(<span class="fl">0.15</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb88-53"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-53" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> (<span class="dv">13</span>)),</span>
<span id="cb88-54"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-54" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb88-55"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-55" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)</span>
<span id="cb88-56"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb88-56" tabindex="-1"></a>    )</span></code></pre></div>
<p><img src="_main_files/figure-html/volcano%20plot-1.png" width="672" /></p>
</div>
<div id="heat-maps-1" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Heat maps<a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Another common way to represent differential expression results is through a heat map. The package <a href="https://bioconductor.org/packages/release/bioc/html/ComplexHeatmap.html"><em>ComplexHeatmap</em></a> offers a flexible toolkit to easily create heat maps with row and column annotations, a feature of particular value to plot expression data of genes across samples with multiple biological and technical differences. Although initially all genes in your data can be plotted, frequently only DEGs are included as they tend to show clearer gene expression patterns.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ComplexHeatmap&quot;</span>)</span>
<span id="cb89-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-2" tabindex="-1"></a></span>
<span id="cb89-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-3" tabindex="-1"></a><span class="do">## We plot lognorm counts</span></span>
<span id="cb89-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-4" tabindex="-1"></a>lognorm_data <span class="ot">&lt;-</span> <span class="fu">assays</span>(rse_gene_filt)<span class="sc">$</span>logcounts</span>
<span id="cb89-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-5" tabindex="-1"></a></span>
<span id="cb89-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-6" tabindex="-1"></a><span class="do">## Subset to DEGs only</span></span>
<span id="cb89-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-7" tabindex="-1"></a>lognorm_data <span class="ot">&lt;-</span> lognorm_data[<span class="fu">rownames</span>(de_genes), ]</span>
<span id="cb89-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-8" tabindex="-1"></a></span>
<span id="cb89-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-9" tabindex="-1"></a><span class="do">## Define column (sample) names to display</span></span>
<span id="cb89-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb89-10" tabindex="-1"></a><span class="fu">colnames</span>(lognorm_data) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Pup_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(lognorm_data)[<span class="dv">2</span>])</span></code></pre></div>
<div style="background-color:#EDEDED; padding:20px; font-family: sans-serif; border: 1px solid black">
<p>🗒️ <strong>Notes</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>It is sometimes convenient to regress out the technical variables’ contributions on gene expression to see more clearly the effects of interest. This can happen, for instance, when the logFCs are too small to see any significant differences in the plots or when there are other strong confounding factors. Functions such as <a href="https://rdrr.io/github/LieberInstitute/jaffelab/man/cleaningY.html"><code>cleaningY()</code></a> of <em>jaffelab</em> can be used for this purpose.</p></li>
<li><p>The lognorm counts have to be correctly scaled and centered (around zero) to make the differences in the expression of the genes more notorious in the heat map. A simple way to do that is substracting from each lognorm count <span class="math inline">\(y_{gi}\)</span> (from the gene <span class="math inline">\(g\)</span> and sample <span class="math inline">\(i\)</span>) the mean expression of the gene* and dividing by the standard deviation (<span class="math inline">\(\sigma\)</span>) of the same gene expression values. This is formally called the <em>z</em>-score: the number of standard deviations away from the mean.</p></li>
</ol>
<p><span class="math display">\[
z=\frac{y_{gi} - \frac{\sum_{k=1}^{n}{y_{gk}}}{n}}{\sigma},
\]</span>
<span class="math inline">\(n\)</span> is the number of samples.</p>
<p><font size="2"> * This can also be done by columns (samples), not only by rows (genes). </font></p>
<p class="link" style="background-color:#EDEDED">
👉🏼 For more on centering and scaling, see this video:
</p>
<p class="link" style="background-color:#EDEDED; text-align: center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/c8ffeFVUzk8" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen>
</iframe>
</p>
</div>
<p>
</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-1" tabindex="-1"></a><span class="do">## Center and scale the data to make differences more evident</span></span>
<span id="cb90-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-2" tabindex="-1"></a>lognorm_data <span class="ot">&lt;-</span> (lognorm_data <span class="sc">-</span> <span class="fu">rowMeans</span>(lognorm_data)) <span class="sc">/</span> <span class="fu">rowSds</span>(lognorm_data)</span>
<span id="cb90-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-3" tabindex="-1"></a></span>
<span id="cb90-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-4" tabindex="-1"></a><span class="do">## Sample annotation: Sex, Group, and library size</span></span>
<span id="cb90-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-5" tabindex="-1"></a>col_anno <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(</span>
<span id="cb90-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-6" tabindex="-1"></a>    <span class="at">df =</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(rse_gene_filt)[, <span class="fu">c</span>(<span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Group&quot;</span>)]),</span>
<span id="cb90-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-7" tabindex="-1"></a>    <span class="at">library_size =</span> <span class="fu">anno_barplot</span>(<span class="fu">colData</span>(rse_gene_filt)<span class="sc">$</span>sum, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">&quot;lightyellow2&quot;</span>)),</span>
<span id="cb90-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-8" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">list</span>(</span>
<span id="cb90-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-9" tabindex="-1"></a>        <span class="st">&quot;Sex&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink1&quot;</span>, <span class="st">&quot;M&quot;</span> <span class="ot">=</span> <span class="st">&quot;dodgerblue&quot;</span>),</span>
<span id="cb90-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-10" tabindex="-1"></a>        <span class="st">&quot;Group&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span> <span class="ot">=</span> <span class="st">&quot;gray68&quot;</span>, <span class="st">&quot;Experimental&quot;</span> <span class="ot">=</span> <span class="st">&quot;gold2&quot;</span>)</span>
<span id="cb90-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-11" tabindex="-1"></a>    )</span>
<span id="cb90-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-12" tabindex="-1"></a>)</span>
<span id="cb90-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-13" tabindex="-1"></a></span>
<span id="cb90-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-14" tabindex="-1"></a><span class="do">## Gene annotation: logFC and biotype</span></span>
<span id="cb90-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-15" tabindex="-1"></a>de_genes<span class="sc">$</span>logFC_binary <span class="ot">&lt;-</span> <span class="fu">sapply</span>(de_genes<span class="sc">$</span>logFC, <span class="cf">function</span>(x) {</span>
<span id="cb90-16"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-16" tabindex="-1"></a>    <span class="cf">if</span> (x <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb90-17"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-17" tabindex="-1"></a>        <span class="st">&quot;&gt;0&quot;</span></span>
<span id="cb90-18"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-18" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb90-19"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-19" tabindex="-1"></a>        <span class="st">&quot;&lt;0&quot;</span></span>
<span id="cb90-20"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-20" tabindex="-1"></a>    }</span>
<span id="cb90-21"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-21" tabindex="-1"></a>})</span>
<span id="cb90-22"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-22" tabindex="-1"></a>de_genes<span class="sc">$</span>protein_coding_gene <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(rse_gene_filt[<span class="fu">rownames</span>(de_genes), ])<span class="sc">$</span>gene_type, <span class="cf">function</span>(x) {</span>
<span id="cb90-23"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-23" tabindex="-1"></a>    <span class="cf">if</span> (x <span class="sc">==</span> <span class="st">&quot;protein_coding&quot;</span>) {</span>
<span id="cb90-24"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-24" tabindex="-1"></a>        <span class="st">&quot;TRUE&quot;</span></span>
<span id="cb90-25"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-25" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb90-26"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-26" tabindex="-1"></a>        <span class="st">&quot;FALSE&quot;</span></span>
<span id="cb90-27"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-27" tabindex="-1"></a>    }</span>
<span id="cb90-28"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-28" tabindex="-1"></a>})</span>
<span id="cb90-29"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-29" tabindex="-1"></a></span>
<span id="cb90-30"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-30" tabindex="-1"></a>gene_anno <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(</span>
<span id="cb90-31"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-31" tabindex="-1"></a>    <span class="at">df =</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(</span>
<span id="cb90-32"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-32" tabindex="-1"></a>        <span class="st">&quot;logFC&quot;</span> <span class="ot">=</span> de_genes<span class="sc">$</span>logFC_binary,</span>
<span id="cb90-33"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-33" tabindex="-1"></a>        <span class="st">&quot;protein_coding_gene&quot;</span> <span class="ot">=</span> de_genes<span class="sc">$</span>protein_coding_gene</span>
<span id="cb90-34"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-34" tabindex="-1"></a>    )),</span>
<span id="cb90-35"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-35" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">list</span>(</span>
<span id="cb90-36"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-36" tabindex="-1"></a>        <span class="st">&quot;logFC&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;&lt;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;deepskyblue3&quot;</span>, <span class="st">&quot;&gt;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;brown2&quot;</span>),</span>
<span id="cb90-37"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-37" tabindex="-1"></a>        <span class="st">&quot;protein_coding_gene&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;TRUE&quot;</span> <span class="ot">=</span> <span class="st">&quot;darkseagreen3&quot;</span>, <span class="st">&quot;FALSE&quot;</span> <span class="ot">=</span> <span class="st">&quot;magenta&quot;</span>)</span>
<span id="cb90-38"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-38" tabindex="-1"></a>    )</span>
<span id="cb90-39"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb90-39" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;circlize&quot;</span>)</span>
<span id="cb91-2"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-3" tabindex="-1"></a><span class="do">## Plot</span></span>
<span id="cb91-4"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-4" tabindex="-1"></a><span class="fu">Heatmap</span>(lognorm_data,</span>
<span id="cb91-5"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-5" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;lognorm counts&quot;</span>,</span>
<span id="cb91-6"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-6" tabindex="-1"></a>    <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb91-7"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-7" tabindex="-1"></a>    <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb91-8"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-8" tabindex="-1"></a>    <span class="at">left_annotation =</span> gene_anno,</span>
<span id="cb91-9"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-9" tabindex="-1"></a>    <span class="at">row_km =</span> <span class="dv">2</span>,</span>
<span id="cb91-10"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-10" tabindex="-1"></a>    <span class="at">column_km =</span> <span class="dv">2</span>,</span>
<span id="cb91-11"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-11" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="sc">-</span><span class="fl">0.0001</span>, <span class="dv">00001</span>, <span class="dv">4</span>), <span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;lightsalmon&quot;</span>, <span class="st">&quot;darkred&quot;</span>)),</span>
<span id="cb91-12"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-12" tabindex="-1"></a>    <span class="at">row_title =</span> <span class="st">&quot;DEGs&quot;</span>,</span>
<span id="cb91-13"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-13" tabindex="-1"></a>    <span class="at">column_title =</span> <span class="st">&quot;Samples&quot;</span>,</span>
<span id="cb91-14"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-14" tabindex="-1"></a>    <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>),</span>
<span id="cb91-15"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-15" tabindex="-1"></a>    <span class="at">heatmap_width =</span> <span class="fu">unit</span>(<span class="fl">12.5</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb91-16"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-16" tabindex="-1"></a>    <span class="at">heatmap_height =</span> <span class="fu">unit</span>(<span class="fl">12.5</span>, <span class="st">&quot;cm&quot;</span>)</span>
<span id="cb91-17"><a href="differential-gene-expression-analysis-with-limma-voom.html#cb91-17" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/heat%20map-1.png" width="960" /></p>
<p class="exercise">
📝 <strong>Exercise 7</strong>: obtain the number of DEGs you got and represent them in a volcano plot and a heat map. Include all the sample and gene information you consider relevant in the latter.
</p>
</div>
</div>
<div id="references-2" class="section level2 unnumbered hasAnchor">
<h2>References<a href="differential-gene-expression-analysis-with-limma-voom.html#references-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>Li, W. V., &amp; Li, J. J. (2018). Modeling and analysis of RNA‐seq data: a review from a statistical perspective. Quantitative Biology, 6(3), 195-209.</p></li>
<li><p>Law, C. W., Chen, Y., Shi, W., &amp; Smyth, G. K. (2014). voom: Precision weights unlock linear model analysis tools for RNA-seq read counts. Genome biology, 15(2), 1-17.</p></li>
<li><p>Smyth, G. K., Ritchie, M., Thorne, N., Wettenhall, J., Shi, W., &amp; Hu, Y. (2002). limma: linear models for microarray and RNA-Seq data user’s guide. Bioinformatics Division, The Walter and Eliza Hall Institute of Medical Research, Melbourne, Australia.</p></li>
<li><p>van den Berg, S. M. (2022). Analysing data using linear models. Web site: <a href="https://bookdown.org/pingapang9/linear_models_bookdown/" class="uri">https://bookdown.org/pingapang9/linear_models_bookdown/</a></p></li>
<li><p>Wikipedia. (n.d.). Ordinary least squares. Web site: <a href="https://en.wikipedia.org/wiki/Ordinary_least_squares" class="uri">https://en.wikipedia.org/wiki/Ordinary_least_squares</a></p></li>
<li><p>Taboga, Marco (2021). “Generalized least squares”, Lectures on probability theory and mathematical statistics. Kindle Direct Publishing. Online appendix. <a href="https://www.statlect.com/fundamentals-of-statistics/generalized-least-squares" class="uri">https://www.statlect.com/fundamentals-of-statistics/generalized-least-squares</a>.</p></li>
<li><p>Documentation for lmFit: <a href="https://rdrr.io/bioc/limma/man/lmFit.html" class="uri">https://rdrr.io/bioc/limma/man/lmFit.html</a></p></li>
<li><p>The Pennsylvania State University. (2018). Statistical Analysis of Genomics Data. Web site: <a href="https://online.stat.psu.edu/stat555/node/36/" class="uri">https://online.stat.psu.edu/stat555/node/36/</a></p></li>
<li><p>Tushe, M. (2021). A Simple Trick to Understand the t-test. Web site: <a href="https://miroslavtushev.medium.com/a-simple-trick-to-understand-the-t-test-2c2a9e7f1dc5" class="uri">https://miroslavtushev.medium.com/a-simple-trick-to-understand-the-t-test-2c2a9e7f1dc5</a></p></li>
<li><p>Wikipedia. (n.d.). Empirical Bayes method. Web site: <a href="https://en.wikipedia.org/wiki/Empirical_Bayes_method#" class="uri">https://en.wikipedia.org/wiki/Empirical_Bayes_method#</a>:~:text=Empirical Bayes methods are procedures,before any data are observed.</p></li>
<li><p>Smyth, G. K. (2004). Linear models and empirical bayes methods for assessing differential expression in microarray experiments. Statistical applications in genetics and molecular biology, 3(1).</p></li>
<li><p>Documentation for topTable: <a href="https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/toptable" class="uri">https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/toptable</a></p></li>
<li><p>Robinson, D. (2014). How to interpret a p-value histogram. Web site: <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/" class="uri">http://varianceexplained.org/statistics/interpreting-pvalue-histogram/</a></p></li>
</ol>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-gene-expression-analysis-overview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interpreting-model-coefficients-with-exploremodelmatrix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
