<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>18 Downloading public Visium spatial data and visualizing it with spatialLIBD | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="18 Downloading public Visium spatial data and visualizing it with spatialLIBD | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="18 Downloading public Visium spatial data and visualizing it with spatialLIBD | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="18 Downloading public Visium spatial data and visualizing it with spatialLIBD | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"/>
<link rel="next" href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html"><i class="fa fa-check"></i><b>1</b> SPEAQeasy introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#overview-slides"><i class="fa fa-check"></i><b>1.1</b> 2022-04-20 overview slides</a></li>
<li class="chapter" data-level="1.2" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#speaqeasy-main-links"><i class="fa fa-check"></i><b>1.2</b> SPEAQeasy main links</a></li>
<li class="chapter" data-level="1.3" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#pipeline-outputs"><i class="fa fa-check"></i><b>1.3</b> Pipeline outputs</a></li>
<li class="chapter" data-level="1.4" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#exercises"><i class="fa fa-check"></i><b>1.4</b> Exercises</a></li>
<li class="chapter" data-level="1.5" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#solutions"><i class="fa fa-check"></i><b>1.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html"><i class="fa fa-check"></i><b>2</b> smokingMouse study</a>
<ul>
<li class="chapter" data-level="2.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#overview-1"><i class="fa fa-check"></i><b>2.2</b> Overview</a></li>
<li class="chapter" data-level="2.3" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#goals"><i class="fa fa-check"></i><b>2.3</b> Goals</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#main-objective"><i class="fa fa-check"></i><b>2.3.1</b> Main objective</a></li>
<li class="chapter" data-level="2.3.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#secondary-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Secondary objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#study-design"><i class="fa fa-check"></i><b>2.4</b> Study design</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html"><i class="fa fa-check"></i><b>3</b> SummarizedExperiment review</a>
<ul>
<li class="chapter" data-level="3.1" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#quiz"><i class="fa fa-check"></i><b>3.2</b> Quiz</a></li>
<li class="chapter" data-level="3.3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#exercises-1"><i class="fa fa-check"></i><b>3.3</b> Exercises</a></li>
<li class="chapter" data-level="3.4" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#solutions-1"><i class="fa fa-check"></i><b>3.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html"><i class="fa fa-check"></i><b>4</b> smokingMouse RSE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#download-data"><i class="fa fa-check"></i><b>4.1</b> Download data</a></li>
<li class="chapter" data-level="4.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#data-overview"><i class="fa fa-check"></i><b>4.2</b> Data overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#assays"><i class="fa fa-check"></i><b>4.2.1</b> Assays</a></li>
<li class="chapter" data-level="4.2.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#sample-data"><i class="fa fa-check"></i><b>4.2.2</b> Sample data</a></li>
<li class="chapter" data-level="4.2.3" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#gene-information"><i class="fa fa-check"></i><b>4.2.3</b> Gene Information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="isee-introduction.html"><a href="isee-introduction.html"><i class="fa fa-check"></i><b>5</b> iSEE introduction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="isee-introduction.html"><a href="isee-introduction.html#toy-rse-data"><i class="fa fa-check"></i><b>5.1</b> Toy RSE data</a></li>
<li class="chapter" data-level="5.2" data-path="isee-introduction.html"><a href="isee-introduction.html#isee"><i class="fa fa-check"></i><b>5.2</b> iSEE</a></li>
<li class="chapter" data-level="5.3" data-path="isee-introduction.html"><a href="isee-introduction.html#exercise-with-data-from-spatiallibd"><i class="fa fa-check"></i><b>5.3</b> Exercise with data from spatialLIBD</a></li>
<li class="chapter" data-level="5.4" data-path="isee-introduction.html"><a href="isee-introduction.html#community"><i class="fa fa-check"></i><b>5.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html"><i class="fa fa-check"></i><b>6</b> RSE plotting basics with smokingMouse</a>
<ul>
<li class="chapter" data-level="6.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#download-the-smokingmouse-data"><i class="fa fa-check"></i><b>6.1</b> Download the smokingMouse data</a></li>
<li class="chapter" data-level="6.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-preparation"><i class="fa fa-check"></i><b>6.2</b> Data preparation</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-normalization"><i class="fa fa-check"></i><b>6.2.1</b> Data normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#gene-filtering"><i class="fa fa-check"></i><b>6.2.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>6.3</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="6.4" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#quality-control-analysis"><i class="fa fa-check"></i><b>6.4</b> Quality Control Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#evaluate-qc-metrics-for-groups-of-samples"><i class="fa fa-check"></i><b>6.4.1</b> Evaluate QC metrics for groups of samples</a></li>
<li class="chapter" data-level="6.4.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#examine-relationships-between-qc-metrics-of-samples"><i class="fa fa-check"></i><b>6.4.2</b> Examine relationships between QC metrics of samples</a></li>
<li class="chapter" data-level="6.4.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#qc-sample-filtering"><i class="fa fa-check"></i><b>6.4.3</b> QC sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#miscellaneous-about-cpm-values"><i class="fa fa-check"></i><b>6.5</b> Miscellaneous about CPM values</a></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>7</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="7.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>7.1</b> Related past materials</a></li>
<li class="chapter" data-level="7.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>7.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="7.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>7.3</b> Live demo</a></li>
<li class="chapter" data-level="7.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-1"><i class="fa fa-check"></i><b>7.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>8</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>8.1</b> recount projects</a></li>
<li class="chapter" data-level="8.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>8.2</b> Using recount3</a></li>
<li class="chapter" data-level="8.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>8.3</b> Exercise</a></li>
<li class="chapter" data-level="8.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-2"><i class="fa fa-check"></i><b>8.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>9</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="9.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>9.1</b> Model objects in R</a></li>
<li class="chapter" data-level="9.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>9.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>9.3</b> Example 1</a></li>
<li class="chapter" data-level="9.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>9.4</b> Example 2</a></li>
<li class="chapter" data-level="9.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>9.5</b> Example 3</a></li>
<li class="chapter" data-level="9.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>9.6</b> Exercise</a></li>
<li class="chapter" data-level="9.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>9.7</b> To learn more</a></li>
<li class="chapter" data-level="9.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-3"><i class="fa fa-check"></i><b>9.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html"><i class="fa fa-check"></i><b>10</b> Differential gene expression analysis with limma</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#srp045638-data"><i class="fa fa-check"></i><b>10.1</b> SRP045638 data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#data-normalization-1"><i class="fa fa-check"></i><b>10.2</b> Data normalization</a></li>
<li class="chapter" data-level="10.3" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#differential-expression"><i class="fa fa-check"></i><b>10.3</b> Differential expression</a></li>
<li class="chapter" data-level="10.4" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#visualizing-degs"><i class="fa fa-check"></i><b>10.4</b> Visualizing DEGs</a></li>
<li class="chapter" data-level="10.5" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#community-4"><i class="fa fa-check"></i><b>10.5</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>11</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>11.1</b> Recap</a></li>
<li class="chapter" data-level="11.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>11.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>12</b> DGE model building with variancePartition</a>
<ul>
<li class="chapter" data-level="12.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#download-the-smokingmouse-data-and-subset-it"><i class="fa fa-check"></i><b>12.1</b> Download the smokingMouse data and subset it</a></li>
<li class="chapter" data-level="12.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#explore-gene-level-effects"><i class="fa fa-check"></i><b>12.2</b> Explore gene-level effects</a></li>
<li class="chapter" data-level="12.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#variance-partition-analysis"><i class="fa fa-check"></i><b>12.3</b> Variance Partition Analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>12.3.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="12.3.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>12.3.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="12.3.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>12.3.3</b> Examine expression of most affected genes by each sample variable</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#solutions-2"><i class="fa fa-check"></i><b>12.4</b> Solutions</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#exercise-1-1"><i class="fa fa-check"></i><b>12.4.1</b> Exercise 1</a></li>
<li class="chapter" data-level="12.4.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#exercise-2-1"><i class="fa fa-check"></i><b>12.4.2</b> Exercise 2</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>13</b> Expression heatmaps with ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="13.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#introduction-1"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#complexheatmap"><i class="fa fa-check"></i><b>13.2</b> ComplexHeatmap</a></li>
<li class="chapter" data-level="13.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#heatmap-with-expression-data"><i class="fa fa-check"></i><b>13.3</b> Heatmap with expression data</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#download-data-1"><i class="fa fa-check"></i><b>13.3.1</b> Download data</a></li>
<li class="chapter" data-level="13.3.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#prepare-the-data"><i class="fa fa-check"></i><b>13.3.2</b> Prepare the data</a></li>
<li class="chapter" data-level="13.3.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#create-annotations"><i class="fa fa-check"></i><b>13.3.3</b> Create annotations</a></li>
<li class="chapter" data-level="13.3.4" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#plot-our-heatmap"><i class="fa fa-check"></i><b>13.3.4</b> Plot our heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="complexheatmap-exercise-solution.html"><a href="complexheatmap-exercise-solution.html"><i class="fa fa-check"></i><b>14</b> ComplexHeatmap exercise solution</a></li>
<li class="chapter" data-level="15" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html"><i class="fa fa-check"></i><b>15</b> Exercise with variancePartition and ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="15.1" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html#steps"><i class="fa fa-check"></i><b>15.1</b> Steps:</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>16</b> Miscellaneous</a>
<ul>
<li class="chapter" data-level="16.1" data-path="miscellaneous.html"><a href="miscellaneous.html#postcards"><i class="fa fa-check"></i><b>16.1</b> postcards</a></li>
<li class="chapter" data-level="16.2" data-path="miscellaneous.html"><a href="miscellaneous.html#organizing-your-work"><i class="fa fa-check"></i><b>16.2</b> Organizing your work</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><i class="fa fa-check"></i><b>17</b> Introduction to the world of spatially-resolved transcriptomics with Visium data</a>
<ul>
<li class="chapter" data-level="17.1" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html#community-5"><i class="fa fa-check"></i><b>17.1</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><i class="fa fa-check"></i><b>18</b> Downloading public Visium spatial data and visualizing it with spatialLIBD</a>
<ul>
<li class="chapter" data-level="18.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#download-data-from-10x-genomics"><i class="fa fa-check"></i><b>18.1</b> Download data from 10x Genomics</a></li>
<li class="chapter" data-level="18.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#add-gene-annotation-information"><i class="fa fa-check"></i><b>18.2</b> Add gene annotation information</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-10x"><i class="fa fa-check"></i><b>18.2.1</b> From 10x</a></li>
<li class="chapter" data-level="18.2.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-gencode"><i class="fa fa-check"></i><b>18.2.2</b> From Gencode</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#wrapper-functions-for-reading-the-data"><i class="fa fa-check"></i><b>18.3</b> Wrapper functions for reading the data</a></li>
<li class="chapter" data-level="18.4" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#visualize-spatial-plots"><i class="fa fa-check"></i><b>18.4</b> Visualize spatial plots</a></li>
<li class="chapter" data-level="18.5" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#exercise-3"><i class="fa fa-check"></i><b>18.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>19</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="19.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>19.1</b> Spatial registration</a></li>
<li class="chapter" data-level="19.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-4"><i class="fa fa-check"></i><b>19.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
<li class="chapter" data-level="20" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>20</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="20.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>20.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>20.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="20.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis-1"><i class="fa fa-check"></i><b>20.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>20.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="20.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>20.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="20.4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-2"><i class="fa fa-check"></i><b>20.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>21</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="21.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>21.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="21.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>21.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>21.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="21.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>21.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="21.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>21.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="21.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>21.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="21.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>21.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>21.3</b> DE visualization</a></li>
<li class="chapter" data-level="21.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-3"><i class="fa fa-check"></i><b>21.4</b> References</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd" class="section level1 hasAnchor" number="18">
<h1><span class="header-section-number">18</span> Downloading public Visium spatial data and visualizing it with spatialLIBD<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Leo</p>
<div id="download-data-from-10x-genomics" class="section level2 hasAnchor" number="18.1">
<h2><span class="header-section-number">18.1</span> Download data from 10x Genomics<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#download-data-from-10x-genomics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll run some code adapted from <a href="http://research.libd.org/spatialLIBD/articles/TenX_data_download.html" class="uri">http://research.libd.org/spatialLIBD/articles/TenX_data_download.html</a> (also available at <a href="https://bioconductor.org/packages/devel/data/experiment/vignettes/spatialLIBD/inst/doc/TenX_data_download.html" class="uri">https://bioconductor.org/packages/devel/data/experiment/vignettes/spatialLIBD/inst/doc/TenX_data_download.html</a>). We’ll explore spatially resolved transcriptomics data from the <a href="https://www.10xgenomics.com/products/spatial-gene-expression">Visium platform by 10x Genomics</a>.</p>
<p>We’ll use <em><a href="https://bioconductor.org/packages/3.18/BiocFileCache">BiocFileCache</a></em> to keep the data in a local cache in case we want to run this example again and don’t want to re-download the data from the web.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-1" tabindex="-1"></a><span class="do">## Download and save a local cache of the data provided by 10x Genomics</span></span>
<span id="cb128-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-2" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> BiocFileCache<span class="sc">::</span><span class="fu">BiocFileCache</span>()</span>
<span id="cb128-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-3" tabindex="-1"></a>lymph.url <span class="ot">&lt;-</span></span>
<span id="cb128-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-4" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb128-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-5" tabindex="-1"></a>        <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/&quot;</span>,</span>
<span id="cb128-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-6" tabindex="-1"></a>        <span class="st">&quot;1.1.0/V1_Human_Lymph_Node/&quot;</span>,</span>
<span id="cb128-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-7" tabindex="-1"></a>        <span class="fu">c</span>(</span>
<span id="cb128-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-8" tabindex="-1"></a>            <span class="st">&quot;V1_Human_Lymph_Node_filtered_feature_bc_matrix.tar.gz&quot;</span>,</span>
<span id="cb128-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-9" tabindex="-1"></a>            <span class="st">&quot;V1_Human_Lymph_Node_spatial.tar.gz&quot;</span>,</span>
<span id="cb128-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-10" tabindex="-1"></a>            <span class="st">&quot;V1_Human_Lymph_Node_analysis.tar.gz&quot;</span></span>
<span id="cb128-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-11" tabindex="-1"></a>        )</span>
<span id="cb128-12"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-12" tabindex="-1"></a>    )</span>
<span id="cb128-13"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-13" tabindex="-1"></a>lymph.data <span class="ot">&lt;-</span> <span class="fu">sapply</span>(lymph.url, BiocFileCache<span class="sc">::</span>bfcrpath, <span class="at">x =</span> bfc)</span>
<span id="cb128-14"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-14" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Human_Lymph_Node/V1_Human_Lymph_Node_filtered_feature_bc_matrix.tar.gz&#39;</span></span>
<span id="cb128-15"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-15" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Human_Lymph_Node/V1_Human_Lymph_Node_spatial.tar.gz&#39;</span></span>
<span id="cb128-16"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb128-16" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Human_Lymph_Node/V1_Human_Lymph_Node_analysis.tar.gz&#39;</span></span></code></pre></div>
<p>10x Genomics provides the files in compressed tarballs (<code>.tar.gz</code> file extension). Which is why we’ll need to use <code>utils::untar()</code> to decompress the files. This will create new directories and we will use <code>list.files()</code> to see what files these directories contain.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-1" tabindex="-1"></a><span class="do">## Extract the files to a temporary location</span></span>
<span id="cb129-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-2" tabindex="-1"></a><span class="do">## (they&#39;ll be deleted once you close your R session)</span></span>
<span id="cb129-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-3" tabindex="-1"></a>xx <span class="ot">&lt;-</span> <span class="fu">sapply</span>(lymph.data, utils<span class="sc">::</span>untar, <span class="at">exdir =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">&quot;outs&quot;</span>))</span>
<span id="cb129-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-4" tabindex="-1"></a><span class="do">## The names are the URLs, which are long and thus too wide to be shown here,</span></span>
<span id="cb129-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-5" tabindex="-1"></a><span class="do">## so we shorten them to only show the file name prior to displaying the</span></span>
<span id="cb129-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-6" tabindex="-1"></a><span class="do">## utils::untar() output status</span></span>
<span id="cb129-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-7" tabindex="-1"></a><span class="fu">names</span>(xx) <span class="ot">&lt;-</span> <span class="fu">basename</span>(<span class="fu">names</span>(xx))</span>
<span id="cb129-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-8" tabindex="-1"></a>xx</span>
<span id="cb129-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-9" tabindex="-1"></a><span class="co">#&gt; V1_Human_Lymph_Node_filtered_feature_bc_matrix.tar.gz.BFC4                    V1_Human_Lymph_Node_spatial.tar.gz.BFC5 </span></span>
<span id="cb129-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-10" tabindex="-1"></a><span class="co">#&gt;                                                          0                                                          0 </span></span>
<span id="cb129-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-11" tabindex="-1"></a><span class="co">#&gt;                   V1_Human_Lymph_Node_analysis.tar.gz.BFC6 </span></span>
<span id="cb129-12"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-12" tabindex="-1"></a><span class="co">#&gt;                                                          0</span></span>
<span id="cb129-13"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-13" tabindex="-1"></a></span>
<span id="cb129-14"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-14" tabindex="-1"></a><span class="do">## List the files we downloaded and extracted</span></span>
<span id="cb129-15"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-15" tabindex="-1"></a><span class="do">## These files are typically SpaceRanger outputs</span></span>
<span id="cb129-16"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-16" tabindex="-1"></a>lymph.dirs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(</span>
<span id="cb129-17"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-17" tabindex="-1"></a>    <span class="fu">tempdir</span>(), <span class="st">&quot;outs&quot;</span>,</span>
<span id="cb129-18"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-18" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">&quot;filtered_feature_bc_matrix&quot;</span>, <span class="st">&quot;spatial&quot;</span>, <span class="st">&quot;raw_feature_bc_matrix&quot;</span>, <span class="st">&quot;analysis&quot;</span>)</span>
<span id="cb129-19"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-19" tabindex="-1"></a>)</span>
<span id="cb129-20"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-20" tabindex="-1"></a><span class="fu">list.files</span>(lymph.dirs)</span>
<span id="cb129-21"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-21" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;aligned_fiducials.jpg&quot;     &quot;barcodes.tsv.gz&quot;           &quot;clustering&quot;                &quot;detected_tissue_image.jpg&quot;</span></span>
<span id="cb129-22"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-22" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;diffexp&quot;                   &quot;features.tsv.gz&quot;           &quot;matrix.mtx.gz&quot;             &quot;pca&quot;                      </span></span>
<span id="cb129-23"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-23" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;scalefactors_json.json&quot;    &quot;tissue_hires_image.png&quot;    &quot;tissue_lowres_image.png&quot;   &quot;tissue_positions_list.csv&quot;</span></span>
<span id="cb129-24"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb129-24" tabindex="-1"></a><span class="co">#&gt; [13] &quot;tsne&quot;                      &quot;umap&quot;</span></span></code></pre></div>
</div>
<div id="add-gene-annotation-information" class="section level2 hasAnchor" number="18.2">
<h2><span class="header-section-number">18.2</span> Add gene annotation information<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#add-gene-annotation-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>These files do not include much information about the genes, such as their chromosomes, coordinates, and other gene annotation information. We thus recommend that you read in this information from a gene annotation file: typically a <code>gtf</code> file. For a real case scenario, you’ll mostly likely have access to the GTF file provided by 10x Genomics. However, we cannot download that file without downloading other files for this example.</p>
<div id="from-10x" class="section level3 hasAnchor" number="18.2.1">
<h3><span class="header-section-number">18.2.1</span> From 10x<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-10x" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Depending on the version of <code>spaceranger</code> you used, you might have used different GTF files 10x Genomics has made available at <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest</a> and described at <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build</a>. These files are too big though and we won’t download them in this example. For instance, <em>References - 2020-A (July 7, 2020)</em> for <em>Human reference (GRCh38)</em> is 11 GB in size and contains files we do not need here. If you did have the file locally, we would just need the path to this GTF file later.</p>
<p>For example, in our computing cluster this GTF file is located at the following path and is 1.4 GB in size:</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb130-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb130-1" tabindex="-1"></a><span class="ex">$</span> cd /dcs04/lieber/lcolladotor/annotationFiles_LIBD001/10x/refdata-gex-GRCh38-2020-A</span>
<span id="cb130-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb130-2" tabindex="-1"></a><span class="ex">$</span> du <span class="at">-sh</span> <span class="at">--apparent-size</span> genes/genes.gtf</span>
<span id="cb130-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb130-3" tabindex="-1"></a><span class="ex">1.4G</span>    genes/genes.gtf</span></code></pre></div>
</div>
<div id="from-gencode" class="section level3 hasAnchor" number="18.2.2">
<h3><span class="header-section-number">18.2.2</span> From Gencode<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-gencode" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In the absence of the GTF file used by 10x Genomics, we’ll use the GTF file from Gencode v32 which we can download and that is much smaller. That’s because the build notes from <em>References - 2020-A (July 7, 2020)</em> and <em>Human reference, GRCh38 (GENCODE v32/Ensembl 98)</em> at <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38_2020A" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38_2020A</a> show that 10x Genomics used Gencode v32. They also used Ensembl version 98 which is why a few genes we have in our object are going to be missing. Let’s download this GTF file.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-1" tabindex="-1"></a><span class="do">## Download the Gencode v32 GTF file and cache it</span></span>
<span id="cb131-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-2" tabindex="-1"></a>gtf_cache <span class="ot">&lt;-</span> BiocFileCache<span class="sc">::</span><span class="fu">bfcrpath</span>(</span>
<span id="cb131-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-3" tabindex="-1"></a>    bfc,</span>
<span id="cb131-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-4" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb131-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-5" tabindex="-1"></a>        <span class="st">&quot;ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/&quot;</span>,</span>
<span id="cb131-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-6" tabindex="-1"></a>        <span class="st">&quot;release_32/gencode.v32.annotation.gtf.gz&quot;</span></span>
<span id="cb131-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-7" tabindex="-1"></a>    )</span>
<span id="cb131-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-8" tabindex="-1"></a>)</span>
<span id="cb131-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-9" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.annotation.gtf.gz&#39;</span></span>
<span id="cb131-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-10" tabindex="-1"></a></span>
<span id="cb131-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-11" tabindex="-1"></a><span class="do">## Show the GTF cache location</span></span>
<span id="cb131-12"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-12" tabindex="-1"></a>gtf_cache</span>
<span id="cb131-13"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-13" tabindex="-1"></a><span class="co">#&gt;                                                                            BFC7 </span></span>
<span id="cb131-14"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb131-14" tabindex="-1"></a><span class="co">#&gt; &quot;/github/home/.cache/R/BiocFileCache/25c7925984a_gencode.v32.annotation.gtf.gz&quot;</span></span></code></pre></div>
</div>
</div>
<div id="wrapper-functions-for-reading-the-data" class="section level2 hasAnchor" number="18.3">
<h2><span class="header-section-number">18.3</span> Wrapper functions for reading the data<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#wrapper-functions-for-reading-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To facilitate reading in the data and preparing it to visualize it interactively using <code>spatialLIBD::run_app()</code>, we implemented <code>read10xVisiumWrapper()</code> which expands <code>SpatialExperiment::read10xVisium()</code> and performs the steps described in more detail at <a href="http://research.libd.org/spatialLIBD/articles/TenX_data_download.html" class="uri">http://research.libd.org/spatialLIBD/articles/TenX_data_download.html</a>. In this example, we’ll load all four images created by SpaceRanger: lowres, hires, detected, and aligned. That way we can toggle between them on the web application.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-1" tabindex="-1"></a><span class="do">## Import the data as a SpatialExperiment object using wrapper functions</span></span>
<span id="cb132-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-2" tabindex="-1"></a><span class="do">## provided by spatialLIBD</span></span>
<span id="cb132-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;spatialLIBD&quot;</span>)</span>
<span id="cb132-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-4" tabindex="-1"></a>spe_wrapper <span class="ot">&lt;-</span> <span class="fu">read10xVisiumWrapper</span>(</span>
<span id="cb132-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-5" tabindex="-1"></a>    <span class="at">samples =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">&quot;outs&quot;</span>),</span>
<span id="cb132-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-6" tabindex="-1"></a>    <span class="at">sample_id =</span> <span class="st">&quot;lymph&quot;</span>,</span>
<span id="cb132-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-7" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">&quot;sparse&quot;</span>, <span class="at">data =</span> <span class="st">&quot;filtered&quot;</span>,</span>
<span id="cb132-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-8" tabindex="-1"></a>    <span class="at">images =</span> <span class="fu">c</span>(<span class="st">&quot;lowres&quot;</span>, <span class="st">&quot;hires&quot;</span>, <span class="st">&quot;detected&quot;</span>, <span class="st">&quot;aligned&quot;</span>), <span class="at">load =</span> <span class="cn">TRUE</span>,</span>
<span id="cb132-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-9" tabindex="-1"></a>    <span class="at">reference_gtf =</span> gtf_cache</span>
<span id="cb132-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-10" tabindex="-1"></a>)</span>
<span id="cb132-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-11" tabindex="-1"></a><span class="co">#&gt; 2024-05-16 02:43:24.074376 SpatialExperiment::read10xVisium: reading basic data from SpaceRanger</span></span>
<span id="cb132-12"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-12" tabindex="-1"></a><span class="co">#&gt; 2024-05-16 02:43:37.535698 read10xVisiumAnalysis: reading analysis output from SpaceRanger</span></span>
<span id="cb132-13"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-13" tabindex="-1"></a><span class="co">#&gt; 2024-05-16 02:43:37.644975 add10xVisiumAnalysis: adding analysis output from SpaceRanger</span></span>
<span id="cb132-14"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-14" tabindex="-1"></a><span class="co">#&gt; 2024-05-16 02:43:37.933207 rtracklayer::import: reading the reference GTF file</span></span>
<span id="cb132-15"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-15" tabindex="-1"></a><span class="co">#&gt; 2024-05-16 02:44:05.569883 adding gene information to the SPE object</span></span>
<span id="cb132-16"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-16" tabindex="-1"></a><span class="co">#&gt; Warning: Gene IDs did not match. This typically happens when you are not using the same GTF file as the one that was</span></span>
<span id="cb132-17"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-17" tabindex="-1"></a><span class="co">#&gt; used by SpaceRanger. For example, one file uses GENCODE IDs and the other one ENSEMBL IDs. read10xVisiumWrapper() will</span></span>
<span id="cb132-18"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-18" tabindex="-1"></a><span class="co">#&gt; try to convert them to ENSEMBL IDs.</span></span>
<span id="cb132-19"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-19" tabindex="-1"></a><span class="co">#&gt; Warning: Dropping 29 out of 36601 genes for which we don&#39;t have information on the reference GTF file. This typically</span></span>
<span id="cb132-20"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-20" tabindex="-1"></a><span class="co">#&gt; happens when you are not using the same GTF file as the one that was used by SpaceRanger.</span></span>
<span id="cb132-21"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-21" tabindex="-1"></a><span class="co">#&gt; 2024-05-16 02:44:06.111727 adding information used by spatialLIBD</span></span>
<span id="cb132-22"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-22" tabindex="-1"></a></span>
<span id="cb132-23"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-23" tabindex="-1"></a><span class="do">## Explore the resulting SpatialExperiment object</span></span>
<span id="cb132-24"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-24" tabindex="-1"></a>spe_wrapper</span>
<span id="cb132-25"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-25" tabindex="-1"></a><span class="co">#&gt; class: SpatialExperiment </span></span>
<span id="cb132-26"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-26" tabindex="-1"></a><span class="co">#&gt; dim: 36572 4035 </span></span>
<span id="cb132-27"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-27" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb132-28"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-28" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb132-29"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-29" tabindex="-1"></a><span class="co">#&gt; rownames(36572): ENSG00000243485.5 ENSG00000237613.2 ... ENSG00000198695.2 ENSG00000198727.2</span></span>
<span id="cb132-30"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-30" tabindex="-1"></a><span class="co">#&gt; rowData names(6): source type ... gene_type gene_search</span></span>
<span id="cb132-31"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-31" tabindex="-1"></a><span class="co">#&gt; colnames(4035): AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 ... TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1</span></span>
<span id="cb132-32"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-32" tabindex="-1"></a><span class="co">#&gt; colData names(20): sample_id in_tissue ... expr_chrM_ratio ManualAnnotation</span></span>
<span id="cb132-33"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-33" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(3): 10x_pca 10x_tsne 10x_umap</span></span>
<span id="cb132-34"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-34" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb132-35"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-35" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span>
<span id="cb132-36"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-36" tabindex="-1"></a><span class="co">#&gt; spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span id="cb132-37"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-37" tabindex="-1"></a><span class="co">#&gt; imgData names(4): sample_id image_id data scaleFactor</span></span>
<span id="cb132-38"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-38" tabindex="-1"></a></span>
<span id="cb132-39"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-39" tabindex="-1"></a><span class="do">## Size of the data</span></span>
<span id="cb132-40"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-40" tabindex="-1"></a>lobstr<span class="sc">::</span><span class="fu">obj_size</span>(spe_wrapper)</span>
<span id="cb132-41"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb132-41" tabindex="-1"></a><span class="co">#&gt; 410.19 MB</span></span></code></pre></div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-1" tabindex="-1"></a><span class="do">## Run our shiny app</span></span>
<span id="cb133-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-2" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">interactive</span>()) {</span>
<span id="cb133-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-3" tabindex="-1"></a>    vars <span class="ot">&lt;-</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spe_wrapper))</span>
<span id="cb133-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-4" tabindex="-1"></a></span>
<span id="cb133-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-5" tabindex="-1"></a>    <span class="fu">run_app</span>(</span>
<span id="cb133-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-6" tabindex="-1"></a>        spe_wrapper,</span>
<span id="cb133-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-7" tabindex="-1"></a>        <span class="at">sce_layer =</span> <span class="cn">NULL</span>,</span>
<span id="cb133-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-8" tabindex="-1"></a>        <span class="at">modeling_results =</span> <span class="cn">NULL</span>,</span>
<span id="cb133-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-9" tabindex="-1"></a>        <span class="at">sig_genes =</span> <span class="cn">NULL</span>,</span>
<span id="cb133-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-10" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">&quot;spatialLIBD: human lymph node by 10x Genomics (made with wrapper)&quot;</span>,</span>
<span id="cb133-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-11" tabindex="-1"></a>        <span class="at">spe_discrete_vars =</span> <span class="fu">c</span>(vars[<span class="fu">grep</span>(<span class="st">&quot;10x_&quot;</span>, vars)], <span class="st">&quot;ManualAnnotation&quot;</span>),</span>
<span id="cb133-12"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-12" tabindex="-1"></a>        <span class="at">spe_continuous_vars =</span> <span class="fu">c</span>(<span class="st">&quot;sum_umi&quot;</span>, <span class="st">&quot;sum_gene&quot;</span>, <span class="st">&quot;expr_chrM&quot;</span>, <span class="st">&quot;expr_chrM_ratio&quot;</span>),</span>
<span id="cb133-13"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-13" tabindex="-1"></a>        <span class="at">default_cluster =</span> <span class="st">&quot;10x_graphclust&quot;</span></span>
<span id="cb133-14"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-14" tabindex="-1"></a>    )</span>
<span id="cb133-15"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb133-15" tabindex="-1"></a>}</span></code></pre></div>
<p>The result should look pretty much like <a href="https://libd.shinyapps.io/spatialLIBD_Human_Lymph_Node_10x/" class="uri">https://libd.shinyapps.io/spatialLIBD_Human_Lymph_Node_10x/</a>.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/cCUHRdI4NSc" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen>
</iframe>
<p>The apps showcased above are linked to and described at <a href="http://research.libd.org/spatialDLPFC/#interactive-websites" class="uri">http://research.libd.org/spatialDLPFC/#interactive-websites</a>. This video was part of Nicholas J. Eagles recent LIBD seminar presentation.</p>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
Congrats Nick for successfully completing your 1st🥇 <a href="https://twitter.com/LieberInstitute?ref_src=twsrc%5Etfw"><span class="citation">@LieberInstitute</span></a> seminar 🙌🏽<br><br>Nick presented <a href="https://twitter.com/hashtag/spatialDLPFC?src=hash&amp;ref_src=twsrc%5Etfw">#spatialDLPFC</a> spot deconvolution results <a href="https://twitter.com/10xGenomics?ref_src=twsrc%5Etfw"><span class="citation">@10xGenomics</span></a> <a href="https://twitter.com/hashtag/snRNAseq?src=hash&amp;ref_src=twsrc%5Etfw">#snRNAseq</a> <a href="https://twitter.com/hashtag/Visium?src=hash&amp;ref_src=twsrc%5Etfw">#Visium</a> <a href="https://twitter.com/hashtag/VisiumSPG?src=hash&amp;ref_src=twsrc%5Etfw">#VisiumSPG</a><br><br>🛝 <a href="https://t.co/2wMLSkTLyW">https://t.co/2wMLSkTLyW</a><br><br>Interact with the data <a href="https://t.co/ya4oyiMVsN">https://t.co/ya4oyiMVsN</a><a href="https://t.co/cTdGNswysy">https://t.co/cTdGNswysy</a> <a href="https://t.co/Ggmnnmj2nA">pic.twitter.com/Ggmnnmj2nA</a>
</p>
— 🇲🇽 Leonardo Collado-Torres (<span class="citation">@lcolladotor</span>) <a href="https://twitter.com/lcolladotor/status/1673831490965106691?ref_src=twsrc%5Etfw">June 27, 2023</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
</div>
<div id="visualize-spatial-plots" class="section level2 hasAnchor" number="18.4">
<h2><span class="header-section-number">18.4</span> Visualize spatial plots<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#visualize-spatial-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have explored the spatially resolved transcriptomics data with the <code>spatialLIBD::run_app()</code> <code>shiny</code> website, lets make some plots ourselves.</p>
<p>Check out the documentation for these two functions:</p>
<ul>
<li><code>spatialLIBD::vis_gene()</code> <a href="http://research.libd.org/spatialLIBD/reference/vis_gene.html" class="uri">http://research.libd.org/spatialLIBD/reference/vis_gene.html</a> for continuous variables, including gene expression values</li>
<li><code>spatialLIBD::vis_clus()</code> <a href="http://research.libd.org/spatialLIBD/reference/vis_clus.html" class="uri">http://research.libd.org/spatialLIBD/reference/vis_clus.html</a> for discrete variables</li>
</ul>
</div>
<div id="exercise-3" class="section level2 hasAnchor" number="18.5">
<h2><span class="header-section-number">18.5</span> Exercise<a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#exercise-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}
</style>
<p class="exercise">
<strong>Exercise 1</strong>:
</p>
<div class="alert alert-info">
<ul>
<li>Plot a <strong>discrete</strong> variable of your choice with and without the histology image on the background.
<ul>
<li>For example, visualize the k-means with <span class="math inline">\(k = 10\)</span> clustering results computed by <code>SpaceRanger</code>.</li>
</ul></li>
<li>Use custom colors of your choice. See <a href="https://emilhvitfeldt.github.io/r-color-palettes/discrete.html" class="uri">https://emilhvitfeldt.github.io/r-color-palettes/discrete.html</a> to choose a color palette.
<ul>
<li>For example, use the <code>Polychrome::palette36()</code> colors.</li>
</ul></li>
</ul>
</div>
<style>
p.solution  {
background-color: #C093D6;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}
</style>
<p class="solution">
<strong>Solution 1</strong>:
The function for plotting discrete variables is <code>spatialLIBD::vis_clus()</code> which is documented at <a href="http://research.libd.org/spatialLIBD/reference/vis_clus.html" class="uri">http://research.libd.org/spatialLIBD/reference/vis_clus.html</a>. After reading the documentation website for this function, we are ready to write the code to solve our exercise problem.
</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-1" tabindex="-1"></a><span class="do">## Basic spatial graph visualizing a discrete variable that we provided to</span></span>
<span id="cb134-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-2" tabindex="-1"></a><span class="do">## the argument &quot;clustervar&quot;. Here we chose to visualize</span></span>
<span id="cb134-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-3" tabindex="-1"></a><span class="do">## &quot;10x_kmeans_10_clusters&quot; which contains the clustering results from the</span></span>
<span id="cb134-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-4" tabindex="-1"></a><span class="do">## K-means algorithm when k = 10.</span></span>
<span id="cb134-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-5" tabindex="-1"></a><span class="do">## We are plotting the one sample we have called &quot;lymph&quot;. This is the same</span></span>
<span id="cb134-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-6" tabindex="-1"></a><span class="do">## name we chose earlier when we ran spatialLIBD::read10xVisiumWrapper().</span></span>
<span id="cb134-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-7" tabindex="-1"></a><span class="fu">vis_clus</span>(</span>
<span id="cb134-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-8" tabindex="-1"></a>    <span class="at">spe =</span> spe_wrapper,</span>
<span id="cb134-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-9" tabindex="-1"></a>    <span class="at">sampleid =</span> <span class="st">&quot;lymph&quot;</span>,</span>
<span id="cb134-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-10" tabindex="-1"></a>    <span class="at">clustervar =</span> <span class="st">&quot;10x_kmeans_10_clusters&quot;</span></span>
<span id="cb134-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb134-11" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-63-1.png" width="672" /></p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-1" tabindex="-1"></a></span>
<span id="cb135-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-2" tabindex="-1"></a><span class="do">## Next we ignore the histology image and stop plotting it by setting</span></span>
<span id="cb135-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-3" tabindex="-1"></a><span class="do">## &quot;spatial = FALSE&quot;</span></span>
<span id="cb135-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-4" tabindex="-1"></a><span class="fu">vis_clus</span>(</span>
<span id="cb135-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-5" tabindex="-1"></a>    <span class="at">spe =</span> spe_wrapper,</span>
<span id="cb135-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-6" tabindex="-1"></a>    <span class="at">sampleid =</span> <span class="st">&quot;lymph&quot;</span>,</span>
<span id="cb135-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-7" tabindex="-1"></a>    <span class="at">clustervar =</span> <span class="st">&quot;10x_kmeans_10_clusters&quot;</span>,</span>
<span id="cb135-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-8" tabindex="-1"></a>    <span class="at">spatial =</span> <span class="cn">FALSE</span></span>
<span id="cb135-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb135-9" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-63-2.png" width="672" /></p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-1" tabindex="-1"></a></span>
<span id="cb136-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-2" tabindex="-1"></a><span class="do">## Finally, we use the &quot;colors&quot; argument to specify our own colors. We use</span></span>
<span id="cb136-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-3" tabindex="-1"></a><span class="do">## the general setNames() R function to create a named vector that has as</span></span>
<span id="cb136-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-4" tabindex="-1"></a><span class="do">## values the colors and as names the same names we have for our &quot;clustervar&quot;.</span></span>
<span id="cb136-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-5" tabindex="-1"></a><span class="fu">vis_clus</span>(</span>
<span id="cb136-6"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-6" tabindex="-1"></a>    <span class="at">spe =</span> spe_wrapper,</span>
<span id="cb136-7"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-7" tabindex="-1"></a>    <span class="at">sampleid =</span> <span class="st">&quot;lymph&quot;</span>,</span>
<span id="cb136-8"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-8" tabindex="-1"></a>    <span class="at">clustervar =</span> <span class="st">&quot;10x_kmeans_10_clusters&quot;</span>,</span>
<span id="cb136-9"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-9" tabindex="-1"></a>    <span class="at">spatial =</span> <span class="cn">FALSE</span>,</span>
<span id="cb136-10"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-10" tabindex="-1"></a>    <span class="at">colors =</span> <span class="fu">setNames</span>(Polychrome<span class="sc">::</span><span class="fu">palette36.colors</span>(<span class="dv">10</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb136-11"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb136-11" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-63-3.png" width="672" /></p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb137-1" tabindex="-1"></a></span>
<span id="cb137-2"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb137-2" tabindex="-1"></a><span class="do">## Here we can see the vector we provided as input to &quot;colors&quot;:</span></span>
<span id="cb137-3"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb137-3" tabindex="-1"></a><span class="fu">setNames</span>(Polychrome<span class="sc">::</span><span class="fu">palette36.colors</span>(<span class="dv">10</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb137-4"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb137-4" tabindex="-1"></a><span class="co">#&gt;         1         2         3         4         5         6         7         8         9        10 </span></span>
<span id="cb137-5"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#cb137-5" tabindex="-1"></a><span class="co">#&gt; &quot;#5A5156&quot; &quot;#E4E1E3&quot; &quot;#F6222E&quot; &quot;#FE00FA&quot; &quot;#16FF32&quot; &quot;#3283FE&quot; &quot;#FEAF16&quot; &quot;#B00068&quot; &quot;#1CFFCE&quot; &quot;#90AD1C&quot;</span></span></code></pre></div>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
