<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Expression heatmaps with ComplexHeatmap | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="9 Expression heatmaps with ComplexHeatmap | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Expression heatmaps with ComplexHeatmap | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Expression heatmaps with ComplexHeatmap | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="differential-gene-expression-exercise.html"/>
<link rel="next" href="complexheatmap-exercise-solution.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#quiz"><i class="fa fa-check"></i><b>1.2</b> Quiz</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.3</b> Exercises</a></li>
<li class="chapter" data-level="1.4" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.5</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-1"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>9</b> Expression heatmaps with ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="9.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#introduction"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#complexheatmap"><i class="fa fa-check"></i><b>9.2</b> ComplexHeatmap</a></li>
<li class="chapter" data-level="9.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#heatmap-with-expression-data"><i class="fa fa-check"></i><b>9.3</b> Heatmap with expression data</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#download-data"><i class="fa fa-check"></i><b>9.3.1</b> Download data</a></li>
<li class="chapter" data-level="9.3.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#prepare-the-data"><i class="fa fa-check"></i><b>9.3.2</b> Prepare the data</a></li>
<li class="chapter" data-level="9.3.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#create-annotations"><i class="fa fa-check"></i><b>9.3.3</b> Create annotations</a></li>
<li class="chapter" data-level="9.3.4" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#plot-our-heatmap"><i class="fa fa-check"></i><b>9.3.4</b> Plot our heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="complexheatmap-exercise-solution.html"><a href="complexheatmap-exercise-solution.html"><i class="fa fa-check"></i><b>10</b> ComplexHeatmap exercise solution</a></li>
<li class="chapter" data-level="11" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>11</b> Research talks</a>
<ul>
<li class="chapter" data-level="11.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>11.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="11.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>11.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="11.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>11.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>12</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="12.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>12.1</b> Related past materials</a></li>
<li class="chapter" data-level="12.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>12.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="12.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>12.3</b> Live demo</a></li>
<li class="chapter" data-level="12.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-2"><i class="fa fa-check"></i><b>12.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>13</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="13.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing-overview"><i class="fa fa-check"></i><b>13.1</b> Single cell RNA sequencing overview</a></li>
<li class="chapter" data-level="13.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>13.2</b> The SingleCellExperiment class</a></li>
<li class="chapter" data-level="13.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>13.3</b> Quality Control</a></li>
<li class="chapter" data-level="13.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>13.4</b> Normalization</a></li>
<li class="chapter" data-level="13.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>13.5</b> Feature selection</a></li>
<li class="chapter" data-level="13.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>13.6</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="13.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>13.7</b> Clustering</a></li>
<li class="chapter" data-level="13.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>13.8</b> Marker gene detection</a></li>
<li class="chapter" data-level="13.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>13.9</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>14</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>14.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>14.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>14.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="14.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>14.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>15</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="15.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>15.1</b> Spatial registration</a></li>
<li class="chapter" data-level="15.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>15.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>16</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="16.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>16.1</b> here</a></li>
<li class="chapter" data-level="16.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>16.2</b> Usethis</a></li>
<li class="chapter" data-level="16.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>16.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>16.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="16.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>16.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="16.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>16.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="16.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>16.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>16.4</b> R websites</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>16.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="16.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>16.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="16.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>16.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="16.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>16.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>16.5</b> postcards</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>16.5.1</b> Installation</a></li>
<li class="chapter" data-level="16.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>16.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>16.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>16.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="16.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>16.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="16.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>16.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="16.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>16.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="16.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>16.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-4"><i class="fa fa-check"></i><b>16.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="expression-heatmaps-with-complexheatmap" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> Expression heatmaps with ComplexHeatmap<a href="expression-heatmaps-with-complexheatmap.html#expression-heatmaps-with-complexheatmap" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Renee</p>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
Congrats to <a href="https://twitter.com/rodriguez_lion?ref_src=twsrc%5Etfw"><span class="citation">@rodriguez_lion</span></a> - final chapter of his PhD thesis - as well as co-1st author <a href="https://twitter.com/mattntran?ref_src=twsrc%5Etfw"><span class="citation">@mattntran</span></a>, and <a href="https://twitter.com/SubmarineGene?ref_src=twsrc%5Etfw"><span class="citation">@SubmarineGene</span></a> who joined the project late, but was invaluable to getting it finished, and much thanks to <a href="https://twitter.com/CerceoPage?ref_src=twsrc%5Etfw"><span class="citation">@CerceoPage</span></a> <a href="https://twitter.com/lcolladotor?ref_src=twsrc%5Etfw"><span class="citation">@lcolladotor</span></a> for co-supervising!
</p>
— Keri Martinowich (<span class="citation">@martinowk</span>) <a href="https://twitter.com/martinowk/status/1675881303940423680?ref_src=twsrc%5Etfw">July 3, 2023</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
<div id="introduction" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Introduction<a href="expression-heatmaps-with-complexheatmap.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Heatmaps are efficient to visualize associations between different sources of data sets and reveal potential patterns (e.g. patterns of expression in your genes). There are multiple packages in R that allow you to create heatmaps, the most widely use include:</p>
<ul>
<li><p><a href="https://r-graph-gallery.com/79-levelplot-with-ggplot2.html">ggplot2 heatmaps</a></p></li>
<li><p><a href="https://davetang.org/muse/2018/05/15/making-a-heatmap-in-r-with-the-pheatmap-package/">pheatmap</a></p></li>
<li><p><a href="https://jokergoo.github.io/ComplexHeatmap/">ComplexHeatmap</a></p></li>
</ul>
<p>All of these are good and useful, but adding extra information to your heatmaps can get really messy depending on the package you are using.</p>
<p>For example, in the next heatmap we are showing the expression of marker genes in multiple cell types. The columns are separated by the group from which they are markers. Then the rows (cell types), have two different classifications: <code>group</code> and <code>Population</code>, plus a barplot that shows the number of nuclei that cell type contains.</p>
<figure>
<img src="Figures/LS_heatmap.png" width="700px" align=center />
</figure>
</div>
<div id="complexheatmap" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> ComplexHeatmap<a href="expression-heatmaps-with-complexheatmap.html#complexheatmap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The ComplexHeatmap package provides a highly flexible way to arrange multiple heatmaps and supports self-defined annotation graphics.The ComplexHeatmap package is implemented in an object-oriented way. To describe a heatmap list, there are following classes:</p>
<p><strong>Heatmap class</strong>: a single heatmap containing heatmap body, row/column names, titles, dendrograms and row/column annotations.</p>
<p><strong>HeatmapAnnotation class</strong>: defines a list of row annotations and column annotations. The heatmap annotations can be components of heatmap, also they can be independent as heatmaps.</p>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}

</style>
<p class="exercise">
<strong>Exercise 1</strong>:
Explore the <a href="https://bioconductor.org/packages/release/bioc/html/ComplexHeatmap.html">Bioconductor page for this package</a>, go to the vignette and find a function that would allow you to add the barplot on the right side of the heatmap previously shown.
</p>
</div>
<div id="heatmap-with-expression-data" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Heatmap with expression data<a href="expression-heatmaps-with-complexheatmap.html#heatmap-with-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now, we are gonna make our own heatmap with the SmokingMouse data. In this heatmap we want to see the difference of expression between both <code>Control</code> and <code>Experimental</code> for all the DEG in <code>Pup</code>, as well as the <code>Sex</code> of the samples and the FDR of the genes.</p>
<p>Let’s download and prepare our data first.</p>
<div id="download-data" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Download data<a href="expression-heatmaps-with-complexheatmap.html#download-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="expression-heatmaps-with-complexheatmap.html#cb102-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</span>
<span id="cb102-2"><a href="expression-heatmaps-with-complexheatmap.html#cb102-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ComplexHeatmap&quot;</span>)</span>
<span id="cb102-3"><a href="expression-heatmaps-with-complexheatmap.html#cb102-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;circlize&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="expression-heatmaps-with-complexheatmap.html#cb103-1" tabindex="-1"></a><span class="do">## Download and cache the file</span></span>
<span id="cb103-2"><a href="expression-heatmaps-with-complexheatmap.html#cb103-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;BiocFileCache&quot;</span>)</span>
<span id="cb103-3"><a href="expression-heatmaps-with-complexheatmap.html#cb103-3" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> BiocFileCache<span class="sc">::</span><span class="fu">BiocFileCache</span>()</span>
<span id="cb103-4"><a href="expression-heatmaps-with-complexheatmap.html#cb103-4" tabindex="-1"></a>cached_rse_gene <span class="ot">&lt;-</span> BiocFileCache<span class="sc">::</span><span class="fu">bfcrpath</span>(</span>
<span id="cb103-5"><a href="expression-heatmaps-with-complexheatmap.html#cb103-5" tabindex="-1"></a>    <span class="at">x =</span> bfc,</span>
<span id="cb103-6"><a href="expression-heatmaps-with-complexheatmap.html#cb103-6" tabindex="-1"></a>    <span class="st">&quot;https://github.com/LieberInstitute/SPEAQeasyWorkshop2023/raw/devel/provisional_data/rse_gene_mouse_RNAseq_nic-smo.Rdata&quot;</span></span>
<span id="cb103-7"><a href="expression-heatmaps-with-complexheatmap.html#cb103-7" tabindex="-1"></a>)</span>
<span id="cb103-8"><a href="expression-heatmaps-with-complexheatmap.html#cb103-8" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;https://github.com/LieberInstitute/SPEAQeasyWorkshop2023/raw/devel/provisional_data/rse_gene_mouse_RNAseq_nic-smo.Rdata&#39;</span></span></code></pre></div>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="expression-heatmaps-with-complexheatmap.html#cb104-1" tabindex="-1"></a></span>
<span id="cb104-2"><a href="expression-heatmaps-with-complexheatmap.html#cb104-2" tabindex="-1"></a><span class="do">## Check the local path on our cache</span></span>
<span id="cb104-3"><a href="expression-heatmaps-with-complexheatmap.html#cb104-3" tabindex="-1"></a>cached_rse_gene</span>
<span id="cb104-4"><a href="expression-heatmaps-with-complexheatmap.html#cb104-4" tabindex="-1"></a><span class="co">#&gt;                                                                                  BFC1 </span></span>
<span id="cb104-5"><a href="expression-heatmaps-with-complexheatmap.html#cb104-5" tabindex="-1"></a><span class="co">#&gt; &quot;/github/home/.cache/R/BiocFileCache/37313fbd784_rse_gene_mouse_RNAseq_nic-smo.Rdata&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="expression-heatmaps-with-complexheatmap.html#cb105-1" tabindex="-1"></a></span>
<span id="cb105-2"><a href="expression-heatmaps-with-complexheatmap.html#cb105-2" tabindex="-1"></a><span class="do">## Load the rse_gene object</span></span>
<span id="cb105-3"><a href="expression-heatmaps-with-complexheatmap.html#cb105-3" tabindex="-1"></a><span class="fu">load</span>(cached_rse_gene, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-4"><a href="expression-heatmaps-with-complexheatmap.html#cb105-4" tabindex="-1"></a><span class="co">#&gt; Loading objects:</span></span>
<span id="cb105-5"><a href="expression-heatmaps-with-complexheatmap.html#cb105-5" tabindex="-1"></a><span class="co">#&gt;   rse_gene</span></span></code></pre></div>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="expression-heatmaps-with-complexheatmap.html#cb106-1" tabindex="-1"></a></span>
<span id="cb106-2"><a href="expression-heatmaps-with-complexheatmap.html#cb106-2" tabindex="-1"></a><span class="do">## General overview of the object</span></span>
<span id="cb106-3"><a href="expression-heatmaps-with-complexheatmap.html#cb106-3" tabindex="-1"></a>rse_gene</span>
<span id="cb106-4"><a href="expression-heatmaps-with-complexheatmap.html#cb106-4" tabindex="-1"></a><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span id="cb106-5"><a href="expression-heatmaps-with-complexheatmap.html#cb106-5" tabindex="-1"></a><span class="co">#&gt; dim: 55401 208 </span></span>
<span id="cb106-6"><a href="expression-heatmaps-with-complexheatmap.html#cb106-6" tabindex="-1"></a><span class="co">#&gt; metadata(1): Obtained_from</span></span>
<span id="cb106-7"><a href="expression-heatmaps-with-complexheatmap.html#cb106-7" tabindex="-1"></a><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span id="cb106-8"><a href="expression-heatmaps-with-complexheatmap.html#cb106-8" tabindex="-1"></a><span class="co">#&gt; rownames(55401): ENSMUSG00000102693.1 ENSMUSG00000064842.1 ... ENSMUSG00000064371.1 ENSMUSG00000064372.1</span></span>
<span id="cb106-9"><a href="expression-heatmaps-with-complexheatmap.html#cb106-9" tabindex="-1"></a><span class="co">#&gt; rowData names(13): Length gencodeID ... DE_in_pup_brain_nicotine DE_in_pup_brain_smoking</span></span>
<span id="cb106-10"><a href="expression-heatmaps-with-complexheatmap.html#cb106-10" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb106-11"><a href="expression-heatmaps-with-complexheatmap.html#cb106-11" tabindex="-1"></a><span class="co">#&gt; colData names(71): SAMPLE_ID FQCbasicStats ... retained_after_QC_sample_filtering</span></span>
<span id="cb106-12"><a href="expression-heatmaps-with-complexheatmap.html#cb106-12" tabindex="-1"></a><span class="co">#&gt;   retained_after_manual_sample_filtering</span></span></code></pre></div>
</div>
<div id="prepare-the-data" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Prepare the data<a href="expression-heatmaps-with-complexheatmap.html#prepare-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="expression-heatmaps-with-complexheatmap.html#cb107-1" tabindex="-1"></a><span class="do">## Extract genes and samples of interest</span></span>
<span id="cb107-2"><a href="expression-heatmaps-with-complexheatmap.html#cb107-2" tabindex="-1"></a>rse_gene_pup_nic <span class="ot">&lt;-</span> rse_gene[</span>
<span id="cb107-3"><a href="expression-heatmaps-with-complexheatmap.html#cb107-3" tabindex="-1"></a>    <span class="fu">rowData</span>(rse_gene)<span class="sc">$</span>DE_in_pup_brain_nicotine <span class="sc">==</span> <span class="cn">TRUE</span>,</span>
<span id="cb107-4"><a href="expression-heatmaps-with-complexheatmap.html#cb107-4" tabindex="-1"></a>    rse_gene<span class="sc">$</span>Age <span class="sc">==</span> <span class="st">&quot;Pup&quot;</span> <span class="sc">&amp;</span> rse_gene<span class="sc">$</span>Expt <span class="sc">==</span> <span class="st">&quot;Nicotine&quot;</span></span>
<span id="cb107-5"><a href="expression-heatmaps-with-complexheatmap.html#cb107-5" tabindex="-1"></a>]</span>
<span id="cb107-6"><a href="expression-heatmaps-with-complexheatmap.html#cb107-6" tabindex="-1"></a></span>
<span id="cb107-7"><a href="expression-heatmaps-with-complexheatmap.html#cb107-7" tabindex="-1"></a><span class="do">## Extract logcounts and add name columns</span></span>
<span id="cb107-8"><a href="expression-heatmaps-with-complexheatmap.html#cb107-8" tabindex="-1"></a>logs_pup_nic <span class="ot">&lt;-</span> <span class="fu">assay</span>(rse_gene_pup_nic, <span class="dv">2</span>)</span>
<span id="cb107-9"><a href="expression-heatmaps-with-complexheatmap.html#cb107-9" tabindex="-1"></a><span class="fu">colnames</span>(logs_pup_nic) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Pup_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(rse_gene_pup_nic)[<span class="dv">2</span>])</span>
<span id="cb107-10"><a href="expression-heatmaps-with-complexheatmap.html#cb107-10" tabindex="-1"></a></span>
<span id="cb107-11"><a href="expression-heatmaps-with-complexheatmap.html#cb107-11" tabindex="-1"></a><span class="do">## Create function to remove technical variables&#39; contributions,</span></span>
<span id="cb107-12"><a href="expression-heatmaps-with-complexheatmap.html#cb107-12" tabindex="-1"></a><span class="do">## this is from jaffelab package (https://github.com/LieberInstitute/jaffelab)</span></span>
<span id="cb107-13"><a href="expression-heatmaps-with-complexheatmap.html#cb107-13" tabindex="-1"></a>cleaningY <span class="ot">&lt;-</span> <span class="cf">function</span>(y, mod, P) {</span>
<span id="cb107-14"><a href="expression-heatmaps-with-complexheatmap.html#cb107-14" tabindex="-1"></a>    <span class="fu">stopifnot</span>(P <span class="sc">&lt;=</span> <span class="fu">ncol</span>(mod))</span>
<span id="cb107-15"><a href="expression-heatmaps-with-complexheatmap.html#cb107-15" tabindex="-1"></a>    <span class="fu">stopifnot</span>(<span class="st">`</span><span class="at">Input matrix is not full rank</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">qr</span>(mod)<span class="sc">$</span>rank <span class="sc">==</span></span>
<span id="cb107-16"><a href="expression-heatmaps-with-complexheatmap.html#cb107-16" tabindex="-1"></a>        <span class="fu">ncol</span>(mod))</span>
<span id="cb107-17"><a href="expression-heatmaps-with-complexheatmap.html#cb107-17" tabindex="-1"></a>    Hat <span class="ot">&lt;-</span> <span class="fu">solve</span>(<span class="fu">t</span>(mod) <span class="sc">%*%</span> mod) <span class="sc">%*%</span> <span class="fu">t</span>(mod)</span>
<span id="cb107-18"><a href="expression-heatmaps-with-complexheatmap.html#cb107-18" tabindex="-1"></a>    ty <span class="ot">&lt;-</span> <span class="fu">t</span>(y)</span>
<span id="cb107-19"><a href="expression-heatmaps-with-complexheatmap.html#cb107-19" tabindex="-1"></a>    ty[<span class="fu">is.na</span>(ty)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb107-20"><a href="expression-heatmaps-with-complexheatmap.html#cb107-20" tabindex="-1"></a>    beta <span class="ot">&lt;-</span> (Hat <span class="sc">%*%</span> ty)</span>
<span id="cb107-21"><a href="expression-heatmaps-with-complexheatmap.html#cb107-21" tabindex="-1"></a>    cleany <span class="ot">&lt;-</span> y <span class="sc">-</span> <span class="fu">t</span>(<span class="fu">as.matrix</span>(mod[, <span class="sc">-</span><span class="fu">c</span>(<span class="fu">seq_len</span>(P))]) <span class="sc">%*%</span> beta[<span class="sc">-</span><span class="fu">seq_len</span>(P), ])</span>
<span id="cb107-22"><a href="expression-heatmaps-with-complexheatmap.html#cb107-22" tabindex="-1"></a>    <span class="fu">return</span>(cleany)</span>
<span id="cb107-23"><a href="expression-heatmaps-with-complexheatmap.html#cb107-23" tabindex="-1"></a>}</span>
<span id="cb107-24"><a href="expression-heatmaps-with-complexheatmap.html#cb107-24" tabindex="-1"></a></span>
<span id="cb107-25"><a href="expression-heatmaps-with-complexheatmap.html#cb107-25" tabindex="-1"></a><span class="do">## Remove contribution of technical variables</span></span>
<span id="cb107-26"><a href="expression-heatmaps-with-complexheatmap.html#cb107-26" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="er">~</span> Group <span class="sc">+</span> Sex <span class="sc">+</span> plate <span class="sc">+</span> flowcell <span class="sc">+</span> rRNA_rate <span class="sc">+</span> totalAssignedGene <span class="sc">+</span> ERCCsumLogErr <span class="sc">+</span></span>
<span id="cb107-27"><a href="expression-heatmaps-with-complexheatmap.html#cb107-27" tabindex="-1"></a>    overallMapRate <span class="sc">+</span> mitoRate</span>
<span id="cb107-28"><a href="expression-heatmaps-with-complexheatmap.html#cb107-28" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(formula, <span class="at">data =</span> <span class="fu">colData</span>(rse_gene_pup_nic))</span>
<span id="cb107-29"><a href="expression-heatmaps-with-complexheatmap.html#cb107-29" tabindex="-1"></a>logs_pup_nic <span class="ot">&lt;-</span> <span class="fu">cleaningY</span>(logs_pup_nic, model, <span class="at">P =</span> <span class="dv">2</span>)</span>
<span id="cb107-30"><a href="expression-heatmaps-with-complexheatmap.html#cb107-30" tabindex="-1"></a></span>
<span id="cb107-31"><a href="expression-heatmaps-with-complexheatmap.html#cb107-31" tabindex="-1"></a><span class="do">## Center the data to make differences more evident</span></span>
<span id="cb107-32"><a href="expression-heatmaps-with-complexheatmap.html#cb107-32" tabindex="-1"></a>logs_pup_nic <span class="ot">&lt;-</span> (logs_pup_nic <span class="sc">-</span> <span class="fu">rowMeans</span>(logs_pup_nic)) <span class="sc">/</span> <span class="fu">rowSds</span>(logs_pup_nic)</span></code></pre></div>
<p>More for on centering and scaling, see this video:</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/c8ffeFVUzk8" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen>
</iframe>
</div>
<div id="create-annotations" class="section level3 hasAnchor" number="9.3.3">
<h3><span class="header-section-number">9.3.3</span> Create annotations<a href="expression-heatmaps-with-complexheatmap.html#create-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="expression-heatmaps-with-complexheatmap.html#cb108-1" tabindex="-1"></a><span class="do">## Prepare annotation for our heatmap</span></span>
<span id="cb108-2"><a href="expression-heatmaps-with-complexheatmap.html#cb108-2" tabindex="-1"></a><span class="do">## For this heatmap I want to be able to see the Group to which each sample belongs</span></span>
<span id="cb108-3"><a href="expression-heatmaps-with-complexheatmap.html#cb108-3" tabindex="-1"></a><span class="do">## as well as the Sex of the pup</span></span>
<span id="cb108-4"><a href="expression-heatmaps-with-complexheatmap.html#cb108-4" tabindex="-1"></a></span>
<span id="cb108-5"><a href="expression-heatmaps-with-complexheatmap.html#cb108-5" tabindex="-1"></a>top_ans <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(</span>
<span id="cb108-6"><a href="expression-heatmaps-with-complexheatmap.html#cb108-6" tabindex="-1"></a>    <span class="at">df =</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(rse_gene_pup_nic)[, <span class="fu">c</span>(<span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Group&quot;</span>)]),</span>
<span id="cb108-7"><a href="expression-heatmaps-with-complexheatmap.html#cb108-7" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">list</span>(</span>
<span id="cb108-8"><a href="expression-heatmaps-with-complexheatmap.html#cb108-8" tabindex="-1"></a>        <span class="st">&quot;Sex&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span> <span class="ot">=</span> <span class="st">&quot;hotpink1&quot;</span>, <span class="st">&quot;M&quot;</span> <span class="ot">=</span> <span class="st">&quot;dodgerblue&quot;</span>),</span>
<span id="cb108-9"><a href="expression-heatmaps-with-complexheatmap.html#cb108-9" tabindex="-1"></a>        <span class="st">&quot;Group&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span> <span class="ot">=</span> <span class="st">&quot;brown2&quot;</span>, <span class="st">&quot;Experimental&quot;</span> <span class="ot">=</span> <span class="st">&quot;deepskyblue3&quot;</span>)</span>
<span id="cb108-10"><a href="expression-heatmaps-with-complexheatmap.html#cb108-10" tabindex="-1"></a>    )</span>
<span id="cb108-11"><a href="expression-heatmaps-with-complexheatmap.html#cb108-11" tabindex="-1"></a>)</span>
<span id="cb108-12"><a href="expression-heatmaps-with-complexheatmap.html#cb108-12" tabindex="-1"></a></span>
<span id="cb108-13"><a href="expression-heatmaps-with-complexheatmap.html#cb108-13" tabindex="-1"></a><span class="do">## Also, I want to add the FDR associated to each gene</span></span>
<span id="cb108-14"><a href="expression-heatmaps-with-complexheatmap.html#cb108-14" tabindex="-1"></a><span class="do">## Even though we do have that data, for this particular exercise we are gonna simulate it</span></span>
<span id="cb108-15"><a href="expression-heatmaps-with-complexheatmap.html#cb108-15" tabindex="-1"></a></span>
<span id="cb108-16"><a href="expression-heatmaps-with-complexheatmap.html#cb108-16" tabindex="-1"></a><span class="fu">rowData</span>(rse_gene_pup_nic)<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">sample</span>(</span>
<span id="cb108-17"><a href="expression-heatmaps-with-complexheatmap.html#cb108-17" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">5000</span>,</span>
<span id="cb108-18"><a href="expression-heatmaps-with-complexheatmap.html#cb108-18" tabindex="-1"></a>    <span class="at">size =</span> <span class="fu">dim</span>(rse_gene_pup_nic)[<span class="dv">1</span>],</span>
<span id="cb108-19"><a href="expression-heatmaps-with-complexheatmap.html#cb108-19" tabindex="-1"></a>    <span class="at">prob =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="fl">0.0012</span>, <span class="dv">1001</span>), <span class="fu">rep</span>(<span class="fl">0.00005</span>, <span class="dv">4000</span>))</span>
<span id="cb108-20"><a href="expression-heatmaps-with-complexheatmap.html#cb108-20" tabindex="-1"></a>) <span class="sc">/</span> <span class="dv">10000</span></span>
<span id="cb108-21"><a href="expression-heatmaps-with-complexheatmap.html#cb108-21" tabindex="-1"></a></span>
<span id="cb108-22"><a href="expression-heatmaps-with-complexheatmap.html#cb108-22" tabindex="-1"></a>left_ans <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(</span>
<span id="cb108-23"><a href="expression-heatmaps-with-complexheatmap.html#cb108-23" tabindex="-1"></a>    <span class="at">FDR =</span> <span class="fu">rowData</span>(rse_gene_pup_nic)<span class="sc">$</span>FDR,</span>
<span id="cb108-24"><a href="expression-heatmaps-with-complexheatmap.html#cb108-24" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">list</span>(<span class="at">FDR =</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.049</span>), <span class="fu">c</span>(<span class="st">&quot;#ecf39e&quot;</span>, <span class="st">&quot;#4f772d&quot;</span>))),</span>
<span id="cb108-25"><a href="expression-heatmaps-with-complexheatmap.html#cb108-25" tabindex="-1"></a>    <span class="at">annotation_legend_param =</span> <span class="fu">list</span>(<span class="at">FDR =</span> <span class="fu">list</span>(<span class="at">at =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.01</span>, <span class="fl">0.02</span>, <span class="fl">0.03</span>, <span class="fl">0.04</span>, <span class="fl">0.05</span>)))</span>
<span id="cb108-26"><a href="expression-heatmaps-with-complexheatmap.html#cb108-26" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="plot-our-heatmap" class="section level3 hasAnchor" number="9.3.4">
<h3><span class="header-section-number">9.3.4</span> Plot our heatmap<a href="expression-heatmaps-with-complexheatmap.html#plot-our-heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="expression-heatmaps-with-complexheatmap.html#cb109-1" tabindex="-1"></a><span class="do">## Finally, let&#39;s plot!</span></span>
<span id="cb109-2"><a href="expression-heatmaps-with-complexheatmap.html#cb109-2" tabindex="-1"></a><span class="fu">Heatmap</span>(logs_pup_nic,</span>
<span id="cb109-3"><a href="expression-heatmaps-with-complexheatmap.html#cb109-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;logcounts&quot;</span>,</span>
<span id="cb109-4"><a href="expression-heatmaps-with-complexheatmap.html#cb109-4" tabindex="-1"></a>    <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb109-5"><a href="expression-heatmaps-with-complexheatmap.html#cb109-5" tabindex="-1"></a>    <span class="at">top_annotation =</span> top_ans,</span>
<span id="cb109-6"><a href="expression-heatmaps-with-complexheatmap.html#cb109-6" tabindex="-1"></a>    <span class="at">left_annotation =</span> left_ans,</span>
<span id="cb109-7"><a href="expression-heatmaps-with-complexheatmap.html#cb109-7" tabindex="-1"></a>    <span class="at">row_km =</span> <span class="dv">2</span>,</span>
<span id="cb109-8"><a href="expression-heatmaps-with-complexheatmap.html#cb109-8" tabindex="-1"></a>    <span class="at">column_km =</span> <span class="dv">2</span>,</span>
<span id="cb109-9"><a href="expression-heatmaps-with-complexheatmap.html#cb109-9" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="sc">-</span><span class="fl">0.0001</span>, <span class="dv">00001</span>, <span class="dv">4</span>), <span class="fu">c</span>(<span class="st">&quot;darkblue&quot;</span>, <span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;lightsalmon&quot;</span>, <span class="st">&quot;darkred&quot;</span>)),</span>
<span id="cb109-10"><a href="expression-heatmaps-with-complexheatmap.html#cb109-10" tabindex="-1"></a>    <span class="at">row_title =</span> <span class="cn">NULL</span>,</span>
<span id="cb109-11"><a href="expression-heatmaps-with-complexheatmap.html#cb109-11" tabindex="-1"></a>    <span class="at">column_title =</span> <span class="cn">NULL</span>,</span>
<span id="cb109-12"><a href="expression-heatmaps-with-complexheatmap.html#cb109-12" tabindex="-1"></a>    <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>)</span>
<span id="cb109-13"><a href="expression-heatmaps-with-complexheatmap.html#cb109-13" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-29-1.png" width="1000px" /></p>
<p class="exercise">
<strong>Exercise 2</strong>:
<strong>a)</strong> Add a barplot as a <code>topAnnotation</code> with the library sizes (<code>sum</code>) of each sample. Fill the bars with the color of your choice
<strong>b)</strong> Classify the genes according to whether they are <code>proteing_coding</code> or not (<code>not_proteing_coding</code>).
</p>
</div>
</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-gene-expression-exercise.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="complexheatmap-exercise-solution.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
