<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Final R Session | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="Final R Session | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Final R Session | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Final R Session | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="making-your-own-website-with-postcards.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.2</b> Exercises</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.3</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
<li class="chapter" data-level="2.1.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment"><i class="fa fa-check"></i><b>2.1.3</b> SpatialExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
<li class="chapter" data-level="2.4.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#explore-the-data"><i class="fa fa-check"></i><b>2.4.2</b> Explore the Data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features"><i class="fa fa-check"></i><b>2.5</b> Introduction to Advanced iSEE Features</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression"><i class="fa fa-check"></i><b>2.5.1</b> Selecting Cells Based on a Single Gene Expression</a></li>
<li class="chapter" data-level="2.5.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression"><i class="fa fa-check"></i><b>2.5.2</b> Using a Single Plot for Two Gene Co-Expression</a></li>
<li class="chapter" data-level="2.5.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes"><i class="fa fa-check"></i><b>2.5.3</b> Selecting Cells Based on the Co-Expression of Two or more Genes</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.6</b> References</a></li>
<li class="chapter" data-level="2.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#community"><i class="fa fa-check"></i><b>2.7</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-1"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-2"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>9</b> Research talks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>9.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="9.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>9.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="9.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>9.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>10</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>10.1</b> Related past materials</a></li>
<li class="chapter" data-level="10.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>10.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="10.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>10.3</b> Live demo</a></li>
<li class="chapter" data-level="10.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-3"><i class="fa fa-check"></i><b>10.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>11</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="11.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing"><i class="fa fa-check"></i><b>11.1</b> Single cell RNA sequencing</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#pre-processing-of-scrna-seq-data-before-r"><i class="fa fa-check"></i><b>11.1.1</b> Pre-processing of scRNA-seq Data (Before R)</a></li>
<li class="chapter" data-level="11.1.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#different-technologies"><i class="fa fa-check"></i><b>11.1.2</b> Different Technologies</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basic-workflow"><i class="fa fa-check"></i><b>11.2</b> Basic Workflow</a></li>
<li class="chapter" data-level="11.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>11.3</b> The SingleCellExperiment class</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#data-loading"><i class="fa fa-check"></i><b>11.3.1</b> Data Loading</a></li>
<li class="chapter" data-level="11.3.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#basics-of-your-sce"><i class="fa fa-check"></i><b>11.3.2</b> Basics of your SCE</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>11.4</b> Quality Control</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#common-choices-of-qc-metrics"><i class="fa fa-check"></i><b>11.4.1</b> Common choices of QC metrics</a></li>
<li class="chapter" data-level="11.4.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#identifying-low-quality-cells"><i class="fa fa-check"></i><b>11.4.2</b> Identifying low-quality cells</a></li>
<li class="chapter" data-level="11.4.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#checking-diagnostic-plots"><i class="fa fa-check"></i><b>11.4.3</b> Checking diagnostic plots</a></li>
<li class="chapter" data-level="11.4.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#removing-low-quality-cells"><i class="fa fa-check"></i><b>11.4.4</b> Removing low-quality cells</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>11.5</b> Normalization</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#library-size-normalization"><i class="fa fa-check"></i><b>11.5.1</b> Library size normalization</a></li>
<li class="chapter" data-level="11.5.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-deconvolution"><i class="fa fa-check"></i><b>11.5.2</b> Normalization by deconvolution</a></li>
<li class="chapter" data-level="11.5.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization-by-spike-ins"><i class="fa fa-check"></i><b>11.5.3</b> Normalization by spike-ins</a></li>
<li class="chapter" data-level="11.5.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#scaling-and-log-transforming"><i class="fa fa-check"></i><b>11.5.4</b> Scaling and log-transforming</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>11.6</b> Feature selection</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-per-gene-variation"><i class="fa fa-check"></i><b>11.6.1</b> Quantifying per-gene variation</a></li>
<li class="chapter" data-level="11.6.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-spike-ins"><i class="fa fa-check"></i><b>11.6.2</b> Quantifying technical noise (spike-ins)</a></li>
<li class="chapter" data-level="11.6.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quantifying-technical-noise-mean-variance-trend"><i class="fa fa-check"></i><b>11.6.3</b> Quantifying technical noise (mean-variance trend)</a></li>
<li class="chapter" data-level="11.6.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#handling-batch-effects"><i class="fa fa-check"></i><b>11.6.4</b> Handling batch effects</a></li>
<li class="chapter" data-level="11.6.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#selecting-highly-variable-genes"><i class="fa fa-check"></i><b>11.6.5</b> Selecting highly variable genes</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>11.7</b> Dimensionality reduction</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#principal-components-analysis"><i class="fa fa-check"></i><b>11.7.1</b> Principal components analysis</a></li>
<li class="chapter" data-level="11.7.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#choosing-the-number-of-pcs"><i class="fa fa-check"></i><b>11.7.2</b> Choosing the number of PCs</a></li>
<li class="chapter" data-level="11.7.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#visualizing-the-pcs"><i class="fa fa-check"></i><b>11.7.3</b> Visualizing the PCs</a></li>
<li class="chapter" data-level="11.7.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#non-linear-methods-for-visualization"><i class="fa fa-check"></i><b>11.7.4</b> Non-linear methods for visualization</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>11.8</b> Clustering</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#graph-based-clustering"><i class="fa fa-check"></i><b>11.8.1</b> Graph-based clustering</a></li>
<li class="chapter" data-level="11.8.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#adjusting-the-parameters"><i class="fa fa-check"></i><b>11.8.2</b> Adjusting the parameters</a></li>
<li class="chapter" data-level="11.8.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#hierarchical-clustering"><i class="fa fa-check"></i><b>11.8.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="11.8.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#subclustering"><i class="fa fa-check"></i><b>11.8.4</b> Subclustering</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>11.9</b> Marker gene detection</a></li>
<li class="chapter" data-level="11.10" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>11.10</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>12</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>12.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>12.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>12.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>12.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>13</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="13.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>13.1</b> Spatial registration</a></li>
<li class="chapter" data-level="13.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>13.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>14</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="14.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>14.1</b> here</a></li>
<li class="chapter" data-level="14.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>14.2</b> Usethis</a></li>
<li class="chapter" data-level="14.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>14.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>14.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="14.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>14.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="14.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>14.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="14.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>14.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>14.4</b> R websites</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>14.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="14.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>14.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="14.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>14.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="14.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>14.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>14.5</b> postcards</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>14.5.1</b> Installation</a></li>
<li class="chapter" data-level="14.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>14.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>14.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>14.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="14.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>14.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="14.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>14.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="14.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>14.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="14.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>14.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-4"><i class="fa fa-check"></i><b>14.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="final-r-session" class="section level1 unnumbered hasAnchor">
<h1>Final R Session<a href="final-r-session.html#final-r-session" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This is the final R session after all the code in this book is run sequentially.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value
#&gt;  version  R version 4.4.0 (2024-04-24)
#&gt;  os       Ubuntu 22.04.4 LTS
#&gt;  system   x86_64, linux-gnu
#&gt;  ui       X11
#&gt;  language (EN)
#&gt;  collate  en_US.UTF-8
#&gt;  ctype    en_US.UTF-8
#&gt;  tz       UTC
#&gt;  date     2024-06-09
#&gt;  pandoc   3.1.13 @ /usr/bin/ (via rmarkdown)
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package              * version     date (UTC) lib source
#&gt;  abind                  1.4-5       2016-07-21 [1] RSPM (R 4.4.0)
#&gt;  airway               * 1.24.0      2024-05-02 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  alabaster.base         1.4.1       2024-05-03 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  alabaster.matrix       1.4.0       2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  alabaster.ranges       1.4.1       2024-05-21 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  alabaster.sce          1.4.0       2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  alabaster.schemas      1.4.0       2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  alabaster.se           1.4.1       2024-05-21 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  AnnotationDbi        * 1.66.0      2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  AnnotationFilter     * 1.28.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  AnnotationHub        * 3.12.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  aod                    1.3.3       2023-12-13 [1] RSPM (R 4.4.0)
#&gt;  askpass                1.2.0       2023-09-03 [2] RSPM (R 4.4.0)
#&gt;  attempt                0.3.1       2020-05-03 [1] RSPM (R 4.4.0)
#&gt;  backports              1.5.0       2024-05-23 [1] RSPM (R 4.4.0)
#&gt;  base64enc              0.1-3       2015-07-28 [2] RSPM (R 4.4.0)
#&gt;  beachmat               2.20.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  beeswarm               0.4.0       2021-06-01 [1] RSPM (R 4.4.0)
#&gt;  benchmarkme            1.0.8       2022-06-12 [1] RSPM (R 4.4.0)
#&gt;  benchmarkmeData        1.0.4       2020-04-23 [1] RSPM (R 4.4.0)
#&gt;  bibtex                 0.5.1       2023-01-26 [1] RSPM (R 4.4.0)
#&gt;  Biobase              * 2.64.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocFileCache        * 2.12.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocGenerics         * 0.50.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocIO                 1.14.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocManager            1.30.23     2024-05-04 [2] CRAN (R 4.4.0)
#&gt;  BiocNeighbors          1.22.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocParallel         * 1.38.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocSingular           1.20.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocStyle            * 2.32.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  biocthis             * 1.14.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  BiocVersion            3.19.1      2024-04-17 [2] Bioconductor 3.19 (R 4.4.0)
#&gt;  Biostrings             2.72.1      2024-06-02 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  bit                    4.0.5       2022-11-15 [1] RSPM (R 4.4.0)
#&gt;  bit64                  4.0.5       2020-08-30 [1] RSPM (R 4.4.0)
#&gt;  bitops                 1.0-7       2021-04-24 [1] RSPM (R 4.4.0)
#&gt;  blob                   1.2.4       2023-03-17 [1] RSPM (R 4.4.0)
#&gt;  bluster              * 1.14.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  bookdown               0.39        2024-04-15 [1] RSPM (R 4.4.0)
#&gt;  boot                   1.3-30      2024-02-26 [3] CRAN (R 4.4.0)
#&gt;  broom                  1.0.6       2024-05-17 [1] RSPM (R 4.4.0)
#&gt;  bslib                  0.7.0       2024-03-29 [2] RSPM (R 4.4.0)
#&gt;  cachem                 1.1.0       2024-05-16 [2] RSPM (R 4.4.0)
#&gt;  Cairo                  1.6-2       2023-11-28 [1] RSPM (R 4.4.0)
#&gt;  caTools                1.18.2      2021-03-28 [1] RSPM (R 4.4.0)
#&gt;  checkmate              2.3.1       2023-12-04 [1] RSPM (R 4.4.0)
#&gt;  circlize             * 0.4.16      2024-02-20 [1] RSPM (R 4.4.0)
#&gt;  cli                    3.6.2       2023-12-11 [2] RSPM (R 4.4.0)
#&gt;  clue                   0.3-65      2023-09-23 [1] RSPM (R 4.4.0)
#&gt;  cluster                2.1.6       2023-12-01 [3] CRAN (R 4.4.0)
#&gt;  codetools              0.2-20      2024-03-31 [3] CRAN (R 4.4.0)
#&gt;  colorspace             2.1-0       2023-01-23 [1] RSPM (R 4.4.0)
#&gt;  colourpicker           1.3.0       2023-08-21 [1] RSPM (R 4.4.0)
#&gt;  ComplexHeatmap       * 2.20.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  config                 0.3.2       2023-08-30 [1] RSPM (R 4.4.0)
#&gt;  corpcor                1.6.10      2021-09-16 [1] RSPM (R 4.4.0)
#&gt;  cowplot              * 1.1.3       2024-01-22 [1] RSPM (R 4.4.0)
#&gt;  crayon                 1.5.2       2022-09-29 [2] RSPM (R 4.4.0)
#&gt;  credentials            2.0.1       2023-09-06 [2] RSPM (R 4.4.0)
#&gt;  curl                   5.2.1       2024-03-01 [1] RSPM (R 4.4.0)
#&gt;  data.table             1.15.4      2024-03-30 [1] RSPM (R 4.4.0)
#&gt;  DBI                    1.2.3       2024-06-02 [1] RSPM (R 4.4.0)
#&gt;  dbplyr               * 2.5.0       2024-03-19 [1] RSPM (R 4.4.0)
#&gt;  DelayedArray           0.30.1      2024-05-07 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  DelayedMatrixStats     1.26.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  dendextend           * 1.17.1      2023-03-25 [1] RSPM (R 4.4.0)
#&gt;  desc                   1.4.3       2023-12-10 [2] RSPM (R 4.4.0)
#&gt;  digest                 0.6.35      2024-03-11 [2] RSPM (R 4.4.0)
#&gt;  doParallel             1.0.17      2022-02-07 [1] RSPM (R 4.4.0)
#&gt;  dotCall64              1.1-1       2023-11-28 [1] RSPM (R 4.4.0)
#&gt;  dplyr                  1.1.4       2023-11-17 [1] RSPM (R 4.4.0)
#&gt;  dqrng                  0.4.1       2024-05-28 [1] RSPM (R 4.4.0)
#&gt;  DT                     0.33        2024-04-04 [1] RSPM (R 4.4.0)
#&gt;  dynamicTreeCut       * 1.63-1      2016-03-11 [1] RSPM (R 4.4.0)
#&gt;  edgeR                * 4.2.0       2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  ensembldb            * 2.28.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  EnvStats               2.8.1       2023-08-22 [1] RSPM (R 4.4.0)
#&gt;  evaluate               0.23        2023-11-01 [2] RSPM (R 4.4.0)
#&gt;  ExperimentHub        * 2.12.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  ExploreModelMatrix   * 1.16.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  fANCOVA                0.6-1       2020-11-13 [1] RSPM (R 4.4.0)
#&gt;  fansi                  1.0.6       2023-12-08 [2] RSPM (R 4.4.0)
#&gt;  farver                 2.1.2       2024-05-13 [1] RSPM (R 4.4.0)
#&gt;  fastmap                1.2.0       2024-05-15 [2] RSPM (R 4.4.0)
#&gt;  fields                 15.2        2023-08-17 [1] RSPM (R 4.4.0)
#&gt;  filelock               1.0.3       2023-12-11 [1] RSPM (R 4.4.0)
#&gt;  FNN                    1.1.4       2024-01-12 [1] RSPM (R 4.4.0)
#&gt;  foreach                1.5.2       2022-02-02 [1] RSPM (R 4.4.0)
#&gt;  foreign                0.8-86      2023-11-28 [3] CRAN (R 4.4.0)
#&gt;  Formula                1.2-5       2023-02-24 [1] RSPM (R 4.4.0)
#&gt;  fs                     1.6.4       2024-04-25 [2] RSPM (R 4.4.0)
#&gt;  generics               0.1.3       2022-07-05 [1] RSPM (R 4.4.0)
#&gt;  GenomeInfoDb         * 1.40.1      2024-05-24 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  GenomeInfoDbData       1.2.12      2024-05-26 [1] Bioconductor
#&gt;  GenomicAlignments      1.40.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  GenomicFeatures      * 1.56.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  GenomicRanges        * 1.56.0      2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  gert                 * 2.0.1       2023-12-04 [2] RSPM (R 4.4.0)
#&gt;  GetoptLong             1.0.5       2020-12-15 [1] RSPM (R 4.4.0)
#&gt;  ggbeeswarm             0.7.2       2023-04-29 [1] RSPM (R 4.4.0)
#&gt;  ggplot2              * 3.5.1       2024-04-23 [1] RSPM (R 4.4.0)
#&gt;  ggrepel              * 0.9.5       2024-01-10 [1] RSPM (R 4.4.0)
#&gt;  gh                   * 1.4.1       2024-03-28 [2] RSPM (R 4.4.0)
#&gt;  gitcreds             * 0.1.2       2022-09-08 [2] RSPM (R 4.4.0)
#&gt;  GlobalOptions          0.1.2       2020-06-10 [1] RSPM (R 4.4.0)
#&gt;  glue                   1.7.0       2024-01-09 [2] RSPM (R 4.4.0)
#&gt;  golem                  0.4.1       2023-06-05 [1] RSPM (R 4.4.0)
#&gt;  gplots                 3.1.3.1     2024-02-02 [1] RSPM (R 4.4.0)
#&gt;  gridExtra              2.3         2017-09-09 [1] RSPM (R 4.4.0)
#&gt;  gtable                 0.3.5       2024-04-22 [1] RSPM (R 4.4.0)
#&gt;  gtools                 3.9.5       2023-11-20 [1] RSPM (R 4.4.0)
#&gt;  gypsum                 1.0.1       2024-05-08 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  HDF5Array              1.32.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  here                 * 1.0.1       2020-12-13 [1] RSPM (R 4.4.0)
#&gt;  highr                  0.11        2024-05-26 [2] RSPM (R 4.4.0)
#&gt;  Hmisc                * 5.1-3       2024-05-28 [1] RSPM (R 4.4.0)
#&gt;  htmlTable              2.4.2       2023-10-29 [1] RSPM (R 4.4.0)
#&gt;  htmltools              0.5.8.1     2024-04-04 [2] RSPM (R 4.4.0)
#&gt;  htmlwidgets            1.6.4       2023-12-06 [2] RSPM (R 4.4.0)
#&gt;  httpuv                 1.6.15      2024-03-26 [2] RSPM (R 4.4.0)
#&gt;  httr                   1.4.7       2023-08-15 [2] RSPM (R 4.4.0)
#&gt;  httr2                  1.0.1       2024-04-01 [2] RSPM (R 4.4.0)
#&gt;  igraph                 2.0.3       2024-03-13 [1] RSPM (R 4.4.0)
#&gt;  IRanges              * 2.38.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  irlba                  2.3.5.1     2022-10-03 [1] RSPM (R 4.4.0)
#&gt;  iSEE                 * 2.16.0      2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  iterators              1.0.14      2022-02-05 [1] RSPM (R 4.4.0)
#&gt;  jquerylib              0.1.4       2021-04-26 [2] RSPM (R 4.4.0)
#&gt;  jsonlite               1.8.8       2023-12-04 [2] RSPM (R 4.4.0)
#&gt;  KEGGREST               1.44.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  KernSmooth             2.23-24     2024-05-17 [3] RSPM (R 4.4.0)
#&gt;  knitr                  1.47        2024-05-29 [2] RSPM (R 4.4.0)
#&gt;  labeling               0.4.3       2023-08-29 [1] RSPM (R 4.4.0)
#&gt;  later                  1.3.2       2023-12-06 [2] RSPM (R 4.4.0)
#&gt;  lattice                0.22-6      2024-03-20 [3] CRAN (R 4.4.0)
#&gt;  lazyeval               0.2.2       2019-03-15 [1] RSPM (R 4.4.0)
#&gt;  lifecycle              1.0.4       2023-11-07 [2] RSPM (R 4.4.0)
#&gt;  limma                * 3.60.2      2024-05-19 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  listviewer             4.0.0       2023-09-30 [1] RSPM (R 4.4.0)
#&gt;  lme4                   1.1-35.3    2024-04-16 [1] RSPM (R 4.4.0)
#&gt;  lmerTest               3.1-3       2020-10-23 [1] RSPM (R 4.4.0)
#&gt;  lobstr               * 1.1.2       2022-06-22 [1] RSPM (R 4.4.0)
#&gt;  locfit                 1.5-9.9     2024-03-01 [1] RSPM (R 4.4.0)
#&gt;  lubridate              1.9.3       2023-09-27 [1] RSPM (R 4.4.0)
#&gt;  magick                 2.8.3       2024-02-18 [1] RSPM (R 4.4.0)
#&gt;  magrittr               2.0.3       2022-03-30 [2] RSPM (R 4.4.0)
#&gt;  maps                   3.4.2       2023-12-15 [1] RSPM (R 4.4.0)
#&gt;  MASS                   7.3-60.2    2024-05-06 [3] local
#&gt;  Matrix                 1.7-0       2024-03-22 [3] CRAN (R 4.4.0)
#&gt;  MatrixGenerics       * 1.16.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  matrixStats          * 1.3.0       2024-04-11 [1] RSPM (R 4.4.0)
#&gt;  memoise                2.0.1       2021-11-26 [2] RSPM (R 4.4.0)
#&gt;  metapod                1.12.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  mgcv                   1.9-1       2023-12-21 [3] CRAN (R 4.4.0)
#&gt;  mime                   0.12        2021-09-28 [2] RSPM (R 4.4.0)
#&gt;  miniUI                 0.1.1.1     2018-05-18 [2] RSPM (R 4.4.0)
#&gt;  minqa                  1.2.7       2024-05-20 [1] RSPM (R 4.4.0)
#&gt;  munsell                0.5.1       2024-04-01 [1] RSPM (R 4.4.0)
#&gt;  mvtnorm                1.2-5       2024-05-21 [1] RSPM (R 4.4.0)
#&gt;  nlme                   3.1-165     2024-06-06 [3] RSPM (R 4.4.0)
#&gt;  nloptr                 2.0.3       2022-05-26 [1] RSPM (R 4.4.0)
#&gt;  nnet                   7.3-19      2023-05-03 [3] CRAN (R 4.4.0)
#&gt;  numDeriv               2016.8-1.1  2019-06-06 [1] RSPM (R 4.4.0)
#&gt;  openssl                2.2.0       2024-05-16 [2] RSPM (R 4.4.0)
#&gt;  paletteer              1.6.0       2024-01-21 [1] RSPM (R 4.4.0)
#&gt;  patchwork            * 1.2.0       2024-01-08 [1] RSPM (R 4.4.0)
#&gt;  pbkrtest               0.5.2       2023-01-19 [1] RSPM (R 4.4.0)
#&gt;  pheatmap             * 1.0.12      2019-01-04 [1] RSPM (R 4.4.0)
#&gt;  pillar                 1.9.0       2023-03-22 [2] RSPM (R 4.4.0)
#&gt;  pkgconfig              2.0.3       2019-09-22 [2] RSPM (R 4.4.0)
#&gt;  plotly                 4.10.4      2024-01-13 [1] RSPM (R 4.4.0)
#&gt;  plyr                   1.8.9       2023-10-02 [1] RSPM (R 4.4.0)
#&gt;  png                    0.1-8       2022-11-29 [1] RSPM (R 4.4.0)
#&gt;  Polychrome           * 1.5.1       2022-05-03 [1] RSPM (R 4.4.0)
#&gt;  postcards            * 0.2.3       2022-01-07 [1] RSPM (R 4.4.0)
#&gt;  prettyunits            1.2.0       2023-09-24 [2] RSPM (R 4.4.0)
#&gt;  promises               1.3.0       2024-04-05 [2] RSPM (R 4.4.0)
#&gt;  ProtGenerics           1.36.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  purrr                  1.0.2       2023-08-10 [2] RSPM (R 4.4.0)
#&gt;  R.cache                0.16.0      2022-07-21 [1] RSPM (R 4.4.0)
#&gt;  R.methodsS3            1.8.2       2022-06-13 [1] RSPM (R 4.4.0)
#&gt;  R.oo                   1.26.0      2024-01-24 [1] RSPM (R 4.4.0)
#&gt;  R.utils                2.12.3      2023-11-18 [1] RSPM (R 4.4.0)
#&gt;  R6                     2.5.1       2021-08-19 [2] RSPM (R 4.4.0)
#&gt;  rappdirs               0.3.3       2021-01-31 [2] RSPM (R 4.4.0)
#&gt;  rbibutils              2.2.16      2023-10-25 [1] RSPM (R 4.4.0)
#&gt;  RColorBrewer         * 1.1-3       2022-04-03 [1] RSPM (R 4.4.0)
#&gt;  Rcpp                   1.0.12      2024-01-09 [2] RSPM (R 4.4.0)
#&gt;  RCurl                  1.98-1.14   2024-01-09 [1] RSPM (R 4.4.0)
#&gt;  Rdpack                 2.6         2023-11-08 [1] RSPM (R 4.4.0)
#&gt;  recount3             * 1.14.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  RefManageR           * 1.4.0       2022-09-30 [1] RSPM (R 4.4.0)
#&gt;  remaCor                0.0.18      2024-02-08 [1] RSPM (R 4.4.0)
#&gt;  rematch2               2.1.2       2020-05-01 [2] RSPM (R 4.4.0)
#&gt;  reshape2               1.4.4       2020-04-09 [1] RSPM (R 4.4.0)
#&gt;  restfulr               0.0.15      2022-06-16 [1] RSPM (R 4.4.0)
#&gt;  rhdf5                  2.48.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  rhdf5filters           1.16.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  Rhdf5lib               1.26.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  RhpcBLASctl            0.23-42     2023-02-11 [1] RSPM (R 4.4.0)
#&gt;  rintrojs               0.3.4       2024-01-11 [1] RSPM (R 4.4.0)
#&gt;  rjson                  0.2.21      2022-01-09 [1] RSPM (R 4.4.0)
#&gt;  rlang                * 1.1.4       2024-06-04 [2] RSPM (R 4.4.0)
#&gt;  rmarkdown              2.27        2024-05-17 [2] RSPM (R 4.4.0)
#&gt;  rpart                  4.1.23      2023-12-05 [3] CRAN (R 4.4.0)
#&gt;  rprojroot              2.0.4       2023-11-05 [2] RSPM (R 4.4.0)
#&gt;  Rsamtools              2.20.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  RSQLite                2.3.7       2024-05-27 [1] RSPM (R 4.4.0)
#&gt;  rstudioapi             0.16.0      2024-03-24 [2] RSPM (R 4.4.0)
#&gt;  rsvd                   1.0.5       2021-04-16 [1] RSPM (R 4.4.0)
#&gt;  rtracklayer            1.64.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  Rtsne                  0.17        2023-12-07 [1] RSPM (R 4.4.0)
#&gt;  S4Arrays               1.4.1       2024-05-20 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  S4Vectors            * 0.42.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  sass                   0.4.9       2024-03-15 [2] RSPM (R 4.4.0)
#&gt;  ScaledMatrix           1.12.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  scales                 1.3.0       2023-11-28 [1] RSPM (R 4.4.0)
#&gt;  scater               * 1.32.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  scatterplot3d          0.3-44      2023-05-05 [1] RSPM (R 4.4.0)
#&gt;  scran                * 1.32.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  scRNAseq             * 2.18.0      2024-05-02 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  scuttle              * 1.14.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  sessioninfo          * 1.2.2       2021-12-06 [2] RSPM (R 4.4.0)
#&gt;  shape                  1.4.6.1     2024-02-23 [1] RSPM (R 4.4.0)
#&gt;  shiny                  1.8.1.1     2024-04-02 [2] RSPM (R 4.4.0)
#&gt;  shinyAce               0.4.2       2022-05-06 [1] RSPM (R 4.4.0)
#&gt;  shinydashboard         0.7.2       2021-09-30 [1] RSPM (R 4.4.0)
#&gt;  shinyjs                2.1.0       2021-12-23 [1] RSPM (R 4.4.0)
#&gt;  shinyWidgets           0.8.6       2024-04-24 [1] RSPM (R 4.4.0)
#&gt;  SingleCellExperiment * 1.26.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  smokingMouse         * 0.99.91     2024-06-09 [1] Github (LieberInstitute/smokingMouse@96d8480)
#&gt;  spam                   2.10-0      2023-10-23 [1] RSPM (R 4.4.0)
#&gt;  SparseArray            1.4.8       2024-05-24 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  sparseMatrixStats      1.16.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  SpatialExperiment    * 1.14.0      2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  spatialLIBD          * 1.16.2      2024-05-28 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  statmod                1.5.0       2023-01-06 [1] RSPM (R 4.4.0)
#&gt;  stringi                1.8.4       2024-05-06 [2] RSPM (R 4.4.0)
#&gt;  stringr              * 1.5.1       2023-11-14 [2] RSPM (R 4.4.0)
#&gt;  styler                 1.10.3      2024-04-07 [1] RSPM (R 4.4.0)
#&gt;  SummarizedExperiment * 1.34.0      2024-05-01 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  sys                    3.4.2       2023-05-23 [2] RSPM (R 4.4.0)
#&gt;  tibble                 3.2.1       2023-03-20 [2] RSPM (R 4.4.0)
#&gt;  tidyr                  1.3.1       2024-01-24 [1] RSPM (R 4.4.0)
#&gt;  tidyselect             1.2.1       2024-03-11 [1] RSPM (R 4.4.0)
#&gt;  timechange             0.3.0       2024-01-18 [1] RSPM (R 4.4.0)
#&gt;  UCSC.utils             1.0.0       2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  usethis              * 2.2.3       2024-02-19 [2] RSPM (R 4.4.0)
#&gt;  utf8                   1.2.4       2023-10-22 [2] RSPM (R 4.4.0)
#&gt;  uwot                   0.2.2       2024-04-21 [1] RSPM (R 4.4.0)
#&gt;  variancePartition    * 1.34.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  vctrs                  0.6.5       2023-12-01 [2] RSPM (R 4.4.0)
#&gt;  vipor                  0.4.7       2023-12-18 [1] RSPM (R 4.4.0)
#&gt;  viridis                0.6.5       2024-01-29 [1] RSPM (R 4.4.0)
#&gt;  viridisLite            0.4.2       2023-05-02 [1] RSPM (R 4.4.0)
#&gt;  whisker                0.4.1       2022-12-05 [2] RSPM (R 4.4.0)
#&gt;  withr                  3.0.0       2024-01-16 [2] RSPM (R 4.4.0)
#&gt;  xfun                   0.44        2024-05-15 [2] RSPM (R 4.4.0)
#&gt;  XML                    3.99-0.16.1 2024-01-22 [1] RSPM (R 4.4.0)
#&gt;  xml2                   1.3.6       2023-12-04 [2] RSPM (R 4.4.0)
#&gt;  xtable                 1.8-4       2019-04-21 [2] RSPM (R 4.4.0)
#&gt;  XVector                0.44.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt;  yaml                   2.3.8       2023-12-11 [2] RSPM (R 4.4.0)
#&gt;  zlibbioc               1.50.0      2024-04-30 [1] Bioconductor 3.19 (R 4.4.0)
#&gt; 
#&gt;  [1] /__w/_temp/Library
#&gt;  [2] /usr/local/lib/R/site-library
#&gt;  [3] /usr/local/lib/R/library
#&gt; 
#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>
<p>System <code>curl</code> version information:</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="final-r-session.html#cb213-1" tabindex="-1"></a>curl<span class="sc">::</span><span class="fu">curl_version</span>()</span>
<span id="cb213-2"><a href="final-r-session.html#cb213-2" tabindex="-1"></a><span class="co">#&gt; $version</span></span>
<span id="cb213-3"><a href="final-r-session.html#cb213-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;7.81.0&quot;</span></span>
<span id="cb213-4"><a href="final-r-session.html#cb213-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-5"><a href="final-r-session.html#cb213-5" tabindex="-1"></a><span class="co">#&gt; $ssl_version</span></span>
<span id="cb213-6"><a href="final-r-session.html#cb213-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;OpenSSL/3.0.2&quot;</span></span>
<span id="cb213-7"><a href="final-r-session.html#cb213-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-8"><a href="final-r-session.html#cb213-8" tabindex="-1"></a><span class="co">#&gt; $libz_version</span></span>
<span id="cb213-9"><a href="final-r-session.html#cb213-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1.2.11&quot;</span></span>
<span id="cb213-10"><a href="final-r-session.html#cb213-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-11"><a href="final-r-session.html#cb213-11" tabindex="-1"></a><span class="co">#&gt; $libssh_version</span></span>
<span id="cb213-12"><a href="final-r-session.html#cb213-12" tabindex="-1"></a><span class="co">#&gt; [1] &quot;libssh/0.9.6/openssl/zlib&quot;</span></span>
<span id="cb213-13"><a href="final-r-session.html#cb213-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-14"><a href="final-r-session.html#cb213-14" tabindex="-1"></a><span class="co">#&gt; $libidn_version</span></span>
<span id="cb213-15"><a href="final-r-session.html#cb213-15" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2.3.2&quot;</span></span>
<span id="cb213-16"><a href="final-r-session.html#cb213-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-17"><a href="final-r-session.html#cb213-17" tabindex="-1"></a><span class="co">#&gt; $host</span></span>
<span id="cb213-18"><a href="final-r-session.html#cb213-18" tabindex="-1"></a><span class="co">#&gt; [1] &quot;x86_64-pc-linux-gnu&quot;</span></span>
<span id="cb213-19"><a href="final-r-session.html#cb213-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-20"><a href="final-r-session.html#cb213-20" tabindex="-1"></a><span class="co">#&gt; $protocols</span></span>
<span id="cb213-21"><a href="final-r-session.html#cb213-21" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;dict&quot;    &quot;file&quot;    &quot;ftp&quot;     &quot;ftps&quot;    &quot;gopher&quot;  &quot;gophers&quot; &quot;http&quot;    &quot;https&quot;   &quot;imap&quot;    &quot;imaps&quot;   &quot;ldap&quot;   </span></span>
<span id="cb213-22"><a href="final-r-session.html#cb213-22" tabindex="-1"></a><span class="co">#&gt; [12] &quot;ldaps&quot;   &quot;mqtt&quot;    &quot;pop3&quot;    &quot;pop3s&quot;   &quot;rtmp&quot;    &quot;rtsp&quot;    &quot;scp&quot;     &quot;sftp&quot;    &quot;smb&quot;     &quot;smbs&quot;    &quot;smtp&quot;   </span></span>
<span id="cb213-23"><a href="final-r-session.html#cb213-23" tabindex="-1"></a><span class="co">#&gt; [23] &quot;smtps&quot;   &quot;telnet&quot;  &quot;tftp&quot;   </span></span>
<span id="cb213-24"><a href="final-r-session.html#cb213-24" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-25"><a href="final-r-session.html#cb213-25" tabindex="-1"></a><span class="co">#&gt; $ipv6</span></span>
<span id="cb213-26"><a href="final-r-session.html#cb213-26" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb213-27"><a href="final-r-session.html#cb213-27" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-28"><a href="final-r-session.html#cb213-28" tabindex="-1"></a><span class="co">#&gt; $http2</span></span>
<span id="cb213-29"><a href="final-r-session.html#cb213-29" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb213-30"><a href="final-r-session.html#cb213-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb213-31"><a href="final-r-session.html#cb213-31" tabindex="-1"></a><span class="co">#&gt; $idn</span></span>
<span id="cb213-32"><a href="final-r-session.html#cb213-32" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>This interactive book was last updated at 2024-06-09 13:33:07.666333.</p>

</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="making-your-own-website-with-postcards.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
