<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />

<link rel="next" href="speaqeasy-introduction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html"><i class="fa fa-check"></i><b>1</b> SPEAQeasy introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#overview-slides"><i class="fa fa-check"></i><b>1.1</b> 2022-04-20 overview slides</a></li>
<li class="chapter" data-level="1.2" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#speaqeasy-main-links"><i class="fa fa-check"></i><b>1.2</b> SPEAQeasy main links</a></li>
<li class="chapter" data-level="1.3" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#pipeline-outputs"><i class="fa fa-check"></i><b>1.3</b> Pipeline outputs</a></li>
<li class="chapter" data-level="1.4" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#exercises"><i class="fa fa-check"></i><b>1.4</b> Exercises</a></li>
<li class="chapter" data-level="1.5" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#solutions"><i class="fa fa-check"></i><b>1.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html"><i class="fa fa-check"></i><b>2</b> smokingMouse study</a>
<ul>
<li class="chapter" data-level="2.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#overview-1"><i class="fa fa-check"></i><b>2.2</b> Overview</a></li>
<li class="chapter" data-level="2.3" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#goals"><i class="fa fa-check"></i><b>2.3</b> Goals</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#main-objective"><i class="fa fa-check"></i><b>2.3.1</b> Main objective</a></li>
<li class="chapter" data-level="2.3.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#secondary-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Secondary objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#study-design"><i class="fa fa-check"></i><b>2.4</b> Study design</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html"><i class="fa fa-check"></i><b>3</b> SummarizedExperiment review</a>
<ul>
<li class="chapter" data-level="3.1" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#quiz"><i class="fa fa-check"></i><b>3.2</b> Quiz</a></li>
<li class="chapter" data-level="3.3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#exercises-1"><i class="fa fa-check"></i><b>3.3</b> Exercises</a></li>
<li class="chapter" data-level="3.4" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#solutions-1"><i class="fa fa-check"></i><b>3.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html"><i class="fa fa-check"></i><b>4</b> smokingMouse RSE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#download-data"><i class="fa fa-check"></i><b>4.1</b> Download data</a></li>
<li class="chapter" data-level="4.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#data-overview"><i class="fa fa-check"></i><b>4.2</b> Data overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#assays"><i class="fa fa-check"></i><b>4.2.1</b> Assays</a></li>
<li class="chapter" data-level="4.2.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#sample-data"><i class="fa fa-check"></i><b>4.2.2</b> Sample data</a></li>
<li class="chapter" data-level="4.2.3" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#gene-information"><i class="fa fa-check"></i><b>4.2.3</b> Gene Information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="isee-introduction.html"><a href="isee-introduction.html"><i class="fa fa-check"></i><b>5</b> iSEE introduction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="isee-introduction.html"><a href="isee-introduction.html#toy-rse-data"><i class="fa fa-check"></i><b>5.1</b> Toy RSE data</a></li>
<li class="chapter" data-level="5.2" data-path="isee-introduction.html"><a href="isee-introduction.html#isee"><i class="fa fa-check"></i><b>5.2</b> iSEE</a></li>
<li class="chapter" data-level="5.3" data-path="isee-introduction.html"><a href="isee-introduction.html#exercise-with-data-from-spatiallibd"><i class="fa fa-check"></i><b>5.3</b> Exercise with data from spatialLIBD</a></li>
<li class="chapter" data-level="5.4" data-path="isee-introduction.html"><a href="isee-introduction.html#community"><i class="fa fa-check"></i><b>5.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html"><i class="fa fa-check"></i><b>6</b> RSE plotting basics with smokingMouse</a>
<ul>
<li class="chapter" data-level="6.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#download-the-smokingmouse-data"><i class="fa fa-check"></i><b>6.1</b> Download the smokingMouse data</a></li>
<li class="chapter" data-level="6.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-preparation"><i class="fa fa-check"></i><b>6.2</b> Data preparation</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-normalization"><i class="fa fa-check"></i><b>6.2.1</b> Data normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#gene-filtering"><i class="fa fa-check"></i><b>6.2.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>6.3</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="6.4" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#quality-control-analysis"><i class="fa fa-check"></i><b>6.4</b> Quality Control Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#evaluate-qc-metrics-for-groups-of-samples"><i class="fa fa-check"></i><b>6.4.1</b> Evaluate QC metrics for groups of samples</a></li>
<li class="chapter" data-level="6.4.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#examine-relationships-between-qc-metrics-of-samples"><i class="fa fa-check"></i><b>6.4.2</b> Examine relationships between QC metrics of samples</a></li>
<li class="chapter" data-level="6.4.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#qc-sample-filtering"><i class="fa fa-check"></i><b>6.4.3</b> QC sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#miscellaneous-about-cpm-values"><i class="fa fa-check"></i><b>6.5</b> Miscellaneous about CPM values</a></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>7</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="7.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>7.1</b> Related past materials</a></li>
<li class="chapter" data-level="7.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>7.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="7.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>7.3</b> Live demo</a></li>
<li class="chapter" data-level="7.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-1"><i class="fa fa-check"></i><b>7.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>8</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>8.1</b> recount projects</a></li>
<li class="chapter" data-level="8.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>8.2</b> Using recount3</a></li>
<li class="chapter" data-level="8.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>8.3</b> Exercise</a></li>
<li class="chapter" data-level="8.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-2"><i class="fa fa-check"></i><b>8.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>9</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="9.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>9.1</b> Model objects in R</a></li>
<li class="chapter" data-level="9.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>9.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>9.3</b> Example 1</a></li>
<li class="chapter" data-level="9.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>9.4</b> Example 2</a></li>
<li class="chapter" data-level="9.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>9.5</b> Example 3</a></li>
<li class="chapter" data-level="9.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>9.6</b> Exercise</a></li>
<li class="chapter" data-level="9.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>9.7</b> To learn more</a></li>
<li class="chapter" data-level="9.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-3"><i class="fa fa-check"></i><b>9.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html"><i class="fa fa-check"></i><b>10</b> Differential gene expression analysis with limma</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#srp045638-data"><i class="fa fa-check"></i><b>10.1</b> SRP045638 data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#data-normalization-1"><i class="fa fa-check"></i><b>10.2</b> Data normalization</a></li>
<li class="chapter" data-level="10.3" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#differential-expression"><i class="fa fa-check"></i><b>10.3</b> Differential expression</a></li>
<li class="chapter" data-level="10.4" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#visualizing-degs"><i class="fa fa-check"></i><b>10.4</b> Visualizing DEGs</a></li>
<li class="chapter" data-level="10.5" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#community-4"><i class="fa fa-check"></i><b>10.5</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>11</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>11.1</b> Recap</a></li>
<li class="chapter" data-level="11.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>11.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>12</b> DGE model building with variancePartition</a>
<ul>
<li class="chapter" data-level="12.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#download-the-smokingmouse-data-and-subset-it"><i class="fa fa-check"></i><b>12.1</b> Download the smokingMouse data and subset it</a></li>
<li class="chapter" data-level="12.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#explore-gene-level-effects"><i class="fa fa-check"></i><b>12.2</b> Explore gene-level effects</a></li>
<li class="chapter" data-level="12.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#variance-partition-analysis"><i class="fa fa-check"></i><b>12.3</b> Variance Partition Analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>12.3.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="12.3.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>12.3.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="12.3.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>12.3.3</b> Examine expression of most affected genes by each sample variable</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#solutions-2"><i class="fa fa-check"></i><b>12.4</b> Solutions</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#exercise-1-1"><i class="fa fa-check"></i><b>12.4.1</b> Exercise 1</a></li>
<li class="chapter" data-level="12.4.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#exercise-2-1"><i class="fa fa-check"></i><b>12.4.2</b> Exercise 2</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>13</b> Expression heatmaps with ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="13.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#introduction-1"><i class="fa fa-check"></i><b>13.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#complexheatmap"><i class="fa fa-check"></i><b>13.2</b> ComplexHeatmap</a></li>
<li class="chapter" data-level="13.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#heatmap-with-expression-data"><i class="fa fa-check"></i><b>13.3</b> Heatmap with expression data</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#download-data-1"><i class="fa fa-check"></i><b>13.3.1</b> Download data</a></li>
<li class="chapter" data-level="13.3.2" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#prepare-the-data"><i class="fa fa-check"></i><b>13.3.2</b> Prepare the data</a></li>
<li class="chapter" data-level="13.3.3" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#create-annotations"><i class="fa fa-check"></i><b>13.3.3</b> Create annotations</a></li>
<li class="chapter" data-level="13.3.4" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html#plot-our-heatmap"><i class="fa fa-check"></i><b>13.3.4</b> Plot our heatmap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="complexheatmap-exercise-solution.html"><a href="complexheatmap-exercise-solution.html"><i class="fa fa-check"></i><b>14</b> ComplexHeatmap exercise solution</a></li>
<li class="chapter" data-level="15" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html"><i class="fa fa-check"></i><b>15</b> Exercise with variancePartition and ComplexHeatmap</a>
<ul>
<li class="chapter" data-level="15.1" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html#steps"><i class="fa fa-check"></i><b>15.1</b> Steps:</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>16</b> Miscellaneous</a>
<ul>
<li class="chapter" data-level="16.1" data-path="miscellaneous.html"><a href="miscellaneous.html#postcards"><i class="fa fa-check"></i><b>16.1</b> postcards</a></li>
<li class="chapter" data-level="16.2" data-path="miscellaneous.html"><a href="miscellaneous.html#organizing-your-work"><i class="fa fa-check"></i><b>16.2</b> Organizing your work</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><i class="fa fa-check"></i><b>17</b> Introduction to the world of spatially-resolved transcriptomics with Visium data</a>
<ul>
<li class="chapter" data-level="17.1" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html#community-5"><i class="fa fa-check"></i><b>17.1</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><i class="fa fa-check"></i><b>18</b> Downloading public Visium spatial data and visualizing it with spatialLIBD</a>
<ul>
<li class="chapter" data-level="18.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#download-data-from-10x-genomics"><i class="fa fa-check"></i><b>18.1</b> Download data from 10x Genomics</a></li>
<li class="chapter" data-level="18.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#add-gene-annotation-information"><i class="fa fa-check"></i><b>18.2</b> Add gene annotation information</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-10x"><i class="fa fa-check"></i><b>18.2.1</b> From 10x</a></li>
<li class="chapter" data-level="18.2.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-gencode"><i class="fa fa-check"></i><b>18.2.2</b> From Gencode</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#wrapper-functions-for-reading-the-data"><i class="fa fa-check"></i><b>18.3</b> Wrapper functions for reading the data</a></li>
<li class="chapter" data-level="18.4" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#visualize-spatial-plots"><i class="fa fa-check"></i><b>18.4</b> Visualize spatial plots</a></li>
<li class="chapter" data-level="18.5" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#exercise-3"><i class="fa fa-check"></i><b>18.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>19</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="19.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>19.1</b> Spatial registration</a></li>
<li class="chapter" data-level="19.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-4"><i class="fa fa-check"></i><b>19.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
<li class="chapter" data-level="20" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>20</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="20.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>20.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>20.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="20.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis-1"><i class="fa fa-check"></i><b>20.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>20.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="20.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>20.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="20.4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-2"><i class="fa fa-check"></i><b>20.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>21</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="21.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>21.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="21.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>21.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>21.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="21.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>21.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="21.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>21.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="21.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>21.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="21.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>21.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>21.3</b> DE visualization</a></li>
<li class="chapter" data-level="21.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-3"><i class="fa fa-check"></i><b>21.4</b> References</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Statistical Analysis of Genome Scale Data 2024</h1>
<p class="author"><em>Leonardo Collado-Torres</em></p>
</div>
<div id="overview" class="section level1 unnumbered hasAnchor">
<h1>Overview<a href="index.html#overview" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here you can find the files for the June 2024 <em>Statistical Analysis of Genome Scale Data</em> course at <a href="https://meetings.cshl.edu/courses.aspx?course=C-DATA&amp;year=24">CSHL</a> portion taught by Leo and his team (June 9-11).</p>
<!-- See [CSHLData2023](https://seandavi.github.io/CSHLData2023/) for a list of all instructors and related materials. -->
<p>Instructor:</p>
<ul>
<li><a href="http://lcolladotor.github.io/"><strong>Leonardo Collado-Torres</strong></a>, <a href="https://twitter.com/lcolladotor">Twitter</a></li>
</ul>
<p>Teach assistants:</p>
<ul>
<li><a href="https://daianna21.github.io/">Daianna González Padilla</a>, <a href="https://twitter.com/daianna_glez">Twitter</a></li>
<li><a href="http://Melii99.github.io/">Melissa Mayén Quiroz</a>, <a href="http://twitter.com/meli_mq99">Twitter</a></li>
</ul>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
Happy to welcome <a href="https://twitter.com/lcolladotor?ref_src=twsrc%5Etfw"><span class="citation">@lcolladotor</span></a>, <a href="https://twitter.com/SubmarineGene?ref_src=twsrc%5Etfw"><span class="citation">@SubmarineGene</span></a>, and <a href="https://twitter.com/daianna_glez?ref_src=twsrc%5Etfw"><span class="citation">@daianna_glez</span></a> to <a href="https://twitter.com/cshlcourses?ref_src=twsrc%5Etfw"><span class="citation">@cshlcourses</span></a> on Statistical Analysis of Genome Scale Data. Materials here:<a href="https://t.co/0JGzEEf3Yx">https://t.co/0JGzEEf3Yx</a>
</p>
— Sean Davis (<span class="citation">@seandavis12</span>) <a href="https://twitter.com/seandavis12/status/1675926879826485253?ref_src=twsrc%5Etfw">July 3, 2023</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
Interested in learning R/<a href="https://twitter.com/Bioconductor?ref_src=twsrc%5Etfw"><span class="citation">@Bioconductor</span></a> for <a href="https://twitter.com/hashtag/RNAseq?src=hash&amp;ref_src=twsrc%5Etfw">#RNAseq</a> data analysis?<br><br>Check <a href="https://t.co/jcNWOq7VWD">https://t.co/jcNWOq7VWD</a> that <a href="https://twitter.com/daianna_glez?ref_src=twsrc%5Etfw"><span class="citation">@daianna_glez</span></a> <a href="https://twitter.com/SubmarineGene?ref_src=twsrc%5Etfw"><span class="citation">@SubmarineGene</span></a> and I built for a CSHL course organized by <a href="https://twitter.com/seandavis12?ref_src=twsrc%5Etfw"><span class="citation">@seandavis12</span></a> &amp; Vincent Carey<a href="https://twitter.com/hashtag/rstats?src=hash&amp;ref_src=twsrc%5Etfw">#rstats</a> <a href="https://twitter.com/LieberInstitute?ref_src=twsrc%5Etfw"><span class="citation">@LieberInstitute</span></a> <a href="https://twitter.com/cshlcourses?ref_src=twsrc%5Etfw"><span class="citation">@cshlcourses</span></a> <br><br>Source at <a href="https://t.co/aQ8kLMadSl">https://t.co/aQ8kLMadSl</a> ⭐️
</p>
— 🇲🇽 Leonardo Collado-Torres (<span class="citation">@lcolladotor</span>) <a href="https://twitter.com/lcolladotor/status/1678488349307461632?ref_src=twsrc%5Etfw">July 10, 2023</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
<div id="download-course-materials" class="section level2 unnumbered hasAnchor">
<h2>Download course materials<a href="index.html#download-course-materials" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the materials for this course with <code>usethis::use_course('lcolladotor/cshl_rstats_genome_scale_2024')</code> or view online at <a href="http://lcolladotor.github.io/cshl_rstats_genome_scale_2024"><strong>lcolladotor.github.io/cshl_rstats_genome_scale_2024</strong></a>. This command downloads a static version of the course materials.</p>
<p>If you want to be able to easily download updates, we recommend using <code>Git</code>. <a href="https://happygitwithr.com/"><em>Happy Git and GitHub for the useR</em></a> is great for getting your computer ready to use <code>Git</code> and <code>GitHub</code>. If you already have a <code>GitHub</code> account, you can instead use this command to download the course:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="index.html#cb1-1" tabindex="-1"></a><span class="co">## Download it the first time</span></span>
<span id="cb1-2"><a href="index.html#cb1-2" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/lcolladotor/cshl_rstats_genome_scale_2024.git</span>
<span id="cb1-3"><a href="index.html#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="index.html#cb1-4" tabindex="-1"></a><span class="co">## To update the contents, use:</span></span>
<span id="cb1-5"><a href="index.html#cb1-5" tabindex="-1"></a><span class="bu">cd</span> cshl_rstats_genome_scale_2024</span>
<span id="cb1-6"><a href="index.html#cb1-6" tabindex="-1"></a><span class="fu">git</span> pull</span></code></pre></div>
<p>Or you could use the <a href="x-github-client://openRepo/https://github.com/lcolladotor/cshl_rstats_genome_scale_2024">GitHub Desktop application</a>.</p>
</div>
<div id="code-of-conduct" class="section level2 unnumbered hasAnchor">
<h2>Code of Conduct<a href="index.html#code-of-conduct" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll follow the CSHL code of conduct as well as version 1.2.0 of the Bioconductor code of conduct <a href="https://bioconductor.github.io/bioc_coc_multilingual/">bioconductor.github.io/bioc_coc_multilingual/</a>.</p>
<p>For reporting any violations of the code of conduct, report them to the Instructor and/or Course Coordinators.</p>
</div>
<div id="course-schedule" class="section level2 unnumbered hasAnchor">
<h2>Course Schedule<a href="index.html#course-schedule" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>Local times in US Eastern</em></p>
<p>See <a href="https://docs.google.com/document/d/1pdvZFjJBwRUpA_VJOAFu4z-0Eys6Wv2iRAh84T6_yF4/edit?usp=sharing">CSHLData2023</a> for the detailed schedule. Below is a list of the key packages we will use each day:</p>
<ul>
<li>Monday July 3rd: <code>SummarizedExperiment</code>, <code>iSEE</code>, <code>smokingMouse</code>, <code>biocthis</code></li>
<li>Tuesday July 4th: <code>recount3</code>, <code>ExploreModelMatrix</code>, <code>limma</code></li>
<li>Wednesday July 5th: <code>variancePartition</code>, <code>ComplexHeatmap</code></li>
<li>Thursday July 6th: <code>spatialLIBD</code></li>
</ul>
</div>
<div id="external-links" class="section level2 unnumbered hasAnchor">
<h2>External links<a href="index.html#external-links" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<!-- * [CSHLData2023](https://seandavi.github.io/CSHLData2023/) -->
<ul>
<li><a href="https://meetings.cshl.edu/courses.aspx?course=C-DATA&amp;year=24">CSHL course</a></li>
<li><a href="https://github.com/lcolladotor/cshl_rstats_genome_scale_2024">GitHub source code</a></li>
<li><a href="https://cshldata-2023.slack.com">Slack</a></li>
<li>Example real project (by Daianna): <a href="https://github.com/LieberInstitute/smokingMouse_Indirects">LieberInstitute/smokingMouse_Indirects</a></li>
<li><code>SPEAQeasyWorkshop2023</code> for BioC2023 (by Daianna and Renee): <a href="https://github.com/LieberInstitute/SPEAQeasyWorkshop2023">LieberInstitute/SPEAQeasyWorkshop2023</a></li>
<li><a href="https://github.com/LieberInstitute/template_project">LieberInstitute/template_project</a></li>
<li><a href="http://research.libd.org/rstatsclub/">LIBD rstats club</a>: check the <a href="https://docs.google.com/spreadsheets/d/1is8dZSd0FZ9Qi1Zvq1uRhm-P1McnJRd_zxdAfCRoMfA/edit?usp=sharing">public schedule</a></li>
</ul>
</div>
<div id="course-prerequisites" class="section level2 unnumbered hasAnchor">
<h2>Course Prerequisites<a href="index.html#course-prerequisites" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Install R 4.3.x from <a href="https://cran.r-project.org/">CRAN</a> then install the following R packages:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" tabindex="-1"></a><span class="do">## For installing Bioconductor packages</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb2-3"><a href="index.html#cb2-3" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb2-4"><a href="index.html#cb2-4" tabindex="-1"></a>}</span>
<span id="cb2-5"><a href="index.html#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="index.html#cb2-6" tabindex="-1"></a><span class="do">## Install required packages</span></span>
<span id="cb2-7"><a href="index.html#cb2-7" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(</span>
<span id="cb2-8"><a href="index.html#cb2-8" tabindex="-1"></a>    <span class="fu">c</span>(</span>
<span id="cb2-9"><a href="index.html#cb2-9" tabindex="-1"></a>        <span class="st">&quot;usethis&quot;</span>, <span class="do">## Utilities</span></span>
<span id="cb2-10"><a href="index.html#cb2-10" tabindex="-1"></a>        <span class="st">&quot;BiocFileCache&quot;</span>,</span>
<span id="cb2-11"><a href="index.html#cb2-11" tabindex="-1"></a>        <span class="st">&quot;RefManageR&quot;</span>,</span>
<span id="cb2-12"><a href="index.html#cb2-12" tabindex="-1"></a>        <span class="st">&quot;here&quot;</span>,</span>
<span id="cb2-13"><a href="index.html#cb2-13" tabindex="-1"></a>        <span class="st">&quot;Hmisc&quot;</span>,</span>
<span id="cb2-14"><a href="index.html#cb2-14" tabindex="-1"></a>        <span class="st">&quot;biocthis&quot;</span>,</span>
<span id="cb2-15"><a href="index.html#cb2-15" tabindex="-1"></a>        <span class="st">&quot;lobstr&quot;</span>,</span>
<span id="cb2-16"><a href="index.html#cb2-16" tabindex="-1"></a>        <span class="st">&quot;postcards&quot;</span>,</span>
<span id="cb2-17"><a href="index.html#cb2-17" tabindex="-1"></a>        <span class="st">&quot;scater&quot;</span>,</span>
<span id="cb2-18"><a href="index.html#cb2-18" tabindex="-1"></a>        <span class="st">&quot;sessioninfo&quot;</span>,</span>
<span id="cb2-19"><a href="index.html#cb2-19" tabindex="-1"></a>        <span class="st">&quot;smokingMouse&quot;</span>,</span>
<span id="cb2-20"><a href="index.html#cb2-20" tabindex="-1"></a>        <span class="st">&quot;stringr&quot;</span>,</span>
<span id="cb2-21"><a href="index.html#cb2-21" tabindex="-1"></a>        <span class="st">&quot;SummarizedExperiment&quot;</span>, <span class="do">## Main containers / vis</span></span>
<span id="cb2-22"><a href="index.html#cb2-22" tabindex="-1"></a>        <span class="st">&quot;iSEE&quot;</span>,</span>
<span id="cb2-23"><a href="index.html#cb2-23" tabindex="-1"></a>        <span class="st">&quot;edgeR&quot;</span>, <span class="do">## RNA-seq</span></span>
<span id="cb2-24"><a href="index.html#cb2-24" tabindex="-1"></a>        <span class="st">&quot;ExploreModelMatrix&quot;</span>,</span>
<span id="cb2-25"><a href="index.html#cb2-25" tabindex="-1"></a>        <span class="st">&quot;limma&quot;</span>,</span>
<span id="cb2-26"><a href="index.html#cb2-26" tabindex="-1"></a>        <span class="st">&quot;recount3&quot;</span>,</span>
<span id="cb2-27"><a href="index.html#cb2-27" tabindex="-1"></a>        <span class="st">&quot;rlang&quot;</span>,</span>
<span id="cb2-28"><a href="index.html#cb2-28" tabindex="-1"></a>        <span class="st">&quot;pheatmap&quot;</span>, <span class="do">## Visualization</span></span>
<span id="cb2-29"><a href="index.html#cb2-29" tabindex="-1"></a>        <span class="st">&quot;ggplot2&quot;</span>,</span>
<span id="cb2-30"><a href="index.html#cb2-30" tabindex="-1"></a>        <span class="st">&quot;ggrepel&quot;</span>,</span>
<span id="cb2-31"><a href="index.html#cb2-31" tabindex="-1"></a>        <span class="st">&quot;patchwork&quot;</span>,</span>
<span id="cb2-32"><a href="index.html#cb2-32" tabindex="-1"></a>        <span class="st">&quot;RColorBrewer&quot;</span>,</span>
<span id="cb2-33"><a href="index.html#cb2-33" tabindex="-1"></a>        <span class="st">&quot;ComplexHeatmap&quot;</span>,</span>
<span id="cb2-34"><a href="index.html#cb2-34" tabindex="-1"></a>        <span class="st">&quot;cowplot&quot;</span>,</span>
<span id="cb2-35"><a href="index.html#cb2-35" tabindex="-1"></a>        <span class="st">&quot;Polychrome&quot;</span>,</span>
<span id="cb2-36"><a href="index.html#cb2-36" tabindex="-1"></a>        <span class="st">&quot;spatialLIBD&quot;</span>, <span class="do">## Advanced</span></span>
<span id="cb2-37"><a href="index.html#cb2-37" tabindex="-1"></a>        <span class="st">&quot;variancePartition&quot;</span></span>
<span id="cb2-38"><a href="index.html#cb2-38" tabindex="-1"></a>    )</span>
<span id="cb2-39"><a href="index.html#cb2-39" tabindex="-1"></a>)</span>
<span id="cb2-40"><a href="index.html#cb2-40" tabindex="-1"></a></span>
<span id="cb2-41"><a href="index.html#cb2-41" tabindex="-1"></a><span class="do">## Install smokingMouse, which is currently under review at Bioconductor</span></span>
<span id="cb2-42"><a href="index.html#cb2-42" tabindex="-1"></a><span class="do">## at https://github.com/Bioconductor/Contributions/issues/3024.</span></span>
<span id="cb2-43"><a href="index.html#cb2-43" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;LieberInstitute/smokingMouse&quot;</span>)</span>
<span id="cb2-44"><a href="index.html#cb2-44" tabindex="-1"></a></span>
<span id="cb2-45"><a href="index.html#cb2-45" tabindex="-1"></a><span class="do">## Required to access some recent bug fixes on biocthis</span></span>
<span id="cb2-46"><a href="index.html#cb2-46" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;lcolladotor/biocthis&quot;</span>)</span></code></pre></div>
<p>You will also need to install <a href="https://rstudio.com/products/rstudio/download/#download">RStudio</a> version 2023.06.0+421 or newer.</p>
</div>
<div id="r-session-information" class="section level2 unnumbered hasAnchor">
<h2>R session information<a href="index.html#r-session-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Details on the R version used for making this book. The source code is available at <a href="https://github.com/lcolladotor/cshl_rstats_genome_scale_2024"><code>lcolladotor/cshl_rstats_genome_scale_2024</code></a>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1" tabindex="-1"></a><span class="do">## Load the package at the top of your script</span></span>
<span id="cb3-2"><a href="index.html#cb3-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;sessioninfo&quot;</span>)</span>
<span id="cb3-3"><a href="index.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="index.html#cb3-4" tabindex="-1"></a><span class="do">## Utilities</span></span>
<span id="cb3-5"><a href="index.html#cb3-5" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;BiocFileCache&quot;</span>)</span>
<span id="cb3-6"><a href="index.html#cb3-6" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;BiocStyle&quot;</span>)</span>
<span id="cb3-7"><a href="index.html#cb3-7" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;biocthis&quot;</span>)</span>
<span id="cb3-8"><a href="index.html#cb3-8" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;here&quot;</span>)</span>
<span id="cb3-9"><a href="index.html#cb3-9" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;lobstr&quot;</span>)</span>
<span id="cb3-10"><a href="index.html#cb3-10" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;postcards&quot;</span>)</span>
<span id="cb3-11"><a href="index.html#cb3-11" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;usethis&quot;</span>)</span>
<span id="cb3-12"><a href="index.html#cb3-12" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;sessioninfo&quot;</span>)</span>
<span id="cb3-13"><a href="index.html#cb3-13" tabindex="-1"></a></span>
<span id="cb3-14"><a href="index.html#cb3-14" tabindex="-1"></a><span class="do">## Data</span></span>
<span id="cb3-15"><a href="index.html#cb3-15" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;smokingMouse&quot;</span>)</span>
<span id="cb3-16"><a href="index.html#cb3-16" tabindex="-1"></a></span>
<span id="cb3-17"><a href="index.html#cb3-17" tabindex="-1"></a><span class="do">## Main containers / vis</span></span>
<span id="cb3-18"><a href="index.html#cb3-18" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</span>
<span id="cb3-19"><a href="index.html#cb3-19" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;iSEE&quot;</span>)</span>
<span id="cb3-20"><a href="index.html#cb3-20" tabindex="-1"></a></span>
<span id="cb3-21"><a href="index.html#cb3-21" tabindex="-1"></a><span class="do">## RNA-seq</span></span>
<span id="cb3-22"><a href="index.html#cb3-22" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;edgeR&quot;</span>)</span>
<span id="cb3-23"><a href="index.html#cb3-23" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ExploreModelMatrix&quot;</span>)</span>
<span id="cb3-24"><a href="index.html#cb3-24" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;limma&quot;</span>)</span>
<span id="cb3-25"><a href="index.html#cb3-25" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;recount3&quot;</span>)</span>
<span id="cb3-26"><a href="index.html#cb3-26" tabindex="-1"></a></span>
<span id="cb3-27"><a href="index.html#cb3-27" tabindex="-1"></a><span class="do">## QCA</span></span>
<span id="cb3-28"><a href="index.html#cb3-28" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;scater&quot;</span>)</span>
<span id="cb3-29"><a href="index.html#cb3-29" tabindex="-1"></a></span>
<span id="cb3-30"><a href="index.html#cb3-30" tabindex="-1"></a><span class="do">## Variance Partition</span></span>
<span id="cb3-31"><a href="index.html#cb3-31" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;variancePartition&quot;</span>)</span>
<span id="cb3-32"><a href="index.html#cb3-32" tabindex="-1"></a></span>
<span id="cb3-33"><a href="index.html#cb3-33" tabindex="-1"></a><span class="do">## Visualization: plots &amp; text</span></span>
<span id="cb3-34"><a href="index.html#cb3-34" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ComplexHeatmap&quot;</span>)</span>
<span id="cb3-35"><a href="index.html#cb3-35" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb3-36"><a href="index.html#cb3-36" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;patchwork&quot;</span>)</span>
<span id="cb3-37"><a href="index.html#cb3-37" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb3-38"><a href="index.html#cb3-38" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;RColorBrewer&quot;</span>)</span>
<span id="cb3-39"><a href="index.html#cb3-39" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;Hmisc&quot;</span>)</span>
<span id="cb3-40"><a href="index.html#cb3-40" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;stringr&quot;</span>)</span>
<span id="cb3-41"><a href="index.html#cb3-41" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;cowplot&quot;</span>)</span>
<span id="cb3-42"><a href="index.html#cb3-42" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;rlang&quot;</span>)</span>
<span id="cb3-43"><a href="index.html#cb3-43" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggrepel&quot;</span>)</span>
<span id="cb3-44"><a href="index.html#cb3-44" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;Polychrome&quot;</span>)</span>
<span id="cb3-45"><a href="index.html#cb3-45" tabindex="-1"></a></span>
<span id="cb3-46"><a href="index.html#cb3-46" tabindex="-1"></a><span class="do">## Spatial transcriptomics</span></span>
<span id="cb3-47"><a href="index.html#cb3-47" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;spatialLIBD&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="index.html#cb4-1" tabindex="-1"></a><span class="do">## Reproducibility information</span></span>
<span id="cb4-2"><a href="index.html#cb4-2" tabindex="-1"></a><span class="fu">options</span>(<span class="at">width =</span> <span class="dv">120</span>)</span>
<span id="cb4-3"><a href="index.html#cb4-3" tabindex="-1"></a><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.3.3 (2024-02-29)
##  os       Ubuntu 22.04.4 LTS
##  system   x86_64, linux-gnu
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       UTC
##  date     2024-05-17
##  pandoc   3.1.1 @ /usr/local/bin/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package                * version     date (UTC) lib source
##  abind                    1.4-5       2016-07-21 [1] RSPM (R 4.3.0)
##  AnnotationDbi            1.64.1      2023-11-03 [1] Bioconductor
##  AnnotationHub            3.10.1      2024-04-05 [1] Bioconductor 3.18 (R 4.3.2)
##  aod                      1.3.3       2023-12-13 [1] RSPM (R 4.3.0)
##  attempt                  0.3.1       2020-05-03 [1] RSPM (R 4.3.0)
##  backports                1.4.1       2021-12-13 [1] RSPM (R 4.3.0)
##  base64enc                0.1-3       2015-07-28 [2] RSPM (R 4.3.0)
##  beachmat                 2.18.1      2024-02-14 [1] Bioconductor 3.18 (R 4.3.2)
##  beeswarm                 0.4.0       2021-06-01 [1] RSPM (R 4.3.0)
##  benchmarkme              1.0.8       2022-06-12 [1] RSPM (R 4.3.0)
##  benchmarkmeData          1.0.4       2020-04-23 [1] RSPM (R 4.3.0)
##  Biobase                * 2.62.0      2023-10-24 [1] Bioconductor
##  BiocFileCache          * 2.10.2      2024-03-27 [1] Bioconductor 3.18 (R 4.3.2)
##  BiocGenerics           * 0.48.1      2023-11-01 [1] Bioconductor
##  BiocIO                   1.12.0      2023-10-24 [1] Bioconductor
##  BiocManager              1.30.22     2023-08-08 [1] RSPM (R 4.3.0)
##  BiocNeighbors            1.20.2      2024-01-07 [1] Bioconductor 3.18 (R 4.3.2)
##  BiocParallel           * 1.36.0      2023-10-24 [1] Bioconductor
##  BiocSingular             1.18.0      2023-10-24 [1] Bioconductor
##  BiocStyle              * 2.30.0      2023-10-24 [1] Bioconductor
##  biocthis               * 1.12.0      2023-10-24 [1] Bioconductor
##  BiocVersion              3.18.1      2023-11-15 [2] Bioconductor
##  Biostrings               2.70.3      2024-03-13 [1] Bioconductor 3.18 (R 4.3.2)
##  bit                      4.0.5       2022-11-15 [1] RSPM (R 4.3.0)
##  bit64                    4.0.5       2020-08-30 [1] RSPM (R 4.3.0)
##  bitops                   1.0-7       2021-04-24 [1] RSPM (R 4.3.0)
##  blob                     1.2.4       2023-03-17 [1] RSPM (R 4.3.0)
##  bookdown                 0.39        2024-04-15 [1] RSPM (R 4.3.0)
##  boot                     1.3-30      2024-02-26 [3] RSPM (R 4.3.0)
##  broom                    1.0.5       2023-06-09 [1] RSPM (R 4.3.0)
##  bslib                    0.7.0       2024-03-29 [2] RSPM (R 4.3.0)
##  cachem                   1.0.8       2023-05-01 [2] RSPM (R 4.3.0)
##  caTools                  1.18.2      2021-03-28 [1] RSPM (R 4.3.0)
##  checkmate                2.3.1       2023-12-04 [1] RSPM (R 4.3.0)
##  circlize                 0.4.16      2024-02-20 [1] RSPM (R 4.3.0)
##  cli                      3.6.2       2023-12-11 [2] RSPM (R 4.3.0)
##  clue                     0.3-65      2023-09-23 [1] RSPM (R 4.3.0)
##  cluster                  2.1.6       2023-12-01 [3] CRAN (R 4.3.3)
##  codetools                0.2-20      2024-03-31 [3] RSPM (R 4.3.0)
##  colorspace               2.1-0       2023-01-23 [1] RSPM (R 4.3.0)
##  colourpicker             1.3.0       2023-08-21 [1] RSPM (R 4.3.0)
##  ComplexHeatmap         * 2.18.0      2023-10-24 [1] Bioconductor
##  config                   0.3.2       2023-08-30 [1] RSPM (R 4.3.0)
##  corpcor                  1.6.10      2021-09-16 [1] RSPM (R 4.3.0)
##  cowplot                * 1.1.3       2024-01-22 [1] RSPM (R 4.3.0)
##  crayon                   1.5.2       2022-09-29 [2] RSPM (R 4.3.0)
##  curl                     5.2.1       2024-03-01 [2] RSPM (R 4.3.0)
##  data.table               1.15.4      2024-03-30 [1] RSPM (R 4.3.0)
##  DBI                      1.2.2       2024-02-16 [1] RSPM (R 4.3.0)
##  dbplyr                 * 2.5.0       2024-03-19 [1] RSPM (R 4.3.0)
##  DelayedArray             0.28.0      2023-10-24 [1] Bioconductor
##  DelayedMatrixStats       1.24.0      2023-10-24 [1] Bioconductor
##  digest                   0.6.35      2024-03-11 [2] RSPM (R 4.3.0)
##  doParallel               1.0.17      2022-02-07 [1] RSPM (R 4.3.0)
##  dotCall64                1.1-1       2023-11-28 [1] RSPM (R 4.3.0)
##  dplyr                    1.1.4       2023-11-17 [1] RSPM (R 4.3.0)
##  DT                       0.33        2024-04-04 [1] RSPM (R 4.3.0)
##  edgeR                  * 4.0.16      2024-02-18 [1] Bioconductor 3.18 (R 4.3.2)
##  EnvStats                 2.8.1       2023-08-22 [1] RSPM (R 4.3.0)
##  evaluate                 0.23        2023-11-01 [2] RSPM (R 4.3.0)
##  ExperimentHub            2.10.0      2023-10-24 [1] Bioconductor
##  ExploreModelMatrix     * 1.14.0      2023-10-24 [1] Bioconductor
##  fANCOVA                  0.6-1       2020-11-13 [1] RSPM (R 4.3.0)
##  fansi                    1.0.6       2023-12-08 [2] RSPM (R 4.3.0)
##  fastmap                  1.1.1       2023-02-24 [2] RSPM (R 4.3.0)
##  fields                   15.2        2023-08-17 [1] RSPM (R 4.3.0)
##  filelock                 1.0.3       2023-12-11 [1] RSPM (R 4.3.0)
##  foreach                  1.5.2       2022-02-02 [1] RSPM (R 4.3.0)
##  foreign                  0.8-86      2023-11-28 [3] CRAN (R 4.3.3)
##  Formula                  1.2-5       2023-02-24 [1] RSPM (R 4.3.0)
##  fs                       1.6.3       2023-07-20 [2] RSPM (R 4.3.0)
##  generics                 0.1.3       2022-07-05 [1] RSPM (R 4.3.0)
##  GenomeInfoDb           * 1.38.8      2024-03-15 [1] Bioconductor 3.18 (R 4.3.2)
##  GenomeInfoDbData         1.2.11      2024-04-06 [1] Bioconductor
##  GenomicAlignments        1.38.2      2024-01-16 [1] Bioconductor 3.18 (R 4.3.2)
##  GenomicRanges          * 1.54.1      2023-10-29 [1] Bioconductor
##  GetoptLong               1.0.5       2020-12-15 [1] RSPM (R 4.3.0)
##  ggbeeswarm               0.7.2       2023-04-29 [1] RSPM (R 4.3.0)
##  ggplot2                * 3.5.0       2024-02-23 [1] RSPM (R 4.3.0)
##  ggrepel                * 0.9.5       2024-01-10 [1] RSPM (R 4.3.0)
##  GlobalOptions            0.1.2       2020-06-10 [1] RSPM (R 4.3.0)
##  glue                     1.7.0       2024-01-09 [2] RSPM (R 4.3.0)
##  golem                    0.4.1       2023-06-05 [1] RSPM (R 4.3.0)
##  gplots                   3.1.3.1     2024-02-02 [1] RSPM (R 4.3.0)
##  gridExtra                2.3         2017-09-09 [1] RSPM (R 4.3.0)
##  gtable                   0.3.5       2024-04-22 [1] RSPM (R 4.3.0)
##  gtools                   3.9.5       2023-11-20 [1] RSPM (R 4.3.0)
##  here                   * 1.0.1       2020-12-13 [1] RSPM (R 4.3.0)
##  Hmisc                  * 5.1-2       2024-03-11 [1] RSPM (R 4.3.0)
##  htmlTable                2.4.2       2023-10-29 [1] RSPM (R 4.3.0)
##  htmltools                0.5.8.1     2024-04-04 [2] RSPM (R 4.3.0)
##  htmlwidgets              1.6.4       2023-12-06 [2] RSPM (R 4.3.0)
##  httpuv                   1.6.15      2024-03-26 [2] RSPM (R 4.3.0)
##  httr                     1.4.7       2023-08-15 [2] RSPM (R 4.3.0)
##  igraph                   2.0.3       2024-03-13 [1] RSPM (R 4.3.0)
##  interactiveDisplayBase   1.40.0      2023-10-24 [1] Bioconductor
##  IRanges                * 2.36.0      2023-10-24 [1] Bioconductor
##  irlba                    2.3.5.1     2022-10-03 [1] RSPM (R 4.3.0)
##  iSEE                   * 2.14.0      2023-10-24 [1] Bioconductor
##  iterators                1.0.14      2022-02-05 [1] RSPM (R 4.3.0)
##  jquerylib                0.1.4       2021-04-26 [2] RSPM (R 4.3.0)
##  jsonlite                 1.8.8       2023-12-04 [2] RSPM (R 4.3.0)
##  KEGGREST                 1.42.0      2023-10-24 [1] Bioconductor
##  KernSmooth               2.23-22     2023-07-10 [3] CRAN (R 4.3.3)
##  knitr                    1.46        2024-04-06 [2] RSPM (R 4.3.0)
##  later                    1.3.2       2023-12-06 [2] RSPM (R 4.3.0)
##  lattice                  0.22-6      2024-03-20 [3] RSPM (R 4.3.0)
##  lazyeval                 0.2.2       2019-03-15 [1] RSPM (R 4.3.0)
##  lifecycle                1.0.4       2023-11-07 [2] RSPM (R 4.3.0)
##  limma                  * 3.58.1      2023-10-31 [1] Bioconductor
##  lme4                     1.1-35.3    2024-04-16 [1] RSPM (R 4.3.0)
##  lmerTest                 3.1-3       2020-10-23 [1] RSPM (R 4.3.0)
##  lobstr                 * 1.1.2       2022-06-22 [1] RSPM (R 4.3.0)
##  locfit                   1.5-9.9     2024-03-01 [1] RSPM (R 4.3.0)
##  magick                   2.8.3       2024-02-18 [1] RSPM (R 4.3.0)
##  magrittr                 2.0.3       2022-03-30 [2] RSPM (R 4.3.0)
##  maps                     3.4.2       2023-12-15 [1] RSPM (R 4.3.0)
##  MASS                     7.3-60.0.1  2024-01-13 [3] CRAN (R 4.3.3)
##  Matrix                   1.6-5       2024-01-11 [3] CRAN (R 4.3.3)
##  MatrixGenerics         * 1.14.0      2023-10-24 [1] Bioconductor
##  matrixStats            * 1.3.0       2024-04-11 [1] RSPM (R 4.3.0)
##  memoise                  2.0.1       2021-11-26 [2] RSPM (R 4.3.0)
##  mgcv                     1.9-1       2023-12-21 [3] CRAN (R 4.3.3)
##  mime                     0.12        2021-09-28 [2] RSPM (R 4.3.0)
##  miniUI                   0.1.1.1     2018-05-18 [2] RSPM (R 4.3.0)
##  minqa                    1.2.6       2023-09-11 [1] RSPM (R 4.3.0)
##  munsell                  0.5.1       2024-04-01 [1] RSPM (R 4.3.0)
##  mvtnorm                  1.2-4       2023-11-27 [1] RSPM (R 4.3.0)
##  nlme                     3.1-164     2023-11-27 [3] CRAN (R 4.3.3)
##  nloptr                   2.0.3       2022-05-26 [1] RSPM (R 4.3.0)
##  nnet                     7.3-19      2023-05-03 [3] CRAN (R 4.3.3)
##  numDeriv                 2016.8-1.1  2019-06-06 [1] RSPM (R 4.3.0)
##  paletteer                1.6.0       2024-01-21 [1] RSPM (R 4.3.0)
##  patchwork              * 1.2.0       2024-01-08 [1] RSPM (R 4.3.0)
##  pbkrtest                 0.5.2       2023-01-19 [1] RSPM (R 4.3.0)
##  pheatmap               * 1.0.12      2019-01-04 [1] RSPM (R 4.3.0)
##  pillar                   1.9.0       2023-03-22 [2] RSPM (R 4.3.0)
##  pkgconfig                2.0.3       2019-09-22 [2] RSPM (R 4.3.0)
##  plotly                   4.10.4      2024-01-13 [1] RSPM (R 4.3.0)
##  plyr                     1.8.9       2023-10-02 [1] RSPM (R 4.3.0)
##  png                      0.1-8       2022-11-29 [1] RSPM (R 4.3.0)
##  Polychrome             * 1.5.1       2022-05-03 [1] RSPM (R 4.3.0)
##  postcards              * 0.2.3       2022-01-07 [1] RSPM (R 4.3.0)
##  promises                 1.3.0       2024-04-05 [2] RSPM (R 4.3.0)
##  purrr                    1.0.2       2023-08-10 [2] RSPM (R 4.3.0)
##  R.cache                  0.16.0      2022-07-21 [1] RSPM (R 4.3.0)
##  R.methodsS3              1.8.2       2022-06-13 [1] RSPM (R 4.3.0)
##  R.oo                     1.26.0      2024-01-24 [1] RSPM (R 4.3.0)
##  R.utils                  2.12.3      2023-11-18 [1] RSPM (R 4.3.0)
##  R6                       2.5.1       2021-08-19 [2] RSPM (R 4.3.0)
##  rappdirs                 0.3.3       2021-01-31 [2] RSPM (R 4.3.0)
##  rbibutils                2.2.16      2023-10-25 [1] RSPM (R 4.3.0)
##  RColorBrewer           * 1.1-3       2022-04-03 [1] RSPM (R 4.3.0)
##  Rcpp                     1.0.12      2024-01-09 [2] RSPM (R 4.3.0)
##  RCurl                    1.98-1.14   2024-01-09 [1] RSPM (R 4.3.0)
##  Rdpack                   2.6         2023-11-08 [1] RSPM (R 4.3.0)
##  recount3               * 1.12.0      2023-10-24 [1] Bioconductor
##  remaCor                  0.0.18      2024-02-08 [1] RSPM (R 4.3.0)
##  rematch2                 2.1.2       2020-05-01 [2] RSPM (R 4.3.0)
##  reshape2                 1.4.4       2020-04-09 [1] RSPM (R 4.3.0)
##  restfulr                 0.0.15      2022-06-16 [1] RSPM (R 4.3.2)
##  RhpcBLASctl              0.23-42     2023-02-11 [1] RSPM (R 4.3.0)
##  rintrojs                 0.3.4       2024-01-11 [1] RSPM (R 4.3.0)
##  rjson                    0.2.21      2022-01-09 [1] RSPM (R 4.3.0)
##  rlang                  * 1.1.3       2024-01-10 [2] RSPM (R 4.3.0)
##  rmarkdown                2.26        2024-03-05 [2] RSPM (R 4.3.0)
##  rpart                    4.1.23      2023-12-05 [3] CRAN (R 4.3.3)
##  rprojroot                2.0.4       2023-11-05 [2] RSPM (R 4.3.0)
##  Rsamtools                2.18.0      2023-10-24 [1] Bioconductor
##  RSQLite                  2.3.6       2024-03-31 [1] RSPM (R 4.3.0)
##  rstudioapi               0.16.0      2024-03-24 [2] RSPM (R 4.3.0)
##  rsvd                     1.0.5       2021-04-16 [1] RSPM (R 4.3.0)
##  rtracklayer              1.62.0      2023-10-24 [1] Bioconductor
##  S4Arrays                 1.2.1       2024-03-04 [1] Bioconductor 3.18 (R 4.3.2)
##  S4Vectors              * 0.40.2      2023-11-23 [1] Bioconductor 3.18 (R 4.3.2)
##  sass                     0.4.9       2024-03-15 [2] RSPM (R 4.3.0)
##  ScaledMatrix             1.10.0      2023-10-24 [1] Bioconductor
##  scales                   1.3.0       2023-11-28 [1] RSPM (R 4.3.0)
##  scater                 * 1.30.1      2023-11-16 [1] Bioconductor
##  scatterplot3d            0.3-44      2023-05-05 [1] RSPM (R 4.3.0)
##  scuttle                * 1.12.0      2023-10-24 [1] Bioconductor
##  sessioninfo            * 1.2.2       2021-12-06 [2] RSPM (R 4.3.0)
##  shape                    1.4.6.1     2024-02-23 [1] RSPM (R 4.3.0)
##  shiny                    1.8.1.1     2024-04-02 [2] RSPM (R 4.3.0)
##  shinyAce                 0.4.2       2022-05-06 [1] RSPM (R 4.3.0)
##  shinydashboard           0.7.2       2021-09-30 [1] RSPM (R 4.3.0)
##  shinyjs                  2.1.0       2021-12-23 [1] RSPM (R 4.3.0)
##  shinyWidgets             0.8.5       2024-04-17 [1] RSPM (R 4.3.0)
##  SingleCellExperiment   * 1.24.0      2023-10-24 [1] Bioconductor
##  smokingMouse           * 0.99.91     2024-05-17 [1] Github (LieberInstitute/smokingMouse@96d8480)
##  spam                     2.10-0      2023-10-23 [1] RSPM (R 4.3.0)
##  SparseArray              1.2.4       2024-02-11 [1] Bioconductor 3.18 (R 4.3.2)
##  sparseMatrixStats        1.14.0      2023-10-24 [1] Bioconductor
##  SpatialExperiment      * 1.12.0      2023-10-24 [1] Bioconductor
##  spatialLIBD            * 1.14.1      2023-11-30 [1] Bioconductor 3.18 (R 4.3.2)
##  statmod                  1.5.0       2023-01-06 [1] RSPM (R 4.3.0)
##  stringi                  1.8.3       2023-12-11 [2] RSPM (R 4.3.0)
##  stringr                * 1.5.1       2023-11-14 [2] RSPM (R 4.3.0)
##  styler                   1.10.3      2024-04-07 [1] RSPM (R 4.3.0)
##  SummarizedExperiment   * 1.32.0      2023-10-24 [1] Bioconductor
##  tibble                   3.2.1       2023-03-20 [2] RSPM (R 4.3.0)
##  tidyr                    1.3.1       2024-01-24 [1] RSPM (R 4.3.0)
##  tidyselect               1.2.1       2024-03-11 [1] RSPM (R 4.3.0)
##  usethis                * 2.2.3       2024-02-19 [2] RSPM (R 4.3.0)
##  utf8                     1.2.4       2023-10-22 [2] RSPM (R 4.3.0)
##  variancePartition      * 1.32.5      2024-02-16 [1] Bioconductor 3.18 (R 4.3.2)
##  vctrs                    0.6.5       2023-12-01 [2] RSPM (R 4.3.0)
##  vipor                    0.4.7       2023-12-18 [1] RSPM (R 4.3.0)
##  viridis                  0.6.5       2024-01-29 [1] RSPM (R 4.3.0)
##  viridisLite              0.4.2       2023-05-02 [1] RSPM (R 4.3.0)
##  withr                    3.0.0       2024-01-16 [2] RSPM (R 4.3.0)
##  xfun                     0.43        2024-03-25 [2] RSPM (R 4.3.0)
##  XML                      3.99-0.16.1 2024-01-22 [1] RSPM (R 4.3.0)
##  xtable                   1.8-4       2019-04-21 [2] RSPM (R 4.3.0)
##  XVector                  0.42.0      2023-10-24 [1] Bioconductor
##  yaml                     2.3.8       2023-12-11 [2] RSPM (R 4.3.0)
##  zlibbioc                 1.48.2      2024-03-13 [1] Bioconductor 3.18 (R 4.3.2)
## 
##  [1] /__w/_temp/Library
##  [2] /usr/local/lib/R/site-library
##  [3] /usr/local/lib/R/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="index.html#cb6-1" tabindex="-1"></a><span class="fu">proc.time</span>()</span></code></pre></div>
<pre><code>##    user  system elapsed 
##  12.676   0.941  13.308</code></pre>
<p>This interactive book was last updated at 2024-05-17 07:09:24.643485.</p>
</div>
<div id="license" class="section level2 unnumbered hasAnchor">
<h2>License<a href="index.html#license" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</p>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="speaqeasy-introduction.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
