<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Interactive SummarizedExperiment visualizations | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="2 Interactive SummarizedExperiment visualizations | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Interactive SummarizedExperiment visualizations | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Interactive SummarizedExperiment visualizations | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="summarizedexperiment-overview.html"/>
<link rel="next" href="recount3-introduction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.2</b> Exercises</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.3</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
<li class="chapter" data-level="2.1.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment"><i class="fa fa-check"></i><b>2.1.3</b> SpatialExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
<li class="chapter" data-level="2.4.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#explore-the-data"><i class="fa fa-check"></i><b>2.4.2</b> Explore the Data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features"><i class="fa fa-check"></i><b>2.5</b> Introduction to Advanced iSEE Features</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression"><i class="fa fa-check"></i><b>2.5.1</b> Selecting Cells Based on a Single Gene Expression</a></li>
<li class="chapter" data-level="2.5.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression"><i class="fa fa-check"></i><b>2.5.2</b> Using a Single Plot for Two Gene Co-Expression</a></li>
<li class="chapter" data-level="2.5.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes"><i class="fa fa-check"></i><b>2.5.3</b> Selecting Cells Based on the Co-Expression of Two or more Genes</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.6</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-1"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>9</b> Research talks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>9.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="9.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>9.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="9.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>9.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>10</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>10.1</b> Related past materials</a></li>
<li class="chapter" data-level="10.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>10.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="10.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>10.3</b> Live demo</a></li>
<li class="chapter" data-level="10.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-2"><i class="fa fa-check"></i><b>10.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>11</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="11.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing-overview"><i class="fa fa-check"></i><b>11.1</b> Single cell RNA sequencing overview</a></li>
<li class="chapter" data-level="11.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>11.2</b> The SingleCellExperiment class</a></li>
<li class="chapter" data-level="11.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>11.3</b> Quality Control</a></li>
<li class="chapter" data-level="11.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>11.4</b> Normalization</a></li>
<li class="chapter" data-level="11.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>11.5</b> Feature selection</a></li>
<li class="chapter" data-level="11.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>11.6</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="11.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>11.7</b> Clustering</a></li>
<li class="chapter" data-level="11.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>11.8</b> Marker gene detection</a></li>
<li class="chapter" data-level="11.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>11.9</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>12</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>12.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>12.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>12.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>12.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>13</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="13.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>13.1</b> Spatial registration</a></li>
<li class="chapter" data-level="13.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>13.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>14</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="14.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>14.1</b> here</a></li>
<li class="chapter" data-level="14.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>14.2</b> Usethis</a></li>
<li class="chapter" data-level="14.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>14.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>14.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="14.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>14.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="14.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>14.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="14.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>14.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>14.4</b> R websites</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>14.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="14.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>14.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="14.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>14.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="14.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>14.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>14.5</b> postcards</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>14.5.1</b> Installation</a></li>
<li class="chapter" data-level="14.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>14.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>14.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>14.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="14.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>14.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="14.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>14.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="14.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>14.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="14.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>14.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-4"><i class="fa fa-check"></i><b>14.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="interactive-summarizedexperiment-visualizations" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Interactive SummarizedExperiment visualizations<a href="interactive-summarizedexperiment-visualizations.html#interactive-summarizedexperiment-visualizations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Melissa Mayén Quiroz</p>
<blockquote>
<p>How can you make plots from “SummarizedExperiment” objects without having to write any code?</p>
<p>The answer is with “iSEE”</p>
</blockquote>
<ul>
<li><a href="http://bioconductor.org/packages/iSEE" class="uri">http://bioconductor.org/packages/iSEE</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/basic.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/basic.html</a></li>
</ul>
<p>iSEE is a Bioconductor package that provides an interactive Shiny-based graphical user interface for exploring data stored in SummarizedExperiment objects (Rue-Albrecht et al. 2018).</p>
<div id="classes-for-isee" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Classes for iSEE<a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>SummarizedExperiment (SE) and SingleCellExperiment (SCE) are classes in R. Classes serve as templates for creating objects that contain data and methods for manipulating those data.</p>
<div id="summarizedexperiment-class" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> SummarizedExperiment class<a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Assay Data: The primary data matrix containing quantitative measurements, such as gene expression values or read counts. Rows represent features (e.g., genes, transcripts) and columns represent samples (e.g., experimental conditions, individuals).</p></li>
<li><p>Row Metadata (rowData): Additional information about the features in the assay data. This can include annotations, identifiers, genomic coordinates, and other relevant information.</p></li>
<li><p>Column Metadata (colData): Additional information about the samples in the assay data. This can include sample annotations, experimental conditions, treatment groups, and other relevant information.</p></li>
<li><p>Metadata: Additional information about the experiment.</p></li>
</ul>
</div>
<div id="singlecellexperiment" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> SingleCellExperiment<a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This object is specifically designed to store and analyze single-cell RNA sequencing (scRNA-seq) data. It extends the SummarizedExperiment class to include specialized features for single-cell data, such as cell identifiers, dimensionality reduction results, and methods for quality control and normalization.</p>
<ul>
<li><p>Assay Data: The primary data matrix containing gene expression values or other measurements. Rows represent genes and columns represent cells.</p></li>
<li><p>ColData (Column Metadata): Additional information about each cell, such as cell type, experimental condition, or any other relevant metadata.</p></li>
<li><p>RowData (Row Metadata): Additional information about each gene, such as gene symbols, genomic coordinates, or functional annotations.</p></li>
<li><p>ReducedDims: Dimensionality reduction results, such as “principal component analysis” (PCA), “t-distributed stochastic neighbor embedding” (t-SNE), and
“Uniform Manifold Approximation and Projection” (UMAP), used for visualizing and clustering cells.</p></li>
<li><p>SpikeNames and pikeMatrix: Names of spike-in control genes used for normalization and Spike-in counts matrix, if spike-in controls are used</p></li>
<li><p>Metadata: Additional metadata about the experiment.</p></li>
</ul>
</div>
<div id="spatialexperiment" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> SpatialExperiment<a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This object extends the SingleCellExperiment class and is designed to store and analyze spatially-resolved transcriptomics data. Spatial transcriptomics combines gene expression data with spatial information, providing insights into the spatial organization of tissues.</p>
<ul>
<li><p>Assay Data: The primary data matrix containing gene expression values or other measurements. Rows represent genes and columns represent spatial spots or pixels.</p></li>
<li><p>ColData (Column Metadata): Additional information about each spatial spot or pixel, such as spatial coordinates, tissue section, or any other relevant metadata.</p></li>
<li><p>RowData (Row Metadata): Additional information about each gene, such as gene symbols, genomic coordinates, or functional annotations.</p></li>
<li><p>SpatialCoords: A matrix or data frame containing the spatial coordinates (e.g., x and y coordinates) of each spot or pixel, which is crucial for spatial analyses and visualization.</p></li>
<li><p>ImageData: Links to image data associated with the spatial transcriptomics experiment, such as histology images or microscopy images, which provide the spatial context for the transcriptomics data.</p></li>
<li><p>ReducedDims: Dimensionality reduction results for visualizing and clustering spatial spots or pixels, similar to the SingleCellExperiment class.</p></li>
<li><p>Metadata: Additional metadata about the experiment.</p></li>
</ul>
</div>
</div>
<div id="getting-started-with-isee" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Getting Started with iSEE<a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://bioconductor.org/packages/3.19/bioc/manuals/iSEE/man/iSEE.pdf">Reference manual</a></p>
<p>Adapted from <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/basic.html">The iSEE User’s Guide</a></p>
<ul>
<li>Installation (version “4.4”). In this case, the package is already installed so we just need to load it.</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="interactive-summarizedexperiment-visualizations.html#cb26-1" tabindex="-1"></a><span class="co"># if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb26-2"><a href="interactive-summarizedexperiment-visualizations.html#cb26-2" tabindex="-1"></a><span class="co">#     install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb26-3"><a href="interactive-summarizedexperiment-visualizations.html#cb26-3" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb26-4"><a href="interactive-summarizedexperiment-visualizations.html#cb26-4" tabindex="-1"></a><span class="co"># BiocManager::install(&quot;iSEE&quot;)</span></span>
<span id="cb26-5"><a href="interactive-summarizedexperiment-visualizations.html#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="interactive-summarizedexperiment-visualizations.html#cb26-6" tabindex="-1"></a><span class="fu">library</span>(iSEE)</span></code></pre></div>
<ul>
<li>Documentation</li>
</ul>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="interactive-summarizedexperiment-visualizations.html#cb27-1" tabindex="-1"></a><span class="fu">browseVignettes</span>(<span class="st">&quot;iSEE&quot;</span>)</span></code></pre></div>
<ul>
<li>Use (simple launch):</li>
</ul>
<p>If you have a “SummarizedExperiment” object (se) or an instance of a subclass, like a “SingleCellExperiment” object (sce), you can launch an iSEE app by running:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="interactive-summarizedexperiment-visualizations.html#cb28-1" tabindex="-1"></a><span class="do">## Launch iSEE for the se (&quot;SummarizedExperiment&quot; object)</span></span>
<span id="cb28-2"><a href="interactive-summarizedexperiment-visualizations.html#cb28-2" tabindex="-1"></a><span class="fu">iSEE</span>(se)</span>
<span id="cb28-3"><a href="interactive-summarizedexperiment-visualizations.html#cb28-3" tabindex="-1"></a></span>
<span id="cb28-4"><a href="interactive-summarizedexperiment-visualizations.html#cb28-4" tabindex="-1"></a><span class="do">## Launch iSEE for the sce (&quot;SingleCellExperiment&quot; object)</span></span>
<span id="cb28-5"><a href="interactive-summarizedexperiment-visualizations.html#cb28-5" tabindex="-1"></a><span class="fu">iSEE</span>(sce)</span></code></pre></div>
</div>
<div id="description-of-the-user-interface" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Description of the user interface<a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE.png" /></p>
<p>By default, the app starts with a dashboard that contains one panel or table of each type. By opening the collapsible panels named “Data parameters”, “Visual parameters”, and “Selection parameters” under each plot, we can control the content and appearance of each panel.</p>
<p>** introductory tour:</p>
<p>In the upper right corner there is a question mark icon. Clicking it and then on the hand button you can have an introductory tour. During this tour, you will be taken through the different components of the iSEE user interface and learn the basic usage mechanisms by doing small actions guided by the tutorial: the highlighted elements will be responding to your actions, while the rest of the UI will be shaded.
**</p>
<div id="header" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Header<a href="interactive-summarizedexperiment-visualizations.html#header" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The layout of the iSEE user interface uses the “shinydashboard” package. The dashboard header contains four dropdown menus.</p>
<ul>
<li>The “Organization” menu, which is identified by an icon displaying multiple windows</li>
<li>“Export” dropdown menu, which is identified by a download icon</li>
<li>The “Documentation” dropdown menu which is identified by aquestion mark icon</li>
<li>The “Additional Information” dropdown menu which is identified by the information icon (i)</li>
</ul>
<div id="organization-menu" class="section level4 hasAnchor" number="2.3.1.1">
<h4><span class="header-section-number">2.3.1.1</span> Organization menu<a href="interactive-summarizedexperiment-visualizations.html#organization-menu" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The “Organization” dropdown menu, dentified by an icon displaying multiple windows, includes:</p>
<ul>
<li>“Organize panels” button opens a modal window that contains:
<ul>
<li>A selectize input to add, remove, and reorder panels in the main interface.</li>
<li>Two inputs to control the width and height, respectively, of each panel selected above.</li>
</ul></li>
<li>The “Examine panel chart” feature, identified by a chain icon, allows you to visualize the relationships and point selections among your visible plot and table panels. Each panel is represented by a node, color-coded to match the app. (This functionality is particularly useful in sessions with many panels, helping you to see the structure of how panels send and receive data point selections).</li>
</ul>
<p>** #### Export dropdown menu</p>
<p>The “Export” dropdown menu, identified by a download icon, includes:</p>
<ul>
<li><p>The “Download panel output” feature that allows you to download a zip folder containing the currently displayed panel content, including high-resolution figures and table contents as CSV files.</p></li>
<li><p>The “Extract the R code” feature which provides a way to record the exact code that reproduces the current state of each plot. Clicking on this button opens a popup window with a text editor displaying the formatted code with syntax highlighting. You can copy this code, including initial lines and sessionInfo() commands, to your clipboard for inclusion in your analysis report or script. This code can then be further edited for publication.</p></li>
<li><p>“Display panel settings” lets you export the code defining the current state of the panels in the interface. This is useful for pre-configuring an iSEE instance to start in the current state rather than with the default set of panels.
**</p></li>
</ul>
</div>
<div id="documentation-menu" class="section level4 hasAnchor" number="2.3.1.2">
<h4><span class="header-section-number">2.3.1.2</span> Documentation Menu<a href="interactive-summarizedexperiment-visualizations.html#documentation-menu" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The “Documentation” dropdown, accessible through the question mark icon, includes:</p>
<ul>
<li><p>Interactive Tour: Launches a guided tour of iSEE, teaching basic usage interactively.</p></li>
<li><p>Open Vignette: Displays the iSEE vignette, either locally or from the Bioconductor project site.</p></li>
</ul>
</div>
<div id="additional-information-menu" class="section level4 hasAnchor" number="2.3.1.3">
<h4><span class="header-section-number">2.3.1.3</span> Additional Information Menu<a href="interactive-summarizedexperiment-visualizations.html#additional-information-menu" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The “Additional Information” dropdown, accessible through the information icon (i), includes:</p>
<ul>
<li>About this Session: Shows the output of the sessionInfo() function in a popup.</li>
<li>About iSEE: Provides information on the development team, licensing, citation, and links to the GitHub repository for following development and contributing suggestions.</li>
</ul>
</div>
</div>
<div id="panel-types" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Panel types<a href="interactive-summarizedexperiment-visualizations.html#panel-types" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The main element in the body of iSEE is the combination of panels, generated (and optionally linked to one another) according to your actions. There are currently eight standard panel types that can be generated with iSEE:</p>
<ul>
<li>Reduced dimension plot</li>
<li>Column data table</li>
<li>Column data plot</li>
<li>Feature assay plot</li>
<li>Row data table</li>
<li>Row data plot</li>
<li>Sample assay plot</li>
<li>Complex heatmap</li>
</ul>
<p>In addition, custom panel types can be defined.</p>
</div>
<div id="parameter-sets" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Parameter sets<a href="interactive-summarizedexperiment-visualizations.html#parameter-sets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For each standard plot panel, three different sets of parameters will be available in collapsible boxes:</p>
<ul>
<li><p>“Data parameters”, to control parameters specific to each type of plot.</p></li>
<li><p>“Visual parameters”, to specify parameter s that will determine the aspect of the plot, in terms of coloring, point features, and more (e.g., legend placement, font size).</p></li>
<li><p>“Selection parameters” to control the incoming point selection and link relationships to other plots.</p></li>
</ul>
</div>
<div id="reduced-dimension-plots" class="section level3 hasAnchor" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> Reduced dimension plots<a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If a “SingleCellExperiment” object is supplied to the iSEE function, reduced dimension results are extracted from the reducedDim slot.
Examples include low-dimensional embeddings from principal components analysis (PCA) or t-distributed stochastic neighbour embedding (t-SNE). These results are used to construct a two-dimensional Reduced dimension plot where each point is a sample, to facilitate efficient exploration of high-dimensional datasets. The “Data parameters” control the reducedDim slot to be displayed, as well as the two dimensions to plot against each other.</p>
<p>** Note that this builtin panel does not compute reduced dimension embeddings; they must be precomputed and available in the object provided to the iSEE function. Nevertheless, custom panels - such as the iSEEu DynamicReducedDimensionPlot can be developed and used to enable such features. **</p>
</div>
<div id="column-data-plots" class="section level3 hasAnchor" number="2.3.5">
<h3><span class="header-section-number">2.3.5</span> Column data plots<a href="interactive-summarizedexperiment-visualizations.html#column-data-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Column data plot visualizes sample metadata contained in column metadata. Different fields can be used for the x- and y-axes by selecting appropriate values in the “Data parameters” box. This plot can assume various forms, depending on the nature of the data on the x- and y-axes:</p>
<ul>
<li><p>If the y-axis is continuous and the x-axis is categorical, violin plots are generated (grouped by the x-axis factor).</p></li>
<li><p>If the y-axis is categorical and the x-axis is continuous, horizontal violin plots are generated (grouped by the y-axis factor).</p></li>
<li><p>If both axes are continuous, a scatter plot is generated. This enables the use of contours that are overlaid on top of the plot, check the “Other” box to see the available options.</p></li>
<li><p>If both axes are categorical, a plot of squares (Hinton plot) is generated where the area of each square is proportional to the number of samples within each combination of factor levels.</p></li>
</ul>
</div>
<div id="feature-assay-plots" class="section level3 hasAnchor" number="2.3.6">
<h3><span class="header-section-number">2.3.6</span> Feature assay plots<a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Feature assay plot visualizes the assayed values (e.g., gene expression) for a particular feature (e.g., gene) across the samples on the y-axis. This usually results in a (grouped) violin plot, if the x-axis is set to “None” or a categorical variable; or a scatter plot, if the x-axis is another continuous variable.</p>
<p>** Gene selection for the y-axis can be achieved by using a linked row data table in another panel. Clicking on a row in the table automatically changes the assayed values plotted on the y-axis. Alternatively, the row name can be directly entered as text that corresponds to an entry of rownames(se). (This is not effective if se does not contain row names.) **</p>
</div>
<div id="row-data-plots" class="section level3 hasAnchor" number="2.3.7">
<h3><span class="header-section-number">2.3.7</span> Row data plots<a href="interactive-summarizedexperiment-visualizations.html#row-data-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Row data plot allows the visualization of information stored in the rowData slot of a “SummarizedExperiment” object. Its behavior mirrors the implementation for the Column data plot, and correspondingly this plot can assume various forms depending on whether the data are categorical or continuous.</p>
</div>
<div id="sample-assay-plots" class="section level3 hasAnchor" number="2.3.8">
<h3><span class="header-section-number">2.3.8</span> Sample assay plots<a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Sample assay plot visualizes the assayed values (e.g., gene expression) for a particular sample (e.g., cell) across the features on the y-axis.</p>
<p>This usually results in a (grouped) violin plot, if the x-axis is set to “None” or a categorical variable (e.g., gene biotype); or a scatter plot, if the x-axis is another continuous variable.</p>
<p>Notably, the x-axis covariate can also be set to:</p>
<ul>
<li><p>A discrete row data covariates (e.g., gene biotype), to stratify the distribution of assayed values</p></li>
<li><p>A continuous row data covariate (e.g., count of cells expressing each gene)</p></li>
<li><p>Another sample, to visualize and compare the assayed values in any two samples.</p></li>
</ul>
</div>
<div id="row-data-tables" class="section level3 hasAnchor" number="2.3.9">
<h3><span class="header-section-number">2.3.9</span> Row data tables<a href="interactive-summarizedexperiment-visualizations.html#row-data-tables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Row data table contains the values of the rowData slot. If none are available, a column named Present is added and set to TRUE for all features, to avoid issues with ‘DT::datatable’ and an ‘empty DataFrame’.</p>
<p>** Typically, these tables are used to link to other plots to determine the features to use for plotting or coloring. **</p>
</div>
<div id="column-data-tables" class="section level3 hasAnchor" number="2.3.10">
<h3><span class="header-section-number">2.3.10</span> Column data tables<a href="interactive-summarizedexperiment-visualizations.html#column-data-tables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Column data table contains the values of the colData slot. Its behavior mirrors the implementation for the Row data table. Correspondingly, if none are available, a column named Present is added and set to TRUE for all samples.</p>
<p>** Typically, these tables are used to link to other plots to determine the samples to use for plotting or coloring. **</p>
</div>
<div id="heat-maps" class="section level3 hasAnchor" number="2.3.11">
<h3><span class="header-section-number">2.3.11</span> Heat maps<a href="interactive-summarizedexperiment-visualizations.html#heat-maps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Heat map panels provide a compact overview of the data for multiple features in the form of color-coded matrices. These correspond to the assays stored in the SCE/SE object, where features (e.g., genes) are the rows and samples are the columns.</p>
<p>User can select features (rows) to display from the selectize widget (which supports autocompletion), or also via other panels, like row data plots or row data tables. In addition, users can rapidly import custom lists of feature names using a modal popup that provides an Ace editor where they can directly type of paste feature names, and a file upload button that accepts text files containing one feature name per line. Users should remember to click the “Apply” button before closing the modal, to update the heat map with the new list of features.</p>
<p>The “Suggest feature order” button clusters the rows, and also rearranges the elements in the selectize according to the clustering.</p>
<p>It is also possible to choose which assay type is displayed (“logcounts” being the default choice, if available).</p>
<p>Samples in the heat map can also be annotated, simply by selecting relevant column metadata.</p>
<p>A zooming functionality is also available, restricted to the y-axis (i.e., allowing closer inspection on the individual features included).</p>
</div>
<div id="description-of-isee-functionality" class="section level3 hasAnchor" number="2.3.12">
<h3><span class="header-section-number">2.3.12</span> Description of iSEE functionality<a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="coloring-plots-by-sample-attributes" class="section level4 hasAnchor" number="2.3.12.1">
<h4><span class="header-section-number">2.3.12.1</span> Coloring plots by sample attributes<a href="interactive-summarizedexperiment-visualizations.html#coloring-plots-by-sample-attributes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="column-based-plots" class="section level5 hasAnchor" number="2.3.12.1.1">
<h5><span class="header-section-number">2.3.12.1.1</span> Column-based plots<a href="interactive-summarizedexperiment-visualizations.html#column-based-plots" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Column-based plots are:</p>
<ul>
<li><p>reduced dimension</p></li>
<li><p>feature assay</p></li>
<li><p>column data plots</p></li>
</ul>
<p>Where each data point represents a sample. Here, data points can be colored in different ways:</p>
<ul>
<li><p>The default is no color scheme (“None” in the radio button).</p></li>
<li><p>Any column of colData(se) can be used. The plot automatically adjusts the scale to use based on whether the chosen column is continuous or categorical.</p></li>
<li><p>The assay values of a particular feature in each sample can be used. The feature can be chosen either via a linked row table or selectize input (as described for the Feature assay plot panel). Users can also specify the assays from which values are extracted.</p></li>
<li><p>The identity of a particular sample can be used, which will be highlighted on the plot in a user-specified color. The sample can be chosen either via a linked column table or via a selectize input.</p></li>
</ul>
</div>
<div id="row-based-plots" class="section level5 hasAnchor" number="2.3.12.1.2">
<h5><span class="header-section-number">2.3.12.1.2</span> Row-based plots<a href="interactive-summarizedexperiment-visualizations.html#row-based-plots" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>For row-based plots (i.e., the sample assay and row data plots), each data point represents a feature. Like the column-based plots, data points can be colored by:</p>
<ul>
<li><p>“None”, yielding data points of fixed color.</p></li>
<li><p>Any column of rowData(se).</p></li>
<li><p>The identity of a particular feature, which is highlighted in the user-specified color.</p></li>
<li><p>Assay values for a particular sample.</p></li>
</ul>
</div>
</div>
<div id="controlling-point-aesthetics" class="section level4 hasAnchor" number="2.3.12.2">
<h4><span class="header-section-number">2.3.12.2</span> Controlling point aesthetics<a href="interactive-summarizedexperiment-visualizations.html#controlling-point-aesthetics" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Data points can be set to different shapes according to categorical factors in colData(se) (for column-based plots) or rowData(se) (for row-based plots). This is achieved by checking the “Shape” box to reveal the shape-setting options. The size and opacity of the data points can be modified via the options available by checking the “Point” box. This may be useful for aesthetically pleasing visualizations when the number of points is very large or small.</p>
</div>
<div id="faceting" class="section level4 hasAnchor" number="2.3.12.3">
<h4><span class="header-section-number">2.3.12.3</span> Faceting<a href="interactive-summarizedexperiment-visualizations.html#faceting" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Each point-based plot can be split into multiple facets using the options in the “Facet” checkbox. Users can facet by row and/or column, using categorical factors in colData(se) (for column-based plots) or rowData(se) (for row-based plots). This provides a convenient way to stratify points in a single plot by multiple factors of interest. Note that point selection can only occur within a single facet at a time; points cannot be selected across facets.</p>
</div>
<div id="zooming-in-and-out" class="section level4 hasAnchor" number="2.3.12.4">
<h4><span class="header-section-number">2.3.12.4</span> Zooming in and out<a href="interactive-summarizedexperiment-visualizations.html#zooming-in-and-out" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Zooming in is possible by first selecting a region of interest in a plot using the brush (drag and select); double-clicking on the brushed area then zooms into the selected area. To zoom out to the original plot, simply double-click at any location in the plot.</p>
</div>
</div>
</div>
<div id="lets-practice" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Let’s practice!<a href="interactive-summarizedexperiment-visualizations.html#lets-practice" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<script defer class="speakerdeck-embed" data-id="32acbf51b837497d92b59ca110ed8f29" data-ratio="1.7777777777777777" src="//speakerdeck.com/assets/embed.js"></script>
<div id="setting-up-the-data" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Setting up the data<a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>We’ll download a <code>SingleCellExperiment</code> object, which is similar to <code>SummarizedExperiment</code> as it extends it.
<ul>
<li><a href="http://bioconductor.org/packages/SingleCellExperiment" class="uri">http://bioconductor.org/packages/SingleCellExperiment</a></li>
<li><a href="http://bioconductor.org/packages/spatialLIBD" class="uri">http://bioconductor.org/packages/spatialLIBD</a></li>
<li><a href="https://doi.org/10.1038/s41593-020-00787-0" class="uri">https://doi.org/10.1038/s41593-020-00787-0</a></li>
<li><a href="https://osca.bioconductor.org/" class="uri">https://osca.bioconductor.org/</a></li>
<li><a href="https://www.nature.com/articles/s41592-019-0654-x" class="uri">https://www.nature.com/articles/s41592-019-0654-x</a> Figures 2 and 3</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="interactive-summarizedexperiment-visualizations.html#cb29-1" tabindex="-1"></a><span class="do">## Lets get some data using spatialLIBD</span></span>
<span id="cb29-2"><a href="interactive-summarizedexperiment-visualizations.html#cb29-2" tabindex="-1"></a>sce_layer <span class="ot">&lt;-</span> spatialLIBD<span class="sc">::</span><span class="fu">fetch_data</span>(<span class="st">&quot;sce_layer&quot;</span>)</span>
<span id="cb29-3"><a href="interactive-summarizedexperiment-visualizations.html#cb29-3" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;https://www.dropbox.com/s/bg8xwysh2vnjwvg/Human_DLPFC_Visium_processedData_sce_scran_sce_layer_spatialLIBD.Rdata?dl=1&#39;</span></span>
<span id="cb29-4"><a href="interactive-summarizedexperiment-visualizations.html#cb29-4" tabindex="-1"></a><span class="co">#&gt; 2024-06-09 01:53:10.1459 loading file /github/home/.cache/R/BiocFileCache/39833cc539a_Human_DLPFC_Visium_processedData_sce_scran_sce_layer_spatialLIBD.Rdata%3Fdl%3D1</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="interactive-summarizedexperiment-visualizations.html#cb30-1" tabindex="-1"></a>sce_layer</span>
<span id="cb30-2"><a href="interactive-summarizedexperiment-visualizations.html#cb30-2" tabindex="-1"></a><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span id="cb30-3"><a href="interactive-summarizedexperiment-visualizations.html#cb30-3" tabindex="-1"></a><span class="co">#&gt; dim: 22331 76 </span></span>
<span id="cb30-4"><a href="interactive-summarizedexperiment-visualizations.html#cb30-4" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb30-5"><a href="interactive-summarizedexperiment-visualizations.html#cb30-5" tabindex="-1"></a><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span id="cb30-6"><a href="interactive-summarizedexperiment-visualizations.html#cb30-6" tabindex="-1"></a><span class="co">#&gt; rownames(22331): ENSG00000243485 ENSG00000238009 ... ENSG00000278384 ENSG00000271254</span></span>
<span id="cb30-7"><a href="interactive-summarizedexperiment-visualizations.html#cb30-7" tabindex="-1"></a><span class="co">#&gt; rowData names(10): source type ... is_top_hvg is_top_hvg_sce_layer</span></span>
<span id="cb30-8"><a href="interactive-summarizedexperiment-visualizations.html#cb30-8" tabindex="-1"></a><span class="co">#&gt; colnames(76): 151507_Layer1 151507_Layer2 ... 151676_Layer6 151676_WM</span></span>
<span id="cb30-9"><a href="interactive-summarizedexperiment-visualizations.html#cb30-9" tabindex="-1"></a><span class="co">#&gt; colData names(13): sample_name layer_guess ... layer_guess_reordered_short spatialLIBD</span></span>
<span id="cb30-10"><a href="interactive-summarizedexperiment-visualizations.html#cb30-10" tabindex="-1"></a><span class="co">#&gt; reducedDimNames(6): PCA TSNE_perplexity5 ... UMAP_neighbors15 PCAsub</span></span>
<span id="cb30-11"><a href="interactive-summarizedexperiment-visualizations.html#cb30-11" tabindex="-1"></a><span class="co">#&gt; mainExpName: NULL</span></span>
<span id="cb30-12"><a href="interactive-summarizedexperiment-visualizations.html#cb30-12" tabindex="-1"></a><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="interactive-summarizedexperiment-visualizations.html#cb31-1" tabindex="-1"></a></span>
<span id="cb31-2"><a href="interactive-summarizedexperiment-visualizations.html#cb31-2" tabindex="-1"></a><span class="do">## We can check how big the object is with lobstr</span></span>
<span id="cb31-3"><a href="interactive-summarizedexperiment-visualizations.html#cb31-3" tabindex="-1"></a>lobstr<span class="sc">::</span><span class="fu">obj_size</span>(sce_layer)</span>
<span id="cb31-4"><a href="interactive-summarizedexperiment-visualizations.html#cb31-4" tabindex="-1"></a><span class="co">#&gt; 33.99 MB</span></span></code></pre></div>
<div style="background-color:#E0EEE0; padding:20px; font-family: sans-serif">
<p>NOTE: if you run into this error:</p>
<pre><code>Error in `BiocFileCache::bfcrpath()`:
! not all &#39;rnames&#39; found or unique.
Backtrace:
 1. spatialLIBD::fetch_data(&quot;sce_layer&quot;)
 3. BiocFileCache::bfcrpath(bfc, url)</code></pre>
<p>check the output of</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="interactive-summarizedexperiment-visualizations.html#cb33-1" tabindex="-1"></a>curl<span class="sc">::</span><span class="fu">curl_version</span>()<span class="sc">$</span>version</span>
<span id="cb33-2"><a href="interactive-summarizedexperiment-visualizations.html#cb33-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;7.81.0&quot;</span></span></code></pre></div>
<p>If it’s version 8.6.0, you likely need to upgrade to version 8.8.0. For macOS users, you can do this via <code>Homebrew</code> with</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="interactive-summarizedexperiment-visualizations.html#cb34-1" tabindex="-1"></a><span class="co">## Install homebrew from https://brew.sh/</span></span>
<span id="cb34-2"><a href="interactive-summarizedexperiment-visualizations.html#cb34-2" tabindex="-1"></a><span class="ex">brew</span> install curl</span></code></pre></div>
<p>then install <code>curl</code> from source with:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="interactive-summarizedexperiment-visualizations.html#cb35-1" tabindex="-1"></a><span class="fu">Sys.setenv</span>(<span class="at">PKG_CONFIG_PATH =</span> <span class="st">&quot;/opt/homebrew/opt/curl/lib/pkgconfig&quot;</span>)</span>
<span id="cb35-2"><a href="interactive-summarizedexperiment-visualizations.html#cb35-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;curl&quot;</span>, <span class="at">type =</span> <span class="st">&quot;source&quot;</span>)</span></code></pre></div>
<p>For all the gory details, check <a href="https://github.com/curl/curl/issues/13725" class="uri">https://github.com/curl/curl/issues/13725</a>, <a href="https://github.com/Bioconductor/BiocFileCache/issues/48" class="uri">https://github.com/Bioconductor/BiocFileCache/issues/48</a>, and related issues.</p>
</div>
</div>
<div id="explore-the-data" class="section level3 hasAnchor" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Explore the Data<a href="interactive-summarizedexperiment-visualizations.html#explore-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now we can deploy <code>iSEE()</code> to explore the data.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="interactive-summarizedexperiment-visualizations.html#cb36-1" tabindex="-1"></a><span class="do">## Load library</span></span>
<span id="cb36-2"><a href="interactive-summarizedexperiment-visualizations.html#cb36-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;iSEE&quot;</span>)</span>
<span id="cb36-3"><a href="interactive-summarizedexperiment-visualizations.html#cb36-3" tabindex="-1"></a></span>
<span id="cb36-4"><a href="interactive-summarizedexperiment-visualizations.html#cb36-4" tabindex="-1"></a><span class="do">## Deploy</span></span>
<span id="cb36-5"><a href="interactive-summarizedexperiment-visualizations.html#cb36-5" tabindex="-1"></a><span class="fu">iSEE</span>(sce_layer)</span></code></pre></div>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}
</style>
<p class="exercise">
<strong>Question 1</strong>:
Which panel Type is displaying the following plot?
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_01_02.PNG" /></p>
<p class="exercise">
<strong>Exercise 1</strong>:
Recreate the following plot.
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_01_02.PNG" /></p>
<p class="exercise">
<strong>Question 2</strong>:
What is different between this 2 plots?
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_03.PNG" /></p>
<p class="exercise">
<strong>Exercise 2</strong>:
Recreate the following plot.
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_04.PNG" /></p>
<p class="exercise">
<strong>Question 3</strong>:
What is different between this 2 plots?
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_05.PNG" /></p>
<p class="exercise">
<strong>Exercise 3</strong>:
Recreate the following plot
Ensembl IDs:
ENSG00000177757
ENSG00000237491
ENSG00000238009
ENSG00000243485
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_06.PNG" /></p>
<p class="exercise">
<p><strong>Exercise 4</strong>:
Recreate the following plot. What would you change from the last one?</p>
<p>Ensembl IDs:
ENSG00000177757
ENSG00000237491
ENSG00000238009
ENSG00000243485</p>
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_07.PNG" /></p>
<p class="exercise">
<strong>Exercise 5</strong>:
Recreate the following plot. What would you change from the last one?
Ensembl IDs:
ENSG00000177757
ENSG00000237491
ENSG00000238009
ENSG00000243485
</p>
<p><img src="https://raw.githubusercontent.com/lcolladotor/cshl_rstats_genome_scale_2024/devel/Figures/iSEE_08.PNG" /></p>
<p class="exercise">
<strong>Exercise 6</strong>:
Download only the last plot (Final HeatMap)
</p>
<p class="exercise">
<strong>Exercise 7</strong>:
Extract the R code only for the last plot.(Final HeatMap)
</p>
</div>
</div>
<div id="introduction-to-advanced-isee-features" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Introduction to Advanced iSEE Features<a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Adapted from the GitHub Issue: <a href="https://github.com/iSEE/iSEE/issues/650" class="uri">https://github.com/iSEE/iSEE/issues/650</a></p>
<p>Beyond its basic functionalities, “iSEE” offers advanced features that allow users to perform complex data manipulations interactively. This includes the ability to subset and filter cells based on gene expression criteria.</p>
<p>To begin with, we will load the necessary libraries and dataset. In this case we will be using “ReprocessedAllenData” from the “scRNAseq” package, a dataset of 379 mouse brain cells from Tasic et al. (2016).
After loading the dataset, we normalize the counts and perform a PCA (Principal Component Analysis) to prepare the data for visualization.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="interactive-summarizedexperiment-visualizations.html#cb37-1" tabindex="-1"></a><span class="fu">library</span>(scRNAseq)</span>
<span id="cb37-2"><a href="interactive-summarizedexperiment-visualizations.html#cb37-2" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb37-3"><a href="interactive-summarizedexperiment-visualizations.html#cb37-3" tabindex="-1"></a><span class="fu">library</span>(iSEE)</span>
<span id="cb37-4"><a href="interactive-summarizedexperiment-visualizations.html#cb37-4" tabindex="-1"></a></span>
<span id="cb37-5"><a href="interactive-summarizedexperiment-visualizations.html#cb37-5" tabindex="-1"></a><span class="co"># Load the dataset</span></span>
<span id="cb37-6"><a href="interactive-summarizedexperiment-visualizations.html#cb37-6" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">ReprocessedAllenData</span>(<span class="at">assays =</span> <span class="st">&quot;tophat_counts&quot;</span>)</span>
<span id="cb37-7"><a href="interactive-summarizedexperiment-visualizations.html#cb37-7" tabindex="-1"></a></span>
<span id="cb37-8"><a href="interactive-summarizedexperiment-visualizations.html#cb37-8" tabindex="-1"></a><span class="co"># Normalize counts and perform PCA</span></span>
<span id="cb37-9"><a href="interactive-summarizedexperiment-visualizations.html#cb37-9" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(sce, <span class="at">exprs_values =</span> <span class="st">&quot;tophat_counts&quot;</span>)</span>
<span id="cb37-10"><a href="interactive-summarizedexperiment-visualizations.html#cb37-10" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(sce, <span class="at">ncomponents =</span> <span class="dv">4</span>)</span></code></pre></div>
<div id="selecting-cells-based-on-a-single-gene-expression" class="section level3 hasAnchor" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Selecting Cells Based on a Single Gene Expression<a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To select cells based on the expression of a single gene using iSEE, we need to <em>create an initial list of panels that will be displayed when we launch iSEE</em>. The first panel in our list is a “FeatureAssayPlot”, which will show the expression levels of the gene “Serpine2”. By visualizing this plot, we can interactively select cells that express “Serpine2”.</p>
<p>To complement this, we add a “ReducedDimensionPlot” to our panel list. This plot will visualize the PCA and highlight the cells that we selected based on “Serpine2” expression. The linkage between these two panels allows us to see how the selected cells are distributed in the reduced dimensional space (PCA).</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="interactive-summarizedexperiment-visualizations.html#cb38-1" tabindex="-1"></a><span class="do">## Initial settings for a single gene expression</span></span>
<span id="cb38-2"><a href="interactive-summarizedexperiment-visualizations.html#cb38-2" tabindex="-1"></a>initial_single <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb38-3"><a href="interactive-summarizedexperiment-visualizations.html#cb38-3" tabindex="-1"></a>    <span class="fu">FeatureAssayPlot</span>(<span class="at">Assay =</span> <span class="st">&quot;logcounts&quot;</span>, <span class="at">YAxisFeatureName =</span> <span class="st">&quot;Serpine2&quot;</span>),</span>
<span id="cb38-4"><a href="interactive-summarizedexperiment-visualizations.html#cb38-4" tabindex="-1"></a>    <span class="fu">ReducedDimensionPlot</span>(<span class="at">Type =</span> <span class="st">&quot;PCA&quot;</span>, <span class="at">ColorBy =</span> <span class="st">&quot;Column selection&quot;</span>, <span class="at">ColumnSelectionSource =</span> <span class="st">&quot;FeatureAssayPlot1&quot;</span>)</span>
<span id="cb38-5"><a href="interactive-summarizedexperiment-visualizations.html#cb38-5" tabindex="-1"></a>)</span>
<span id="cb38-6"><a href="interactive-summarizedexperiment-visualizations.html#cb38-6" tabindex="-1"></a></span>
<span id="cb38-7"><a href="interactive-summarizedexperiment-visualizations.html#cb38-7" tabindex="-1"></a><span class="do">## Launch iSEE with the initial settings</span></span>
<span id="cb38-8"><a href="interactive-summarizedexperiment-visualizations.html#cb38-8" tabindex="-1"></a><span class="co"># iSEE(sce, initial = initial_single)</span></span></code></pre></div>
</div>
<div id="using-a-single-plot-for-two-gene-co-expression" class="section level3 hasAnchor" number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Using a Single Plot for Two Gene Co-Expression<a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To select cells based on the expression of two gene, we can use a single “FeatureAssayPlot” panel. In this setup, one gene is plotted on the x-axis and the other gene on the y-axis. This method allows us to directly visualize and select cells that express both genes simultaneously.</p>
<p>By adding a “ReducedDimensionPlot” to our initial panel list, we can again see how these selected cells are distributed in the PCA plot. This approach is simpler when dealing with only two genes and provides an intuitive way to explore co-expression patterns.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="interactive-summarizedexperiment-visualizations.html#cb39-1" tabindex="-1"></a><span class="do">## Initial settings for 2 genes expression on the same &quot;FeatureAssayPlot&quot;</span></span>
<span id="cb39-2"><a href="interactive-summarizedexperiment-visualizations.html#cb39-2" tabindex="-1"></a>initial_combined <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb39-3"><a href="interactive-summarizedexperiment-visualizations.html#cb39-3" tabindex="-1"></a>    <span class="fu">FeatureAssayPlot</span>(<span class="at">Assay =</span> <span class="st">&quot;logcounts&quot;</span>, <span class="at">XAxis =</span> <span class="st">&quot;Feature name&quot;</span>, <span class="at">XAxisFeatureName =</span> <span class="st">&quot;Serpine2&quot;</span>, <span class="at">YAxisFeatureName =</span> <span class="st">&quot;Bcl6&quot;</span>),</span>
<span id="cb39-4"><a href="interactive-summarizedexperiment-visualizations.html#cb39-4" tabindex="-1"></a>    <span class="fu">ReducedDimensionPlot</span>(<span class="at">Type =</span> <span class="st">&quot;PCA&quot;</span>, <span class="at">ColorBy =</span> <span class="st">&quot;Column selection&quot;</span>, <span class="at">ColumnSelectionSource =</span> <span class="st">&quot;FeatureAssayPlot1&quot;</span>)</span>
<span id="cb39-5"><a href="interactive-summarizedexperiment-visualizations.html#cb39-5" tabindex="-1"></a>)</span>
<span id="cb39-6"><a href="interactive-summarizedexperiment-visualizations.html#cb39-6" tabindex="-1"></a></span>
<span id="cb39-7"><a href="interactive-summarizedexperiment-visualizations.html#cb39-7" tabindex="-1"></a><span class="do">## Launch iSEE with the initial settings</span></span>
<span id="cb39-8"><a href="interactive-summarizedexperiment-visualizations.html#cb39-8" tabindex="-1"></a><span class="co"># iSEE(sce, initial = initial_combined)</span></span></code></pre></div>
</div>
<div id="selecting-cells-based-on-the-co-expression-of-two-or-more-genes" class="section level3 hasAnchor" number="2.5.3">
<h3><span class="header-section-number">2.5.3</span> Selecting Cells Based on the Co-Expression of Two or more Genes<a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In situations where we want to select cells based on the expression of two or more genes, we need to chain multiple “FeatureAssayPlot” panels together. For instance, if we are interested in cells that express both “Serpine2” and “Bcl6”, we start by creating a “FeatureAssayPlot” for “Serpine2”. Then, we add another “FeatureAssayPlot” for “Bcl6”, but this time we specify that the selection source for this plot is the “FeatureAssayPlot” for “Serpine2”. This setup ensures that only cells that were selected in the first plot (based on “Serpine2”) are displayed in the second plot (for” Bcl6”).</p>
<p>Finally, we include a ReducedDimensionPlot to visualize the PCA, highlighting the cells that meet both criteria. This chained selection process allows for more refined filtering based on multiple gene expressions.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="interactive-summarizedexperiment-visualizations.html#cb40-1" tabindex="-1"></a><span class="do">## Initial settings chainning multiple &quot;FeatureAssayPlot&quot;</span></span>
<span id="cb40-2"><a href="interactive-summarizedexperiment-visualizations.html#cb40-2" tabindex="-1"></a>initial_double <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb40-3"><a href="interactive-summarizedexperiment-visualizations.html#cb40-3" tabindex="-1"></a>    <span class="fu">FeatureAssayPlot</span>(<span class="at">Assay =</span> <span class="st">&quot;logcounts&quot;</span>, <span class="at">YAxisFeatureName =</span> <span class="st">&quot;Serpine2&quot;</span>),</span>
<span id="cb40-4"><a href="interactive-summarizedexperiment-visualizations.html#cb40-4" tabindex="-1"></a>    <span class="fu">FeatureAssayPlot</span>(<span class="at">Assay =</span> <span class="st">&quot;logcounts&quot;</span>, <span class="at">YAxisFeatureName =</span> <span class="st">&quot;Bcl6&quot;</span>, <span class="at">ColumnSelectionSource =</span> <span class="st">&quot;FeatureAssayPlot1&quot;</span>, <span class="at">ColumnSelectionRestrict =</span> <span class="cn">TRUE</span>),</span>
<span id="cb40-5"><a href="interactive-summarizedexperiment-visualizations.html#cb40-5" tabindex="-1"></a>    <span class="fu">ReducedDimensionPlot</span>(<span class="at">Type =</span> <span class="st">&quot;PCA&quot;</span>, <span class="at">ColorBy =</span> <span class="st">&quot;Column selection&quot;</span>, <span class="at">ColumnSelectionSource =</span> <span class="st">&quot;FeatureAssayPlot2&quot;</span>)</span>
<span id="cb40-6"><a href="interactive-summarizedexperiment-visualizations.html#cb40-6" tabindex="-1"></a>)</span>
<span id="cb40-7"><a href="interactive-summarizedexperiment-visualizations.html#cb40-7" tabindex="-1"></a></span>
<span id="cb40-8"><a href="interactive-summarizedexperiment-visualizations.html#cb40-8" tabindex="-1"></a><span class="do">## Launch iSEE with the initial settings</span></span>
<span id="cb40-9"><a href="interactive-summarizedexperiment-visualizations.html#cb40-9" tabindex="-1"></a><span class="co"># iSEE(sce, initial = initial_double)</span></span></code></pre></div>
</div>
</div>
<div id="references" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> References<a href="interactive-summarizedexperiment-visualizations.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/iSEE.html" class="uri">https://www.bioconductor.org/packages/release/bioc/html/iSEE.html</a>
<a href="https://github.com/iSEE/iSEE" class="uri">https://github.com/iSEE/iSEE</a>
<a href="https://shiny.posit.co/r/gallery/life-sciences/isee/" class="uri">https://shiny.posit.co/r/gallery/life-sciences/isee/</a>
<a href="https://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/basic.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/basic.html</a>
<a href="https://github.com/iSEE/iSEE/issues/650" class="uri">https://github.com/iSEE/iSEE/issues/650</a></p>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="summarizedexperiment-overview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="recount3-introduction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
