<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1 SummarizedExperiment overview | Statistical Analysis of Genome Scale Data 2024</title>
  <meta name="description" content="1 SummarizedExperiment overview | Statistical Analysis of Genome Scale Data 2024" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="1 SummarizedExperiment overview | Statistical Analysis of Genome Scale Data 2024" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1 SummarizedExperiment overview | Statistical Analysis of Genome Scale Data 2024" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="index.html"/>
<link rel="next" href="interactive-summarizedexperiment-visualizations.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html"><i class="fa fa-check"></i><b>1</b> SummarizedExperiment overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#overview-1"><i class="fa fa-check"></i><b>1.1</b> Overview</a></li>
<li class="chapter" data-level="1.2" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#exercises"><i class="fa fa-check"></i><b>1.2</b> Exercises</a></li>
<li class="chapter" data-level="1.3" data-path="summarizedexperiment-overview.html"><a href="summarizedexperiment-overview.html#solutions"><i class="fa fa-check"></i><b>1.3</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html"><i class="fa fa-check"></i><b>2</b> Interactive SummarizedExperiment visualizations</a>
<ul>
<li class="chapter" data-level="2.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#classes-for-isee"><i class="fa fa-check"></i><b>2.1</b> Classes for iSEE</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#summarizedexperiment-class"><i class="fa fa-check"></i><b>2.1.1</b> SummarizedExperiment class</a></li>
<li class="chapter" data-level="2.1.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#singlecellexperiment"><i class="fa fa-check"></i><b>2.1.2</b> SingleCellExperiment</a></li>
<li class="chapter" data-level="2.1.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#spatialexperiment"><i class="fa fa-check"></i><b>2.1.3</b> SpatialExperiment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#getting-started-with-isee"><i class="fa fa-check"></i><b>2.2</b> Getting Started with iSEE</a></li>
<li class="chapter" data-level="2.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-the-user-interface"><i class="fa fa-check"></i><b>2.3</b> Description of the user interface</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#header"><i class="fa fa-check"></i><b>2.3.1</b> Header</a></li>
<li class="chapter" data-level="2.3.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#panel-types"><i class="fa fa-check"></i><b>2.3.2</b> Panel types</a></li>
<li class="chapter" data-level="2.3.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#parameter-sets"><i class="fa fa-check"></i><b>2.3.3</b> Parameter sets</a></li>
<li class="chapter" data-level="2.3.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#reduced-dimension-plots"><i class="fa fa-check"></i><b>2.3.4</b> Reduced dimension plots</a></li>
<li class="chapter" data-level="2.3.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-plots"><i class="fa fa-check"></i><b>2.3.5</b> Column data plots</a></li>
<li class="chapter" data-level="2.3.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#feature-assay-plots"><i class="fa fa-check"></i><b>2.3.6</b> Feature assay plots</a></li>
<li class="chapter" data-level="2.3.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-plots"><i class="fa fa-check"></i><b>2.3.7</b> Row data plots</a></li>
<li class="chapter" data-level="2.3.8" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#sample-assay-plots"><i class="fa fa-check"></i><b>2.3.8</b> Sample assay plots</a></li>
<li class="chapter" data-level="2.3.9" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#row-data-tables"><i class="fa fa-check"></i><b>2.3.9</b> Row data tables</a></li>
<li class="chapter" data-level="2.3.10" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#column-data-tables"><i class="fa fa-check"></i><b>2.3.10</b> Column data tables</a></li>
<li class="chapter" data-level="2.3.11" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#heat-maps"><i class="fa fa-check"></i><b>2.3.11</b> Heat maps</a></li>
<li class="chapter" data-level="2.3.12" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#description-of-isee-functionality"><i class="fa fa-check"></i><b>2.3.12</b> Description of iSEE functionality</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#lets-practice"><i class="fa fa-check"></i><b>2.4</b> Let’s practice!</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#setting-up-the-data"><i class="fa fa-check"></i><b>2.4.1</b> Setting up the data</a></li>
<li class="chapter" data-level="2.4.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#explore-the-data"><i class="fa fa-check"></i><b>2.4.2</b> Explore the Data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#introduction-to-advanced-isee-features"><i class="fa fa-check"></i><b>2.5</b> Introduction to Advanced iSEE Features</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-a-single-gene-expression"><i class="fa fa-check"></i><b>2.5.1</b> Selecting Cells Based on a Single Gene Expression</a></li>
<li class="chapter" data-level="2.5.2" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#using-a-single-plot-for-two-gene-co-expression"><i class="fa fa-check"></i><b>2.5.2</b> Using a Single Plot for Two Gene Co-Expression</a></li>
<li class="chapter" data-level="2.5.3" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#selecting-cells-based-on-the-co-expression-of-two-or-more-genes"><i class="fa fa-check"></i><b>2.5.3</b> Selecting Cells Based on the Co-Expression of Two or more Genes</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#references"><i class="fa fa-check"></i><b>2.6</b> References</a></li>
<li class="chapter" data-level="2.7" data-path="interactive-summarizedexperiment-visualizations.html"><a href="interactive-summarizedexperiment-visualizations.html#community"><i class="fa fa-check"></i><b>2.7</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>3</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>3.1</b> recount projects</a></li>
<li class="chapter" data-level="3.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>3.2</b> Using recount3</a></li>
<li class="chapter" data-level="3.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>3.3</b> Exercise</a></li>
<li class="chapter" data-level="3.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-1"><i class="fa fa-check"></i><b>3.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html"><i class="fa fa-check"></i><b>4</b> Differential Gene Expression analysis overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#preliminary-steps"><i class="fa fa-check"></i><b>4.1</b> Preliminary steps</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#rna-seq-data-processing"><i class="fa fa-check"></i><b>4.1.1</b> RNA-seq data processing</a></li>
<li class="chapter" data-level="4.1.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.1.2</b> Exploratory Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#differential-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Differential Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#downstream-analyses"><i class="fa fa-check"></i><b>4.3</b> Downstream analyses</a></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-overview.html"><a href="differential-gene-expression-analysis-overview.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html"><i class="fa fa-check"></i><b>5</b> Differential Gene Expression analysis with <em>limma</em>-<em>voom</em></a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#nb-based-dge-methods"><i class="fa fa-check"></i><b>5.1</b> NB-based DGE methods?</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#limma-voom-pipeline"><i class="fa fa-check"></i><b>5.2</b> <em>limma</em>-<em>voom</em> pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#model.matrix"><i class="fa fa-check"></i><b>5.2.1</b> <code>model.matrix()</code></a></li>
<li class="chapter" data-level="5.2.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#voom"><i class="fa fa-check"></i><b>5.2.2</b> <code>voom()</code></a></li>
<li class="chapter" data-level="5.2.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#lmfit"><i class="fa fa-check"></i><b>5.2.3</b> <code>lmFit()</code></a></li>
<li class="chapter" data-level="5.2.4" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#ebayes"><i class="fa fa-check"></i><b>5.2.4</b> <code>eBayes()</code></a></li>
<li class="chapter" data-level="5.2.5" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#toptable"><i class="fa fa-check"></i><b>5.2.5</b> <code>topTable()</code></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#de-visualization"><i class="fa fa-check"></i><b>5.3</b> DE visualization</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#heat-maps-1"><i class="fa fa-check"></i><b>5.3.2</b> Heat maps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="differential-gene-expression-analysis-with-limma-voom.html"><a href="differential-gene-expression-analysis-with-limma-voom.html#references-2"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>6</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>6.1</b> Model objects in R</a></li>
<li class="chapter" data-level="6.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>6.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="6.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>6.3</b> Example 1</a></li>
<li class="chapter" data-level="6.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>6.4</b> Example 2</a></li>
<li class="chapter" data-level="6.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>6.5</b> Example 3</a></li>
<li class="chapter" data-level="6.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>6.6</b> Exercise</a></li>
<li class="chapter" data-level="6.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>6.7</b> To learn more</a></li>
<li class="chapter" data-level="6.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-2"><i class="fa fa-check"></i><b>6.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>7</b> DGE model building with <em>variancePartition</em></a>
<ul>
<li class="chapter" data-level="7.1" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>7.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="7.2" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>7.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="7.3" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#examine-the-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>7.3</b> Examine the expression of most affected genes by each sample variable</a></li>
<li class="chapter" data-level="" data-path="dge-model-building-with-variancepartition.html"><a href="dge-model-building-with-variancepartition.html#references-3"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html"><i class="fa fa-check"></i><b>8</b> Differential gene expression exercise</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#recap"><i class="fa fa-check"></i><b>8.1</b> Recap</a></li>
<li class="chapter" data-level="8.2" data-path="differential-gene-expression-exercise.html"><a href="differential-gene-expression-exercise.html#exercise-2"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="research-talks.html"><a href="research-talks.html"><i class="fa fa-check"></i><b>9</b> Research talks</a>
<ul>
<li class="chapter" data-level="9.1" data-path="research-talks.html"><a href="research-talks.html#fentanyl-rat-study"><i class="fa fa-check"></i><b>9.1</b> Fentanyl rat study</a></li>
<li class="chapter" data-level="9.2" data-path="research-talks.html"><a href="research-talks.html#cg-hb-cell-projectors-study"><i class="fa fa-check"></i><b>9.2</b> Cg Hb cell projectors study</a></li>
<li class="chapter" data-level="9.3" data-path="research-talks.html"><a href="research-talks.html#deconvolution-benchmark"><i class="fa fa-check"></i><b>9.3</b> deconvolution-benchmark</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>10</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="10.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>10.1</b> Related past materials</a></li>
<li class="chapter" data-level="10.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>10.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="10.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#live-demo"><i class="fa fa-check"></i><b>10.3</b> Live demo</a></li>
<li class="chapter" data-level="10.4" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-3"><i class="fa fa-check"></i><b>10.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html"><i class="fa fa-check"></i><b>11</b> scRNA-seq data analysis overview</a>
<ul>
<li class="chapter" data-level="11.1" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#single-cell-rna-sequencing-overview"><i class="fa fa-check"></i><b>11.1</b> Single cell RNA sequencing overview</a></li>
<li class="chapter" data-level="11.2" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>11.2</b> The SingleCellExperiment class</a></li>
<li class="chapter" data-level="11.3" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#quality-control"><i class="fa fa-check"></i><b>11.3</b> Quality Control</a></li>
<li class="chapter" data-level="11.4" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#normalization"><i class="fa fa-check"></i><b>11.4</b> Normalization</a></li>
<li class="chapter" data-level="11.5" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#feature-selection"><i class="fa fa-check"></i><b>11.5</b> Feature selection</a></li>
<li class="chapter" data-level="11.6" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#dimensionality-reduction"><i class="fa fa-check"></i><b>11.6</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="11.7" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#clustering"><i class="fa fa-check"></i><b>11.7</b> Clustering</a></li>
<li class="chapter" data-level="11.8" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#marker-gene-detection"><i class="fa fa-check"></i><b>11.8</b> Marker gene detection</a></li>
<li class="chapter" data-level="11.9" data-path="scrna-seq-data-analysis-overview.html"><a href="scrna-seq-data-analysis-overview.html#cell-type-annotation"><i class="fa fa-check"></i><b>11.9</b> Cell type annotation</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>12</b> Introduction to spatial transcriptomics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#visium-spatial-technology"><i class="fa fa-check"></i><b>12.1</b> 3’ Visium spatial technology</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spatial-data-visualization"><i class="fa fa-check"></i><b>12.2</b> Spatial data visualization</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#spot-level-data-exploration"><i class="fa fa-check"></i><b>12.2.1</b> Spot-level data exploration</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#layer-level-data-exploration"><i class="fa fa-check"></i><b>12.2.2</b> Layer-level data exploration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-spatial-transcriptomics.html"><a href="introduction-to-spatial-transcriptomics.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>13</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="13.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>13.1</b> Spatial registration</a></li>
<li class="chapter" data-level="13.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-3"><i class="fa fa-check"></i><b>13.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html"><i class="fa fa-check"></i><b>14</b> Making your own website with postcards</a>
<ul>
<li class="chapter" data-level="14.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#here"><i class="fa fa-check"></i><b>14.1</b> here</a></li>
<li class="chapter" data-level="14.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#usethis"><i class="fa fa-check"></i><b>14.2</b> Usethis</a></li>
<li class="chapter" data-level="14.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#git-github"><i class="fa fa-check"></i><b>14.3</b> Git + GitHub</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#prerequisites"><i class="fa fa-check"></i><b>14.3.1</b> Prerequisites</a></li>
<li class="chapter" data-level="14.3.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#creating-a-personal-access-token-pat"><i class="fa fa-check"></i><b>14.3.2</b> Creating a personal access token (PAT)</a></li>
<li class="chapter" data-level="14.3.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#initialize-git-and-github-repository"><i class="fa fa-check"></i><b>14.3.3</b> Initialize Git and GitHub repository</a></li>
<li class="chapter" data-level="14.3.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#some-other-gert-commands"><i class="fa fa-check"></i><b>14.3.4</b> Some other gert commands</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#r-websites"><i class="fa fa-check"></i><b>14.4</b> R websites</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-_site.yml"><i class="fa fa-check"></i><b>14.4.1</b> 1. Set Up _site.yml</a></li>
<li class="chapter" data-level="14.4.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-index.rmd-for-the-homepage"><i class="fa fa-check"></i><b>14.4.2</b> 2. Create index.Rmd for the Homepage</a></li>
<li class="chapter" data-level="14.4.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#render-the-site"><i class="fa fa-check"></i><b>14.4.3</b> 3. Render the Site</a></li>
<li class="chapter" data-level="14.4.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#publish-the-website"><i class="fa fa-check"></i><b>14.4.4</b> 4. Publish the Website</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#postcards"><i class="fa fa-check"></i><b>14.5</b> postcards</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#installation"><i class="fa fa-check"></i><b>14.5.1</b> Installation</a></li>
<li class="chapter" data-level="14.5.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#templates"><i class="fa fa-check"></i><b>14.5.2</b> Templates</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-your-own-website-with-postcards"><i class="fa fa-check"></i><b>14.6</b> Create your own website with postcards!</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#create-a-new-project-in-rstudio-interactive-selection"><i class="fa fa-check"></i><b>14.6.1</b> Create a New Project in RStudio (Interactive Selection)</a></li>
<li class="chapter" data-level="14.6.2" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#set-up-git-and-github"><i class="fa fa-check"></i><b>14.6.2</b> Set Up Git and GitHub</a></li>
<li class="chapter" data-level="14.6.3" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#choose-a-template"><i class="fa fa-check"></i><b>14.6.3</b> Choose a Template</a></li>
<li class="chapter" data-level="14.6.4" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#edit-with-your-information"><i class="fa fa-check"></i><b>14.6.4</b> Edit with Your Information</a></li>
<li class="chapter" data-level="14.6.5" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#deploy-the-page"><i class="fa fa-check"></i><b>14.6.5</b> Deploy the Page</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="making-your-own-website-with-postcards.html"><a href="making-your-own-website-with-postcards.html#references-4"><i class="fa fa-check"></i><b>14.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="summarizedexperiment-overview" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> SummarizedExperiment overview<a href="summarizedexperiment-overview.html#summarizedexperiment-overview" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Leo</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/lqxtgpD-heM" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p><a href="https://docs.google.com/document/d/1umDODmdQldf5w2lNDoFe-unmezHPonpCiKD270VwkrQ/edit?usp=sharing"><em>LIBD rstats club notes</em></a></p>
<div id="overview-1" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Overview<a href="summarizedexperiment-overview.html#overview-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>SummarizedExperiment</code> class is used to store experimental results in the form of matrixes. Objects of this class include observations (features) of the samples, as well as additional metadata. Usually, this type of object is automatically generated as the output of other software (ie. <a href="https://doi.org/10.1186/s12859-021-04142-3"><code>SPEAQeasy</code></a>), but you can also build them.</p>
<p>One of the main characteristics of <code>SummarizedExperiment</code> is that it allows you to handle you data in a “coordinated” way. For example, if you want to subset your data, with <code>SummarizedExperiment</code> you can do so without worrying about keeping your assays and metadata synched.</p>
<figure>
<img src="Figures/se_structure.png" width="700px" align=center />
</figure>
</div>
<div id="exercises" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Exercises<a href="summarizedexperiment-overview.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are gonna use the sample data set from the <code>airway</code> library.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="summarizedexperiment-overview.html#cb10-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</span>
<span id="cb10-2"><a href="summarizedexperiment-overview.html#cb10-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;airway&quot;</span>)</span>
<span id="cb10-3"><a href="summarizedexperiment-overview.html#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="summarizedexperiment-overview.html#cb10-4" tabindex="-1"></a><span class="fu">data</span>(airway, <span class="at">package =</span> <span class="st">&quot;airway&quot;</span>)</span>
<span id="cb10-5"><a href="summarizedexperiment-overview.html#cb10-5" tabindex="-1"></a>se <span class="ot">&lt;-</span> airway</span></code></pre></div>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}

</style>
<p class="exercise">
<strong>Exercise 1</strong>:
<strong>a)</strong> How many genes do we have in this object? And samples?
<strong>b)</strong> How many samples come from donors treated (<code>trt</code>) with dexamethasone (<code>dex</code>)?
</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="summarizedexperiment-overview.html#cb11-1" tabindex="-1"></a><span class="do">## For a) you could only print the summary of the object but since the idea is</span></span>
<span id="cb11-2"><a href="summarizedexperiment-overview.html#cb11-2" tabindex="-1"></a><span class="do">## to understand how to explore the object find other function that gives</span></span>
<span id="cb11-3"><a href="summarizedexperiment-overview.html#cb11-3" tabindex="-1"></a><span class="do">## you the answer.</span></span>
<span id="cb11-4"><a href="summarizedexperiment-overview.html#cb11-4" tabindex="-1"></a>se</span>
<span id="cb11-5"><a href="summarizedexperiment-overview.html#cb11-5" tabindex="-1"></a><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span id="cb11-6"><a href="summarizedexperiment-overview.html#cb11-6" tabindex="-1"></a><span class="co">#&gt; dim: 63677 8 </span></span>
<span id="cb11-7"><a href="summarizedexperiment-overview.html#cb11-7" tabindex="-1"></a><span class="co">#&gt; metadata(1): &#39;&#39;</span></span>
<span id="cb11-8"><a href="summarizedexperiment-overview.html#cb11-8" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb11-9"><a href="summarizedexperiment-overview.html#cb11-9" tabindex="-1"></a><span class="co">#&gt; rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492 ENSG00000273493</span></span>
<span id="cb11-10"><a href="summarizedexperiment-overview.html#cb11-10" tabindex="-1"></a><span class="co">#&gt; rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span id="cb11-11"><a href="summarizedexperiment-overview.html#cb11-11" tabindex="-1"></a><span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span id="cb11-12"><a href="summarizedexperiment-overview.html#cb11-12" tabindex="-1"></a><span class="co">#&gt; colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="summarizedexperiment-overview.html#cb12-1" tabindex="-1"></a></span>
<span id="cb12-2"><a href="summarizedexperiment-overview.html#cb12-2" tabindex="-1"></a><span class="do">## Same thing for b, you could just print the colData and count the samples,</span></span>
<span id="cb12-3"><a href="summarizedexperiment-overview.html#cb12-3" tabindex="-1"></a><span class="do">## but this is not efficient when our data consists in hundreds of samples.</span></span>
<span id="cb12-4"><a href="summarizedexperiment-overview.html#cb12-4" tabindex="-1"></a><span class="do">## Find the answer using other tools.</span></span>
<span id="cb12-5"><a href="summarizedexperiment-overview.html#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="summarizedexperiment-overview.html#cb12-6" tabindex="-1"></a><span class="fu">colData</span>(se)</span>
<span id="cb12-7"><a href="summarizedexperiment-overview.html#cb12-7" tabindex="-1"></a><span class="co">#&gt; DataFrame with 8 rows and 9 columns</span></span>
<span id="cb12-8"><a href="summarizedexperiment-overview.html#cb12-8" tabindex="-1"></a><span class="co">#&gt;            SampleName     cell      dex    albut        Run avgLength Experiment    Sample    BioSample</span></span>
<span id="cb12-9"><a href="summarizedexperiment-overview.html#cb12-9" tabindex="-1"></a><span class="co">#&gt;              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;   &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span></span>
<span id="cb12-10"><a href="summarizedexperiment-overview.html#cb12-10" tabindex="-1"></a><span class="co">#&gt; SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126  SRX384345 SRS508568 SAMN02422669</span></span>
<span id="cb12-11"><a href="summarizedexperiment-overview.html#cb12-11" tabindex="-1"></a><span class="co">#&gt; SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126  SRX384346 SRS508567 SAMN02422675</span></span>
<span id="cb12-12"><a href="summarizedexperiment-overview.html#cb12-12" tabindex="-1"></a><span class="co">#&gt; SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126  SRX384349 SRS508571 SAMN02422678</span></span>
<span id="cb12-13"><a href="summarizedexperiment-overview.html#cb12-13" tabindex="-1"></a><span class="co">#&gt; SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87  SRX384350 SRS508572 SAMN02422670</span></span>
<span id="cb12-14"><a href="summarizedexperiment-overview.html#cb12-14" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 4 rows ]</span></span></code></pre></div>
<p class="exercise">
<strong>Exercise 2</strong>:
Add another assay that has the log10 of your original counts
</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="summarizedexperiment-overview.html#cb13-1" tabindex="-1"></a><span class="do">## In our object, if you look at the part that says assays, we can see that</span></span>
<span id="cb13-2"><a href="summarizedexperiment-overview.html#cb13-2" tabindex="-1"></a><span class="do">## at the moment we only have one with the name &quot;counts&quot;.</span></span>
<span id="cb13-3"><a href="summarizedexperiment-overview.html#cb13-3" tabindex="-1"></a></span>
<span id="cb13-4"><a href="summarizedexperiment-overview.html#cb13-4" tabindex="-1"></a>se</span>
<span id="cb13-5"><a href="summarizedexperiment-overview.html#cb13-5" tabindex="-1"></a><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span id="cb13-6"><a href="summarizedexperiment-overview.html#cb13-6" tabindex="-1"></a><span class="co">#&gt; dim: 63677 8 </span></span>
<span id="cb13-7"><a href="summarizedexperiment-overview.html#cb13-7" tabindex="-1"></a><span class="co">#&gt; metadata(1): &#39;&#39;</span></span>
<span id="cb13-8"><a href="summarizedexperiment-overview.html#cb13-8" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb13-9"><a href="summarizedexperiment-overview.html#cb13-9" tabindex="-1"></a><span class="co">#&gt; rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492 ENSG00000273493</span></span>
<span id="cb13-10"><a href="summarizedexperiment-overview.html#cb13-10" tabindex="-1"></a><span class="co">#&gt; rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span id="cb13-11"><a href="summarizedexperiment-overview.html#cb13-11" tabindex="-1"></a><span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span id="cb13-12"><a href="summarizedexperiment-overview.html#cb13-12" tabindex="-1"></a><span class="co">#&gt; colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="summarizedexperiment-overview.html#cb14-1" tabindex="-1"></a></span>
<span id="cb14-2"><a href="summarizedexperiment-overview.html#cb14-2" tabindex="-1"></a><span class="do">## To see the data that&#39;s stored in that assay you can do either one of the</span></span>
<span id="cb14-3"><a href="summarizedexperiment-overview.html#cb14-3" tabindex="-1"></a><span class="do">## next commands.</span></span>
<span id="cb14-4"><a href="summarizedexperiment-overview.html#cb14-4" tabindex="-1"></a><span class="fu">assay</span>(se)</span>
<span id="cb14-5"><a href="summarizedexperiment-overview.html#cb14-5" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb14-6"><a href="summarizedexperiment-overview.html#cb14-6" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb14-7"><a href="summarizedexperiment-overview.html#cb14-7" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb14-8"><a href="summarizedexperiment-overview.html#cb14-8" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb14-9"><a href="summarizedexperiment-overview.html#cb14-9" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb14-10"><a href="summarizedexperiment-overview.html#cb14-10" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb14-11"><a href="summarizedexperiment-overview.html#cb14-11" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb14-12"><a href="summarizedexperiment-overview.html#cb14-12" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="summarizedexperiment-overview.html#cb15-1" tabindex="-1"></a><span class="fu">assays</span>(se)<span class="sc">$</span>counts</span>
<span id="cb15-2"><a href="summarizedexperiment-overview.html#cb15-2" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb15-3"><a href="summarizedexperiment-overview.html#cb15-3" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb15-4"><a href="summarizedexperiment-overview.html#cb15-4" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb15-5"><a href="summarizedexperiment-overview.html#cb15-5" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb15-6"><a href="summarizedexperiment-overview.html#cb15-6" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb15-7"><a href="summarizedexperiment-overview.html#cb15-7" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb15-8"><a href="summarizedexperiment-overview.html#cb15-8" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb15-9"><a href="summarizedexperiment-overview.html#cb15-9" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="summarizedexperiment-overview.html#cb16-1" tabindex="-1"></a></span>
<span id="cb16-2"><a href="summarizedexperiment-overview.html#cb16-2" tabindex="-1"></a><span class="do">## Note that assay() does not support $ operator</span></span>
<span id="cb16-3"><a href="summarizedexperiment-overview.html#cb16-3" tabindex="-1"></a><span class="co"># assay(se)$counts</span></span>
<span id="cb16-4"><a href="summarizedexperiment-overview.html#cb16-4" tabindex="-1"></a></span>
<span id="cb16-5"><a href="summarizedexperiment-overview.html#cb16-5" tabindex="-1"></a><span class="do">## We would have to do:</span></span>
<span id="cb16-6"><a href="summarizedexperiment-overview.html#cb16-6" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="dv">1</span>)</span>
<span id="cb16-7"><a href="summarizedexperiment-overview.html#cb16-7" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb16-8"><a href="summarizedexperiment-overview.html#cb16-8" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb16-9"><a href="summarizedexperiment-overview.html#cb16-9" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb16-10"><a href="summarizedexperiment-overview.html#cb16-10" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb16-11"><a href="summarizedexperiment-overview.html#cb16-11" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb16-12"><a href="summarizedexperiment-overview.html#cb16-12" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb16-13"><a href="summarizedexperiment-overview.html#cb16-13" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb16-14"><a href="summarizedexperiment-overview.html#cb16-14" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="summarizedexperiment-overview.html#cb17-1" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb17-2"><a href="summarizedexperiment-overview.html#cb17-2" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb17-3"><a href="summarizedexperiment-overview.html#cb17-3" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb17-4"><a href="summarizedexperiment-overview.html#cb17-4" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb17-5"><a href="summarizedexperiment-overview.html#cb17-5" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb17-6"><a href="summarizedexperiment-overview.html#cb17-6" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb17-7"><a href="summarizedexperiment-overview.html#cb17-7" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb17-8"><a href="summarizedexperiment-overview.html#cb17-8" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb17-9"><a href="summarizedexperiment-overview.html#cb17-9" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="summarizedexperiment-overview.html#cb18-1" tabindex="-1"></a></span>
<span id="cb18-2"><a href="summarizedexperiment-overview.html#cb18-2" tabindex="-1"></a><span class="do">## If you use assays() without specifying the element you want to see it</span></span>
<span id="cb18-3"><a href="summarizedexperiment-overview.html#cb18-3" tabindex="-1"></a><span class="do">## shows you the length of the list and the name of each element.</span></span>
<span id="cb18-4"><a href="summarizedexperiment-overview.html#cb18-4" tabindex="-1"></a><span class="fu">assays</span>(se)</span>
<span id="cb18-5"><a href="summarizedexperiment-overview.html#cb18-5" tabindex="-1"></a><span class="co">#&gt; List of length 1</span></span>
<span id="cb18-6"><a href="summarizedexperiment-overview.html#cb18-6" tabindex="-1"></a><span class="co">#&gt; names(1): counts</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="summarizedexperiment-overview.html#cb19-1" tabindex="-1"></a></span>
<span id="cb19-2"><a href="summarizedexperiment-overview.html#cb19-2" tabindex="-1"></a><span class="do">## To obtain a list of names as a vector you can use:</span></span>
<span id="cb19-3"><a href="summarizedexperiment-overview.html#cb19-3" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span>
<span id="cb19-4"><a href="summarizedexperiment-overview.html#cb19-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;counts&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="summarizedexperiment-overview.html#cb20-1" tabindex="-1"></a></span>
<span id="cb20-2"><a href="summarizedexperiment-overview.html#cb20-2" tabindex="-1"></a><span class="do">## Which can also be use to change the name of the assays</span></span>
<span id="cb20-3"><a href="summarizedexperiment-overview.html#cb20-3" tabindex="-1"></a><span class="fu">assayNames</span>(se)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;foo&quot;</span></span>
<span id="cb20-4"><a href="summarizedexperiment-overview.html#cb20-4" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span>
<span id="cb20-5"><a href="summarizedexperiment-overview.html#cb20-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;foo&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="summarizedexperiment-overview.html#cb21-1" tabindex="-1"></a><span class="fu">assayNames</span>(se)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;counts&quot;</span></span></code></pre></div>
<p class="exercise">
<strong>Exercise 3</strong>:
Explore the metadata and add a new column that has the library size of each sample.
</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="summarizedexperiment-overview.html#cb22-1" tabindex="-1"></a><span class="do">## To calculate the library size use</span></span>
<span id="cb22-2"><a href="summarizedexperiment-overview.html#cb22-2" tabindex="-1"></a></span>
<span id="cb22-3"><a href="summarizedexperiment-overview.html#cb22-3" tabindex="-1"></a><span class="fu">apply</span>(<span class="fu">assay</span>(se), <span class="dv">2</span>, sum)</span>
<span id="cb22-4"><a href="summarizedexperiment-overview.html#cb22-4" tabindex="-1"></a><span class="co">#&gt; SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521 </span></span>
<span id="cb22-5"><a href="summarizedexperiment-overview.html#cb22-5" tabindex="-1"></a><span class="co">#&gt;   20637971   18809481   25348649   15163415   24448408   30818215   19126151   21164133</span></span></code></pre></div>
</div>
<div id="solutions" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Solutions<a href="summarizedexperiment-overview.html#solutions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<style>
p.solution  {
background-color: #C093D6;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}
</style>
<p class="solution">
<strong>Solution 1</strong>:
</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="summarizedexperiment-overview.html#cb23-1" tabindex="-1"></a><span class="do">## For a), dim() gives the desired answer</span></span>
<span id="cb23-2"><a href="summarizedexperiment-overview.html#cb23-2" tabindex="-1"></a></span>
<span id="cb23-3"><a href="summarizedexperiment-overview.html#cb23-3" tabindex="-1"></a><span class="fu">dim</span>(se)</span>
<span id="cb23-4"><a href="summarizedexperiment-overview.html#cb23-4" tabindex="-1"></a><span class="co">#&gt; [1] 63677     8</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="summarizedexperiment-overview.html#cb24-1" tabindex="-1"></a></span>
<span id="cb24-2"><a href="summarizedexperiment-overview.html#cb24-2" tabindex="-1"></a><span class="do">## For b),</span></span>
<span id="cb24-3"><a href="summarizedexperiment-overview.html#cb24-3" tabindex="-1"></a></span>
<span id="cb24-4"><a href="summarizedexperiment-overview.html#cb24-4" tabindex="-1"></a><span class="fu">colData</span>(se)[<span class="fu">colData</span>(se)<span class="sc">$</span>dex <span class="sc">==</span> <span class="st">&quot;trt&quot;</span>, ]</span>
<span id="cb24-5"><a href="summarizedexperiment-overview.html#cb24-5" tabindex="-1"></a><span class="co">#&gt; DataFrame with 4 rows and 9 columns</span></span>
<span id="cb24-6"><a href="summarizedexperiment-overview.html#cb24-6" tabindex="-1"></a><span class="co">#&gt;            SampleName     cell      dex    albut        Run avgLength Experiment    Sample    BioSample</span></span>
<span id="cb24-7"><a href="summarizedexperiment-overview.html#cb24-7" tabindex="-1"></a><span class="co">#&gt;              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;   &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span></span>
<span id="cb24-8"><a href="summarizedexperiment-overview.html#cb24-8" tabindex="-1"></a><span class="co">#&gt; SRR1039509 GSM1275863  N61311       trt    untrt SRR1039509       126  SRX384346 SRS508567 SAMN02422675</span></span>
<span id="cb24-9"><a href="summarizedexperiment-overview.html#cb24-9" tabindex="-1"></a><span class="co">#&gt; SRR1039513 GSM1275867  N052611      trt    untrt SRR1039513        87  SRX384350 SRS508572 SAMN02422670</span></span>
<span id="cb24-10"><a href="summarizedexperiment-overview.html#cb24-10" tabindex="-1"></a><span class="co">#&gt; SRR1039517 GSM1275871  N080611      trt    untrt SRR1039517       126  SRX384354 SRS508576 SAMN02422673</span></span>
<span id="cb24-11"><a href="summarizedexperiment-overview.html#cb24-11" tabindex="-1"></a><span class="co">#&gt; SRR1039521 GSM1275875  N061011      trt    untrt SRR1039521        98  SRX384358 SRS508580 SAMN02422677</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="summarizedexperiment-overview.html#cb25-1" tabindex="-1"></a><span class="fu">colData</span>(se)[se<span class="sc">$</span>dex <span class="sc">==</span> <span class="st">&quot;trt&quot;</span>, ]</span>
<span id="cb25-2"><a href="summarizedexperiment-overview.html#cb25-2" tabindex="-1"></a><span class="co">#&gt; DataFrame with 4 rows and 9 columns</span></span>
<span id="cb25-3"><a href="summarizedexperiment-overview.html#cb25-3" tabindex="-1"></a><span class="co">#&gt;            SampleName     cell      dex    albut        Run avgLength Experiment    Sample    BioSample</span></span>
<span id="cb25-4"><a href="summarizedexperiment-overview.html#cb25-4" tabindex="-1"></a><span class="co">#&gt;              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;   &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span></span>
<span id="cb25-5"><a href="summarizedexperiment-overview.html#cb25-5" tabindex="-1"></a><span class="co">#&gt; SRR1039509 GSM1275863  N61311       trt    untrt SRR1039509       126  SRX384346 SRS508567 SAMN02422675</span></span>
<span id="cb25-6"><a href="summarizedexperiment-overview.html#cb25-6" tabindex="-1"></a><span class="co">#&gt; SRR1039513 GSM1275867  N052611      trt    untrt SRR1039513        87  SRX384350 SRS508572 SAMN02422670</span></span>
<span id="cb25-7"><a href="summarizedexperiment-overview.html#cb25-7" tabindex="-1"></a><span class="co">#&gt; SRR1039517 GSM1275871  N080611      trt    untrt SRR1039517       126  SRX384354 SRS508576 SAMN02422673</span></span>
<span id="cb25-8"><a href="summarizedexperiment-overview.html#cb25-8" tabindex="-1"></a><span class="co">#&gt; SRR1039521 GSM1275875  N061011      trt    untrt SRR1039521        98  SRX384358 SRS508580 SAMN02422677</span></span></code></pre></div>
<p class="solution">
<strong>Solution 2</strong>:
</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="summarizedexperiment-overview.html#cb26-1" tabindex="-1"></a><span class="do">## There are multiple ways to do it</span></span>
<span id="cb26-2"><a href="summarizedexperiment-overview.html#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="summarizedexperiment-overview.html#cb26-3" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="st">&quot;logcounts&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">log10</span>(<span class="fu">assay</span>(se, <span class="st">&quot;counts&quot;</span>))</span>
<span id="cb26-4"><a href="summarizedexperiment-overview.html#cb26-4" tabindex="-1"></a></span>
<span id="cb26-5"><a href="summarizedexperiment-overview.html#cb26-5" tabindex="-1"></a><span class="fu">assays</span>(se)<span class="sc">$</span>logcounts_v2 <span class="ot">&lt;-</span> <span class="fu">log10</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts)</span></code></pre></div>
<p class="solution">
<strong>Solution 3</strong>:
</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="summarizedexperiment-overview.html#cb27-1" tabindex="-1"></a><span class="do">## To add the library size we an use..</span></span>
<span id="cb27-2"><a href="summarizedexperiment-overview.html#cb27-2" tabindex="-1"></a></span>
<span id="cb27-3"><a href="summarizedexperiment-overview.html#cb27-3" tabindex="-1"></a><span class="fu">colData</span>(se)<span class="sc">$</span>library_size <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(se), <span class="dv">2</span>, sum)</span>
<span id="cb27-4"><a href="summarizedexperiment-overview.html#cb27-4" tabindex="-1"></a></span>
<span id="cb27-5"><a href="summarizedexperiment-overview.html#cb27-5" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(se))</span>
<span id="cb27-6"><a href="summarizedexperiment-overview.html#cb27-6" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;SampleName&quot;   &quot;cell&quot;         &quot;dex&quot;          &quot;albut&quot;        &quot;Run&quot;          &quot;avgLength&quot;    &quot;Experiment&quot;  </span></span>
<span id="cb27-7"><a href="summarizedexperiment-overview.html#cb27-7" tabindex="-1"></a><span class="co">#&gt;  [8] &quot;Sample&quot;       &quot;BioSample&quot;    &quot;library_size&quot;</span></span></code></pre></div>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interactive-summarizedexperiment-visualizations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
